{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ac95f0f5-7770-4628-b5cd-ed145c0bb515",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import glob\n",
    "import os\n",
    "from os.path import exists\n",
    "import re\n",
    "#from multiprocessing.dummy import Pool as ThreadPool\n",
    "from multiprocessing import Pool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "41bdbe86-4da0-406b-ba58-e1c37752571b",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "fastq_dir = '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "044c1236-9126-4d18-ab2f-47876684b8af",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "61b08369-c60e-4ff5-9354-efef4907d2dc",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'DM2_FKDL230359836-1A_22CNLCLT3_L4_1.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM2/DM2_FKDL230359836-1A_22CNLCLT3_L4_1.fq.gz',\n",
       " 'DM2_FKDL230359836-1A_22CNLCLT3_L4_2.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM2/DM2_FKDL230359836-1A_22CNLCLT3_L4_2.fq.gz',\n",
       " 'DM1_FKDL230359835-1A_22CNLCLT3_L4_2.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM1/DM1_FKDL230359835-1A_22CNLCLT3_L4_2.fq.gz',\n",
       " 'DM1_FKDL230359835-1A_22CNLCLT3_L4_1.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM1/DM1_FKDL230359835-1A_22CNLCLT3_L4_1.fq.gz',\n",
       " 'DM3_FKDL230359837-1A_22CNLCLT3_L4_2.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM3/DM3_FKDL230359837-1A_22CNLCLT3_L4_2.fq.gz',\n",
       " 'DM3_FKDL230359837-1A_22CNLCLT3_L4_1.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM3/DM3_FKDL230359837-1A_22CNLCLT3_L4_1.fq.gz',\n",
       " 'C12C_FKDL230359834-1A_22CNLCLT3_L8_1.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12C/C12C_FKDL230359834-1A_22CNLCLT3_L8_1.fq.gz',\n",
       " 'C12C_FKDL230359834-1A_22CNLCLT3_L8_2.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12C/C12C_FKDL230359834-1A_22CNLCLT3_L8_2.fq.gz',\n",
       " 'C12B_FKDL230359833-1A_22CNLCLT3_L8_1.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12B/C12B_FKDL230359833-1A_22CNLCLT3_L8_1.fq.gz',\n",
       " 'C12B_FKDL230359833-1A_22CNLCLT3_L8_2.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12B/C12B_FKDL230359833-1A_22CNLCLT3_L8_2.fq.gz',\n",
       " 'C12A_FKDL230359832-1A_22CT5CLT3_L2_1.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12A/C12A_FKDL230359832-1A_22CT5CLT3_L2_1.fq.gz',\n",
       " 'C12A_FKDL230359832-1A_22CT5CLT3_L2_2.fq.gz': '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12A/C12A_FKDL230359832-1A_22CT5CLT3_L2_2.fq.gz'}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_of_fq = {}\n",
    "for (dirpath, dirnames, filenames) in os.walk(fastq_dir):\n",
    "    for filename in filenames:\n",
    "        if filename.endswith('fq.gz'):\n",
    "            list_of_fq[filename] = os.sep.join([dirpath, filename])\n",
    "\n",
    "list_of_fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "17ecaac5-3830-47a7-866d-011e836d00e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'DM3_FKDL230359837-1A_22CNLCLT3_L4', 'C12A_FKDL230359832-1A_22CT5CLT3_L2', 'C12C_FKDL230359834-1A_22CNLCLT3_L8', 'C12B_FKDL230359833-1A_22CNLCLT3_L8', 'DM2_FKDL230359836-1A_22CNLCLT3_L4', 'DM1_FKDL230359835-1A_22CNLCLT3_L4'}\n",
      "6\n"
     ]
    }
   ],
   "source": [
    "import re\n",
    "\n",
    "list_of_sample = set([re.sub('_\\d\\.fq\\.gz$', '', fq) for fq in list_of_fq.keys()])\n",
    "\n",
    "print(list_of_sample)  # This will print something like: {'DM2', 'DM1', 'C12C', ...}\n",
    "print(len(list_of_sample))  # This will print the number of unique samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "312544cb-c4f4-419a-b54e-ccef1de5562f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "41c11679-8a22-4738-8ca0-da0436cc94d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "QC_output_dir = '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/'\n",
    "# Ensure the output directory exists\n",
    "os.makedirs(QC_output_dir, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6ae5dbf1-9900-4f1f-8326-7fb5176a0ab0",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Read1 before filtering:\n",
      "total reads: 80663152\n",
      "total bases: 12099472800\n",
      "Q20 bases: 11457706049(94.6959%)\n",
      "Q30 bases: 10856820785(89.7297%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 80663152\n",
      "total bases: 12099472800\n",
      "Q20 bases: 11876305922(98.1556%)\n",
      "Q30 bases: 11658547351(96.3558%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 80131180\n",
      "total bases: 9077033498\n",
      "Q20 bases: 8935716599(98.4431%)\n",
      "Q30 bases: 8828769181(97.2649%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 80131180\n",
      "total bases: 9077033498\n",
      "Q20 bases: 8947445343(98.5724%)\n",
      "Q30 bases: 8822127515(97.1917%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 160262360\n",
      "reads failed due to low quality: 1001670\n",
      "reads failed due to too many N: 62274\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 100474336\n",
      "bases trimmed due to adapters: 5893663962\n",
      "\n",
      "Duplication rate: 9.45277%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 97\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM1_FKDL230359835-1A_22CNLCLT3_L4_report.html\n",
      "\n",
      "fastp -i /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM1/DM1_FKDL230359835-1A_22CNLCLT3_L4_1.fq.gz -I /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM1/DM1_FKDL230359835-1A_22CNLCLT3_L4_2.fq.gz -o /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM1_FKDL230359835-1A_22CNLCLT3_L4_1_clean.fq.gz -O /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM1_FKDL230359835-1A_22CNLCLT3_L4_2_clean.fq.gz -h /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM1_FKDL230359835-1A_22CNLCLT3_L4_report.html \n",
      "fastp v0.22.0, time used: 640 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 88253817\n",
      "total bases: 13238072550\n",
      "Q20 bases: 12294721369(92.874%)\n",
      "Q30 bases: 11432680936(86.3621%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 88253817\n",
      "total bases: 13238072550\n",
      "Q20 bases: 12943254863(97.773%)\n",
      "Q30 bases: 12664915981(95.6704%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 87521233\n",
      "total bases: 9235295204\n",
      "Q20 bases: 9055544622(98.0537%)\n",
      "Q30 bases: 8932670902(96.7232%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 87521233\n",
      "total bases: 9235295204\n",
      "Q20 bases: 9078262837(98.2996%)\n",
      "Q30 bases: 8934986998(96.7483%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 175042466\n",
      "reads failed due to low quality: 1438184\n",
      "reads failed due to too many N: 26984\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 123006914\n",
      "bases trimmed due to adapters: 7798861026\n",
      "\n",
      "Duplication rate: 16.1569%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 66\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12C_FKDL230359834-1A_22CNLCLT3_L8_report.html\n",
      "\n",
      "fastp -i /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12C/C12C_FKDL230359834-1A_22CNLCLT3_L8_1.fq.gz -I /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12C/C12C_FKDL230359834-1A_22CNLCLT3_L8_2.fq.gz -o /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12C_FKDL230359834-1A_22CNLCLT3_L8_1_clean.fq.gz -O /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12C_FKDL230359834-1A_22CNLCLT3_L8_2_clean.fq.gz -h /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12C_FKDL230359834-1A_22CNLCLT3_L8_report.html \n",
      "fastp v0.22.0, time used: 731 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 94168651\n",
      "total bases: 14125297650\n",
      "Q20 bases: 13352812347(94.5312%)\n",
      "Q30 bases: 12610394749(89.2752%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 94168651\n",
      "total bases: 14125297650\n",
      "Q20 bases: 13877726828(98.2473%)\n",
      "Q30 bases: 13628495380(96.4829%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 93628083\n",
      "total bases: 10499164042\n",
      "Q20 bases: 10352983560(98.6077%)\n",
      "Q30 bases: 10237944934(97.512%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 93628083\n",
      "total bases: 10499164042\n",
      "Q20 bases: 10355393751(98.6307%)\n",
      "Q30 bases: 10210818054(97.2536%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 187256166\n",
      "reads failed due to low quality: 1015384\n",
      "reads failed due to too many N: 65752\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 117424516\n",
      "bases trimmed due to adapters: 7097906824\n",
      "\n",
      "Duplication rate: 9.19929%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 86\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM2_FKDL230359836-1A_22CNLCLT3_L4_report.html\n",
      "\n",
      "fastp -i /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM2/DM2_FKDL230359836-1A_22CNLCLT3_L4_1.fq.gz -I /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM2/DM2_FKDL230359836-1A_22CNLCLT3_L4_2.fq.gz -o /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM2_FKDL230359836-1A_22CNLCLT3_L4_1_clean.fq.gz -O /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM2_FKDL230359836-1A_22CNLCLT3_L4_2_clean.fq.gz -h /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM2_FKDL230359836-1A_22CNLCLT3_L4_report.html \n",
      "fastp v0.22.0, time used: 771 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 83684942\n",
      "total bases: 12552741300\n",
      "Q20 bases: 11826821042(94.217%)\n",
      "Q30 bases: 11111788399(88.5208%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 83684942\n",
      "total bases: 12552741300\n",
      "Q20 bases: 12315767543(98.1122%)\n",
      "Q30 bases: 12083671882(96.2632%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 83122685\n",
      "total bases: 9150649836\n",
      "Q20 bases: 9005216805(98.4107%)\n",
      "Q30 bases: 8894234091(97.1978%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 83122685\n",
      "total bases: 9150649836\n",
      "Q20 bases: 9016852602(98.5378%)\n",
      "Q30 bases: 8887681405(97.1262%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 166245370\n",
      "reads failed due to low quality: 1062414\n",
      "reads failed due to too many N: 62100\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 109792852\n",
      "bases trimmed due to adapters: 6644534666\n",
      "\n",
      "Duplication rate: 8.42085%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 78\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM3_FKDL230359837-1A_22CNLCLT3_L4_report.html\n",
      "\n",
      "fastp -i /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM3/DM3_FKDL230359837-1A_22CNLCLT3_L4_1.fq.gz -I /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/DM3/DM3_FKDL230359837-1A_22CNLCLT3_L4_2.fq.gz -o /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM3_FKDL230359837-1A_22CNLCLT3_L4_1_clean.fq.gz -O /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM3_FKDL230359837-1A_22CNLCLT3_L4_2_clean.fq.gz -h /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/DM3_FKDL230359837-1A_22CNLCLT3_L4_report.html \n",
      "fastp v0.22.0, time used: 697 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 109910407\n",
      "total bases: 16486561050\n",
      "Q20 bases: 15332631642(93.0008%)\n",
      "Q30 bases: 14374360111(87.1883%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 109910407\n",
      "total bases: 16486561050\n",
      "Q20 bases: 16220987402(98.3892%)\n",
      "Q30 bases: 15948490531(96.7363%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 109115222\n",
      "total bases: 12020079094\n",
      "Q20 bases: 11751277786(97.7637%)\n",
      "Q30 bases: 11554464796(96.1264%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 109115222\n",
      "total bases: 12020079094\n",
      "Q20 bases: 11880379949(98.8378%)\n",
      "Q30 bases: 11741260506(97.6804%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 218230444\n",
      "reads failed due to low quality: 1555034\n",
      "reads failed due to too many N: 35336\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 145360114\n",
      "bases trimmed due to adapters: 8706331906\n",
      "\n",
      "Duplication rate: 19.1532%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 86\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12B_FKDL230359833-1A_22CNLCLT3_L8_report.html\n",
      "\n",
      "fastp -i /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12B/C12B_FKDL230359833-1A_22CNLCLT3_L8_1.fq.gz -I /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12B/C12B_FKDL230359833-1A_22CNLCLT3_L8_2.fq.gz -o /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12B_FKDL230359833-1A_22CNLCLT3_L8_1_clean.fq.gz -O /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12B_FKDL230359833-1A_22CNLCLT3_L8_2_clean.fq.gz -h /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12B_FKDL230359833-1A_22CNLCLT3_L8_report.html \n",
      "fastp v0.22.0, time used: 896 seconds\n",
      "Read1 before filtering:\n",
      "total reads: 146472042\n",
      "total bases: 21970806300\n",
      "Q20 bases: 20801566460(94.6782%)\n",
      "Q30 bases: 19824927174(90.233%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 146472042\n",
      "total bases: 21970806300\n",
      "Q20 bases: 21661470362(98.5921%)\n",
      "Q30 bases: 21339752635(97.1278%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 145517883\n",
      "total bases: 17221784142\n",
      "Q20 bases: 16898817533(98.1247%)\n",
      "Q30 bases: 16644838083(96.6499%)\n",
      "\n",
      "Read2 after filtering:\n",
      "total reads: 145517883\n",
      "total bases: 17221784142\n",
      "Q20 bases: 17038475660(98.9356%)\n",
      "Q30 bases: 16849519668(97.8384%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 291035766\n",
      "reads failed due to low quality: 1851650\n",
      "reads failed due to too many N: 56668\n",
      "reads failed due to too short: 0\n",
      "reads with adapter trimmed: 163271998\n",
      "bases trimmed due to adapters: 9221648408\n",
      "\n",
      "Duplication rate: 24.8153%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 97\n",
      "\n",
      "JSON report: fastp.json\n",
      "HTML report: /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12A_FKDL230359832-1A_22CT5CLT3_L2_report.html\n",
      "\n",
      "fastp -i /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12A/C12A_FKDL230359832-1A_22CT5CLT3_L2_1.fq.gz -I /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/01.RawData/C12A/C12A_FKDL230359832-1A_22CT5CLT3_L2_2.fq.gz -o /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12A_FKDL230359832-1A_22CT5CLT3_L2_1_clean.fq.gz -O /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12A_FKDL230359832-1A_22CT5CLT3_L2_2_clean.fq.gz -h /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/QC_fastp/C12A_FKDL230359832-1A_22CT5CLT3_L2_report.html \n",
      "fastp v0.22.0, time used: 1163 seconds\n"
     ]
    }
   ],
   "source": [
    "def run_fastp(sample):\n",
    "    r1_input = list_of_fq[f'{sample}_1.fq.gz']\n",
    "    r2_input = list_of_fq[f'{sample}_2.fq.gz']\n",
    "    r1_output = os.path.join(QC_output_dir, f'{sample}_1_clean.fq.gz')\n",
    "    r2_output = os.path.join(QC_output_dir, f'{sample}_2_clean.fq.gz')\n",
    "    html_report = os.path.join(QC_output_dir, f'{sample}_report.html')\n",
    "    \n",
    "    command = [\n",
    "        'fastp',\n",
    "        '-i', r1_input,\n",
    "        '-I', r2_input,\n",
    "        '-o', r1_output,\n",
    "        '-O', r2_output,\n",
    "        '-h', html_report\n",
    "    ]\n",
    "    \n",
    "    subprocess.run(command, check=True)\n",
    "\n",
    "# Convert list_of_sample from a set to a list, as Pool.map requires a list\n",
    "list_of_sample = list(list_of_sample)\n",
    "\n",
    "# Use a pool to run fastp on multiple samples in parallel\n",
    "with Pool(3) as p:\n",
    "    p.map(run_fastp, list_of_sample)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea9d29ea-6a69-44d1-b19c-57a653f7a779",
   "metadata": {},
   "source": [
    "#### tools like Bowtie2 or BWA are more commonly used for ATAC-seq data as they are optimized for speed and efficiency in aligning non-spliced reads."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18973e2d-b9f8-483a-9819-ffd5ebcdf5ad",
   "metadata": {},
   "source": [
    "#### hisat2 might introduce gap in alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "db87db8e-1276-4620-b366-96ce057fb372",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "output_dir = '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir'\n",
    "if not os.path.exists(output_dir):\n",
    "    os.makedirs(output_dir)\n",
    "    \n",
    "sam_path = output_dir + '/1_sam/'\n",
    "bam_path = output_dir + '/2_bam/'\n",
    "if not os.path.exists(sam_path):\n",
    "    os.makedirs(sam_path)\n",
    "    \n",
    "if not os.path.exists(bam_path):\n",
    "    os.makedirs(bam_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "fab20028-0c6f-4eae-92bb-4f6d24a26e0a",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Settings:\n",
      "  Output files: \"/media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  /media/HDD2/Genomes/Ath_Ensembl56/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:01\n",
      "bmax according to bmaxDivN setting: 29870385\n",
      "Using parameters --bmax 22402789 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 22402789 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:01\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 6; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 1; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Split 1, merged 1; iterating...\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.70688e+07 (target: 22402788)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 7\n",
      "  Reserving size (22402789) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 17731233 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 17731234 for bucket 1\n",
      "Getting block 2 of 7\n",
      "  Reserving size (22402789) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 17090348 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 17090349 for bucket 2\n",
      "Getting block 3 of 7\n",
      "  Reserving size (22402789) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 16230961 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 16230962 for bucket 3\n",
      "Getting block 4 of 7\n",
      "  Reserving size (22402789) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 21059564 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:03\n",
      "Returning block of 21059565 for bucket 4\n",
      "Getting block 5 of 7\n",
      "  Reserving size (22402789) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 21279068 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:03\n",
      "Returning block of 21279069 for bucket 5\n",
      "Getting block 6 of 7\n",
      "  Reserving size (22402789) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 20672363 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 20672364 for bucket 6\n",
      "Getting block 7 of 7\n",
      "  Reserving size (22402789) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 5418000 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:00\n",
      "Returning block of 5418001 for bucket 7\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 38223602\n",
      "fchr[G]: 59775041\n",
      "fchr[T]: 81303691\n",
      "fchr[$]: 119481543\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 44026387 bytes to primary EBWT file: /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.1.bt2.tmp\n",
      "Wrote 29870392 bytes to secondary EBWT file: /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 119481543\n",
      "    bwtLen: 119481544\n",
      "    sz: 29870386\n",
      "    bwtSz: 29870386\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 7467597\n",
      "    offsSz: 29870388\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 622300\n",
      "    numLines: 622300\n",
      "    ebwtTotLen: 39827200\n",
      "    ebwtTotSz: 39827200\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:40\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 29870385\n",
      "Using parameters --bmax 22402789 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 22402789 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:01\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.70688e+07 (target: 22402788)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 7\n",
      "  Reserving size (22402789) for bucket 1\n",
      "  Calculating Z arrays for bucket 1\n",
      "  Entering block accumulator loop for bucket 1:\n",
      "  bucket 1: 10%\n",
      "  bucket 1: 20%\n",
      "  bucket 1: 30%\n",
      "  bucket 1: 40%\n",
      "  bucket 1: 50%\n",
      "  bucket 1: 60%\n",
      "  bucket 1: 70%\n",
      "  bucket 1: 80%\n",
      "  bucket 1: 90%\n",
      "  bucket 1: 100%\n",
      "  Sorting block of length 22266455 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:03\n",
      "Returning block of 22266456 for bucket 1\n",
      "Getting block 2 of 7\n",
      "  Reserving size (22402789) for bucket 2\n",
      "  Calculating Z arrays for bucket 2\n",
      "  Entering block accumulator loop for bucket 2:\n",
      "  bucket 2: 10%\n",
      "  bucket 2: 20%\n",
      "  bucket 2: 30%\n",
      "  bucket 2: 40%\n",
      "  bucket 2: 50%\n",
      "  bucket 2: 60%\n",
      "  bucket 2: 70%\n",
      "  bucket 2: 80%\n",
      "  bucket 2: 90%\n",
      "  bucket 2: 100%\n",
      "  Sorting block of length 16123163 for bucket 2\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 16123164 for bucket 2\n",
      "Getting block 3 of 7\n",
      "  Reserving size (22402789) for bucket 3\n",
      "  Calculating Z arrays for bucket 3\n",
      "  Entering block accumulator loop for bucket 3:\n",
      "  bucket 3: 10%\n",
      "  bucket 3: 20%\n",
      "  bucket 3: 30%\n",
      "  bucket 3: 40%\n",
      "  bucket 3: 50%\n",
      "  bucket 3: 60%\n",
      "  bucket 3: 70%\n",
      "  bucket 3: 80%\n",
      "  bucket 3: 90%\n",
      "  bucket 3: 100%\n",
      "  Sorting block of length 13320170 for bucket 3\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 13320171 for bucket 3\n",
      "Getting block 4 of 7\n",
      "  Reserving size (22402789) for bucket 4\n",
      "  Calculating Z arrays for bucket 4\n",
      "  Entering block accumulator loop for bucket 4:\n",
      "  bucket 4: 10%\n",
      "  bucket 4: 20%\n",
      "  bucket 4: 30%\n",
      "  bucket 4: 40%\n",
      "  bucket 4: 50%\n",
      "  bucket 4: 60%\n",
      "  bucket 4: 70%\n",
      "  bucket 4: 80%\n",
      "  bucket 4: 90%\n",
      "  bucket 4: 100%\n",
      "  Sorting block of length 12560658 for bucket 4\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 12560659 for bucket 4\n",
      "Getting block 5 of 7\n",
      "  Reserving size (22402789) for bucket 5\n",
      "  Calculating Z arrays for bucket 5\n",
      "  Entering block accumulator loop for bucket 5:\n",
      "  bucket 5: 10%\n",
      "  bucket 5: 20%\n",
      "  bucket 5: 30%\n",
      "  bucket 5: 40%\n",
      "  bucket 5: 50%\n",
      "  bucket 5: 60%\n",
      "  bucket 5: 70%\n",
      "  bucket 5: 80%\n",
      "  bucket 5: 90%\n",
      "  bucket 5: 100%\n",
      "  Sorting block of length 16526718 for bucket 5\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 16526719 for bucket 5\n",
      "Getting block 6 of 7\n",
      "  Reserving size (22402789) for bucket 6\n",
      "  Calculating Z arrays for bucket 6\n",
      "  Entering block accumulator loop for bucket 6:\n",
      "  bucket 6: 10%\n",
      "  bucket 6: 20%\n",
      "  bucket 6: 30%\n",
      "  bucket 6: 40%\n",
      "  bucket 6: 50%\n",
      "  bucket 6: 60%\n",
      "  bucket 6: 70%\n",
      "  bucket 6: 80%\n",
      "  bucket 6: 90%\n",
      "  bucket 6: 100%\n",
      "  Sorting block of length 16842938 for bucket 6\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 16842939 for bucket 6\n",
      "Getting block 7 of 7\n",
      "  Reserving size (22402789) for bucket 7\n",
      "  Calculating Z arrays for bucket 7\n",
      "  Entering block accumulator loop for bucket 7:\n",
      "  bucket 7: 10%\n",
      "  bucket 7: 20%\n",
      "  bucket 7: 30%\n",
      "  bucket 7: 40%\n",
      "  bucket 7: 50%\n",
      "  bucket 7: 60%\n",
      "  bucket 7: 70%\n",
      "  bucket 7: 80%\n",
      "  bucket 7: 90%\n",
      "  bucket 7: 100%\n",
      "  Sorting block of length 21841435 for bucket 7\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:03\n",
      "Returning block of 21841436 for bucket 7\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 38223602\n",
      "fchr[G]: 59775041\n",
      "fchr[T]: 81303691\n",
      "fchr[$]: 119481543\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 44026387 bytes to primary EBWT file: /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.rev.1.bt2.tmp\n",
      "Wrote 29870392 bytes to secondary EBWT file: /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.rev.2.bt2.tmp\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 119481543\n",
      "    bwtLen: 119481544\n",
      "    sz: 29870386\n",
      "    bwtSz: 29870386\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 7467597\n",
      "    offsSz: 29870388\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 622300\n",
      "    numLines: 622300\n",
      "    ebwtTotLen: 39827200\n",
      "    ebwtTotSz: 39827200\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:35\n",
      "Renaming /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.3.bt2.tmp to /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.3.bt2\n",
      "Renaming /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.4.bt2.tmp to /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.4.bt2\n",
      "Renaming /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.1.bt2.tmp to /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.1.bt2\n",
      "Renaming /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.2.bt2.tmp to /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.2.bt2\n",
      "Renaming /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.rev.1.bt2.tmp to /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.rev.1.bt2\n",
      "Renaming /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.rev.2.bt2.tmp to /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index.rev.2.bt2\n"
     ]
    }
   ],
   "source": [
    "! /home/hu/Downloads/bowtie2-2.5.1-linux-x86_64/bowtie2-build /media/HDD2/Genomes/Ath_Ensembl56/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa /media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "764e712b-0f94-4048-a8d2-3edbedfebf67",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "80131180 reads; of these:\n",
      "  80131180 (100.00%) were paired; of these:\n",
      "    1626271 (2.03%) aligned concordantly 0 times\n",
      "    50650802 (63.21%) aligned concordantly exactly 1 time\n",
      "    27854107 (34.76%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1626271 pairs aligned concordantly 0 times; of these:\n",
      "      582436 (35.81%) aligned discordantly 1 time\n",
      "    ----\n",
      "    1043835 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      2087670 mates make up the pairs; of these:\n",
      "        1404524 (67.28%) aligned 0 times\n",
      "        204914 (9.82%) aligned exactly 1 time\n",
      "        478232 (22.91%) aligned >1 times\n",
      "99.12% overall alignment rate\n",
      "87521233 reads; of these:\n",
      "  87521233 (100.00%) were paired; of these:\n",
      "    1545221 (1.77%) aligned concordantly 0 times\n",
      "    52838444 (60.37%) aligned concordantly exactly 1 time\n",
      "    33137568 (37.86%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1545221 pairs aligned concordantly 0 times; of these:\n",
      "      526862 (34.10%) aligned discordantly 1 time\n",
      "    ----\n",
      "    1018359 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      2036718 mates make up the pairs; of these:\n",
      "        1317175 (64.67%) aligned 0 times\n",
      "        211734 (10.40%) aligned exactly 1 time\n",
      "        507809 (24.93%) aligned >1 times\n",
      "99.25% overall alignment rate\n",
      "93628083 reads; of these:\n",
      "  93628083 (100.00%) were paired; of these:\n",
      "    2199562 (2.35%) aligned concordantly 0 times\n",
      "    59798956 (63.87%) aligned concordantly exactly 1 time\n",
      "    31629565 (33.78%) aligned concordantly >1 times\n",
      "    ----\n",
      "    2199562 pairs aligned concordantly 0 times; of these:\n",
      "      913319 (41.52%) aligned discordantly 1 time\n",
      "    ----\n",
      "    1286243 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      2572486 mates make up the pairs; of these:\n",
      "        1609049 (62.55%) aligned 0 times\n",
      "        290900 (11.31%) aligned exactly 1 time\n",
      "        672537 (26.14%) aligned >1 times\n",
      "99.14% overall alignment rate\n",
      "109115222 reads; of these:\n",
      "  109115222 (100.00%) were paired; of these:\n",
      "    1758009 (1.61%) aligned concordantly 0 times\n",
      "    64160662 (58.80%) aligned concordantly exactly 1 time\n",
      "    43196551 (39.59%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1758009 pairs aligned concordantly 0 times; of these:\n",
      "      534975 (30.43%) aligned discordantly 1 time\n",
      "    ----\n",
      "    1223034 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      2446068 mates make up the pairs; of these:\n",
      "        1638479 (66.98%) aligned 0 times\n",
      "        177825 (7.27%) aligned exactly 1 time\n",
      "        629764 (25.75%) aligned >1 times\n",
      "99.25% overall alignment rate\n",
      "83122685 reads; of these:\n",
      "  83122685 (100.00%) were paired; of these:\n",
      "    1870149 (2.25%) aligned concordantly 0 times\n",
      "    53636058 (64.53%) aligned concordantly exactly 1 time\n",
      "    27616478 (33.22%) aligned concordantly >1 times\n",
      "    ----\n",
      "    1870149 pairs aligned concordantly 0 times; of these:\n",
      "      738322 (39.48%) aligned discordantly 1 time\n",
      "    ----\n",
      "    1131827 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      2263654 mates make up the pairs; of these:\n",
      "        1513981 (66.88%) aligned 0 times\n",
      "        226289 (10.00%) aligned exactly 1 time\n",
      "        523384 (23.12%) aligned >1 times\n",
      "99.09% overall alignment rate\n",
      "145517883 reads; of these:\n",
      "  145517883 (100.00%) were paired; of these:\n",
      "    4134147 (2.84%) aligned concordantly 0 times\n",
      "    85720214 (58.91%) aligned concordantly exactly 1 time\n",
      "    55663522 (38.25%) aligned concordantly >1 times\n",
      "    ----\n",
      "    4134147 pairs aligned concordantly 0 times; of these:\n",
      "      2001998 (48.43%) aligned discordantly 1 time\n",
      "    ----\n",
      "    2132149 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      4264298 mates make up the pairs; of these:\n",
      "        1758201 (41.23%) aligned 0 times\n",
      "        506694 (11.88%) aligned exactly 1 time\n",
      "        1999403 (46.89%) aligned >1 times\n",
      "99.40% overall alignment rate\n"
     ]
    }
   ],
   "source": [
    "def run_bowtie2(sample):\n",
    "    r1_input = os.path.join(QC_output_dir, f'{sample}_1_clean.fq.gz')\n",
    "    r2_input = os.path.join(QC_output_dir, f'{sample}_2_clean.fq.gz')\n",
    "    # now you want to split it and take just the first part ('DM2')\n",
    "    simplified_sample = re.split('_', sample)[0]\n",
    "    sam_output = os.path.join(sam_path, f'{simplified_sample}.sam')\n",
    "    \n",
    "    command = [\n",
    "        '/home/hu/Downloads/bowtie2-2.5.1-linux-x86_64/bowtie2',\n",
    "        '-x', '/media/HDD2/Genomes/Ath_Ensembl56/Ath_bowtie2_index',  # replace with your actual reference path\n",
    "        '-1', r1_input,\n",
    "        '-2', r2_input,\n",
    "        '-S', sam_output,\n",
    "        '-p', '31' # threads\n",
    "    ]\n",
    "    \n",
    "    subprocess.run(command, check=True)\n",
    "\n",
    "# Run Bowtie2 alignment on all samples in parallel\n",
    "with Pool(4) as p:\n",
    "    p.map(run_bowtie2, list_of_sample)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e1f054df-ed52-4476-a601-41fe1d6fe3ce",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glob.glob(os.path.join(sam_path, '*.sam'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "905fedf9-7887-4a33-a341-19563bd5f322",
   "metadata": {},
   "outputs": [],
   "source": [
    "sam_files = glob.glob(os.path.join(sam_path, '*.sam'))\n",
    "\n",
    "# Extracting the sample names from the file paths\n",
    "list_of_sample = [os.path.basename(sam_file).replace('.sam', '') for sam_file in sam_files]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e08c1772-715b-4736-91ed-df70eaf11ce1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_of_sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "a4bd0317-a96b-449a-aa80-c1757fc1d240",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 60 files and 30 in-memory blocks...\n",
      "[bam_sort_core] merging from 60 files and 30 in-memory blocks...\n",
      "[bam_sort_core] merging from 60 files and 30 in-memory blocks...\n",
      "[bam_sort_core] merging from 90 files and 30 in-memory blocks...\n",
      "[bam_sort_core] merging from 60 files and 30 in-memory blocks...\n",
      "[bam_sort_core] merging from 60 files and 30 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "def process_sam(sample):\n",
    "    sam_input = os.path.join(sam_path, f'{sample}.sam')\n",
    "    bam_output = os.path.join(bam_path, f'{sample}_sorted.bam')\n",
    "    \n",
    "    # Convert SAM to BAM\n",
    "    command1 = [\n",
    "        'samtools',\n",
    "        'view',\n",
    "        '-@', '30',  # using multi threads\n",
    "        '-bS', sam_input\n",
    "    ]\n",
    "    \n",
    "    # Sort BAM\n",
    "    command2 = [\n",
    "        'samtools',\n",
    "        'sort',\n",
    "        '-@', '30',  # using multi threads\n",
    "        '-o', bam_output\n",
    "    ]\n",
    "    \n",
    "    # Index BAM\n",
    "    command3 = [\n",
    "        'samtools',\n",
    "        'index',\n",
    "        '-@', '30',  # using multi threads\n",
    "        bam_output\n",
    "    ]\n",
    "    \n",
    "    # Run commands\n",
    "    with subprocess.Popen(command1, stdout=subprocess.PIPE) as p1:\n",
    "        with subprocess.Popen(command2, stdin=p1.stdout) as p2:\n",
    "            p2.wait()\n",
    "    subprocess.run(command3, check=True)\n",
    "    \n",
    "    \n",
    "# Process SAM files for all samples in parallel\n",
    "with Pool(4) as p:\n",
    "    p.map(process_sam, list_of_sample)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "9943f476-7b64-4727-90f6-010839a71e00",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setting up directories for cleaned BAM files and merged BAM file\n",
    "clean_bam_path = os.path.join(output_dir, '3_clean_bam')\n",
    "merged_bam_path = os.path.join(output_dir, '4_merged_bam')\n",
    "\n",
    "os.makedirs(clean_bam_path, exist_ok=True)\n",
    "os.makedirs(merged_bam_path, exist_ok=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5a28af21-1bf4-444d-8159-3bcdaba783a0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/2_bam/'"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6fdfeca5-9f20-413c-9498-54b0f5958dcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "bam_files = glob.glob(os.path.join(bam_path, '*.bam'))\n",
    "\n",
    "# Extracting the sample names from the file paths\n",
    "list_of_sample = [os.path.basename(bam_file).replace('.bam', '') for bam_file in bam_files]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "f8fdf889-4136-481b-8e29-a01f442d4096",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['DM3', 'DM1', 'C12C', 'DM2', 'C12A', 'C12B']"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_of_sample = [sample.replace('_sorted', '') for sample in list_of_sample]\n",
    "list_of_sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0dedf375-6e88-4801-8164-7122be693f2f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "640e24fd-1389-4470-a180-87d4416a8a3c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "finding positions of the duplicate reads in the file...\n",
      "finding positions of the duplicate reads in the file...\n",
      "finding positions of the duplicate reads in the file...\n",
      "finding positions of the duplicate reads in the file...\n",
      "  sorted 79309995 end pairs\n",
      "     and 0 single ends (among them 0 unmatched pairs)\n",
      "  collecting indices of duplicate reads...   sorted 82244886 end pairs\n",
      "     and 0 single ends (among them 0 unmatched pairs)\n",
      "  collecting indices of duplicate reads...   done in 7338 ms\n",
      "  found 42538752 duplicates\n",
      "collected list of positions in 1 min 33 sec\n",
      "marking duplicates...\n",
      "  done in 6896 ms\n",
      "  found 42807056 duplicates\n",
      "collected list of positions in 1 min 34 sec\n",
      "marking duplicates...\n",
      "  sorted 86732844 end pairs\n",
      "     and 0 single ends (among them 0 unmatched pairs)\n",
      "  collecting indices of duplicate reads...   done in 8599 ms\n",
      "  found 75319478 duplicates\n",
      "collected list of positions in 2 min 4 sec\n",
      "marking duplicates...\n",
      "total time elapsed: 2 min 48 sec\n",
      "  sorted 92664430 end pairs\n",
      "     and 0 single ends (among them 0 unmatched pairs)\n",
      "  collecting indices of duplicate reads... total time elapsed: 2 min 51 sec\n",
      "  done in 8268 ms\n",
      "  found 51209818 duplicates\n",
      "collected list of positions in 1 min 57 sec\n",
      "marking duplicates...\n",
      "total time elapsed: 4 min 15 sec\n",
      "total time elapsed: 3 min 41 sec\n",
      "finding positions of the duplicate reads in the file...\n",
      "finding positions of the duplicate reads in the file...\n",
      "  sorted 108181660 end pairs\n",
      "     and 0 single ends (among them 0 unmatched pairs)\n",
      "  collecting indices of duplicate reads...   done in 10335 ms\n",
      "  found 100158090 duplicates\n",
      "collected list of positions in 2 min 46 sec\n",
      "marking duplicates...\n",
      "total time elapsed: 4 min 24 sec\n",
      "  sorted 144380491 end pairs\n",
      "     and 0 single ends (among them 0 unmatched pairs)\n",
      "  collecting indices of duplicate reads...   done in 15362 ms\n",
      "  found 135338762 duplicates\n",
      "collected list of positions in 4 min 10 sec\n",
      "marking duplicates...\n",
      "total time elapsed: 9 min 32 sec\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from multiprocessing import Pool\n",
    "\n",
    "def filter_bam(sample):\n",
    "    bam_input = os.path.join(bam_path, f'{sample}_sorted.bam')\n",
    "    \n",
    "    bam_output_temp1 = os.path.join(clean_bam_path, f'{sample}_temp1.bam')\n",
    "    bam_output_temp2 = os.path.join(clean_bam_path, f'{sample}_temp2.bam')\n",
    "    bam_output_final = os.path.join(clean_bam_path, f'{sample}_clean.bam')\n",
    "\n",
    "    # Filter using Sambamba\n",
    "    command1 = [\n",
    "        'sambamba',\n",
    "        'view',\n",
    "        '-f', 'bam',\n",
    "        '-F', 'not (unmapped or mate_is_unmapped or secondary_alignment or failed_quality_control)',\n",
    "        '--nthreads', '20',\n",
    "        '-o', bam_output_temp1,\n",
    "        bam_input\n",
    "    ]\n",
    "\n",
    "    # Mark duplicates using Sambamba\n",
    "    command2 = [\n",
    "        'sambamba',\n",
    "        'markdup',\n",
    "        '--nthreads', '20',\n",
    "        bam_output_temp1,\n",
    "        bam_output_temp2\n",
    "    ]\n",
    "\n",
    "    # Filter low-quality reads using Samtools\n",
    "    command3 = [\n",
    "        'samtools',\n",
    "        'view',\n",
    "        '-q', '20',\n",
    "        '-b',\n",
    "        bam_output_temp2,\n",
    "        '-@', '25',\n",
    "        '-o', bam_output_final\n",
    "    ]\n",
    "\n",
    "    # Indexing the final bam file using Samtools\n",
    "    command4 = [\n",
    "        # 'samtools',\n",
    "        'sambamba',\n",
    "        'index',\n",
    "        '-@', '20',\n",
    "        bam_output_final\n",
    "    ]\n",
    "\n",
    "    # Execute the commands\n",
    "    subprocess.run(command1, check=True)\n",
    "    subprocess.run(command2, check=True)\n",
    "    subprocess.run(command3, check=True)\n",
    "    subprocess.run(command4, check=True)\n",
    "\n",
    "    # Clean up temporary files\n",
    "    os.remove(bam_output_temp1)\n",
    "    os.remove(bam_output_temp2)\n",
    "\n",
    "    # Also remove the associated .bai index files if they exist\n",
    "    for temp_file in [bam_output_temp1, bam_output_temp2]:\n",
    "        bai_file = temp_file + \".bai\"\n",
    "        if os.path.exists(bai_file):\n",
    "            os.remove(bai_file)\n",
    "\n",
    "# Filter BAM files for all samples in parallel\n",
    "with Pool(4) as p:\n",
    "    p.map(filter_bam, list_of_sample)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "22829f57-f506-4349-8f6e-2ff25382441c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "a64b9eec-0171-42fe-a489-040f80adac07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\t30427671\t28389619\t0\n",
      "2\t19698289\t18428894\t0\n",
      "3\t23459830\t22719066\t0\n",
      "4\t18585056\t17390074\t0\n",
      "5\t26975502\t25169344\t0\n",
      "Mt\t366924\t1344801\t0\n",
      "Pt\t154478\t4735664\t0\n",
      "*\t0\t0\t0\n"
     ]
    }
   ],
   "source": [
    "! samtools idxstats /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/3_clean_bam/DM1_clean.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0549cc74-30f0-4310-878e-8bfa33395c23",
   "metadata": {},
   "source": [
    "##### keep_regions.bed as\n",
    "\"\"\"1\t0\t30427671\n",
    "2\t0\t19698289\n",
    "3\t0\t23459830\n",
    "4\t0\t18585056\n",
    "5\t0\t26975502\n",
    "\"\"\" are created"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "a4ef3d13-b87c-4b94-b9e3-bb43e11723d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/3_clean_bam'"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_bam_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "255719ab-9e8e-49a6-b430-ec6a77678859",
   "metadata": {},
   "outputs": [],
   "source": [
    "bed_file = '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/3_clean_bam/keep_regions.bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "37f31cb1-9145-4d13-8c61-d9f69d5a99a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "intersect_bam_path = os.path.join(output_dir, '4_intersect_bam')\n",
    "os.makedirs(intersect_bam_path, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "978ad112-abfe-4a12-97c9-5513b9ac42f3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def intersect_bam(sample):\n",
    "    bam_input = os.path.join(clean_bam_path, f'{sample}_clean.bam')\n",
    "    output_file = os.path.join(intersect_bam_path, f'{sample}_clean_filtered.bam')\n",
    "\n",
    "    command = [\n",
    "        'bedtools',\n",
    "        'intersect',\n",
    "        '-a', bam_input,\n",
    "        '-b', bed_file,\n",
    "        '-wa'\n",
    "    ]\n",
    "    \n",
    "    # Capture standard output and error\n",
    "    with open(output_file, 'wb') as out_file:\n",
    "        result = subprocess.run(command, stdout=out_file, stderr=subprocess.PIPE)\n",
    "    \n",
    "    # If there's an error, print the error message\n",
    "    if result.returncode != 0:\n",
    "        print(f\"Error processing {sample}:\\n{result.stderr.decode('utf-8')}\")\n",
    "        return  # Exit the function or handle the error as needed\n",
    "\n",
    "# Filter BAM files for all samples in parallel\n",
    "with Pool(4) as p:\n",
    "    p.map(intersect_bam, list_of_sample)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "8a39925f-3f52-433f-9505-4d35783b8f44",
   "metadata": {},
   "outputs": [],
   "source": [
    "def index_bam(sample):\n",
    "    bam_file_to_index = os.path.join(intersect_bam_path, f'{sample}_clean_filtered.bam')\n",
    "    command = [\n",
    "        'sambamba',\n",
    "        'index',\n",
    "        '-t', '20',  # Number of threads\n",
    "        bam_file_to_index\n",
    "    ]\n",
    "    \n",
    "    # Execute the command\n",
    "    result = subprocess.run(command, capture_output=True, text=True)\n",
    "\n",
    "    # If there's an error, print the error message\n",
    "    if result.returncode != 0:\n",
    "        print(f\"Error indexing {sample}:\\n{result.stderr}\")\n",
    "        return  # Exit the function or handle the error as needed\n",
    "\n",
    "# Index BAM files for all samples in parallel\n",
    "with Pool(4) as p:\n",
    "    p.map(index_bam, list_of_sample)\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "ec9a0c42-3fe0-480f-b93b-fcaeb40506b8",
   "metadata": {},
   "source": [
    "def merge_bams(group_samples):\n",
    "    group, samples = group_samples  # Unpack tuple of group name and samples list\n",
    "    input_files = [os.path.join(clean_bam_path, f'{sample}_clean.bam') for sample in samples]\n",
    "    merged_output = os.path.join(merged_bam_path, f'{group}_merged.bam')\n",
    "    \n",
    "    command = [\n",
    "        'samtools',\n",
    "        'merge',\n",
    "        '-@', '51',  # Adjust the number of threads used per merging operation\n",
    "        merged_output,\n",
    "        *input_files  # Unpack list of input files into separate arguments\n",
    "    ]\n",
    "    \n",
    "    subprocess.run(command, check=True)\n",
    "\n",
    "# Define groups\n",
    "OK_group = [sample for sample in list_of_sample if sample.startswith('OK')]\n",
    "C_group = [sample for sample in list_of_sample if sample.startswith('C')]\n",
    "\n",
    "# Create a list of tuples, each containing a group name and the list of samples for that group\n",
    "groups_to_merge = [('OK', OK_group), ('C', C_group)]\n",
    "\n",
    "# Merge BAM files for each group in parallel\n",
    "with Pool(2) as p:  # Create a pool with 2 processes, one for each group\n",
    "    p.map(merge_bams, groups_to_merge)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "c61b6b4d-50a1-481b-99c9-51d9010988fa",
   "metadata": {},
   "source": [
    "def index_and_stats(bam_file):\n",
    "    # Generate index for the BAM file\n",
    "    command1 = ['samtools', 'index',\n",
    "        '-@', '51',  # Adjust the number of threads used per merging operation                \n",
    "                bam_file]\n",
    "    subprocess.run(command1, check=True)\n",
    "\n",
    "\n",
    "\n",
    "# List all BAM files in the specified directory\n",
    "bam_files = [os.path.join(merged_bam_path, f) for f in os.listdir(merged_bam_path) if f.endswith('.bam')]\n",
    "print(bam_files)\n",
    "# Use a Pool to process BAM files in parallel\n",
    "with Pool() as p:\n",
    "    results = p.map(index_and_stats, bam_files)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
