{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "900aeda1-d458-48d8-9f0a-663f4e2c632e",
   "metadata": {},
   "source": [
    "#### create an env##\n",
    "conda create -n macs2_env -c conda-forge -c bioconda -c anaconda macs2 python=3.8 r-irkernel ipykernel mamba"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ac95f0f5-7770-4628-b5cd-ed145c0bb515",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import glob\n",
    "import os\n",
    "from os.path import exists\n",
    "import re\n",
    "#from multiprocessing.dummy import Pool as ThreadPool\n",
    "from multiprocessing import Pool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1d73b111-f61d-406c-8f14-bb63f247b4b3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "db87db8e-1276-4620-b366-96ce057fb372",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "output_dir = '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "52f605de-0112-4a11-aea6-d62e5a3c1430",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cleaned_bam_dir = os.path.join(output_dir, '4_intersect_bam')\n",
    "bam_files = glob.glob(os.path.join(cleaned_bam_dir, '*.bam'))\n",
    "bam_files"
   ]
  },
  {
   "cell_type": "raw",
   "id": "110d67b9-d4db-4868-9e00-33eae5b79b78",
   "metadata": {},
   "source": [
    "# Setting up directories for cleaned BAM files and merged BAM file\n",
    "peaks_dir = os.path.join(output_dir, '5_macs2peaks')\n",
    "\n",
    "os.makedirs(peaks_dir, exist_ok=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7443ee74-f65a-4fd1-9c5a-b6927834c08e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setting up directories for cleaned BAM files and not merged BAM file\n",
    "peaks_dir = os.path.join(output_dir, '5_macs2peaks_simple')\n",
    "\n",
    "os.makedirs(peaks_dir, exist_ok=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2ca77a1a-3799-4d16-b52d-f44e91b36fa3",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO  @ Tue, 31 Oct 2023 20:37:23: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12C_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple --format BAMPE -g 1.2e+8 -B\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12C_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: #1 read fragment files... \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: #1 read treatment fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12A_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple --format BAMPE -g 1.2e+8 -B\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12A_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: #1 read fragment files... \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: #1 read treatment fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM2_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple --format BAMPE -g 1.2e+8 -B\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM2_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: #1 read fragment files... \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:23: #1 read treatment fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:24:  1000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:25:  1000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:25:  1000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:26:  2000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:26:  2000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:27:  2000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:28:  3000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:28:  3000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:28:  3000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:30:  4000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:30:  4000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:30:  4000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:32:  5000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:32:  5000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:32:  5000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:34:  6000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:34:  6000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:34:  6000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:36:  7000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:36:  7000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:36:  7000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:37:  8000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:38:  8000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:38:  8000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:39:  9000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:39:  9000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:40:  9000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:41:  10000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:41:  10000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:42:  10000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:43:  11000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:43:  11000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:44:  11000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:45:  12000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:45:  12000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:46:  12000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:47:  13000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:47:  13000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:48:  13000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:48:  14000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:49:  14000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:50:  14000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:50:  15000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:51:  15000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:53:  16000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:53:  15000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:53:  16000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:55:  17000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:55:  16000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:55:  17000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:56:  18000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:57:  17000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:57:  18000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:58:  19000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:59:  18000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:37:59:  19000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:00:  20000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:02:  19000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:02:  20000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:03:  21000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:04:  20000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:05:  21000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:06:  22000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:07:  21000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:07:  22000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:08:  23000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:09:  22000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:09:  23000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:10:  24000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:11:  23000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:11:  24000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:11:  25000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:13:  24000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:13:  25000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:13:  26000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:15:  25000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:15:  27000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:15:  26000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:17:  26000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:17:  28000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:17:  27000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:19:  27000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:19:  29000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:19:  28000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:21:  28000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:21:  30000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:21:  29000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:23:  29000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:23:  31000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:23:  30000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:25:  30000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:25:  32000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:25:  31000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:27:  31000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:27:  33000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:27:  32000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:28:  32000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:28:  34000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:29:  33000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:30:  35000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:30:  33000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:31:  34000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:32:  36000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:32:  34000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:33:  35000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:34:  37000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:34:  35000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:35:  36000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:36:  38000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:36:  36000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:37:  37000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:38:  39000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:38:  37000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:39:  38000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:40:  40000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:40:  38000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:41:  39000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:42:  41000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:42:  39000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:43:  40000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:43: 40305828 fragments have been read. \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:43:  42000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:44:  40000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:45:  43000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:46:  41000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:47:  44000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:48:  42000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:49:  45000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:51:  43000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:51:  46000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:53:  44000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:53:  47000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:55:  45000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:55:  48000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:57:  46000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:57:  49000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:59:  50000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:38:59:  47000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:02:  51000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:02:  48000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:02: #1 mean fragment size is determined as 131.7 bp from treatment \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:02: #1 fragment size = 131.7 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:02: #1  total fragments in treatment: 40305828 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:02: #1 user defined the maximum fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:02: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:04:  52000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:04:  49000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:14:  53000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:16:  50000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:22:  54000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:37:  51000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:38:  55000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:39: #1  fragments after filtering in treatment: 36665136 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:39: #1  Redundant rate of treatment: 0.09 \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:39: #1 finished! \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:39: #2 Build Peak Model... \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:39: #2 Skipped... \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:39: #3 Call peaks... \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:39: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Tue, 31 Oct 2023 20:39:50:  52000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:03:  56000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:04:  53000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:16:  54000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:17:  57000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:27:  55000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:32:  58000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:40: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:40: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12C_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:40: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12C_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:40: #3   Pileup will be based on sequencing depth in treatment. \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:40: #3 Call peaks for each chromosome... \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:52:  56000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:58:  57000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:40:59:  59000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:00:  58000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:02:  59000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:04:  60000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:06:  61000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:07:  60000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:08:  62000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:10:  63000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:12:  64000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:15:  65000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:16:  61000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:17:  66000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:19:  67000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:20: 67925379 fragments have been read. \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:23:  62000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:29:  63000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:32:  64000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:34:  65000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:34: 65204380 fragments have been read. \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:35: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12C_clean_filtered_peaks.xls \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:35: #4 Write peak in narrowPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12C_clean_filtered_peaks.narrowPeak \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:36: #4 Write summits bed file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12C_clean_filtered_summits.bed \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:36: Done! \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:36: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM1_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple --format BAMPE -g 1.2e+8 -B\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM1_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:36: #1 read fragment files... \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:36: #1 read treatment fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:38:  1000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:47:  2000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:50:  3000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:53:  4000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:53: #1 mean fragment size is determined as 163.3 bp from treatment \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:53: #1 fragment size = 163.3 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:53: #1  total fragments in treatment: 67925379 \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:53: #1 user defined the maximum fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:53: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Tue, 31 Oct 2023 20:41:58:  5000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:02:  6000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:06: #1 mean fragment size is determined as 149.9 bp from treatment \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:06: #1 fragment size = 149.9 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:06: #1  total fragments in treatment: 65204380 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:06: #1 user defined the maximum fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:06: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:06:  7000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:09:  8000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:11:  9000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:13:  10000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:15:  11000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:17:  12000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:19:  13000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:21:  14000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:23:  15000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:25:  16000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:27:  17000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:29:  18000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:31:  19000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:33:  20000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:35:  21000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:36:  22000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:38:  23000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:40:  24000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:42:  25000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:44:  26000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:46:  27000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:48:  28000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:50:  29000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53:  30000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53: #1  fragments after filtering in treatment: 58249353 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53: #1  Redundant rate of treatment: 0.14 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53: #1 finished! \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53: #2 Build Peak Model... \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53: #2 Skipped... \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53: #3 Call peaks... \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:53: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:55:  31000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:57:  32000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:42:59:  33000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:01:  34000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:03:  35000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05:  36000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05: #1  fragments after filtering in treatment: 56179824 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05: #1  Redundant rate of treatment: 0.14 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05: #1 finished! \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05: #2 Build Peak Model... \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05: #2 Skipped... \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05: #3 Call peaks... \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:05: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:06:  37000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:08:  38000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:10:  39000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:12:  40000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:14:  41000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:16:  42000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:18:  43000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:20:  44000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:26:  45000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:28:  46000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:30:  47000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:33:  48000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:35:  49000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:37:  50000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:39:  51000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:41:  52000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:43:  53000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:45:  54000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:51:  55000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:43:52: 55403324 fragments have been read. \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:18: #1 mean fragment size is determined as 149.1 bp from treatment \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:18: #1 fragment size = 149.1 \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:18: #1  total fragments in treatment: 55403324 \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:18: #1 user defined the maximum fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:18: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:25: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:25: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM2_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:25: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM2_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:25: #3   Pileup will be based on sequencing depth in treatment. \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:25: #3 Call peaks for each chromosome... \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:32: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:32: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12A_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:32: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12A_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:32: #3   Pileup will be based on sequencing depth in treatment. \n",
      "INFO  @ Tue, 31 Oct 2023 20:44:32: #3 Call peaks for each chromosome... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:08: #1  fragments after filtering in treatment: 48316193 \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:08: #1  Redundant rate of treatment: 0.13 \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:08: #1 finished! \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:08: #2 Build Peak Model... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:08: #2 Skipped... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:08: #3 Call peaks... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:08: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:38: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM2_clean_filtered_peaks.xls \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:38: #4 Write peak in narrowPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM2_clean_filtered_peaks.narrowPeak \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:38: #4 Write summits bed file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM2_clean_filtered_summits.bed \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:38: Done! \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:39: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM3_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple --format BAMPE -g 1.2e+8 -B\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM3_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:39: #1 read fragment files... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:39: #1 read treatment fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:42:  1000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:45:  2000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:47: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12A_clean_filtered_peaks.xls \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:47: #4 Write peak in narrowPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12A_clean_filtered_peaks.narrowPeak \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:48: #4 Write summits bed file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12A_clean_filtered_summits.bed \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:48: Done! \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:48: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12B_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple --format BAMPE -g 1.2e+8 -B\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12B_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:48: #1 read fragment files... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:48: #1 read treatment fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:50:  3000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:52:  1000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:58:  2000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:45:59:  4000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:03:  5000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:04:  3000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:08:  6000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:09:  4000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:13:  7000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:17:  8000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:17:  5000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:21:  9000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:22: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:22: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM1_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:22: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM1_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:22: #3   Pileup will be based on sequencing depth in treatment. \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:22: #3 Call peaks for each chromosome... \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:24:  6000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:24:  10000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:27:  11000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:28:  7000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:29:  12000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:30:  8000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:31:  13000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:33:  9000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:34:  14000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:36:  10000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:36:  15000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:39:  16000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:41:  17000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:42:  11000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:44:  18000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:45:  12000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:47:  19000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:49:  13000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:50:  20000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:52:  14000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:54:  21000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:56:  15000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:46:57:  22000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:01:  16000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:01:  23000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:03:  24000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:03:  17000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:06:  25000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:06:  18000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:08:  26000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:08:  19000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:10:  27000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:10:  20000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:12:  28000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:12:  21000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:14:  29000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:14:  22000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:16:  30000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:16:  23000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:18:  24000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:23:  31000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:25:  32000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:27:  33000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:28: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM1_clean_filtered_peaks.xls \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:28: #4 Write peak in narrowPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM1_clean_filtered_peaks.narrowPeak \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:28: #4 Write summits bed file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM1_clean_filtered_summits.bed \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:28: Done! \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:29:  25000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:29:  34000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:31:  26000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:31:  35000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:33:  36000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:33:  27000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:34:  37000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:36:  38000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:38:  28000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:38:  39000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:40:  40000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:40:  29000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:42:  41000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:42:  30000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:44:  31000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:44:  42000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:46:  32000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:46:  43000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:48:  33000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:48:  44000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:49:  34000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:50:  45000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:51:  35000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:52:  46000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:53:  36000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:54:  47000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:55:  37000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:56:  48000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:57:  38000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:58:  49000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:59:  39000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:47:59:  50000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:00:  40000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:01:  51000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:02:  41000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:04:  52000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:05:  42000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:06:  53000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:07:  43000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:08:  54000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:09:  44000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:11:  55000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:11:  45000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:13:  56000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:13:  46000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:15:  57000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:15:  47000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:17:  58000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:17: 58034780 fragments have been read. \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:17:  48000000 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:18: 48679779 fragments have been read. \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:41: #1 mean fragment size is determined as 137.7 bp from treatment \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:41: #1 fragment size = 137.7 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:41: #1  total fragments in treatment: 48679779 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:41: #1 user defined the maximum fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:41: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:44: #1 mean fragment size is determined as 142.8 bp from treatment \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:44: #1 fragment size = 142.8 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:44: #1  total fragments in treatment: 58034780 \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:44: #1 user defined the maximum fragments... \n",
      "INFO  @ Tue, 31 Oct 2023 20:48:44: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:25: #1  fragments after filtering in treatment: 43623919 \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:25: #1  Redundant rate of treatment: 0.10 \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:25: #1 finished! \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:25: #2 Build Peak Model... \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:25: #2 Skipped... \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:25: #3 Call peaks... \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:25: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:39: #1  fragments after filtering in treatment: 50824887 \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:39: #1  Redundant rate of treatment: 0.12 \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:39: #1 finished! \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:39: #2 Build Peak Model... \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:39: #2 Skipped... \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:39: #3 Call peaks... \n",
      "INFO  @ Tue, 31 Oct 2023 20:49:39: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:40: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:40: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12B_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:40: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12B_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:40: #3   Pileup will be based on sequencing depth in treatment. \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:40: #3 Call peaks for each chromosome... \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:55: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:55: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM3_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:55: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM3_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:55: #3   Pileup will be based on sequencing depth in treatment. \n",
      "INFO  @ Tue, 31 Oct 2023 20:50:55: #3 Call peaks for each chromosome... \n",
      "INFO  @ Tue, 31 Oct 2023 20:51:45: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12B_clean_filtered_peaks.xls \n",
      "INFO  @ Tue, 31 Oct 2023 20:51:46: #4 Write peak in narrowPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12B_clean_filtered_peaks.narrowPeak \n",
      "INFO  @ Tue, 31 Oct 2023 20:51:46: #4 Write summits bed file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/C12B_clean_filtered_summits.bed \n",
      "INFO  @ Tue, 31 Oct 2023 20:51:46: Done! \n",
      "INFO  @ Tue, 31 Oct 2023 20:52:04: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM3_clean_filtered_peaks.xls \n",
      "INFO  @ Tue, 31 Oct 2023 20:52:04: #4 Write peak in narrowPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM3_clean_filtered_peaks.narrowPeak \n",
      "INFO  @ Tue, 31 Oct 2023 20:52:04: #4 Write summits bed file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/5_macs2peaks_simple/DM3_clean_filtered_summits.bed \n",
      "INFO  @ Tue, 31 Oct 2023 20:52:04: Done! \n"
     ]
    }
   ],
   "source": [
    "# Define the function to call peaks\n",
    "def call_peaks(bam_file):\n",
    "    output_prefix = os.path.join(peaks_dir, os.path.basename(bam_file).replace('.bam', ''))\n",
    "\n",
    "    # Construct the MACS2 command\n",
    "    command = [\n",
    "        'macs2',\n",
    "        'callpeak',\n",
    "        '-t', bam_file,\n",
    "        '-n', output_prefix,\n",
    "        '--outdir', peaks_dir,\n",
    "        '--format', 'BAMPE',\n",
    "        '-g', '1.2e+8',  # genome size 120M for Ath genome\n",
    "        '-B',  # This option tells MACS2 to save the pileup signal in a bedGraph file\n",
    "    ]\n",
    "    \n",
    "    # Run the MACS2 command\n",
    "    subprocess.run(command, check=True)\n",
    "\n",
    "# Define the number of parallel processes\n",
    "num_processes = 3 # Change this number as desired\n",
    "\n",
    "# Use a Pool to process BAM files in parallel\n",
    "with Pool(num_processes) as p:\n",
    "    p.map(call_peaks, bam_files)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "322508c7-7d2d-4c76-81ad-a3f96f926390",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setting up directories for cleaned BAM files and not merged BAM file\n",
    "broad_peaks_dir = os.path.join(output_dir, '12_macs2peaks_broad2')\n",
    "\n",
    "os.makedirs(broad_peaks_dir, exist_ok=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "35fc4699-3865-4b3b-8b32-c94ba614bdc1",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO  @ Wed, 24 Jan 2024 14:18:53: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12C_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2 --format BAMPE -g 1.2e+8 -B --nomodel --shift -100 --extsize 200 --broad\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12C_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: #1 read fragment files... \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: #1 read treatment fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12A_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2 --format BAMPE -g 1.2e+8 -B --nomodel --shift -100 --extsize 200 --broad\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12A_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: #1 read fragment files... \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: #1 read treatment fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM2_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2 --format BAMPE -g 1.2e+8 -B --nomodel --shift -100 --extsize 200 --broad\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM2_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: #1 read fragment files... \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:53: #1 read treatment fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:54:  1000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:54:  1000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:54:  1000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:56:  2000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:56:  2000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:56:  2000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:58:  3000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:58:  3000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:58:  3000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:59:  4000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:59:  4000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:18:59:  4000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:01:  5000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:01:  5000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:01:  5000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:03:  6000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:03:  6000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:03:  6000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:04:  7000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:05:  7000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:05:  7000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:06:  8000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:06:  8000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:06:  8000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:08:  9000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:08:  9000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:08:  9000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:09:  10000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:10:  10000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:10:  10000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:11:  11000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:11:  11000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:11:  11000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:13:  12000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:13:  12000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:13:  12000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:15:  13000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:15:  13000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:15:  13000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:16:  14000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:16:  14000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:16:  14000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:18:  15000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:18:  15000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:18:  15000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:20:  16000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:20:  16000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:20:  16000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:21:  17000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:21:  17000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:21:  17000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:23:  18000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:23:  18000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:23:  18000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:25:  19000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:25:  19000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:25:  19000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:26:  20000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:26:  20000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:27:  20000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:28:  21000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:28:  21000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:28:  21000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:30:  22000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:30:  22000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:30:  22000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:31:  23000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:31:  23000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:32:  23000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:33:  24000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:33:  24000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:33:  24000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:35:  25000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:35:  25000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:35:  25000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:37:  26000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:37:  26000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:37:  26000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:38:  27000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:38:  27000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:39:  27000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:40:  28000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:40:  28000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:40:  28000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:42:  29000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:42:  29000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:42:  29000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:43:  30000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:43:  30000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:44:  30000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:45:  31000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:45:  31000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:45:  31000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:47:  32000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:47:  32000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:47:  32000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:48:  33000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:48:  33000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:49:  33000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:50:  34000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:50:  34000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:51:  34000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:52:  35000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:52:  35000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:52:  35000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:53:  36000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:53:  36000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:54:  36000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:55:  37000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:55:  37000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:56:  37000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:57:  38000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:57:  38000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:57:  38000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:58:  39000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:58:  39000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:19:59:  39000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:00:  40000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:00:  40000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:01: 40305828 fragments have been read. \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:01:  40000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:02:  41000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:02:  41000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:03:  42000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:04:  42000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:05:  43000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:06:  43000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:07:  44000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:07:  44000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:09:  45000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:09:  45000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:10:  46000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:11:  46000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:12:  47000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:13:  47000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:14:  48000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:14:  48000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:15:  49000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:16:  49000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:17:  50000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:18:  50000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:18: #1 mean fragment size is determined as 131.7 bp from treatment \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:18: #1 fragment size = 131.7 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:18: #1  total fragments in treatment: 40305828 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:18: #1 user defined the maximum fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:18: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:19:  51000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:19:  51000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:20:  52000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:21:  52000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:22:  53000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:23:  53000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:24:  54000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:25:  54000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:26:  55000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:26:  55000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:27:  56000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:28:  56000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:29:  57000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:30:  57000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:31:  58000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:31:  58000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:32:  59000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:33:  59000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:34:  60000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:35:  60000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:36:  61000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:36:  61000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:37:  62000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:38:  62000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:39:  63000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:40:  63000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:41:  64000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:42:  64000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:43:  65000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:43: 65204380 fragments have been read. \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:43:  65000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:45:  66000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:47:  67000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:48: 67925379 fragments have been read. \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #1  fragments after filtering in treatment: 36665136 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #1  Redundant rate of treatment: 0.09 \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #1 finished! \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #2 Build Peak Model... \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #2 Skipped... \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #3 Call peaks... \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Wed, 24 Jan 2024 14:20:51: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:12: #1 mean fragment size is determined as 149.9 bp from treatment \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:12: #1 fragment size = 149.9 \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:12: #1  total fragments in treatment: 65204380 \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:12: #1 user defined the maximum fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:12: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:19: #1 mean fragment size is determined as 163.3 bp from treatment \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:19: #1 fragment size = 163.3 \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:19: #1  total fragments in treatment: 67925379 \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:19: #1 user defined the maximum fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:19: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:43: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:43: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12C_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:43: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12C_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:21:43: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #1  fragments after filtering in treatment: 56179824 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #1  Redundant rate of treatment: 0.14 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #1 finished! \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #2 Build Peak Model... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #2 Skipped... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #3 Call peaks... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:04: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #1  fragments after filtering in treatment: 58249353 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #1  Redundant rate of treatment: 0.14 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #1 finished! \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #2 Build Peak Model... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #2 Skipped... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #3 Call peaks... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:13: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:50: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12C_clean_filtered_peaks.xls \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:50: #4 Write broad peak in broadPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12C_clean_filtered_peaks.broadPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:50: #4 Write broad peak in bed12/gappedPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12C_clean_filtered_peaks.gappedPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:50: Done! \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:50: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM1_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2 --format BAMPE -g 1.2e+8 -B --nomodel --shift -100 --extsize 200 --broad\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM1_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:50: #1 read fragment files... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:50: #1 read treatment fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:52:  1000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:54:  2000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:55:  3000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:57:  4000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:22:59:  5000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:00:  6000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:02:  7000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:04:  8000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:05:  9000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:07:  10000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:09:  11000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:10:  12000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:12:  13000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:13: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:13: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM2_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:13: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM2_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:13: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:14:  14000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:15:  15000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:17:  16000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:19:  17000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:20:  18000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:22:  19000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:24:  20000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:25:  21000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:27:  22000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:29:  23000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:30:  24000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:32:  25000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:34:  26000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:35:  27000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:37:  28000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:39:  29000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:39: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:39: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12A_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:39: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12A_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:39: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:40:  30000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:42:  31000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:44:  32000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:45:  33000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:47:  34000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:49:  35000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:50:  36000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:52:  37000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:54:  38000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:55:  39000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:57:  40000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:23:59:  41000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:00:  42000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:02:  43000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:04:  44000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:05:  45000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:07:  46000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:09:  47000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:10:  48000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:12:  49000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:14:  50000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:15:  51000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:17:  52000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:19:  53000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:21:  54000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:22:  55000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:23: 55403324 fragments have been read. \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:40: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM2_clean_filtered_peaks.xls \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:40: #4 Write broad peak in broadPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM2_clean_filtered_peaks.broadPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:40: #4 Write broad peak in bed12/gappedPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM2_clean_filtered_peaks.gappedPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:40: Done! \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:41: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM3_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2 --format BAMPE -g 1.2e+8 -B --nomodel --shift -100 --extsize 200 --broad\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM3_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:41: #1 read fragment files... \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:41: #1 read treatment fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:42:  1000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:44:  2000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:46:  3000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:47:  4000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:48: #1 mean fragment size is determined as 149.1 bp from treatment \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:48: #1 fragment size = 149.1 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:48: #1  total fragments in treatment: 55403324 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:48: #1 user defined the maximum fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:48: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:49:  5000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:51:  6000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:52:  7000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:54:  8000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:56:  9000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:57:  10000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:24:59:  11000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:01:  12000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:02:  13000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:04:  14000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:06:  15000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:08:  16000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:09:  17000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:11:  18000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:13:  19000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:14:  20000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:16:  21000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:18:  22000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:19:  23000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:20: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12A_clean_filtered_peaks.xls \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:20: #4 Write broad peak in broadPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12A_clean_filtered_peaks.broadPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:20: #4 Write broad peak in bed12/gappedPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12A_clean_filtered_peaks.gappedPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:20: Done! \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:21: \n",
      "# Command line: callpeak -t /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam -n /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12B_clean_filtered --outdir /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2 --format BAMPE -g 1.2e+8 -B --nomodel --shift -100 --extsize 200 --broad\n",
      "# ARGUMENTS LIST:\n",
      "# name = /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12B_clean_filtered\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+08\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
      "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is on\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:21: #1 read fragment files... \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:21: #1 read treatment fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:21:  24000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:22:  1000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:23:  25000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:24:  2000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:24:  26000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:26:  3000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:26:  27000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:27:  4000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:28:  28000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:29:  5000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:29:  29000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:31:  6000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:31:  30000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:32:  7000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:33:  31000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34:  8000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34:  32000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #1  fragments after filtering in treatment: 48316193 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #1  Redundant rate of treatment: 0.13 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #1 finished! \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #2 Build Peak Model... \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #2 Skipped... \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #3 Call peaks... \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:34: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:36:  9000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:36:  33000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:37:  10000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:38:  34000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:39:  11000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:39:  35000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:41:  12000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:41:  36000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:42:  13000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:43:  37000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:44:  14000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:44:  38000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:46:  15000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:46:  39000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:47:  16000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:48:  40000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:49:  17000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:49:  41000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:51:  18000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:51:  42000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:52:  19000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:53:  43000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:54:  20000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:54:  44000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:56:  21000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:56:  45000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:57:  22000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:58:  46000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:59:  23000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:25:59:  47000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:01:  24000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:01:  48000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:02:  25000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:03:  49000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:04:  26000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:04:  50000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:06:  27000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:06:  51000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:07:  28000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:08:  52000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:09:  29000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:09:  53000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:11:  30000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:11:  54000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:12:  31000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:13:  55000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:14:  32000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:14:  56000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:16:  33000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:16:  57000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:17:  34000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:18:  58000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:18: 58034780 fragments have been read. \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:19:  35000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:21:  36000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:22:  37000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:24:  38000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:26:  39000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:28:  40000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:29:  41000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:31:  42000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:33:  43000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:34:  44000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:35: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:35: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM1_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:35: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM1_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:35: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:36:  45000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:38:  46000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:39:  47000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:41:  48000000 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:42: 48679779 fragments have been read. \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:43: #1 mean fragment size is determined as 142.8 bp from treatment \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:43: #1 fragment size = 142.8 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:43: #1  total fragments in treatment: 58034780 \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:43: #1 user defined the maximum fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:26:43: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:04: #1 mean fragment size is determined as 137.7 bp from treatment \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:04: #1 fragment size = 137.7 \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:04: #1  total fragments in treatment: 48679779 \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:04: #1 user defined the maximum fragments... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:04: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #1  fragments after filtering in treatment: 50824887 \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #1  Redundant rate of treatment: 0.12 \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #1 finished! \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #2 Build Peak Model... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #2 Skipped... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #3 Call peaks... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:29: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #1  fragments after filtering in treatment: 43623919 \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #1  Redundant rate of treatment: 0.10 \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #1 finished! \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #2 Build Peak Model... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #2 Skipped... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #3 Call peaks... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:42: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:53: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM1_clean_filtered_peaks.xls \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:53: #4 Write broad peak in broadPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM1_clean_filtered_peaks.broadPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:53: #4 Write broad peak in bed12/gappedPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM1_clean_filtered_peaks.gappedPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:27:54: Done! \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:31: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:31: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM3_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:31: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM3_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:31: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:45: #3 In the peak calling step, the following will be performed simultaneously: \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:45: #3   Write bedGraph files for treatment pileup (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12B_clean_filtered_treat_pileup.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:45: #3   Write bedGraph files for control lambda (after scaling if necessary)... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12B_clean_filtered_control_lambda.bdg \n",
      "INFO  @ Wed, 24 Jan 2024 14:28:45: #3 Call peaks for each chromosome... \n",
      "INFO  @ Wed, 24 Jan 2024 14:29:50: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM3_clean_filtered_peaks.xls \n",
      "INFO  @ Wed, 24 Jan 2024 14:29:50: #4 Write broad peak in broadPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM3_clean_filtered_peaks.broadPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:29:50: #4 Write broad peak in bed12/gappedPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/DM3_clean_filtered_peaks.gappedPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:29:50: Done! \n",
      "INFO  @ Wed, 24 Jan 2024 14:30:04: #4 Write output xls file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12B_clean_filtered_peaks.xls \n",
      "INFO  @ Wed, 24 Jan 2024 14:30:04: #4 Write broad peak in broadPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12B_clean_filtered_peaks.broadPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:30:04: #4 Write broad peak in bed12/gappedPeak format file... /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/12_macs2peaks_broad2/C12B_clean_filtered_peaks.gappedPeak \n",
      "INFO  @ Wed, 24 Jan 2024 14:30:05: Done! \n"
     ]
    }
   ],
   "source": [
    "# Define the function to call peaks\n",
    "def call_peaks_broad(bam_file):\n",
    "    output_prefix = os.path.join(broad_peaks_dir, os.path.basename(bam_file).replace('.bam', ''))\n",
    "\n",
    "    # Construct the MACS2 command\n",
    "    command = [\n",
    "        'macs2',\n",
    "        'callpeak',\n",
    "        '-t', bam_file,\n",
    "        '-n', output_prefix,\n",
    "        '--outdir', broad_peaks_dir,\n",
    "        '--format', 'BAMPE',\n",
    "        '-g', '1.2e+8',  # genome size 120M for Ath genome\n",
    "        # '-B',  # This option tells MACS2 to save the pileup signal in a bedGraph file\n",
    "        '--nomodel', \n",
    "        '--shift',  '-100',\n",
    "        '--extsize', '200', \n",
    "        '--broad',\n",
    "    ]\n",
    "    \n",
    "    # Run the MACS2 command\n",
    "    subprocess.run(command, check=True)\n",
    "\n",
    "# Define the number of parallel processes\n",
    "num_processes = 3 # Change this number as desired\n",
    "\n",
    "# Use a Pool to process BAM files in parallel\n",
    "with Pool(num_processes) as p:\n",
    "    p.map(call_peaks_broad, bam_files)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca696d12-9f3c-464d-8463-0ad8f0913563",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:work2] *",
   "language": "python",
   "name": "conda-env-work2-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
