{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3f8c14ff-0b23-42ae-bbf7-b227ef42824c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import glob\n",
    "import os\n",
    "from os.path import exists\n",
    "import re\n",
    "#from multiprocessing.dummy import Pool as ThreadPool\n",
    "from multiprocessing import Pool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4f0046d8-22f9-4304-b9a7-d81a8be249f0",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: plotProfile -m matrix.gz\n",
      "help: plotProfile -h / plotProfile --help\n",
      "plotProfile: error: the following arguments are required: --outFileName/-out/-o\n"
     ]
    }
   ],
   "source": [
    "! plotProfile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b6d51151-757a-4493-83b7-b78a254618ec",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "output_dir = '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "52f605de-0112-4a11-aea6-d62e5a3c1430",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bams_output = os.path.join(output_dir, '4_intersect_bam')\n",
    "bam_files = glob.glob(os.path.join(bams_output, '*.bam'))\n",
    "bam_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2400e9db-d152-4371-b3be-ac8311e1a5c0",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: RPKM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 1\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: RPKM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 1\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: RPKM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 1\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: RPKM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 1\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: RPKM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 1\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: RPKM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 1\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    }
   ],
   "source": [
    "def bam_to_bigWig(bam_file, output_dir):\n",
    "    bigwig_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        'bamCoverage', \n",
    "        '-b', bam_file, \n",
    "        '-o', bigwig_file,\n",
    "        '--normalizeUsing', 'RPKM',  # Normalize the coverage (you can use other methods as well)\n",
    "        '--binSize', '10'  # The size of the bins, in bases\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "    return bigwig_file\n",
    "\n",
    "# Wrapper function to pass multiple arguments to the mapped function\n",
    "def wrapper_bam_to_bigWig(args):\n",
    "    return bam_to_bigWig(*args)\n",
    "\n",
    "bw_output = os.path.join(output_dir, '6_bw')\n",
    "if not os.path.exists(bw_output):\n",
    "    os.makedirs(bw_output)\n",
    "\n",
    "\n",
    "# Use a Pool to process BAM files in parallel\n",
    "with Pool(4) as p:  # Here 4 represents the number of processes to run concurrently\n",
    "    p.map(wrapper_bam_to_bigWig, [(bam_file, bw_output) for bam_file in bam_files])\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "f1b75697-d4e5-4d25-a315-0f6540b98eed",
   "metadata": {},
   "source": [
    "global_exception = None\n",
    "def bam_to_bigWig(bam_file, output_dir):\n",
    "    bigwig_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        'bamCoverage',\n",
    "        '-b', bam_file,\n",
    "        '-o', bigwig_file,\n",
    "        '--normalizeUsing', 'RPKM',  # Normalize the coverage (you can use other methods as well)\n",
    "        '--binSize', '10'  # The size of the bins, in bases\n",
    "    ]\n",
    "    try:\n",
    "        subprocess.run(cmd, check=True)\n",
    "        return bigwig_file\n",
    "    except subprocess.CalledProcessError as e:\n",
    "        global_exception = e\n",
    "        print(f\"Error processing {bam_file}: {e}\")\n",
    "        return None\n",
    "\n",
    "# Wrapper function to pass multiple arguments to the mapped function\n",
    "def wrapper_bam_to_bigWig(args):\n",
    "    return bam_to_bigWig(*args)\n",
    "\n",
    "bw_output = os.path.join(output_dir, '6_bw')\n",
    "if not os.path.exists(bw_output):\n",
    "    os.makedirs(bw_output)\n",
    "\n",
    "# Sample list of bam files, replace with your list or a glob to get them dynamically\n",
    "bam_files = glob.glob(os.path.join(output_dir, '3_clean_bam', '*.bam'))\n",
    "\n",
    "# Use a Pool to process BAM files in parallel\n",
    "with Pool(2) as p:  # Here 2 represents the number of processes to run concurrently\n",
    "    p.map(wrapper_bam_to_bigWig, [(bam_file, bw_output) for bam_file in bam_files])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e6803cd-504b-430f-bb7e-cbddf4bfdb4d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "eafe3e30-a3dc-400f-814a-0dcf8527d5ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "peaks_output = os.path.join(output_dir, '5_macs2peaks_simple')\n",
    "bed_files = glob.glob(os.path.join(peaks_output, '*.bed'))\n",
    "bw_files = glob.glob(os.path.join(output_dir, '6_bw', '*.bw'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5c8be325-c035-4c51-9765-2a5c0e5c07d9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/6_bw/C12B_clean_filtered.bw',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/6_bw/C12C_clean_filtered.bw',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/6_bw/DM2_clean_filtered.bw',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/6_bw/DM1_clean_filtered.bw',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/6_bw/C12A_clean_filtered.bw',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/6_bw/DM3_clean_filtered.bw']"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bw_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "346eb022-4ccc-4981-a7dd-9bf52d6f9aa9",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "    sys.exit(main())\n",
      "    sys.exit(main())\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "    sys.exit(main())\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "    res = list(map(func, TASKS))\n",
      "    res = list(map(func, TASKS))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "    res = list(map(func, TASKS))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "    res = list(map(func, TASKS))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 729, in coverage_from_big_wig\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 729, in coverage_from_big_wig\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 719, in coverage_from_big_wig\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 729, in coverage_from_big_wig\n",
      "    return heatmapper.coverage_from_array(values_array, zones,\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 627, in coverage_from_array\n",
      "    return heatmapper.coverage_from_array(values_array, zones,\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 627, in coverage_from_array\n",
      "    return heatmapper.coverage_from_array(values_array, zones,\n",
      "    values_array[startIdx:endIdx] = bigwig.values(chrom, start, end)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 627, in coverage_from_array\n",
      "KeyboardInterrupt\n",
      "    counts_list.append(heatmapper.my_average(valuesArray[idxStart:idxEnd], avgType))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 739, in my_average\n",
      "    counts_list.append(heatmapper.my_average(valuesArray[idxStart:idxEnd], avgType))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 738, in my_average\n",
      "    counts_list.append(heatmapper.my_average(valuesArray[idxStart:idxEnd], avgType))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 739, in my_average\n",
      "    avg = np.ma.__getattribute__(avgType)(valuesArray)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 6882, in __call__\n",
      "    valuesArray = np.ma.masked_invalid(valuesArray)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 2360, in masked_invalid\n",
      "    avg = np.ma.__getattribute__(avgType)(valuesArray)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 6882, in __call__\n",
      "    res = masked_where(~(np.isfinite(a)), a, copy=copy)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 1940, in masked_where\n",
      "    result = a.view(cls)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 2984, in __array_finalize__\n",
      "    return method(marr, *args, **params)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 5343, in mean\n",
      "    return method(marr, *args, **params)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 5344, in mean\n",
      "    self._update_from(obj)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 2967, in _update_from\n",
      "    _dict = dict(_fill_value=getattr(obj, '_fill_value', None),\n",
      "KeyboardInterrupt\n",
      "    dsum = self.sum(axis=axis, dtype=dtype, **kwargs)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 5175, in sum\n",
      "    cnt = self.count(axis=axis, **kwargs)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 4622, in count\n",
      "    return (~m).sum(axis=axis, dtype=np.intp, **kwargs)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/core/_methods.py\", line 49, in _sum\n",
      "    result = self.filled(0).sum(axis, dtype=dtype, **kwargs)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 3846, in filled\n",
      "    return umr_sum(a, axis, dtype, out, keepdims, initial, where)\n",
      "KeyboardInterrupt\n",
      "    elif not m.any():\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/core/_methods.py\", line 58, in _any\n",
      "    return umr_any(a, axis, dtype, out, keepdims)\n",
      "KeyboardInterrupt\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[15], line 56\u001b[0m\n\u001b[1;32m     52\u001b[0m     os\u001b[38;5;241m.\u001b[39mmakedirs(bedtools_output)\n\u001b[1;32m     55\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m Pool(\u001b[38;5;241m4\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m p:  \u001b[38;5;66;03m# Using 4 concurrent processes\u001b[39;00m\n\u001b[0;32m---> 56\u001b[0m     \u001b[43mp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmap\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwrapper_run_deepTools\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m[\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbed_file\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbw_file\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbedtools_output\u001b[49m\u001b[43m)\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mbed_file\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbw_file\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;28;43mzip\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mbed_files\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mbw_files\u001b[49m\u001b[43m)\u001b[49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/multiprocessing/pool.py:367\u001b[0m, in \u001b[0;36mPool.map\u001b[0;34m(self, func, iterable, chunksize)\u001b[0m\n\u001b[1;32m    362\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mmap\u001b[39m(\u001b[38;5;28mself\u001b[39m, func, iterable, chunksize\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    363\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m'''\u001b[39;00m\n\u001b[1;32m    364\u001b[0m \u001b[38;5;124;03m    Apply `func` to each element in `iterable`, collecting the results\u001b[39;00m\n\u001b[1;32m    365\u001b[0m \u001b[38;5;124;03m    in a list that is returned.\u001b[39;00m\n\u001b[1;32m    366\u001b[0m \u001b[38;5;124;03m    '''\u001b[39;00m\n\u001b[0;32m--> 367\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_map_async\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfunc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43miterable\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmapstar\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mchunksize\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/multiprocessing/pool.py:768\u001b[0m, in \u001b[0;36mApplyResult.get\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    767\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(\u001b[38;5;28mself\u001b[39m, timeout\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m--> 768\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    769\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mready():\n\u001b[1;32m    770\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTimeoutError\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/multiprocessing/pool.py:765\u001b[0m, in \u001b[0;36mApplyResult.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    764\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwait\u001b[39m(\u001b[38;5;28mself\u001b[39m, timeout\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m--> 765\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_event\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/threading.py:607\u001b[0m, in \u001b[0;36mEvent.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    605\u001b[0m signaled \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_flag\n\u001b[1;32m    606\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m signaled:\n\u001b[0;32m--> 607\u001b[0m     signaled \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cond\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    608\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m signaled\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/threading.py:320\u001b[0m, in \u001b[0;36mCondition.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    318\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:    \u001b[38;5;66;03m# restore state no matter what (e.g., KeyboardInterrupt)\u001b[39;00m\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 320\u001b[0m         \u001b[43mwaiter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43macquire\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    321\u001b[0m         gotit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m    322\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "def run_deepTools(bed_file, bigwig_file, output_dir):\n",
    "    sorted_bed = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_sorted.bed'))\n",
    "    matrix_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_matrix.gz'))\n",
    "    heatmap_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_heatmap.pdf'))\n",
    "    profile_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_profile.pdf'))\n",
    "\n",
    "    # Sort the BED file\n",
    "    cmd = ['sort', '-k1,1', '-k2,2n', bed_file, '-o', sorted_bed]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Compute the matrix\n",
    "    cmd = [\n",
    "        'computeMatrix', 'reference-point',\n",
    "        '--referencePoint', 'center',\n",
    "        '-R', sorted_bed,\n",
    "        '-S', bigwig_file,\n",
    "        '-a', '1000',\n",
    "        '-b', '1000',\n",
    "        '-out', matrix_file\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Plot the heatmap\n",
    "    cmd = [\n",
    "        'plotHeatmap', '-m', matrix_file,\n",
    "        '-out', heatmap_file,\n",
    "        '--colorMap', 'RdBu_r',\n",
    "        # '--zMin', '-20',\n",
    "        # '--zMax', '20',\n",
    "        '--sortUsing', 'mean',\n",
    "        '--missingDataColor', '0.5',\n",
    "        '--heatmapHeight', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Plot the profile\n",
    "    cmd = [\n",
    "        'plotProfile', '-m', matrix_file,\n",
    "        '-out', profile_file,\n",
    "        '--numPlotsPerRow', '1',\n",
    "        '--plotHeight', '5',\n",
    "        '--plotWidth', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "def wrapper_run_deepTools(args):\n",
    "    return run_deepTools(*args)\n",
    "\n",
    "\n",
    "bedtools_output = os.path.join(output_dir, '7_sorted_bed_and_matrix_auto2')\n",
    "if not os.path.exists(bedtools_output):\n",
    "    os.makedirs(bedtools_output)\n",
    "\n",
    "\n",
    "with Pool(4) as p:  # Using 4 concurrent processes\n",
    "    p.map(wrapper_run_deepTools, [(bed_file, bw_file, bedtools_output) for bed_file, bw_file in zip(bed_files, bw_files)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "44b2dc65-69b4-47da-8c8f-9d916bcdfc9c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "def run_deepTools(bed_file, bigwig_file, output_dir):\n",
    "    sorted_bed = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_sorted.bed'))\n",
    "    matrix_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_matrix.gz'))\n",
    "    heatmap_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_heatmap.pdf'))\n",
    "    profile_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_profile.pdf'))\n",
    "\n",
    "    # Sort the BED file\n",
    "    cmd = ['sort', '-k1,1', '-k2,2n', bed_file, '-o', sorted_bed]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Compute the matrix\n",
    "    cmd = [\n",
    "        'computeMatrix', 'reference-point',\n",
    "        '--referencePoint', 'center',\n",
    "        '-R', sorted_bed,\n",
    "        '-S', bigwig_file,\n",
    "        '-a', '1000',\n",
    "        '-b', '1000',\n",
    "        '-out', matrix_file\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Plot the heatmap\n",
    "    cmd = [\n",
    "        'plotHeatmap', '-m', matrix_file,\n",
    "        '-out', heatmap_file,\n",
    "        '--colorMap', 'RdBu_r',\n",
    "        '--zMin', '0',\n",
    "        '--zMax', '380',\n",
    "        '--sortUsing', 'mean',\n",
    "        '--missingDataColor', '0.5',\n",
    "        '--heatmapHeight', '8'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Plot the profile\n",
    "    cmd = [\n",
    "        'plotProfile', '-m', matrix_file,\n",
    "        '-out', profile_file,\n",
    "        '--numPlotsPerRow', '1',\n",
    "        '--plotHeight', '5',\n",
    "        '--plotWidth', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "def wrapper_run_deepTools(args):\n",
    "    return run_deepTools(*args)\n",
    "\n",
    "\n",
    "bedtools_output = os.path.join(output_dir, '7_sorted_bed_and_matrix_zMax380')\n",
    "if not os.path.exists(bedtools_output):\n",
    "    os.makedirs(bedtools_output)\n",
    "\n",
    "\n",
    "with Pool(4) as p:  # Using 4 concurrent processes\n",
    "    p.map(wrapper_run_deepTools, [(bed_file, bw_file, bedtools_output) for bed_file, bw_file in zip(bed_files, bw_files)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "c5a989e9-7035-4e38-9369-6fe4f76b541e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/hu/miniconda3/envs/work2/bin/computeMatrix\", line 10, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "    sys.exit(main())\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "    sys.exit(main())\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "    sys.exit(main())\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/computeMatrix.py\", line 428, in main\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 252, in computeMatrix\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "    res, labels = mapReduce.mapReduce([score_file_list, parameters],\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/mapReduce.py\", line 146, in mapReduce\n",
      "    res = list(map(func, TASKS))\n",
      "    res = list(map(func, TASKS))\n",
      "    res = list(map(func, TASKS))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "    res = list(map(func, TASKS))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 174, in compute_sub_matrix_wrapper\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "    return heatmapper.compute_sub_matrix_worker(*args)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 533, in compute_sub_matrix_worker\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 729, in coverage_from_big_wig\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 729, in coverage_from_big_wig\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 729, in coverage_from_big_wig\n",
      "    cov = heatmapper.coverage_from_big_wig(\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 729, in coverage_from_big_wig\n",
      "    return heatmapper.coverage_from_array(values_array, zones,\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 627, in coverage_from_array\n",
      "    return heatmapper.coverage_from_array(values_array, zones,\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 627, in coverage_from_array\n",
      "    return heatmapper.coverage_from_array(values_array, zones,\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 627, in coverage_from_array\n",
      "    return heatmapper.coverage_from_array(values_array, zones,\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 627, in coverage_from_array\n",
      "    counts_list.append(heatmapper.my_average(valuesArray[idxStart:idxEnd], avgType))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 739, in my_average\n",
      "    counts_list.append(heatmapper.my_average(valuesArray[idxStart:idxEnd], avgType))\n",
      "    counts_list.append(heatmapper.my_average(valuesArray[idxStart:idxEnd], avgType))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 738, in my_average\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 738, in my_average\n",
      "    counts_list.append(heatmapper.my_average(valuesArray[idxStart:idxEnd], avgType))\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/deeptools/heatmapper.py\", line 738, in my_average\n",
      "    avg = np.ma.__getattribute__(avgType)(valuesArray)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 6882, in __call__\n",
      "    valuesArray = np.ma.masked_invalid(valuesArray)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 2360, in masked_invalid\n",
      "    valuesArray = np.ma.masked_invalid(valuesArray)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 2360, in masked_invalid\n",
      "    valuesArray = np.ma.masked_invalid(valuesArray)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 2360, in masked_invalid\n",
      "    res = masked_where(~(np.isfinite(a)), a, copy=copy)\n",
      "    res = masked_where(~(np.isfinite(a)), a, copy=copy)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 1932, in masked_where\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 1940, in masked_where\n",
      "    res = masked_where(~(np.isfinite(a)), a, copy=copy)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 1940, in masked_where\n",
      "    if cshape and cshape != ashape:\n",
      "    result = a.view(cls)\n",
      "KeyboardInterrupt\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 2978, in __array_finalize__\n",
      "    result = a.view(cls)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 3049, in __array_finalize__\n",
      "    return method(marr, *args, **params)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 5343, in mean\n",
      "    def __array_finalize__(self, obj):\n",
      "KeyboardInterrupt\n",
      "    _mask = _mask.view()\n",
      "KeyboardInterrupt\n",
      "    dsum = self.sum(axis=axis, dtype=dtype, **kwargs)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 5172, in sum\n",
      "    newmask = _check_mask_axis(_mask, axis, **kwargs)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/ma/core.py\", line 1820, in _check_mask_axis\n",
      "    return mask.all(axis=axis, **kwargs)\n",
      "  File \"/home/hu/miniconda3/envs/work2/lib/python3.10/site-packages/numpy/core/_methods.py\", line 64, in _all\n",
      "    return umr_all(a, axis, dtype, out, keepdims)\n",
      "KeyboardInterrupt\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[17], line 63\u001b[0m\n\u001b[1;32m     61\u001b[0m tasks \u001b[38;5;241m=\u001b[39m [(bed_file, bw_file, bedtools_output, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTSS\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m bed_file, bw_file \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(bed_files, bw_files)]  \u001b[38;5;66;03m# for TSS\u001b[39;00m\n\u001b[1;32m     62\u001b[0m tasks \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m [(bed_file, bw_file, bedtools_output, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTTS\u001b[39m\u001b[38;5;124m'\u001b[39m) \u001b[38;5;28;01mfor\u001b[39;00m bed_file, bw_file \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(bed_files, bw_files)]  \u001b[38;5;66;03m# for TTS\u001b[39;00m\n\u001b[0;32m---> 63\u001b[0m \u001b[43mp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmap\u001b[49m\u001b[43m(\u001b[49m\u001b[43mwrapper_run_deepTools\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtasks\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/multiprocessing/pool.py:367\u001b[0m, in \u001b[0;36mPool.map\u001b[0;34m(self, func, iterable, chunksize)\u001b[0m\n\u001b[1;32m    362\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mmap\u001b[39m(\u001b[38;5;28mself\u001b[39m, func, iterable, chunksize\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    363\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m'''\u001b[39;00m\n\u001b[1;32m    364\u001b[0m \u001b[38;5;124;03m    Apply `func` to each element in `iterable`, collecting the results\u001b[39;00m\n\u001b[1;32m    365\u001b[0m \u001b[38;5;124;03m    in a list that is returned.\u001b[39;00m\n\u001b[1;32m    366\u001b[0m \u001b[38;5;124;03m    '''\u001b[39;00m\n\u001b[0;32m--> 367\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_map_async\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfunc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43miterable\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmapstar\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mchunksize\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/multiprocessing/pool.py:768\u001b[0m, in \u001b[0;36mApplyResult.get\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    767\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mget\u001b[39m(\u001b[38;5;28mself\u001b[39m, timeout\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m--> 768\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    769\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mready():\n\u001b[1;32m    770\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTimeoutError\u001b[39;00m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/multiprocessing/pool.py:765\u001b[0m, in \u001b[0;36mApplyResult.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    764\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwait\u001b[39m(\u001b[38;5;28mself\u001b[39m, timeout\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m--> 765\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_event\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/threading.py:607\u001b[0m, in \u001b[0;36mEvent.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    605\u001b[0m signaled \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_flag\n\u001b[1;32m    606\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m signaled:\n\u001b[0;32m--> 607\u001b[0m     signaled \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cond\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    608\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m signaled\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/threading.py:320\u001b[0m, in \u001b[0;36mCondition.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    318\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:    \u001b[38;5;66;03m# restore state no matter what (e.g., KeyboardInterrupt)\u001b[39;00m\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 320\u001b[0m         \u001b[43mwaiter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43macquire\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    321\u001b[0m         gotit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m    322\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from multiprocessing import Pool\n",
    "\n",
    "def run_deepTools(bed_file, bigwig_file, output_dir, ref_point='TSS'):\n",
    "    sorted_bed = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', '_sorted.bed'))\n",
    "    matrix_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', f'_{ref_point}_matrix.gz'))\n",
    "    heatmap_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', f'_{ref_point}_heatmap.pdf'))\n",
    "    profile_file = os.path.join(output_dir, os.path.basename(bed_file).replace('.bed', f'_{ref_point}_profile.pdf'))\n",
    "\n",
    "    # Sort the BED file\n",
    "    cmd = ['sort', '-k1,1', '-k2,2n', bed_file, '-o', sorted_bed]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Compute the matrix\n",
    "    ref_point_option = 'TSS' if ref_point == 'TSS' else 'TES'\n",
    "    cmd = [\n",
    "        'computeMatrix', 'reference-point',\n",
    "        '--referencePoint', ref_point_option,\n",
    "        '-R', sorted_bed,\n",
    "        '-S', bigwig_file,\n",
    "        '-a', '1000',\n",
    "        '-b', '1000',\n",
    "        '-out', matrix_file\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Plot the heatmap\n",
    "    cmd = [\n",
    "        'plotHeatmap', '-m', matrix_file,\n",
    "        '-out', heatmap_file,\n",
    "        '--colorMap', 'RdBu_r',\n",
    "        '--zMin', '0',\n",
    "        '--zMax', '380',\n",
    "        '--sortUsing', 'mean',\n",
    "        '--missingDataColor', '0.5',\n",
    "        '--heatmapHeight', '8'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "    # Plot the profile\n",
    "    cmd = [\n",
    "        'plotProfile', '-m', matrix_file,\n",
    "        '-out', profile_file,\n",
    "        '--numPlotsPerRow', '1',\n",
    "        '--plotHeight', '5',\n",
    "        '--plotWidth', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "def wrapper_run_deepTools(args):\n",
    "    return run_deepTools(*args)\n",
    "\n",
    "output_dir = '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir'\n",
    "bedtools_output = os.path.join(output_dir, '7_sorted_bed_and_matrix_zMax380_TSS_TTS')\n",
    "if not os.path.exists(bedtools_output):\n",
    "    os.makedirs(bedtools_output)\n",
    "\n",
    "# Example usage, specifying TSS or TTS\n",
    "with Pool(4) as p:  # Using 4 concurrent processes\n",
    "    tasks = [(bed_file, bw_file, bedtools_output, 'TSS') for bed_file, bw_file in zip(bed_files, bw_files)]  # for TSS\n",
    "    tasks += [(bed_file, bw_file, bedtools_output, 'TTS') for bed_file, bw_file in zip(bed_files, bw_files)]  # for TTS\n",
    "    p.map(wrapper_run_deepTools, tasks)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "383b81b4-b6bc-45ae-ba01-fa362f6d2858",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Get the list of all BAM files in the directory\n",
    "bam_files = [os.path.join(bams_output, f) for f in os.listdir(bams_output) if f.endswith('.bam')]\n",
    "bam_files"
   ]
  },
  {
   "cell_type": "raw",
   "id": "4b997896-ee2e-4047-b1c2-ec2ec1a8fea9",
   "metadata": {
    "tags": []
   },
   "source": [
    "bed_files = glob.glob(os.path.join(bam2bed_output, '*.bed'))\n",
    "bed_files"
   ]
  },
  {
   "cell_type": "raw",
   "id": "1ed76b4f-8d00-469a-84a1-38ab4a500a37",
   "metadata": {
    "tags": []
   },
   "source": [
    "\n",
    "# Separate BED files into two categories based on filename prefix: OK and C (Col-0)\n",
    "DM_files = [f for f in bed_files if os.path.basename(f).startswith(\"DM\")]\n",
    "c_files = [f for f in bed_files if os.path.basename(f).startswith(\"C12\")]\n",
    "print(DM_files)\n",
    "print(c_files)\n",
    "merged_dir = os.path.join(output_dir, '9_merged_bed')\n",
    "if not os.path.exists(merged_dir):\n",
    "    os.makedirs(merged_dir)\n",
    "\n",
    "# Function to merge BED files\n",
    "def merge_bed_files(files, output_file):\n",
    "    # First, concatenate the files\n",
    "    with open(output_file, 'w') as out_file:\n",
    "        for file in files:\n",
    "            with open(file, 'r') as f:\n",
    "                for line in f:\n",
    "                    out_file.write(line)\n",
    "    # Then use bedtools to merge overlapping regions\n",
    "    merged_output = output_file.replace(\".bed\", \"_merged.bed\")\n",
    "    cmd = ['bedtools', 'merge', '-i', output_file, '-o', merged_output]\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "# Merge OK and C (Col-0) files\n",
    "merge_bed_files(DM_files, os.path.join(merged_dir, 'DM1_DM2d_merged.bed'))\n",
    "merge_bed_files(c_files, os.path.join(merged_dir, 'Col-0_12DAS.bed'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "16e63e0b-4f37-47cf-bfa6-8ed1721a56c3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_files = glob.glob(os.path.join(bams_output, '*.bam'))\n",
    "bam_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "827ac693-7bc7-4c7c-8123-b04de1b876bc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "merged_bam_dir = os.path.join(output_dir, '9_merged_bam')\n",
    "if not os.path.exists(merged_bam_dir):\n",
    "    os.makedirs(merged_bam_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "211d45c6-d787-462f-8dcf-0e35585af8d7",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM2_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM1_clean_filtered.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/DM3_clean_filtered.bam']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Separate BAM files into two categories based on filename prefix: C12 and DM\n",
    "c12_files = [f for f in bam_files if os.path.basename(f).startswith(\"C12\")]\n",
    "dm_files = [f for f in bam_files if os.path.basename(f).startswith(\"DM\")]\n",
    "c12_files\n",
    "dm_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a1949075-2910-44a9-9d8c-0b2001088c3b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_merge] File '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam' exists. Please apply '-f' to overwrite. Abort.\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command '['samtools', 'merge', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[19], line 7\u001b[0m\n\u001b[1;32m      4\u001b[0m     subprocess\u001b[38;5;241m.\u001b[39mrun(cmd, check\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m      6\u001b[0m \u001b[38;5;66;03m# Merge C12 and DM files\u001b[39;00m\n\u001b[0;32m----> 7\u001b[0m \u001b[43mmerge_bam_files\u001b[49m\u001b[43m(\u001b[49m\u001b[43mc12_files\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mos\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mpath\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mjoin\u001b[49m\u001b[43m(\u001b[49m\u001b[43mmerged_bam_dir\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mC12_merged.bam\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      8\u001b[0m merge_bam_files(dm_files, os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(merged_bam_dir, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mDM_merged.bam\u001b[39m\u001b[38;5;124m'\u001b[39m))\n",
      "Cell \u001b[0;32mIn[19], line 4\u001b[0m, in \u001b[0;36mmerge_bam_files\u001b[0;34m(files, output_file)\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mmerge_bam_files\u001b[39m(files, output_file):\n\u001b[1;32m      3\u001b[0m     cmd \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m'\u001b[39m\u001b[38;5;124msamtools\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmerge\u001b[39m\u001b[38;5;124m'\u001b[39m,  output_file] \u001b[38;5;241m+\u001b[39m files\n\u001b[0;32m----> 4\u001b[0m     \u001b[43msubprocess\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mrun\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcmd\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcheck\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/work2/lib/python3.10/subprocess.py:526\u001b[0m, in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    524\u001b[0m     retcode \u001b[38;5;241m=\u001b[39m process\u001b[38;5;241m.\u001b[39mpoll()\n\u001b[1;32m    525\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m check \u001b[38;5;129;01mand\u001b[39;00m retcode:\n\u001b[0;32m--> 526\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m CalledProcessError(retcode, process\u001b[38;5;241m.\u001b[39margs,\n\u001b[1;32m    527\u001b[0m                                  output\u001b[38;5;241m=\u001b[39mstdout, stderr\u001b[38;5;241m=\u001b[39mstderr)\n\u001b[1;32m    528\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m CompletedProcess(process\u001b[38;5;241m.\u001b[39margs, retcode, stdout, stderr)\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command '['samtools', 'merge', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12C_clean_filtered.bam', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12A_clean_filtered.bam', '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/4_intersect_bam/C12B_clean_filtered.bam']' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "# Function to merge BAM files using samtools\n",
    "def merge_bam_files(files, output_file):\n",
    "    cmd = ['samtools', 'merge',  output_file] + files\n",
    "    subprocess.run(cmd, check=True)\n",
    "\n",
    "# Merge C12 and DM files\n",
    "merge_bam_files(c12_files, os.path.join(merged_bam_dir, 'C12_merged.bam'))\n",
    "merge_bam_files(dm_files, os.path.join(merged_bam_dir, 'DM_merged.bam'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ab733f1f-fff7-448e-b5b2-b70215be5c90",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "220bce17-19dd-475d-ad80-9206bee00f35",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "merged_bams = glob.glob(os.path.join(merged_bam_dir, '*.bam'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "95c94ad7-e267-42d7-8d75-9e20436c85fb",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "\n",
    "def bamtobed(bam_file, output_dir):\n",
    "    \"\"\"\n",
    "    Converts a BAM file to BED format.\n",
    "    \"\"\"\n",
    "    # Set the output file path\n",
    "    bed_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bed'))\n",
    "    \n",
    "    # Convert BAM to BED\n",
    "    cmd = ['bedtools', 'bamtobed', '-i', bam_file]\n",
    "    with open(bed_file, 'w') as outfile:\n",
    "        subprocess.run(cmd, stdout=outfile, check=True)\n",
    "    \n",
    "    return bed_file\n",
    "\n",
    "\n",
    "def bamtobed_wrapper(args):\n",
    "    return bamtobed(*args)\n",
    "\n",
    "\n",
    "bam2bed_output = os.path.join(output_dir, '10_bam2bed')\n",
    "if not os.path.exists(bam2bed_output):\n",
    "    os.makedirs(bam2bed_output)\n",
    "\n",
    "with Pool(2) as p:  # Using 4 concurrent processes\n",
    "    p.map(bamtobed_wrapper, [(bam_file, bam2bed_output) for bam_file in merged_bams])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f3061fc3-d5a4-474c-831a-0772ae9c7320",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'merged_bam_dir' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[16], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m merged_bams \u001b[38;5;241m=\u001b[39m glob\u001b[38;5;241m.\u001b[39mglob(os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(\u001b[43mmerged_bam_dir\u001b[49m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m*.bam\u001b[39m\u001b[38;5;124m'\u001b[39m))\n\u001b[1;32m      2\u001b[0m merged_bams\n",
      "\u001b[0;31mNameError\u001b[0m: name 'merged_bam_dir' is not defined"
     ]
    }
   ],
   "source": [
    "merged_bams = glob.glob(os.path.join(merged_bam_dir, '*.bam'))\n",
    "merged_bams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "5fb1ee40-5593-4033-814a-37e79df14af7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def index_bam(bam_file):\n",
    "    command = [\n",
    "        'sambamba',\n",
    "        'index',\n",
    "        '-t', '20',  # Number of threads\n",
    "        bam_file\n",
    "    ]\n",
    "    \n",
    "    # Execute the command\n",
    "    result = subprocess.run(command, capture_output=True, text=True)\n",
    "\n",
    "    # If there's an error, print the error message\n",
    "    if result.returncode != 0:\n",
    "        print(f\"Error indexing {sample}:\\n{result.stderr}\")\n",
    "        return  # Exit the function or handle the error as needed\n",
    "\n",
    "# Index BAM files for all samples in parallel\n",
    "with Pool(2) as p:\n",
    "    p.map(index_bam, merged_bams)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38a2f131-cc61-41b7-bd23-27243514a04b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "5089813e-f4cf-4b57-893f-72f51a9f35cb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "bam_files = glob.glob(os.path.join(merged_bam_dir, '*.bam'))\n",
    "bam_files.extend( glob.glob(os.path.join('/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam', '*.bam')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f9a0d51a-7588-4ce5-95f5-05dcd27f0cb0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam']"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_files"
   ]
  },
  {
   "cell_type": "raw",
   "id": "e8a44bd2-01a5-4fdb-a312-a5d68f3acd63",
   "metadata": {
    "tags": []
   },
   "source": [
    "glob.glob(os.path.join('/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam', '*.bam'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "57618351-4729-4e0a-a6ee-6f46178a403e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir'"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "00315dff-8b7d-4818-9461-9cc8cc41187b",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "normalization: 1x (effective genome size 115000000)\n",
      "normalization: 1x (effective genome size 115000000)\n",
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 15\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 15\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 15\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 15\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generated normalized bigWig for /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam at /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/11_normalized_bigwigs/DM_merged.bw\n",
      "Generated normalized bigWig for /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam at /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/11_normalized_bigwigs/C12_merged.bw\n",
      "Generated normalized bigWig for /media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam at /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/11_normalized_bigwigs/C14_merged.bw\n",
      "Generated normalized bigWig for /media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam at /media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/11_normalized_bigwigs/OK_merged.bw\n"
     ]
    }
   ],
   "source": [
    "\n",
    "def generate_normalized_bigwig(bam_file):\n",
    "    norm_bam_dir = os.path.join(output_dir, '11_normalized_bigwigs')\n",
    "    os.makedirs(norm_bam_dir, exist_ok=True)  # Create the directory if it doesn't exist\n",
    "    \n",
    "    bigwig_file = os.path.join(norm_bam_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        'bamCoverage',\n",
    "        '--bam', bam_file,\n",
    "        '-o', bigwig_file,\n",
    "        '--binSize', '10',\n",
    "        '--normalizeUsing',  'RPGC',\n",
    "        '--effectiveGenomeSize', '115000000',\n",
    "        '--scaleFactor', '10',\n",
    "        '--numberOfProcessors', '15'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "    return f'Generated normalized bigWig for {bam_file} at {bigwig_file}'\n",
    "\n",
    "\n",
    "# Initialize the pool and generate normalized bigWigs\n",
    "with Pool(4) as pool:\n",
    "    normalization_results = pool.map(generate_normalized_bigwig, bam_files)\n",
    "\n",
    "for result in normalization_results:\n",
    "    print(result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c64181e-4b67-4716-8581-878c1ea39e5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#just use rpkm "
   ]
  },
  {
   "cell_type": "raw",
   "id": "c9bf2fb9-3a3a-4c1f-bf07-707829dace4a",
   "metadata": {
    "tags": []
   },
   "source": [
    "def bam_to_bigWig(bam_file, output_dir):\n",
    "    bigwig_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        'bamCoverage', \n",
    "        '-b', bam_file, \n",
    "        '-o', bigwig_file,\n",
    "        '--normalizeUsing', 'RPKM',  # Normalize the coverage (you can use other methods as well)\n",
    "        '--binSize', '10',  # The size of the bins, in bases\n",
    "        '--numberOfProcessors', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "    return bigwig_file\n",
    "\n",
    "# Wrapper function to pass multiple arguments to the mapped function\n",
    "def wrapper_bam_to_bigWig(args):\n",
    "    return bam_to_bigWig(*args)\n",
    "\n",
    "merged_bam2bed_output = os.path.join(output_dir, '11_merged_bam2bw')\n",
    "if not os.path.exists(merged_bam2bed_output):\n",
    "    os.makedirs(merged_bam2bed_output)\n",
    "\n",
    "with Pool(2) as p:  # Using 4 concurrent processes\n",
    "    p.map(wrapper_bam_to_bigWig, [(bam_file, merged_bam2bed_output) for bam_file in merged_bams])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "420f2611-4298-450b-a5c8-35720ca146e5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam']"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "ab97d0ff-83d1-4272-aa31-8d3031e41ff1",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: BPM\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    }
   ],
   "source": [
    "def bam_to_bigWig(bam_file, output_dir):\n",
    "    bigwig_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        'bamCoverage', \n",
    "        '-b', bam_file, \n",
    "        '-o', bigwig_file,\n",
    "        '--normalizeUsing', 'BPM',  # similar to TPM in RNA-seq\n",
    "        '--binSize', '10',  # The size of the bins, in bases\n",
    "        '--numberOfProcessors', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "    return bigwig_file\n",
    "\n",
    "# Wrapper function to pass multiple arguments to the mapped function\n",
    "def wrapper_bam_to_bigWig(args):\n",
    "    return bam_to_bigWig(*args)\n",
    "\n",
    "merged_bam2bed_output = os.path.join(output_dir, '11_merged_bam2bw_BPM')\n",
    "if not os.path.exists(merged_bam2bed_output):\n",
    "    os.makedirs(merged_bam2bed_output)\n",
    "\n",
    "with Pool(4) as p:  # Using 4 concurrent processes\n",
    "    p.map(wrapper_bam_to_bigWig, [(bam_file, merged_bam2bed_output) for bam_file in bam_files])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "d3322f3e-2566-4501-9f66-a29bd66297c7",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: BPM\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    }
   ],
   "source": [
    "def bam_to_bigWig(bam_file, output_dir):\n",
    "    bigwig_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        'bamCoverage', \n",
    "        '-b', bam_file, \n",
    "        '-o', bigwig_file,\n",
    "        '--normalizeUsing', 'BPM',  # similar to TPM in RNA-seq\n",
    "        '--binSize', '10',  # The size of the bins, in bases\n",
    "        '--numberOfProcessors', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "    return bigwig_file\n",
    "\n",
    "# Wrapper function to pass multiple arguments to the mapped function\n",
    "def wrapper_bam_to_bigWig(args):\n",
    "    return bam_to_bigWig(*args)\n",
    "\n",
    "merged_bam2bed_output = os.path.join(output_dir, '11_merged_bam2bw_BPM')\n",
    "if not os.path.exists(merged_bam2bed_output):\n",
    "    os.makedirs(merged_bam2bed_output)\n",
    "\n",
    "with Pool(4) as p:  # Using 4 concurrent processes\n",
    "    p.map(wrapper_bam_to_bigWig, [(bam_file, merged_bam2bed_output) for bam_file in bam_files])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "812291b6-af14-426c-bd36-9ccb63e834b9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam',\n",
       " '/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "902105e2-f9ab-4b0c-ade6-df645aba09b8",
   "metadata": {},
   "source": [
    "### larger bin "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d0691ff4-f3f1-4a08-97cb-267e6066a55d",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/C14_merged.bam']\n",
      "binLength: 30\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 30\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/DM_merged.bam']\n",
      "binLength: 30\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 30\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/DM1_DM2d_Novogene/X401SC23101639-Z01-F001/Analysi_dir/9_merged_bam/C12_merged.bam']\n",
      "binLength: 30\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 30\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "normalization: BPM\n",
      "bamFilesList: ['/media/HDD3/bulk_ATAC_DMs/N2315964_30-927980316_Lib_2023-09-30/Analysi_dir/9_merged_bam/OK_merged.bam']\n",
      "binLength: 30\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 10\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 30\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    }
   ],
   "source": [
    "def bam_to_bigWig(bam_file, output_dir):\n",
    "    bigwig_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        'bamCoverage', \n",
    "        '-b', bam_file, \n",
    "        '-o', bigwig_file,\n",
    "        '--normalizeUsing', 'BPM',  # similar to TPM in RNA-seq\n",
    "        '--binSize', '30',  # The size of the bins, in bases\n",
    "        '--numberOfProcessors', '10'\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "    return bigwig_file\n",
    "\n",
    "# Wrapper function to pass multiple arguments to the mapped function\n",
    "def wrapper_bam_to_bigWig(args):\n",
    "    return bam_to_bigWig(*args)\n",
    "\n",
    "merged_bam2bed_output = os.path.join(output_dir, '11_merged_bam2bw_BPM_bin30')\n",
    "if not os.path.exists(merged_bam2bed_output):\n",
    "    os.makedirs(merged_bam2bed_output)\n",
    "\n",
    "with Pool(4) as p:  # Using 4 concurrent processes\n",
    "    p.map(wrapper_bam_to_bigWig, [(bam_file, merged_bam2bed_output) for bam_file in bam_files])\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f42ef40b-fcf1-45ae-ab52-fac116f95a2e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dab98c75-f906-4376-a4d8-5d873a7a5627",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "0db67dc9-7787-4088-ac7f-4251ef931170",
   "metadata": {},
   "source": [
    "## For RNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "5995e1c0-ce7a-4ad6-a6c9-1a8d0f324551",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from multiprocessing import Pool\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "429c98dc-1cdb-4aa5-afa2-b0a25b45dfc1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def merge_bam(group_files, output_dir, group_name):\n",
    "    merged_bam = os.path.join(output_dir, f\"{group_name}_merged.bam\")\n",
    "    cmd = [\"samtools\", \"merge\", merged_bam] + group_files\n",
    "    subprocess.run(cmd, check=True)\n",
    "    print(f\"Merged {group_name}: {merged_bam}\")\n",
    "    return merged_bam\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "c1cec69d-2b52-41c6-b5aa-30747fb1ed31",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['COL',\n",
       "  ['/media/HDD3/3DM/DM1_DM2d/2_bam/COL7_ROO_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/COL3_ROO_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/COL4_SHO_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/COL8_SHO_sorted.bam']],\n",
       " ['DM',\n",
       "  ['/media/HDD3/3DM/DM1_DM2d/2_bam/DM5_ROOT_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM1_ROOT_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM6_SHOO_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM13_ROO_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM2_SHOO_sorted.bam',\n",
       "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM14_SHO_sorted.bam']]]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bam_dir = \"/media/HDD3/3DM/DM1_DM2d/2_bam\"\n",
    "output_dir = \"/media/HDD3/3DM/DM1_DM2d/\"\n",
    "merged_dir = os.path.join(output_dir, \"3_merged_bams\")\n",
    "bigwig_dir = os.path.join(output_dir, \"4_bigwigs\")\n",
    "\n",
    "os.makedirs(merged_dir, exist_ok=True)\n",
    "os.makedirs(bigwig_dir, exist_ok=True)\n",
    "\n",
    "groups = [\"COL\", \"DM\"]\n",
    "grouped_bam_files = [[group, [os.path.join(bam_dir, f) for f in os.listdir(bam_dir) if f.endswith(\".bam\") and f.startswith(group)]] for group in groups]\n",
    "grouped_bam_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b8fdefd0-9c52-420e-be91-f8924ecae7ba",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "grouped_bam_files = [['COL',\n",
    "  [\n",
    "      # '/media/HDD3/3DM/DM1_DM2d/2_bam/COL7_ROO_sorted.bam',\n",
    "   # '/media/HDD3/3DM/DM1_DM2d/2_bam/COL3_ROO_sorted.bam',\n",
    "   '/media/HDD3/3DM/DM1_DM2d/2_bam/COL4_SHO_sorted.bam',\n",
    "   '/media/HDD3/3DM/DM1_DM2d/2_bam/COL8_SHO_sorted.bam']],\n",
    " ['DM',\n",
    "  [\n",
    "      # '/media/HDD3/3DM/DM1_DM2d/2_bam/DM5_ROOT_sorted.bam',\n",
    "   # '/media/HDD3/3DM/DM1_DM2d/2_bam/DM1_ROOT_sorted.bam',\n",
    "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM6_SHOO_sorted.bam',\n",
    "   # '/media/HDD3/3DM/DM1_DM2d/2_bam/DM13_ROO_sorted.bam',\n",
    "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM2_SHOO_sorted.bam',\n",
    "   '/media/HDD3/3DM/DM1_DM2d/2_bam/DM14_SHO_sorted.bam']]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "fc81e50d-dd52-4521-ab6d-a57ad5ae1a10",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Merged COL: /media/HDD3/3DM/DM1_DM2d/3_merged_bams/COL_merged.bam\n",
      "Merged DM: /media/HDD3/3DM/DM1_DM2d/3_merged_bams/DM_merged.bam\n"
     ]
    }
   ],
   "source": [
    "num_processes = 2  # Adjust based on your system\n",
    "\n",
    "with Pool(num_processes) as pool:\n",
    "    merged_bams = pool.starmap(merge_bam, [(files, merged_dir, group) for group, files in grouped_bam_files])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "9d7f4b37-7abe-4e35-b7df-7537fa881f44",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! samtools index /media/HDD3/3DM/DM1_DM2d/3_merged_bams/COL_merged.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "e27ce224-b56e-40e3-987a-2fc0fbb6712e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! samtools index /media/HDD3/3DM/DM1_DM2d/3_merged_bams/DM_merged.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ed882e9-60c0-4de7-b79d-f1c0217dc25d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "16f736c2-b094-46bf-996c-f3af74e94b88",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/3_merged_bams/COL_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    }
   ],
   "source": [
    "! bamCoverage -b /media/HDD3/3DM/DM1_DM2d/3_merged_bams/COL_merged.bam -o\\\n",
    "    /media/HDD3/3DM/DM1_DM2d/4_bigwigs/COL_12_Shoot.bw\\\n",
    "    --binSize 10\\\n",
    "    --normalizeUsing RPGC\\\n",
    "    --effectiveGenomeSize 115000000\\\n",
    "    --scaleFactor 10\\\n",
    "    --numberOfProcessors 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "953403c7-2eb3-4d62-b982-1e0e91ba32bc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/3_merged_bams/DM_merged.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    }
   ],
   "source": [
    "! bamCoverage -b /media/HDD3/3DM/DM1_DM2d/3_merged_bams/DM_merged.bam -o\\\n",
    "    /media/HDD3/3DM/DM1_DM2d/4_bigwigs/DM1_DM2d_Shoot.bw\\\n",
    "    --binSize 10\\\n",
    "    --normalizeUsing RPGC\\\n",
    "    --effectiveGenomeSize 115000000\\\n",
    "    --scaleFactor 10\\\n",
    "    --numberOfProcessors 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "019f1d2c-3c53-4064-9952-ec060348d402",
   "metadata": {},
   "outputs": [],
   "source": [
    "bamCoverage -b /media/HDD3/3DM/DM1_DM2d/3_merged_bams/COL_merged.bam -o\\\n",
    "    /media/HDD3/3DM/DM1_DM2d/4_bigwigs/COL_12_Shoot.bw\\\n",
    "    --binSize 10\\\n",
    "    --normalizeUsing RPGC\\\n",
    "    --effectiveGenomeSize 115000000\\\n",
    "    --scaleFactor 10\\\n",
    "    --numberOfProcessors 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "434a45b4-aa2b-4564-93a6-96d3fd32aabd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b0ac230-8950-4712-a2fe-ec142e475b04",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e6224501-3652-4ec7-9e37-45e0803d0ce3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/media/HDD3/3DM/DM1_DM2d/2_bam/COL7_ROO_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/DM5_ROOT_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/DM1_ROOT_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/COL3_ROO_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/DM6_SHOO_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/DM13_ROO_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/DM2_SHOO_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/COL4_SHO_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/DM14_SHO_sorted.bam',\n",
       " '/media/HDD3/3DM/DM1_DM2d/2_bam/COL8_SHO_sorted.bam']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import subprocess\n",
    "from multiprocessing import Pool\n",
    "\n",
    "def convert_bam_to_bigwig(bam_file, output_dir, effective_genome_size, bin_size=10, scale_factor=10, num_processors=20):\n",
    "    bw_file = os.path.join(output_dir, os.path.basename(bam_file).replace('.bam', '.bw'))\n",
    "    cmd = [\n",
    "        \"bamCoverage\", \"-b\", bam_file, \"-o\", bw_file,\n",
    "        \"--binSize\", str(bin_size),\n",
    "        \"--normalizeUsing\", \"RPGC\",\n",
    "        \"--effectiveGenomeSize\", str(effective_genome_size),\n",
    "        \"--scaleFactor\", str(scale_factor),\n",
    "        \"--numberOfProcessors\", str(num_processors)\n",
    "    ]\n",
    "    subprocess.run(cmd, check=True)\n",
    "    print(f\"Converted {bam_file} to BigWig: {bw_file}\")\n",
    "    \n",
    "bam_dir = \"/media/HDD3/3DM/DM1_DM2d/2_bam\"\n",
    "bw_dir = \"/media/HDD3/3DM/DM1_DM2d/4_bw_individuals\"\n",
    "effective_genome_size = 115000000  # Adjust as needed\n",
    "\n",
    "os.makedirs(bw_dir, exist_ok=True)\n",
    "\n",
    "bam_files = [os.path.join(bam_dir, f) for f in os.listdir(bam_dir) if f.endswith(\".bam\")]\n",
    "bam_files\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "da997772-b994-4cc4-8f53-dc9081896701",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "normalization: 1x (effective genome size 115000000)\n",
      "normalization: 1x (effective genome size 115000000)\n",
      "normalization: 1x (effective genome size 115000000)\n",
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/COL3_ROO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/DM1_ROOT_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/COL7_ROO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/DM5_ROOT_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/DM6_SHOO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/COL7_ROO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/COL7_ROO_sorted.bw\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/COL3_ROO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/COL3_ROO_sorted.bw\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/DM13_ROO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/DM2_SHOO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/DM6_SHOO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/DM6_SHOO_sorted.bw\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/COL4_SHO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/DM1_ROOT_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/DM1_ROOT_sorted.bw\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/DM14_SHO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/DM5_ROOT_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/DM5_ROOT_sorted.bw\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "normalization: 1x (effective genome size 115000000)\n",
      "bamFilesList: ['/media/HDD3/3DM/DM1_DM2d/2_bam/COL8_SHO_sorted.bam']\n",
      "binLength: 10\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 20\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 10\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/DM2_SHOO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/DM2_SHOO_sorted.bw\n",
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/DM13_ROO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/DM13_ROO_sorted.bw\n",
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/DM14_SHO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/DM14_SHO_sorted.bw\n",
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/COL4_SHO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/COL4_SHO_sorted.bw\n",
      "Converted /media/HDD3/3DM/DM1_DM2d/2_bam/COL8_SHO_sorted.bam to BigWig: /media/HDD3/3DM/DM1_DM2d/4_bw_individuals/COL8_SHO_sorted.bw\n"
     ]
    }
   ],
   "source": [
    "num_processes = 5  # Adjust based on your system\n",
    "\n",
    "with Pool(num_processes) as pool:\n",
    "    pool.starmap(convert_bam_to_bigwig, [(bam_file, bw_dir, effective_genome_size) for bam_file in bam_files])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66769632-adb2-446d-9d53-40c81f661081",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9956a38a-ce69-47c5-901c-bdda66c0d920",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:work2] *",
   "language": "python",
   "name": "conda-env-work2-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
