{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "387496df-c5fd-48b6-a68e-67a4ed6152dd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import glob\n",
    "import os\n",
    "from os.path import exists\n",
    "import re\n",
    "import pandas as pd\n",
    "from multiprocessing import Pool\n",
    "import pyreadr"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b157d11-bcca-4a6d-b115-3b2d3032f812",
   "metadata": {},
   "source": [
    "## NLRs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bf09721c-f11f-475b-b3a6-caa0f0fdaa0d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AGI</th>\n",
       "      <th>subClass</th>\n",
       "      <th>Class</th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>Conc</th>\n",
       "      <th>Conc_DM1.DM2d</th>\n",
       "      <th>Conc_Col.0_12DAS</th>\n",
       "      <th>Fold</th>\n",
       "      <th>p.value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>distanceToTSS</th>\n",
       "      <th>log2FC_RNA_DM1_DM2d</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G17750</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>6105278</td>\n",
       "      <td>6105792</td>\n",
       "      <td>515</td>\n",
       "      <td>*</td>\n",
       "      <td>9.169747</td>\n",
       "      <td>8.973674</td>\n",
       "      <td>9.342332</td>\n",
       "      <td>-0.330542</td>\n",
       "      <td>2.302910e-04</td>\n",
       "      <td>8.147028e-04</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>-531.0</td>\n",
       "      <td>0.221663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G55610</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>20786677</td>\n",
       "      <td>20787072</td>\n",
       "      <td>396</td>\n",
       "      <td>*</td>\n",
       "      <td>8.929118</td>\n",
       "      <td>9.337086</td>\n",
       "      <td>8.358192</td>\n",
       "      <td>0.906746</td>\n",
       "      <td>1.433535e-11</td>\n",
       "      <td>1.288136e-10</td>\n",
       "      <td>Promoter (2-3kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-2660.0</td>\n",
       "      <td>5.013700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G55610</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>20784725</td>\n",
       "      <td>20785768</td>\n",
       "      <td>1044</td>\n",
       "      <td>*</td>\n",
       "      <td>10.439106</td>\n",
       "      <td>10.213131</td>\n",
       "      <td>10.634433</td>\n",
       "      <td>-0.394051</td>\n",
       "      <td>3.922709e-07</td>\n",
       "      <td>2.179612e-06</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-708.0</td>\n",
       "      <td>5.013700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G51890</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>19278556</td>\n",
       "      <td>19278730</td>\n",
       "      <td>175</td>\n",
       "      <td>*</td>\n",
       "      <td>7.950350</td>\n",
       "      <td>8.426414</td>\n",
       "      <td>7.234997</td>\n",
       "      <td>1.124701</td>\n",
       "      <td>7.317679e-21</td>\n",
       "      <td>1.231792e-19</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>68.0</td>\n",
       "      <td>5.884220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G51890</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>19280170</td>\n",
       "      <td>19280481</td>\n",
       "      <td>312</td>\n",
       "      <td>*</td>\n",
       "      <td>8.406953</td>\n",
       "      <td>8.584910</td>\n",
       "      <td>8.203916</td>\n",
       "      <td>0.324014</td>\n",
       "      <td>9.571170e-04</td>\n",
       "      <td>2.955329e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-1372.0</td>\n",
       "      <td>5.884220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>469</th>\n",
       "      <td>AT5G48620</td>\n",
       "      <td>CC-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>19714077</td>\n",
       "      <td>19714735</td>\n",
       "      <td>659</td>\n",
       "      <td>*</td>\n",
       "      <td>9.285059</td>\n",
       "      <td>9.408883</td>\n",
       "      <td>9.149601</td>\n",
       "      <td>0.218144</td>\n",
       "      <td>7.537352e-03</td>\n",
       "      <td>1.857587e-02</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>-1369.0</td>\n",
       "      <td>0.429428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>AT5G49140</td>\n",
       "      <td>TIR-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>19921957</td>\n",
       "      <td>19922295</td>\n",
       "      <td>339</td>\n",
       "      <td>*</td>\n",
       "      <td>8.656500</td>\n",
       "      <td>8.855653</td>\n",
       "      <td>8.425384</td>\n",
       "      <td>0.354583</td>\n",
       "      <td>2.092802e-03</td>\n",
       "      <td>5.970124e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>2.013046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>AT5G51630</td>\n",
       "      <td>TIR-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>20970901</td>\n",
       "      <td>20971353</td>\n",
       "      <td>453</td>\n",
       "      <td>*</td>\n",
       "      <td>9.082176</td>\n",
       "      <td>9.193800</td>\n",
       "      <td>8.961186</td>\n",
       "      <td>0.203603</td>\n",
       "      <td>3.644097e-03</td>\n",
       "      <td>9.774699e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>1773.0</td>\n",
       "      <td>3.356119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>AT5G66900</td>\n",
       "      <td>RPW8-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>26717853</td>\n",
       "      <td>26718898</td>\n",
       "      <td>1046</td>\n",
       "      <td>*</td>\n",
       "      <td>10.683632</td>\n",
       "      <td>10.557705</td>\n",
       "      <td>10.799444</td>\n",
       "      <td>-0.225122</td>\n",
       "      <td>3.709744e-05</td>\n",
       "      <td>1.526959e-04</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.184129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>AT5G66910</td>\n",
       "      <td>RPW8-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>26721259</td>\n",
       "      <td>26721969</td>\n",
       "      <td>711</td>\n",
       "      <td>*</td>\n",
       "      <td>10.414788</td>\n",
       "      <td>10.200595</td>\n",
       "      <td>10.601254</td>\n",
       "      <td>-0.370847</td>\n",
       "      <td>2.963537e-06</td>\n",
       "      <td>1.448728e-05</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.096199</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>474 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           AGI      subClass Class seqnames     start       end  width strand  \\\n",
       "0    AT1G17750      LRR-RLKs   LRR        1   6105278   6105792    515      *   \n",
       "1    AT1G55610      LRR-RLKs   LRR        1  20786677  20787072    396      *   \n",
       "2    AT1G55610      LRR-RLKs   LRR        1  20784725  20785768   1044      *   \n",
       "3    AT1G51890      LRR-RLKs   LRR        1  19278556  19278730    175      *   \n",
       "4    AT1G51890      LRR-RLKs   LRR        1  19280170  19280481    312      *   \n",
       "..         ...           ...   ...      ...       ...       ...    ...    ...   \n",
       "469  AT5G48620    CC-NBS-LRR   NLR        5  19714077  19714735    659      *   \n",
       "470  AT5G49140   TIR-NBS-LRR   NLR        5  19921957  19922295    339      *   \n",
       "471  AT5G51630   TIR-NBS-LRR   NLR        5  20970901  20971353    453      *   \n",
       "472  AT5G66900  RPW8-NBS-LRR   NLR        5  26717853  26718898   1046      *   \n",
       "473  AT5G66910  RPW8-NBS-LRR   NLR        5  26721259  26721969    711      *   \n",
       "\n",
       "          Conc  Conc_DM1.DM2d  Conc_Col.0_12DAS      Fold       p.value  \\\n",
       "0     9.169747       8.973674          9.342332 -0.330542  2.302910e-04   \n",
       "1     8.929118       9.337086          8.358192  0.906746  1.433535e-11   \n",
       "2    10.439106      10.213131         10.634433 -0.394051  3.922709e-07   \n",
       "3     7.950350       8.426414          7.234997  1.124701  7.317679e-21   \n",
       "4     8.406953       8.584910          8.203916  0.324014  9.571170e-04   \n",
       "..         ...            ...               ...       ...           ...   \n",
       "469   9.285059       9.408883          9.149601  0.218144  7.537352e-03   \n",
       "470   8.656500       8.855653          8.425384  0.354583  2.092802e-03   \n",
       "471   9.082176       9.193800          8.961186  0.203603  3.644097e-03   \n",
       "472  10.683632      10.557705         10.799444 -0.225122  3.709744e-05   \n",
       "473  10.414788      10.200595         10.601254 -0.370847  2.963537e-06   \n",
       "\n",
       "              FDR        annotation  geneStrand  distanceToTSS  \\\n",
       "0    8.147028e-04  Promoter (<=1kb)           1         -531.0   \n",
       "1    1.288136e-10  Promoter (2-3kb)           2        -2660.0   \n",
       "2    2.179612e-06  Promoter (<=1kb)           2         -708.0   \n",
       "3    1.231792e-19  Promoter (<=1kb)           2           68.0   \n",
       "4    2.955329e-03  Promoter (1-2kb)           2        -1372.0   \n",
       "..            ...               ...         ...            ...   \n",
       "469  1.857587e-02  Promoter (1-2kb)           1        -1369.0   \n",
       "470  5.970124e-03  Promoter (1-2kb)           2         1120.0   \n",
       "471  9.774699e-03  Promoter (1-2kb)           1         1773.0   \n",
       "472  1.526959e-04  Promoter (<=1kb)           2            0.0   \n",
       "473  1.448728e-05  Promoter (<=1kb)           2            0.0   \n",
       "\n",
       "     log2FC_RNA_DM1_DM2d  \n",
       "0               0.221663  \n",
       "1               5.013700  \n",
       "2               5.013700  \n",
       "3               5.884220  \n",
       "4               5.884220  \n",
       "..                   ...  \n",
       "469             0.429428  \n",
       "470             2.013046  \n",
       "471             3.356119  \n",
       "472             1.184129  \n",
       "473             3.096199  \n",
       "\n",
       "[474 rows x 18 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load the RDS file into Python\n",
    "result = pyreadr.read_r(os.path.join('./NLRs_LRRs', \"peakAnno_tb_DM1_DM2d_NLR_LRRs.rds\"))\n",
    "df = result[None]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f722cc32-e104-426a-a0f2-5b8a6c3a4822",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "09b56ef8-f8d5-48c8-8522-7a28045821c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "def add_sequences_to_df(df, genome_path):\n",
    "    # Load the genome fasta file\n",
    "    genome_sequences = SeqIO.to_dict(SeqIO.parse(genome_path, \"fasta\"))\n",
    "\n",
    "    # Create a column to store sequence data\n",
    "    df['sequence'] = ''\n",
    "\n",
    "    # Iterate over rows in the DataFrame\n",
    "    for index, row in df.iterrows():\n",
    "        # Extract sequence details\n",
    "        seq_name = row['seqnames']\n",
    "        start = int(row['start'])\n",
    "        end = int(row['end'])\n",
    "        \n",
    "        # Get the corresponding sequence from the genome\n",
    "        gene_seq = str(genome_sequences[seq_name].seq[start-1:end])\n",
    "\n",
    "        # Assign the sequence to the 'sequence' column\n",
    "        df.at[index, 'sequence'] = gene_seq\n",
    "\n",
    "    return df\n",
    "\n",
    "genome_path = os.path.join('/media/HDD2/Genomes/Ath_Ensembl56', 'Arabidopsis_thaliana.TAIR10.dna.toplevel.fa')\n",
    "\n",
    "# Call the function with the DataFrame and genome path\n",
    "df_with_sequences = add_sequences_to_df(df, genome_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "33836647-cfeb-4d2e-a7d2-fc1abb673f0e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AGI</th>\n",
       "      <th>subClass</th>\n",
       "      <th>Class</th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>Conc</th>\n",
       "      <th>Conc_DM1.DM2d</th>\n",
       "      <th>Conc_Col.0_12DAS</th>\n",
       "      <th>Fold</th>\n",
       "      <th>p.value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>distanceToTSS</th>\n",
       "      <th>log2FC_RNA_DM1_DM2d</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G17750</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>6105278</td>\n",
       "      <td>6105792</td>\n",
       "      <td>515</td>\n",
       "      <td>*</td>\n",
       "      <td>9.169747</td>\n",
       "      <td>8.973674</td>\n",
       "      <td>9.342332</td>\n",
       "      <td>-0.330542</td>\n",
       "      <td>2.302910e-04</td>\n",
       "      <td>8.147028e-04</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>-531.0</td>\n",
       "      <td>0.221663</td>\n",
       "      <td>AGTCTTAGGATGTAGGTGGCCTTGTGATCTTGTGACAAAATGCAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G55610</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>20786677</td>\n",
       "      <td>20787072</td>\n",
       "      <td>396</td>\n",
       "      <td>*</td>\n",
       "      <td>8.929118</td>\n",
       "      <td>9.337086</td>\n",
       "      <td>8.358192</td>\n",
       "      <td>0.906746</td>\n",
       "      <td>1.433535e-11</td>\n",
       "      <td>1.288136e-10</td>\n",
       "      <td>Promoter (2-3kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-2660.0</td>\n",
       "      <td>5.013700</td>\n",
       "      <td>ACGCTTATATATATTTTGATAATTTTGTTTGACATTAAGGACCGAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G55610</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>20784725</td>\n",
       "      <td>20785768</td>\n",
       "      <td>1044</td>\n",
       "      <td>*</td>\n",
       "      <td>10.439106</td>\n",
       "      <td>10.213131</td>\n",
       "      <td>10.634433</td>\n",
       "      <td>-0.394051</td>\n",
       "      <td>3.922709e-07</td>\n",
       "      <td>2.179612e-06</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-708.0</td>\n",
       "      <td>5.013700</td>\n",
       "      <td>ACACCTAACCTATACAACTATACAAAGGAAATGAATATGATAAATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G51890</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>19278556</td>\n",
       "      <td>19278730</td>\n",
       "      <td>175</td>\n",
       "      <td>*</td>\n",
       "      <td>7.950350</td>\n",
       "      <td>8.426414</td>\n",
       "      <td>7.234997</td>\n",
       "      <td>1.124701</td>\n",
       "      <td>7.317679e-21</td>\n",
       "      <td>1.231792e-19</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>68.0</td>\n",
       "      <td>5.884220</td>\n",
       "      <td>AGAGGAGGCTATTCATTTTTCATAGGTTCTTGGTGGGTAAAGATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G51890</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>19280170</td>\n",
       "      <td>19280481</td>\n",
       "      <td>312</td>\n",
       "      <td>*</td>\n",
       "      <td>8.406953</td>\n",
       "      <td>8.584910</td>\n",
       "      <td>8.203916</td>\n",
       "      <td>0.324014</td>\n",
       "      <td>9.571170e-04</td>\n",
       "      <td>2.955329e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-1372.0</td>\n",
       "      <td>5.884220</td>\n",
       "      <td>AGCTTAGAATGAACACGTTTGGTTGTTAATCTTTGTCTTTCTTCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>469</th>\n",
       "      <td>AT5G48620</td>\n",
       "      <td>CC-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>19714077</td>\n",
       "      <td>19714735</td>\n",
       "      <td>659</td>\n",
       "      <td>*</td>\n",
       "      <td>9.285059</td>\n",
       "      <td>9.408883</td>\n",
       "      <td>9.149601</td>\n",
       "      <td>0.218144</td>\n",
       "      <td>7.537352e-03</td>\n",
       "      <td>1.857587e-02</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>-1369.0</td>\n",
       "      <td>0.429428</td>\n",
       "      <td>GGCACAGAGAGAATCGTGTAACAGATACGGTGCAGAATAATAGTAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>AT5G49140</td>\n",
       "      <td>TIR-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>19921957</td>\n",
       "      <td>19922295</td>\n",
       "      <td>339</td>\n",
       "      <td>*</td>\n",
       "      <td>8.656500</td>\n",
       "      <td>8.855653</td>\n",
       "      <td>8.425384</td>\n",
       "      <td>0.354583</td>\n",
       "      <td>2.092802e-03</td>\n",
       "      <td>5.970124e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>2.013046</td>\n",
       "      <td>AGACTAGTTGTGAGCCTTCCTAGTTCAAGTTTCCACCGATCCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>AT5G51630</td>\n",
       "      <td>TIR-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>20970901</td>\n",
       "      <td>20971353</td>\n",
       "      <td>453</td>\n",
       "      <td>*</td>\n",
       "      <td>9.082176</td>\n",
       "      <td>9.193800</td>\n",
       "      <td>8.961186</td>\n",
       "      <td>0.203603</td>\n",
       "      <td>3.644097e-03</td>\n",
       "      <td>9.774699e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>1773.0</td>\n",
       "      <td>3.356119</td>\n",
       "      <td>ACAACTATGGCATGAAGTTGAGTTGGGAAGAGCAGTTTCTGTCAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>AT5G66900</td>\n",
       "      <td>RPW8-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>26717853</td>\n",
       "      <td>26718898</td>\n",
       "      <td>1046</td>\n",
       "      <td>*</td>\n",
       "      <td>10.683632</td>\n",
       "      <td>10.557705</td>\n",
       "      <td>10.799444</td>\n",
       "      <td>-0.225122</td>\n",
       "      <td>3.709744e-05</td>\n",
       "      <td>1.526959e-04</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.184129</td>\n",
       "      <td>CAATGTGTGGTTCATAAAAATGTAGATAGACGAAGGAGACGAATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>AT5G66910</td>\n",
       "      <td>RPW8-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>26721259</td>\n",
       "      <td>26721969</td>\n",
       "      <td>711</td>\n",
       "      <td>*</td>\n",
       "      <td>10.414788</td>\n",
       "      <td>10.200595</td>\n",
       "      <td>10.601254</td>\n",
       "      <td>-0.370847</td>\n",
       "      <td>2.963537e-06</td>\n",
       "      <td>1.448728e-05</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.096199</td>\n",
       "      <td>CAAGGAAGACTATATCAGATGCTTTAGATTTGTTGCAAAGCAATCT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>474 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           AGI      subClass Class seqnames     start       end  width strand  \\\n",
       "0    AT1G17750      LRR-RLKs   LRR        1   6105278   6105792    515      *   \n",
       "1    AT1G55610      LRR-RLKs   LRR        1  20786677  20787072    396      *   \n",
       "2    AT1G55610      LRR-RLKs   LRR        1  20784725  20785768   1044      *   \n",
       "3    AT1G51890      LRR-RLKs   LRR        1  19278556  19278730    175      *   \n",
       "4    AT1G51890      LRR-RLKs   LRR        1  19280170  19280481    312      *   \n",
       "..         ...           ...   ...      ...       ...       ...    ...    ...   \n",
       "469  AT5G48620    CC-NBS-LRR   NLR        5  19714077  19714735    659      *   \n",
       "470  AT5G49140   TIR-NBS-LRR   NLR        5  19921957  19922295    339      *   \n",
       "471  AT5G51630   TIR-NBS-LRR   NLR        5  20970901  20971353    453      *   \n",
       "472  AT5G66900  RPW8-NBS-LRR   NLR        5  26717853  26718898   1046      *   \n",
       "473  AT5G66910  RPW8-NBS-LRR   NLR        5  26721259  26721969    711      *   \n",
       "\n",
       "          Conc  Conc_DM1.DM2d  Conc_Col.0_12DAS      Fold       p.value  \\\n",
       "0     9.169747       8.973674          9.342332 -0.330542  2.302910e-04   \n",
       "1     8.929118       9.337086          8.358192  0.906746  1.433535e-11   \n",
       "2    10.439106      10.213131         10.634433 -0.394051  3.922709e-07   \n",
       "3     7.950350       8.426414          7.234997  1.124701  7.317679e-21   \n",
       "4     8.406953       8.584910          8.203916  0.324014  9.571170e-04   \n",
       "..         ...            ...               ...       ...           ...   \n",
       "469   9.285059       9.408883          9.149601  0.218144  7.537352e-03   \n",
       "470   8.656500       8.855653          8.425384  0.354583  2.092802e-03   \n",
       "471   9.082176       9.193800          8.961186  0.203603  3.644097e-03   \n",
       "472  10.683632      10.557705         10.799444 -0.225122  3.709744e-05   \n",
       "473  10.414788      10.200595         10.601254 -0.370847  2.963537e-06   \n",
       "\n",
       "              FDR        annotation  geneStrand  distanceToTSS  \\\n",
       "0    8.147028e-04  Promoter (<=1kb)           1         -531.0   \n",
       "1    1.288136e-10  Promoter (2-3kb)           2        -2660.0   \n",
       "2    2.179612e-06  Promoter (<=1kb)           2         -708.0   \n",
       "3    1.231792e-19  Promoter (<=1kb)           2           68.0   \n",
       "4    2.955329e-03  Promoter (1-2kb)           2        -1372.0   \n",
       "..            ...               ...         ...            ...   \n",
       "469  1.857587e-02  Promoter (1-2kb)           1        -1369.0   \n",
       "470  5.970124e-03  Promoter (1-2kb)           2         1120.0   \n",
       "471  9.774699e-03  Promoter (1-2kb)           1         1773.0   \n",
       "472  1.526959e-04  Promoter (<=1kb)           2            0.0   \n",
       "473  1.448728e-05  Promoter (<=1kb)           2            0.0   \n",
       "\n",
       "     log2FC_RNA_DM1_DM2d                                           sequence  \n",
       "0               0.221663  AGTCTTAGGATGTAGGTGGCCTTGTGATCTTGTGACAAAATGCAAT...  \n",
       "1               5.013700  ACGCTTATATATATTTTGATAATTTTGTTTGACATTAAGGACCGAC...  \n",
       "2               5.013700  ACACCTAACCTATACAACTATACAAAGGAAATGAATATGATAAATT...  \n",
       "3               5.884220  AGAGGAGGCTATTCATTTTTCATAGGTTCTTGGTGGGTAAAGATTG...  \n",
       "4               5.884220  AGCTTAGAATGAACACGTTTGGTTGTTAATCTTTGTCTTTCTTCCT...  \n",
       "..                   ...                                                ...  \n",
       "469             0.429428  GGCACAGAGAGAATCGTGTAACAGATACGGTGCAGAATAATAGTAA...  \n",
       "470             2.013046  AGACTAGTTGTGAGCCTTCCTAGTTCAAGTTTCCACCGATCCTTGC...  \n",
       "471             3.356119  ACAACTATGGCATGAAGTTGAGTTGGGAAGAGCAGTTTCTGTCAGA...  \n",
       "472             1.184129  CAATGTGTGGTTCATAAAAATGTAGATAGACGAAGGAGACGAATTG...  \n",
       "473             3.096199  CAAGGAAGACTATATCAGATGCTTTAGATTTGTTGCAAAGCAATCT...  \n",
       "\n",
       "[474 rows x 19 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_with_sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3718f420-5f52-4992-ab2f-0fd26ad40a39",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9d80a09c-58ed-4ee5-b68e-085155c9cfec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(84, 19)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_NLRs_up = df[(df['log2FC_RNA_DM1_DM2d'] > 1) & (df['Class'] == 'NLR')]\n",
    "df_NLRs_up.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ac40e73c-8c7e-44fd-9106-8f512502633c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(20, 19)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_NLRs_down = df[(df['log2FC_RNA_DM1_DM2d'] < -1 ) & (df['Class'] == 'NLR')]\n",
    "df_NLRs_down.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f26b524e-dd8f-417f-b854-ccbea8a39ed3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>AGI</th>\n",
       "      <th>subClass</th>\n",
       "      <th>Class</th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>Conc</th>\n",
       "      <th>Conc_DM1.DM2d</th>\n",
       "      <th>Conc_Col.0_12DAS</th>\n",
       "      <th>Fold</th>\n",
       "      <th>p.value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>distanceToTSS</th>\n",
       "      <th>log2FC_RNA_DM1_DM2d</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G17750</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>6105278</td>\n",
       "      <td>6105792</td>\n",
       "      <td>515</td>\n",
       "      <td>*</td>\n",
       "      <td>9.169747</td>\n",
       "      <td>8.973674</td>\n",
       "      <td>9.342332</td>\n",
       "      <td>-0.330542</td>\n",
       "      <td>2.302910e-04</td>\n",
       "      <td>8.147028e-04</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>-531.0</td>\n",
       "      <td>0.221663</td>\n",
       "      <td>AGTCTTAGGATGTAGGTGGCCTTGTGATCTTGTGACAAAATGCAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT1G55610</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>20786677</td>\n",
       "      <td>20787072</td>\n",
       "      <td>396</td>\n",
       "      <td>*</td>\n",
       "      <td>8.929118</td>\n",
       "      <td>9.337086</td>\n",
       "      <td>8.358192</td>\n",
       "      <td>0.906746</td>\n",
       "      <td>1.433535e-11</td>\n",
       "      <td>1.288136e-10</td>\n",
       "      <td>Promoter (2-3kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-2660.0</td>\n",
       "      <td>5.013700</td>\n",
       "      <td>ACGCTTATATATATTTTGATAATTTTGTTTGACATTAAGGACCGAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT1G55610</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>20784725</td>\n",
       "      <td>20785768</td>\n",
       "      <td>1044</td>\n",
       "      <td>*</td>\n",
       "      <td>10.439106</td>\n",
       "      <td>10.213131</td>\n",
       "      <td>10.634433</td>\n",
       "      <td>-0.394051</td>\n",
       "      <td>3.922709e-07</td>\n",
       "      <td>2.179612e-06</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-708.0</td>\n",
       "      <td>5.013700</td>\n",
       "      <td>ACACCTAACCTATACAACTATACAAAGGAAATGAATATGATAAATT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G51890</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>19278556</td>\n",
       "      <td>19278730</td>\n",
       "      <td>175</td>\n",
       "      <td>*</td>\n",
       "      <td>7.950350</td>\n",
       "      <td>8.426414</td>\n",
       "      <td>7.234997</td>\n",
       "      <td>1.124701</td>\n",
       "      <td>7.317679e-21</td>\n",
       "      <td>1.231792e-19</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>68.0</td>\n",
       "      <td>5.884220</td>\n",
       "      <td>AGAGGAGGCTATTCATTTTTCATAGGTTCTTGGTGGGTAAAGATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT1G51890</td>\n",
       "      <td>LRR-RLKs</td>\n",
       "      <td>LRR</td>\n",
       "      <td>1</td>\n",
       "      <td>19280170</td>\n",
       "      <td>19280481</td>\n",
       "      <td>312</td>\n",
       "      <td>*</td>\n",
       "      <td>8.406953</td>\n",
       "      <td>8.584910</td>\n",
       "      <td>8.203916</td>\n",
       "      <td>0.324014</td>\n",
       "      <td>9.571170e-04</td>\n",
       "      <td>2.955329e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>-1372.0</td>\n",
       "      <td>5.884220</td>\n",
       "      <td>AGCTTAGAATGAACACGTTTGGTTGTTAATCTTTGTCTTTCTTCCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>469</th>\n",
       "      <td>AT5G48620</td>\n",
       "      <td>CC-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>19714077</td>\n",
       "      <td>19714735</td>\n",
       "      <td>659</td>\n",
       "      <td>*</td>\n",
       "      <td>9.285059</td>\n",
       "      <td>9.408883</td>\n",
       "      <td>9.149601</td>\n",
       "      <td>0.218144</td>\n",
       "      <td>7.537352e-03</td>\n",
       "      <td>1.857587e-02</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>-1369.0</td>\n",
       "      <td>0.429428</td>\n",
       "      <td>GGCACAGAGAGAATCGTGTAACAGATACGGTGCAGAATAATAGTAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>AT5G49140</td>\n",
       "      <td>TIR-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>19921957</td>\n",
       "      <td>19922295</td>\n",
       "      <td>339</td>\n",
       "      <td>*</td>\n",
       "      <td>8.656500</td>\n",
       "      <td>8.855653</td>\n",
       "      <td>8.425384</td>\n",
       "      <td>0.354583</td>\n",
       "      <td>2.092802e-03</td>\n",
       "      <td>5.970124e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>2.013046</td>\n",
       "      <td>AGACTAGTTGTGAGCCTTCCTAGTTCAAGTTTCCACCGATCCTTGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>AT5G51630</td>\n",
       "      <td>TIR-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>20970901</td>\n",
       "      <td>20971353</td>\n",
       "      <td>453</td>\n",
       "      <td>*</td>\n",
       "      <td>9.082176</td>\n",
       "      <td>9.193800</td>\n",
       "      <td>8.961186</td>\n",
       "      <td>0.203603</td>\n",
       "      <td>3.644097e-03</td>\n",
       "      <td>9.774699e-03</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>1773.0</td>\n",
       "      <td>3.356119</td>\n",
       "      <td>ACAACTATGGCATGAAGTTGAGTTGGGAAGAGCAGTTTCTGTCAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>472</th>\n",
       "      <td>AT5G66900</td>\n",
       "      <td>RPW8-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>26717853</td>\n",
       "      <td>26718898</td>\n",
       "      <td>1046</td>\n",
       "      <td>*</td>\n",
       "      <td>10.683632</td>\n",
       "      <td>10.557705</td>\n",
       "      <td>10.799444</td>\n",
       "      <td>-0.225122</td>\n",
       "      <td>3.709744e-05</td>\n",
       "      <td>1.526959e-04</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.184129</td>\n",
       "      <td>CAATGTGTGGTTCATAAAAATGTAGATAGACGAAGGAGACGAATTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>AT5G66910</td>\n",
       "      <td>RPW8-NBS-LRR</td>\n",
       "      <td>NLR</td>\n",
       "      <td>5</td>\n",
       "      <td>26721259</td>\n",
       "      <td>26721969</td>\n",
       "      <td>711</td>\n",
       "      <td>*</td>\n",
       "      <td>10.414788</td>\n",
       "      <td>10.200595</td>\n",
       "      <td>10.601254</td>\n",
       "      <td>-0.370847</td>\n",
       "      <td>2.963537e-06</td>\n",
       "      <td>1.448728e-05</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.096199</td>\n",
       "      <td>CAAGGAAGACTATATCAGATGCTTTAGATTTGTTGCAAAGCAATCT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>474 rows Ã— 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           AGI      subClass Class seqnames     start       end  width strand  \\\n",
       "0    AT1G17750      LRR-RLKs   LRR        1   6105278   6105792    515      *   \n",
       "1    AT1G55610      LRR-RLKs   LRR        1  20786677  20787072    396      *   \n",
       "2    AT1G55610      LRR-RLKs   LRR        1  20784725  20785768   1044      *   \n",
       "3    AT1G51890      LRR-RLKs   LRR        1  19278556  19278730    175      *   \n",
       "4    AT1G51890      LRR-RLKs   LRR        1  19280170  19280481    312      *   \n",
       "..         ...           ...   ...      ...       ...       ...    ...    ...   \n",
       "469  AT5G48620    CC-NBS-LRR   NLR        5  19714077  19714735    659      *   \n",
       "470  AT5G49140   TIR-NBS-LRR   NLR        5  19921957  19922295    339      *   \n",
       "471  AT5G51630   TIR-NBS-LRR   NLR        5  20970901  20971353    453      *   \n",
       "472  AT5G66900  RPW8-NBS-LRR   NLR        5  26717853  26718898   1046      *   \n",
       "473  AT5G66910  RPW8-NBS-LRR   NLR        5  26721259  26721969    711      *   \n",
       "\n",
       "          Conc  Conc_DM1.DM2d  Conc_Col.0_12DAS      Fold       p.value  \\\n",
       "0     9.169747       8.973674          9.342332 -0.330542  2.302910e-04   \n",
       "1     8.929118       9.337086          8.358192  0.906746  1.433535e-11   \n",
       "2    10.439106      10.213131         10.634433 -0.394051  3.922709e-07   \n",
       "3     7.950350       8.426414          7.234997  1.124701  7.317679e-21   \n",
       "4     8.406953       8.584910          8.203916  0.324014  9.571170e-04   \n",
       "..         ...            ...               ...       ...           ...   \n",
       "469   9.285059       9.408883          9.149601  0.218144  7.537352e-03   \n",
       "470   8.656500       8.855653          8.425384  0.354583  2.092802e-03   \n",
       "471   9.082176       9.193800          8.961186  0.203603  3.644097e-03   \n",
       "472  10.683632      10.557705         10.799444 -0.225122  3.709744e-05   \n",
       "473  10.414788      10.200595         10.601254 -0.370847  2.963537e-06   \n",
       "\n",
       "              FDR        annotation  geneStrand  distanceToTSS  \\\n",
       "0    8.147028e-04  Promoter (<=1kb)           1         -531.0   \n",
       "1    1.288136e-10  Promoter (2-3kb)           2        -2660.0   \n",
       "2    2.179612e-06  Promoter (<=1kb)           2         -708.0   \n",
       "3    1.231792e-19  Promoter (<=1kb)           2           68.0   \n",
       "4    2.955329e-03  Promoter (1-2kb)           2        -1372.0   \n",
       "..            ...               ...         ...            ...   \n",
       "469  1.857587e-02  Promoter (1-2kb)           1        -1369.0   \n",
       "470  5.970124e-03  Promoter (1-2kb)           2         1120.0   \n",
       "471  9.774699e-03  Promoter (1-2kb)           1         1773.0   \n",
       "472  1.526959e-04  Promoter (<=1kb)           2            0.0   \n",
       "473  1.448728e-05  Promoter (<=1kb)           2            0.0   \n",
       "\n",
       "     log2FC_RNA_DM1_DM2d                                           sequence  \n",
       "0               0.221663  AGTCTTAGGATGTAGGTGGCCTTGTGATCTTGTGACAAAATGCAAT...  \n",
       "1               5.013700  ACGCTTATATATATTTTGATAATTTTGTTTGACATTAAGGACCGAC...  \n",
       "2               5.013700  ACACCTAACCTATACAACTATACAAAGGAAATGAATATGATAAATT...  \n",
       "3               5.884220  AGAGGAGGCTATTCATTTTTCATAGGTTCTTGGTGGGTAAAGATTG...  \n",
       "4               5.884220  AGCTTAGAATGAACACGTTTGGTTGTTAATCTTTGTCTTTCTTCCT...  \n",
       "..                   ...                                                ...  \n",
       "469             0.429428  GGCACAGAGAGAATCGTGTAACAGATACGGTGCAGAATAATAGTAA...  \n",
       "470             2.013046  AGACTAGTTGTGAGCCTTCCTAGTTCAAGTTTCCACCGATCCTTGC...  \n",
       "471             3.356119  ACAACTATGGCATGAAGTTGAGTTGGGAAGAGCAGTTTCTGTCAGA...  \n",
       "472             1.184129  CAATGTGTGGTTCATAAAAATGTAGATAGACGAAGGAGACGAATTG...  \n",
       "473             3.096199  CAAGGAAGACTATATCAGATGCTTTAGATTTGTTGCAAAGCAATCT...  \n",
       "\n",
       "[474 rows x 19 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8ac50338-0f92-49d0-88c4-ad8b1e001f7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_sequences_to_fasta(df, output_path):\n",
    "    # Open the file in write mode\n",
    "    with open(output_path, 'w') as fasta_file:\n",
    "        # Iterate through the DataFrame\n",
    "        for index, row in df.iterrows():\n",
    "            # Construct the header using geneId, seqnames, start, and end\n",
    "            # Replace 'geneId' with the appropriate column name for gene identifiers in your DataFrame\n",
    "            header = f\">{row['AGI']}_{row['seqnames']}_{row['start']}_{row['end']}\"\n",
    "            # Retrieve the sequence\n",
    "            sequence = row['sequence']  # Ensure 'sequence' matches your DataFrame's column name\n",
    "            \n",
    "            # Write the header and sequence to the file\n",
    "            fasta_file.write(f\"{header}\\n{sequence}\\n\")\n",
    "\n",
    "output_path = './NLRs_LRRs/'\n",
    "write_sequences_to_fasta(df, os.path.join(output_path, 'all_NLRs_LRRs_peaks.fasta'))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "69eb5fb3-34dc-4a81-9e82-f4c119a8be7d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(20, 19)\n",
      "(84, 19)\n",
      "(142, 19)\n",
      "(79, 19)\n"
     ]
    }
   ],
   "source": [
    "df_NLRs_down = df[(df['log2FC_RNA_DM1_DM2d'] < -1 ) & (df['Class'] == 'NLR')]\n",
    "df_NLRs_up = df[(df['log2FC_RNA_DM1_DM2d'] > 1) & (df['Class'] == 'NLR')]\n",
    "\n",
    "df_LRRs_up = df[(df['log2FC_RNA_DM1_DM2d'] > 1) & (df['Class'] == 'LRR')]\n",
    "df_LRRs_down = df[(df['log2FC_RNA_DM1_DM2d'] < -1 ) & (df['Class'] == 'LRR')]\n",
    "#print the nrow of each df\n",
    "print(df_NLRs_down.shape)\n",
    "print(df_NLRs_up.shape)\n",
    "print(df_LRRs_down.shape)\n",
    "print(df_LRRs_up.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c7ec2ac2-db7a-4784-8758-41a682ed5177",
   "metadata": {},
   "outputs": [],
   "source": [
    "write_sequences_to_fasta(df_NLRs_down, os.path.join(output_path, 'all_NLRs_down_peaks.fasta'))\n",
    "write_sequences_to_fasta(df_NLRs_up, os.path.join(output_path, 'all_NLRs_up_peaks.fasta'))\n",
    "\n",
    "write_sequences_to_fasta(df_LRRs_down, os.path.join(output_path, 'all_LRRs_down_peaks.fasta'))\n",
    "write_sequences_to_fasta(df_LRRs_up, os.path.join(output_path, 'all_LRRs_up_peaks.fasta'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84b4319e-fd14-4cfa-be85-aac31115bf43",
   "metadata": {},
   "source": [
    "## prepare TIRs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "3281b6c3-1ee9-4785-99a2-48bd88e5ced1",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Load the RDS file into Python\n",
    "TIRs = pyreadr.read_r(os.path.join('./TIRs', \"peakAnno_tb_DM1_DM2d_TIRs.rds\"))[None]\n",
    "TIRs = TIRs.rename(columns={'geneId': 'AGI'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e99f95f1-02fc-4d94-851a-a13b48608054",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "TIRs = add_sequences_to_df(TIRs, genome_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d329cd4a-4513-4423-b452-111396dc6310",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>Conc</th>\n",
       "      <th>Conc_DM1.DM2d</th>\n",
       "      <th>Conc_Col.0_12DAS</th>\n",
       "      <th>Fold</th>\n",
       "      <th>p.value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneChr</th>\n",
       "      <th>geneStart</th>\n",
       "      <th>geneEnd</th>\n",
       "      <th>geneLength</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>AGI</th>\n",
       "      <th>distanceToTSS</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>21344814</td>\n",
       "      <td>21345638</td>\n",
       "      <td>825</td>\n",
       "      <td>*</td>\n",
       "      <td>10.270645</td>\n",
       "      <td>10.720302</td>\n",
       "      <td>9.613845</td>\n",
       "      <td>1.093734</td>\n",
       "      <td>4.445062e-75</td>\n",
       "      <td>6.314428e-73</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>21345445</td>\n",
       "      <td>21346157</td>\n",
       "      <td>713</td>\n",
       "      <td>1</td>\n",
       "      <td>AT1G57630</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TATCTGAACTGAACCAAATTGCAGAATTGTGACTTGAGAAAAACAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>9495996</td>\n",
       "      <td>9496806</td>\n",
       "      <td>811</td>\n",
       "      <td>*</td>\n",
       "      <td>10.941896</td>\n",
       "      <td>10.536322</td>\n",
       "      <td>11.258116</td>\n",
       "      <td>-0.710798</td>\n",
       "      <td>1.370821e-46</td>\n",
       "      <td>7.652398e-45</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>4</td>\n",
       "      <td>9488466</td>\n",
       "      <td>9495837</td>\n",
       "      <td>7372</td>\n",
       "      <td>2</td>\n",
       "      <td>AT4G16860</td>\n",
       "      <td>-159.0</td>\n",
       "      <td>AGTCTTAGTCTTTGTACAAGACAGAAATAATATTTCAATTTCTAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>9525707</td>\n",
       "      <td>9526712</td>\n",
       "      <td>1006</td>\n",
       "      <td>*</td>\n",
       "      <td>9.776423</td>\n",
       "      <td>10.245661</td>\n",
       "      <td>9.076541</td>\n",
       "      <td>1.139299</td>\n",
       "      <td>3.009211e-40</td>\n",
       "      <td>1.307940e-38</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>4</td>\n",
       "      <td>9517950</td>\n",
       "      <td>9525800</td>\n",
       "      <td>7851</td>\n",
       "      <td>2</td>\n",
       "      <td>AT4G16920</td>\n",
       "      <td>0.0</td>\n",
       "      <td>TGAGATAGAATATCTTTAAAAGAAGTAAATCAGCAATACATAATGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>27427728</td>\n",
       "      <td>27428464</td>\n",
       "      <td>737</td>\n",
       "      <td>*</td>\n",
       "      <td>9.704940</td>\n",
       "      <td>10.117510</td>\n",
       "      <td>9.124900</td>\n",
       "      <td>0.973040</td>\n",
       "      <td>3.086304e-36</td>\n",
       "      <td>1.121710e-34</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>27429916</td>\n",
       "      <td>27432003</td>\n",
       "      <td>2088</td>\n",
       "      <td>1</td>\n",
       "      <td>AT1G72890</td>\n",
       "      <td>-1452.0</td>\n",
       "      <td>TTCCGTGTACAAAAGTAAATTAGAAAAATTTACAGAGAAATTTCAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>9559878</td>\n",
       "      <td>9560086</td>\n",
       "      <td>209</td>\n",
       "      <td>*</td>\n",
       "      <td>9.540817</td>\n",
       "      <td>8.748015</td>\n",
       "      <td>10.049528</td>\n",
       "      <td>-1.254835</td>\n",
       "      <td>1.767202e-32</td>\n",
       "      <td>5.479661e-31</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>4</td>\n",
       "      <td>9558571</td>\n",
       "      <td>9565569</td>\n",
       "      <td>6999</td>\n",
       "      <td>1</td>\n",
       "      <td>AT4G16990</td>\n",
       "      <td>1307.0</td>\n",
       "      <td>TCCTATATTATTTTGGGTGGTGTTGATGAGCCGTTGACGTAGAAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>5</td>\n",
       "      <td>18116728</td>\n",
       "      <td>18118709</td>\n",
       "      <td>1982</td>\n",
       "      <td>*</td>\n",
       "      <td>11.177318</td>\n",
       "      <td>11.114431</td>\n",
       "      <td>11.237578</td>\n",
       "      <td>-0.112480</td>\n",
       "      <td>2.058890e-02</td>\n",
       "      <td>4.444751e-02</td>\n",
       "      <td>Promoter (2-3kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>18114461</td>\n",
       "      <td>18119006</td>\n",
       "      <td>4546</td>\n",
       "      <td>1</td>\n",
       "      <td>AT5G44870</td>\n",
       "      <td>2267.0</td>\n",
       "      <td>TCTTTAGCAATATGTGCAATCTTCGATATCTCAAAATATACAGTTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>5</td>\n",
       "      <td>18326356</td>\n",
       "      <td>18327288</td>\n",
       "      <td>933</td>\n",
       "      <td>*</td>\n",
       "      <td>10.166848</td>\n",
       "      <td>10.246298</td>\n",
       "      <td>10.082766</td>\n",
       "      <td>0.143400</td>\n",
       "      <td>2.082842e-02</td>\n",
       "      <td>4.488388e-02</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>18326203</td>\n",
       "      <td>18332920</td>\n",
       "      <td>6718</td>\n",
       "      <td>1</td>\n",
       "      <td>AT5G45260</td>\n",
       "      <td>153.0</td>\n",
       "      <td>TCTCTGAAGCTCTCCGTCGAAAAGGCATAAATAATGTGGTCGTAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>5</td>\n",
       "      <td>18851768</td>\n",
       "      <td>18852699</td>\n",
       "      <td>932</td>\n",
       "      <td>*</td>\n",
       "      <td>9.875068</td>\n",
       "      <td>9.787873</td>\n",
       "      <td>9.957292</td>\n",
       "      <td>-0.149562</td>\n",
       "      <td>2.171653e-02</td>\n",
       "      <td>4.649379e-02</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>18850645</td>\n",
       "      <td>18853896</td>\n",
       "      <td>3252</td>\n",
       "      <td>1</td>\n",
       "      <td>AT5G46490</td>\n",
       "      <td>1123.0</td>\n",
       "      <td>AAAGATAGATGATCCAACTGCACTGGAAGAGAGGCTAAAGTACCAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>1</td>\n",
       "      <td>17363952</td>\n",
       "      <td>17365555</td>\n",
       "      <td>1604</td>\n",
       "      <td>*</td>\n",
       "      <td>10.944029</td>\n",
       "      <td>11.026880</td>\n",
       "      <td>10.856128</td>\n",
       "      <td>0.148811</td>\n",
       "      <td>2.209196e-02</td>\n",
       "      <td>4.717185e-02</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>1</td>\n",
       "      <td>17365898</td>\n",
       "      <td>17368703</td>\n",
       "      <td>2806</td>\n",
       "      <td>2</td>\n",
       "      <td>AT1G47370</td>\n",
       "      <td>3148.0</td>\n",
       "      <td>GCTCCAAGATATAAAGCACGGTTGGTAGCAAAGGGATTCACACAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>4</td>\n",
       "      <td>10797337</td>\n",
       "      <td>10798413</td>\n",
       "      <td>1077</td>\n",
       "      <td>*</td>\n",
       "      <td>10.166094</td>\n",
       "      <td>10.251047</td>\n",
       "      <td>10.075824</td>\n",
       "      <td>0.149689</td>\n",
       "      <td>2.211670e-02</td>\n",
       "      <td>4.721603e-02</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>4</td>\n",
       "      <td>10797198</td>\n",
       "      <td>10798527</td>\n",
       "      <td>1330</td>\n",
       "      <td>2</td>\n",
       "      <td>AT4G19920</td>\n",
       "      <td>114.0</td>\n",
       "      <td>TGTACAAGTACAAACACGAGCCTGATCATGTCAAAGCACGCGATCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows Ã— 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    seqnames     start       end  width strand       Conc  Conc_DM1.DM2d  \\\n",
       "0          1  21344814  21345638    825      *  10.270645      10.720302   \n",
       "1          4   9495996   9496806    811      *  10.941896      10.536322   \n",
       "2          4   9525707   9526712   1006      *   9.776423      10.245661   \n",
       "3          1  27427728  27428464    737      *   9.704940      10.117510   \n",
       "4          4   9559878   9560086    209      *   9.540817       8.748015   \n",
       "..       ...       ...       ...    ...    ...        ...            ...   \n",
       "133        5  18116728  18118709   1982      *  11.177318      11.114431   \n",
       "134        5  18326356  18327288    933      *  10.166848      10.246298   \n",
       "135        5  18851768  18852699    932      *   9.875068       9.787873   \n",
       "136        1  17363952  17365555   1604      *  10.944029      11.026880   \n",
       "137        4  10797337  10798413   1077      *  10.166094      10.251047   \n",
       "\n",
       "     Conc_Col.0_12DAS      Fold       p.value           FDR  \\\n",
       "0            9.613845  1.093734  4.445062e-75  6.314428e-73   \n",
       "1           11.258116 -0.710798  1.370821e-46  7.652398e-45   \n",
       "2            9.076541  1.139299  3.009211e-40  1.307940e-38   \n",
       "3            9.124900  0.973040  3.086304e-36  1.121710e-34   \n",
       "4           10.049528 -1.254835  1.767202e-32  5.479661e-31   \n",
       "..                ...       ...           ...           ...   \n",
       "133         11.237578 -0.112480  2.058890e-02  4.444751e-02   \n",
       "134         10.082766  0.143400  2.082842e-02  4.488388e-02   \n",
       "135          9.957292 -0.149562  2.171653e-02  4.649379e-02   \n",
       "136         10.856128  0.148811  2.209196e-02  4.717185e-02   \n",
       "137         10.075824  0.149689  2.211670e-02  4.721603e-02   \n",
       "\n",
       "            annotation  geneChr  geneStart   geneEnd  geneLength  geneStrand  \\\n",
       "0     Promoter (<=1kb)        1   21345445  21346157         713           1   \n",
       "1     Promoter (<=1kb)        4    9488466   9495837        7372           2   \n",
       "2     Promoter (<=1kb)        4    9517950   9525800        7851           2   \n",
       "3     Promoter (1-2kb)        1   27429916  27432003        2088           1   \n",
       "4     Promoter (1-2kb)        4    9558571   9565569        6999           1   \n",
       "..                 ...      ...        ...       ...         ...         ...   \n",
       "133   Promoter (2-3kb)        5   18114461  18119006        4546           1   \n",
       "134   Promoter (<=1kb)        5   18326203  18332920        6718           1   \n",
       "135   Promoter (1-2kb)        5   18850645  18853896        3252           1   \n",
       "136  Distal Intergenic        1   17365898  17368703        2806           2   \n",
       "137   Promoter (<=1kb)        4   10797198  10798527        1330           2   \n",
       "\n",
       "           AGI  distanceToTSS  \\\n",
       "0    AT1G57630            0.0   \n",
       "1    AT4G16860         -159.0   \n",
       "2    AT4G16920            0.0   \n",
       "3    AT1G72890        -1452.0   \n",
       "4    AT4G16990         1307.0   \n",
       "..         ...            ...   \n",
       "133  AT5G44870         2267.0   \n",
       "134  AT5G45260          153.0   \n",
       "135  AT5G46490         1123.0   \n",
       "136  AT1G47370         3148.0   \n",
       "137  AT4G19920          114.0   \n",
       "\n",
       "                                              sequence  \n",
       "0    TATCTGAACTGAACCAAATTGCAGAATTGTGACTTGAGAAAAACAG...  \n",
       "1    AGTCTTAGTCTTTGTACAAGACAGAAATAATATTTCAATTTCTAGG...  \n",
       "2    TGAGATAGAATATCTTTAAAAGAAGTAAATCAGCAATACATAATGA...  \n",
       "3    TTCCGTGTACAAAAGTAAATTAGAAAAATTTACAGAGAAATTTCAC...  \n",
       "4    TCCTATATTATTTTGGGTGGTGTTGATGAGCCGTTGACGTAGAAGA...  \n",
       "..                                                 ...  \n",
       "133  TCTTTAGCAATATGTGCAATCTTCGATATCTCAAAATATACAGTTC...  \n",
       "134  TCTCTGAAGCTCTCCGTCGAAAAGGCATAAATAATGTGGTCGTAGA...  \n",
       "135  AAAGATAGATGATCCAACTGCACTGGAAGAGAGGCTAAAGTACCAA...  \n",
       "136  GCTCCAAGATATAAAGCACGGTTGGTAGCAAAGGGATTCACACAGA...  \n",
       "137  TGTACAAGTACAAACACGAGCCTGATCATGTCAAAGCACGCGATCA...  \n",
       "\n",
       "[138 rows x 20 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TIRs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "5fd0a5b4-2922-47ce-89d0-f928555547fa",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "TIR_common_up = ['AT1G09665','AT1G57670','AT1G57830','AT1G60320','AT2G03030','AT2G03300','AT4G04110','AT4G19910','AT1G27170','AT1G27180','AT1G63730','AT1G63880','AT1G69550','AT1G72840','AT4G09430','AT4G12010','AT4G16900','AT4G16930','AT4G16950','AT4G16990','AT4G19500','AT4G19510','AT4G19530','AT5G17680','AT5G17880','AT5G17890','AT5G36930','AT5G38850','AT5G40090','AT5G40100','AT5G45050','AT5G45060','AT5G45210','AT5G45220','AT5G45230','AT5G45240','AT5G45260','AT5G46270','AT5G46450','AT5G48770']\n",
    "\n",
    "\n",
    "TIR_not_up = ['AT1G51270','AT1G57630','AT1G72900','AT5G46520','AT5G45000','AT5G41750','AT5G45090','AT3G04220','AT3G44400','AT2G20142','AT2G32140','AT2G16870','AT4G11170','AT4G14370','AT1G66090','AT4G19520','AT1G72940','AT1G72920']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "198e79c1-729e-4e6d-a6fb-cad24b98fb4c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df_TIR_common_up = TIRs[TIRs['AGI'].isin(TIR_common_up)]\n",
    "df_TIR_not_up = TIRs[TIRs['AGI'].isin(TIR_not_up)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "dfbf684b-ea2f-4526-aac2-bf787cb71f6f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "write_sequences_to_fasta(df_TIR_not_up, os.path.join('TIRs/', 'df_TIR_not_up.fasta'))\n",
    "write_sequences_to_fasta(df_TIR_common_up, os.path.join('TIRs/', 'df_TIR_common_up.fasta'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf62db00-374d-4a23-84f3-71ec42095dd3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "dcf4d79e-ab27-4a6b-9fed-1732ee67330e",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/TIRs/common_up_TIRs_vs_not_up'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Loading control sequences.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (36).\n",
      "MOTIF: 1 SEQ: 53/53\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 53/53\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/TIRs/common_up_TIRs_vs_not_up\" --control \"/data/TIRs/df_TIR_not_up.fasta\" \\\n",
    "  \"/data/TIRs/df_TIR_common_up.fasta\" \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "9e46a3c9-5c8a-4129-9b00-a6911e80fce7",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/TIRs/common_up_TIRs_shuffle'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Creating control sequences by shuffling input sequences preserving 2-mers.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (36).\n",
      "MOTIF: 1 SEQ: 1044/1044\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 1044/1044\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/TIRs/common_up_TIRs_shuffle\" --control --shuffle-- \\\n",
    "  \"/data/TIRs/df_TIR_common_up.fasta\" \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "e2fe1437-155d-44f3-8bc6-4a14e2f51e36",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "The output directory '/data/NLRs_LRRs/all_LRRs_up_vs_down' already exists.\n",
      "Its contents will be overwritten.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Loading control sequences.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (79).\n",
      "MOTIF: 1 SEQ: 221/221\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 221/221\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_LRRs_up_vs_down\" --control \"/data/NLRs_LRRs/all_LRRs_down_peaks.fasta\" \\\n",
    "  \"/data/NLRs_LRRs/all_LRRs_up_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f4a5cdd3-fa8d-41b5-a08a-0b61c98dd99d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rank</th>\n",
       "      <th>motif_DB</th>\n",
       "      <th>motif_ID</th>\n",
       "      <th>motif_alt_ID</th>\n",
       "      <th>consensus</th>\n",
       "      <th>p-value</th>\n",
       "      <th>adj_p-value</th>\n",
       "      <th>E-value</th>\n",
       "      <th>tests</th>\n",
       "      <th>FASTA_max</th>\n",
       "      <th>pos</th>\n",
       "      <th>neg</th>\n",
       "      <th>PWM_min</th>\n",
       "      <th>TP</th>\n",
       "      <th>%TP</th>\n",
       "      <th>FP</th>\n",
       "      <th>%FP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>/data/scan_motif/Ath_TF_binding_motifs_plantTF...</td>\n",
       "      <td>AT1G49480</td>\n",
       "      <td>MP00192</td>\n",
       "      <td>RAMAAAAAAAAAAAA</td>\n",
       "      <td>2.590000e-08</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.00224</td>\n",
       "      <td>140.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1008.0</td>\n",
       "      <td>5.27</td>\n",
       "      <td>13.0</td>\n",
       "      <td>36.11</td>\n",
       "      <td>49.0</td>\n",
       "      <td>4.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>/data/scan_motif/Ath_TF_binding_motifs_plantTF...</td>\n",
       "      <td>AT2G45660</td>\n",
       "      <td>MP00076</td>\n",
       "      <td>TTTYTYTTTYTYTYTYTTTYT</td>\n",
       "      <td>7.010000e-07</td>\n",
       "      <td>0.000217</td>\n",
       "      <td>0.13400</td>\n",
       "      <td>309.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1008.0</td>\n",
       "      <td>16.90</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27.78</td>\n",
       "      <td>34.0</td>\n",
       "      <td>3.37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>/data/scan_motif/Ath_TF_binding_motifs_plantTF...</td>\n",
       "      <td>AT1G72050</td>\n",
       "      <td>MP00229</td>\n",
       "      <td>CYTCCTCCTCCTCCTCCTC</td>\n",
       "      <td>1.580000e-06</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.13600</td>\n",
       "      <td>139.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1008.0</td>\n",
       "      <td>53.70</td>\n",
       "      <td>6.0</td>\n",
       "      <td>16.67</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>/data/scan_motif/Ath_TF_binding_motifs_plantTF...</td>\n",
       "      <td>AT5G17430</td>\n",
       "      <td>MP00610</td>\n",
       "      <td>AADAAARARAAAAARAGARA</td>\n",
       "      <td>2.230000e-06</td>\n",
       "      <td>0.000556</td>\n",
       "      <td>0.34400</td>\n",
       "      <td>250.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1008.0</td>\n",
       "      <td>3.07</td>\n",
       "      <td>17.0</td>\n",
       "      <td>47.22</td>\n",
       "      <td>137.0</td>\n",
       "      <td>13.59</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>/data/scan_motif/Ath_TF_binding_motifs_plantTF...</td>\n",
       "      <td>AT5G63090</td>\n",
       "      <td>MP00581</td>\n",
       "      <td>WCCGCCGCCDYCKCCGCCGCH</td>\n",
       "      <td>7.580000e-05</td>\n",
       "      <td>0.002800</td>\n",
       "      <td>1.73000</td>\n",
       "      <td>37.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1008.0</td>\n",
       "      <td>17.20</td>\n",
       "      <td>4.0</td>\n",
       "      <td>11.11</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  rank                                           motif_DB   motif_ID  \\\n",
       "0    1  /data/scan_motif/Ath_TF_binding_motifs_plantTF...  AT1G49480   \n",
       "1    2  /data/scan_motif/Ath_TF_binding_motifs_plantTF...  AT2G45660   \n",
       "2    3  /data/scan_motif/Ath_TF_binding_motifs_plantTF...  AT1G72050   \n",
       "3    4  /data/scan_motif/Ath_TF_binding_motifs_plantTF...  AT5G17430   \n",
       "4    5  /data/scan_motif/Ath_TF_binding_motifs_plantTF...  AT5G63090   \n",
       "\n",
       "  motif_alt_ID              consensus       p-value  adj_p-value  E-value  \\\n",
       "0      MP00192        RAMAAAAAAAAAAAA  2.590000e-08     0.000004  0.00224   \n",
       "1      MP00076  TTTYTYTTTYTYTYTYTTTYT  7.010000e-07     0.000217  0.13400   \n",
       "2      MP00229    CYTCCTCCTCCTCCTCCTC  1.580000e-06     0.000220  0.13600   \n",
       "3      MP00610   AADAAARARAAAAARAGARA  2.230000e-06     0.000556  0.34400   \n",
       "4      MP00581  WCCGCCGCCDYCKCCGCCGCH  7.580000e-05     0.002800  1.73000   \n",
       "\n",
       "   tests  FASTA_max   pos     neg  PWM_min    TP    %TP     FP    %FP  \n",
       "0  140.0       36.0  36.0  1008.0     5.27  13.0  36.11   49.0   4.86  \n",
       "1  309.0       36.0  36.0  1008.0    16.90  10.0  27.78   34.0   3.37  \n",
       "2  139.0       36.0  36.0  1008.0    53.70   6.0  16.67    7.0   0.69  \n",
       "3  250.0       36.0  36.0  1008.0     3.07  17.0  47.22  137.0  13.59  \n",
       "4   37.0       36.0  36.0  1008.0    17.20   4.0  11.11    4.0   0.40  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TIRs_ame = pd.read_csv('TIRs/common_up_TIRs_shuffle/ame.tsv', sep='\\t')\n",
    "TIRs_ame =  TIRs_ame[TIRs_ame['consensus'].notna()]\n",
    "TIRs_ame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "c5bdf531-5778-47e2-9a58-a162835faeab",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>external_gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT5G16970</td>\n",
       "      <td>AER</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT4G32100</td>\n",
       "      <td>AT4G32100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT2G43120</td>\n",
       "      <td>AT2G43120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT1G30814</td>\n",
       "      <td>AT1G30814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT3G18710</td>\n",
       "      <td>PUB29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32828</th>\n",
       "      <td>AT3G14120</td>\n",
       "      <td>NUP107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32829</th>\n",
       "      <td>AT3G28630</td>\n",
       "      <td>CROLIN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32830</th>\n",
       "      <td>AT1G23140</td>\n",
       "      <td>CAR8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32831</th>\n",
       "      <td>AT5G09780</td>\n",
       "      <td>REM25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32832</th>\n",
       "      <td>AT3G29430</td>\n",
       "      <td>ATGGPS9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32833 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ensembl_gene_id external_gene_name\n",
       "0           AT5G16970                AER\n",
       "1           AT4G32100          AT4G32100\n",
       "2           AT2G43120          AT2G43120\n",
       "3           AT1G30814          AT1G30814\n",
       "4           AT3G18710              PUB29\n",
       "...               ...                ...\n",
       "32828       AT3G14120             NUP107\n",
       "32829       AT3G28630            CROLIN1\n",
       "32830       AT1G23140               CAR8\n",
       "32831       AT5G09780              REM25\n",
       "32832       AT3G29430            ATGGPS9\n",
       "\n",
       "[32833 rows x 2 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# map_id_name <- read.csv('../gene_table_and_other_scripts/bioMartR_Ath_all_genes_info_googled.csv')\n",
    "map_id_name = pd.read_csv('../gene_table_and_other_scripts/bioMartR_Ath_all_genes_info_googled.csv')\n",
    "map_id_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "7bb72241-9ac3-45f5-9369-4f605494919e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Performing a left join\n",
    "TIRs_ame = pd.merge(TIRs_ame, map_id_name, left_on='motif_ID', right_on='ensembl_gene_id', how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "a7d366bd-22de-4a61-9e9a-af8e154a3a47",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "TIRs_ame.to_csv('TIRs/common_up_TIRs_shuffle/ame.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "f5df8340-1d6d-4f25-a367-8debfd7ead1c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "TIRs_ame.to_csv('TIRs/common_up_TIRs_shuffle/ame.csv', sep=',', index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de1c2670-03c9-474d-875d-2b671b73c13c",
   "metadata": {},
   "source": [
    "### MEME installed from conda seems to be broken, use docker version instead"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "7fb39120-83d0-488a-8ae5-503b958e14db",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[sudo] password for hu: \n"
     ]
    }
   ],
   "source": [
    "! sudo chmod -R 777 /media/HDD2/donghui/bulk_ATAC_DM1_DM2d/WGCNA_ATAC_integration/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e7a570fd-fb25-4571-96ec-c023d01810d3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[sudo] password for hu: \n"
     ]
    }
   ],
   "source": [
    "! sudo chmod -R 777 /media/HDD2/donghui/bulk_ATAC_DM1_DM2d/NLRs_LRRs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b4a63a7b-dd7e-470f-bae3-36fa1909bc4a",
   "metadata": {
    "tags": []
   },
   "source": [
    "docker takes sudo, can run visudo and add \"yourusername ALL=(ALL) NOPASSWD: /usr/bin/docker\"\n",
    "then run the script docker_mem_ame.sh, which loops over all fasta files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28220abf-9f99-4cea-a7f1-c364a6783ddd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "4487bab7-e876-4c18-b46c-8bd0ce357ae8",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (1206664557.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[146], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d/:/data memesuite/memesuite ame --oc /data/WGCNA_ATAC_integration/ame_WGCNA_vs_non_variable_peaks/ --control /data/WGCNA_ATAC_integration/fasta_files_and_control/non_variable_peaks_in_RNA_seq.fasta /data/WGCNA_ATAC_integration/fasta_files_and_control/all_WGCNA_seq.fasta /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\u001b[0m\n\u001b[0m         ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d/:/data memesuite/memesuite ame --oc /data/WGCNA_ATAC_integration/ame_WGCNA_vs_non_variable_peaks/ --control /data/WGCNA_ATAC_integration/fasta_files_and_control/non_variable_peaks_in_RNA_seq.fasta /data/WGCNA_ATAC_integration/fasta_files_and_control/all_WGCNA_seq.fasta /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f253142-6a6a-4279-b931-f0c4d5e9f1a0",
   "metadata": {},
   "source": [
    "/media/HDD2/donghui/bulk_ATAC_DM1_DM2d Â» mkdir WGCNA_ATAC_integration/ame_results_with_shuffle/                      hu@chaelab2\n",
    "(work2) -------------------------------------------------------------------------------------------------------------------------\n",
    "/media/HDD2/donghui/bulk_ATAC_DM1_DM2d Â» chmod -R 777 WGCNA_ATAC_integration/ame_results_with_shuffle/   \n",
    "/media/HDD2/donghui/bulk_ATAC_DM1_DM2d Â» bash ./docker_mem_ame_with_shuffle.sh    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "1a30327b-1f01-406a-b38b-d53e6b70fc33",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/NLRs_LRRs/all_LRRs_up'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Creating control sequences by shuffling input sequences preserving 2-mers.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (79).\n",
      "MOTIF: 1 SEQ: 1106/1106\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 1106/1106\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_LRRs_up\" --control --shuffle-- \\\n",
    "  \"/data/NLRs_LRRs/all_LRRs_up_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "362ca580-20c1-452d-b66a-bbc2260ab583",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/NLRs_LRRs/all_LRRs_down'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Creating control sequences by shuffling input sequences preserving 2-mers.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (142).\n",
      "MOTIF: 1 SEQ: 1278/1278\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 1278/1278\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_LRRs_down\" --control --shuffle-- \\\n",
    "  \"/data/NLRs_LRRs/all_LRRs_down_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "a31e32bc-63c6-4fe0-a2ee-c166356727ee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "The output directory '/data/NLRs_LRRs/all_NLRs_up' already exists.\n",
      "Its contents will be overwritten.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Creating control sequences by shuffling input sequences preserving 2-mers.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (84).\n",
      "MOTIF: 1 SEQ: 1092/1092\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 1092/1092\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_NLRs_up\" --control --shuffle-- \\\n",
    "  \"/data/NLRs_LRRs/all_NLRs_up_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a775fb5b-d163-4c9d-9e1c-760cd2beef9d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/NLRs_LRRs/all_NLRs_down'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Creating control sequences by shuffling input sequences preserving 2-mers.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (20).\n",
      "MOTIF: 1 SEQ: 1040/1040\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 1040/1040\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_NLRs_down\" --control --shuffle-- \\\n",
    "  \"/data/NLRs_LRRs/all_NLRs_down_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "26459928-51d3-4837-97d8-ebbd42fdb825",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/NLRs_LRRs/all_NLRs_up_vs_down'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Loading control sequences.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (84).\n",
      "MOTIF: 1 SEQ: 226/226\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 226/226\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_NLRs_up_vs_down\" --control \"/data/NLRs_LRRs/all_LRRs_down_peaks.fasta\" \\\n",
    "  \"/data/NLRs_LRRs/all_NLRs_up_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "c398972b-ccc7-433a-bf74-895a656bede6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/NLRs_LRRs/all_NLRs_down_vs_up'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Loading control sequences.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (20).\n",
      "MOTIF: 1 SEQ: 99/99\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 99/99\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_NLRs_down_vs_up\" --control \"/data/NLRs_LRRs/all_LRRs_up_peaks.fasta\" \\\n",
    "  \"/data/NLRs_LRRs/all_NLRs_down_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "533cd834-3df9-4c52-a869-010cf918a767",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/NLRs_LRRs/all_LRRs_up_vs_down'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Loading control sequences.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (79).\n",
      "MOTIF: 1 SEQ: 221/221\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 221/221\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_LRRs_up_vs_down\" --control \"/data/NLRs_LRRs/all_LRRs_down_peaks.fasta\" \\\n",
    "  \"/data/NLRs_LRRs/all_LRRs_up_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "f0b5d058-3cae-41af-8204-835833bae140",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme to motif_sources which now has 1 file names.\n",
      "Motif file name is /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme.\n",
      "Writing results to output directory '/data/NLRs_LRRs/all_LRRs_down_vs_up'.\n",
      "E-value threshold for reporting results: 10\n",
      "Checking alphabets in 1 motif files.\n",
      "Loading motifs from file '/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
      "Loading primary sequences.\n",
      "Loading control sequences.\n",
      "Not in partition maximization mode. Fixing partition at the number of primary sequences (142).\n",
      "MOTIF: 1 SEQ: 221/221\n",
      "Sorting sequences by sequence PWM score to get PWM ranks; breaking ties to put negatives first.\n",
      "Leaving sequences sorted by PWM score.\n",
      "Optimizing over sequence PWM score threshold.\n",
      "MOTIF: 619 SEQ: 221/221\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  ame --oc \"/data/NLRs_LRRs/all_LRRs_down_vs_up\" --control \"/data/NLRs_LRRs/all_LRRs_up_peaks.fasta\" \\\n",
    "  \"/data/NLRs_LRRs/all_LRRs_down_peaks.fasta\" \\\n",
    "  \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "3a9317bc-ff36-44e3-bf0b-d51e6ac9c7f2",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using motif +AT1G01060 of width 10.\n",
      "Using motif -AT1G01060 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.863\n",
      "#   Estimated pi_0=0.866078\n",
      "Using motif +AT1G01250 of width 15.\n",
      "Using motif -AT1G01250 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G01260 of width 8.\n",
      "Using motif -AT1G01260 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G01720 of width 22.\n",
      "Using motif -AT1G01720 of width 22.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9574\n",
      "#   Estimated pi_0=0.964112\n",
      "Using motif +AT1G02065 of width 16.\n",
      "Using motif -AT1G02065 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8128\n",
      "#   Estimated pi_0=0.816863\n",
      "Using motif +AT1G02230 of width 14.\n",
      "Using motif -AT1G02230 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978017\n",
      "#   Estimated pi_0=0.990196\n",
      "Using motif +AT1G02250 of width 19.\n",
      "Using motif -AT1G02250 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938783\n",
      "#   Estimated pi_0=0.946772\n",
      "Using motif +AT1G03800 of width 21.\n",
      "Using motif -AT1G03800 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G03840 of width 20.\n",
      "Using motif -AT1G03840 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938045\n",
      "#   Estimated pi_0=0.942797\n",
      "Using motif +AT1G06070 of width 11.\n",
      "Using motif -AT1G06070 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06160 of width 8.\n",
      "Using motif -AT1G06160 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06180 of width 15.\n",
      "Using motif -AT1G06180 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961739\n",
      "#   Estimated pi_0=0.972535\n",
      "Using motif +AT1G06280 of width 15.\n",
      "Using motif -AT1G06280 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06850 of width 14.\n",
      "Using motif -AT1G06850 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G08000 of width 8.\n",
      "Using motif -AT1G08000 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993007\n",
      "#   Estimated pi_0=0.998265\n",
      "Using motif +AT1G08010 of width 15.\n",
      "Using motif -AT1G08010 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.981053\n",
      "#   Estimated pi_0=0.997113\n",
      "Using motif +AT1G08320 of width 11.\n",
      "Using motif -AT1G08320 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.995658\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G08810 of width 19.\n",
      "Using motif -AT1G08810 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937822\n",
      "#   Estimated pi_0=0.940364\n",
      "Using motif +AT1G09030 of width 7.\n",
      "Using motif -AT1G09030 of width 7.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.934955\n",
      "#   Estimated pi_0=0.93975\n",
      "Using motif +AT1G09530 of width 10.\n",
      "Using motif -AT1G09530 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00081\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G09540 of width 15.\n",
      "Using motif -AT1G09540 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.965124\n",
      "#   Estimated pi_0=0.979467\n",
      "Using motif +AT1G09770 of width 11.\n",
      "Using motif -AT1G09770 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G10120 of width 21.\n",
      "Using motif -AT1G10120 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G12260 of width 15.\n",
      "Using motif -AT1G12260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.947129\n",
      "#   Estimated pi_0=0.947129\n",
      "Using motif +AT1G12610 of width 15.\n",
      "Using motif -AT1G12610 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G12630 of width 21.\n",
      "Using motif -AT1G12630 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G13260 of width 17.\n",
      "Using motif -AT1G13260 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9192\n",
      "#   Estimated pi_0=0.9192\n",
      "Using motif +AT1G13300 of width 15.\n",
      "Using motif -AT1G13300 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921\n",
      "#   Estimated pi_0=0.932432\n",
      "Using motif +AT1G13450 of width 19.\n",
      "Using motif -AT1G13450 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8938\n",
      "#   Estimated pi_0=0.903704\n",
      "Using motif +AT1G14580 of width 20.\n",
      "Using motif -AT1G14580 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 46 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948175\n",
      "#   Estimated pi_0=0.953064\n",
      "Using motif +AT1G14687 of width 11.\n",
      "Using motif -AT1G14687 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8216\n",
      "#   Estimated pi_0=0.829808\n",
      "Using motif +AT1G15720 of width 8.\n",
      "Using motif -AT1G15720 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G16490 of width 15.\n",
      "Using motif -AT1G16490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9294\n",
      "#   Estimated pi_0=0.937333\n",
      "Using motif +AT1G17950 of width 11.\n",
      "Using motif -AT1G17950 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.95068\n",
      "#   Estimated pi_0=0.960345\n",
      "Using motif +AT1G18330 of width 12.\n",
      "Using motif -AT1G18330 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8332\n",
      "#   Estimated pi_0=0.839279\n",
      "Using motif +AT1G18570 of width 21.\n",
      "Using motif -AT1G18570 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9644\n",
      "#   Estimated pi_0=0.974\n",
      "Using motif +AT1G18960 of width 22.\n",
      "Using motif -AT1G18960 of width 22.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.99354\n",
      "#   Estimated pi_0=0.999397\n",
      "Using motif +AT1G19000 of width 14.\n",
      "Using motif -AT1G19000 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9014\n",
      "#   Estimated pi_0=0.917815\n",
      "Using motif +AT1G19040 of width 16.\n",
      "Using motif -AT1G19040 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971091\n",
      "#   Estimated pi_0=0.976296\n",
      "Using motif +AT1G19210 of width 14.\n",
      "Using motif -AT1G19210 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0007\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G19350 of width 11.\n",
      "Using motif -AT1G19350 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G19790 of width 23.\n",
      "Using motif -AT1G19790 of width 23.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8144\n",
      "#   Estimated pi_0=0.8144\n",
      "Using motif +AT1G19850 of width 10.\n",
      "Using motif -AT1G19850 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.992222\n",
      "#   Estimated pi_0=0.998593\n",
      "Using motif +AT1G20640 of width 15.\n",
      "Using motif -AT1G20640 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999899\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT1G20980 of width 11.\n",
      "Using motif -AT1G20980 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971\n",
      "#   Estimated pi_0=0.982241\n",
      "Using motif +AT1G21910 of width 20.\n",
      "Using motif -AT1G21910 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G22070 of width 14.\n",
      "Using motif -AT1G22070 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.987752\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G22640 of width 11.\n",
      "Using motif -AT1G22640 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.968\n",
      "#   Estimated pi_0=0.980645\n",
      "Using motif +AT1G22810 of width 15.\n",
      "Using motif -AT1G22810 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999095\n",
      "#   Estimated pi_0=0.999095\n",
      "Using motif +AT1G24260 of width 13.\n",
      "Using motif -AT1G24260 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.912941\n",
      "#   Estimated pi_0=0.920367\n",
      "Using motif +AT1G25340 of width 15.\n",
      "Using motif -AT1G25340 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9088\n",
      "#   Estimated pi_0=0.92\n",
      "Using motif +AT1G25550 of width 11.\n",
      "Using motif -AT1G25550 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8928\n",
      "#   Estimated pi_0=0.8928\n",
      "Using motif +AT1G25560 of width 8.\n",
      "Using motif -AT1G25560 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9668\n",
      "#   Estimated pi_0=0.975488\n",
      "Using motif +AT1G27360 of width 11.\n",
      "Using motif -AT1G27360 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966731\n",
      "#   Estimated pi_0=0.971496\n",
      "Using motif +AT1G27730 of width 11.\n",
      "Using motif -AT1G27730 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944314\n",
      "#   Estimated pi_0=0.952951\n",
      "Using motif +AT1G28160 of width 15.\n",
      "Using motif -AT1G28160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28300 of width 8.\n",
      "Using motif -AT1G28300 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28360 of width 8.\n",
      "Using motif -AT1G28360 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00312\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28370 of width 15.\n",
      "Using motif -AT1G28370 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28470 of width 15.\n",
      "Using motif -AT1G28470 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.989924\n",
      "#   Estimated pi_0=0.99798\n",
      "Using motif +AT1G29160 of width 21.\n",
      "Using motif -AT1G29160 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 43 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8928\n",
      "#   Estimated pi_0=0.899832\n",
      "Using motif +AT1G29280 of width 11.\n",
      "Using motif -AT1G29280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940194\n",
      "#   Estimated pi_0=0.950084\n",
      "Using motif +AT1G29860 of width 11.\n",
      "Using motif -AT1G29860 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9522\n",
      "#   Estimated pi_0=0.96747\n",
      "Using motif +AT1G30210 of width 12.\n",
      "Using motif -AT1G30210 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G30490 of width 15.\n",
      "Using motif -AT1G30490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.85802\n",
      "#   Estimated pi_0=0.863551\n",
      "Using motif +AT1G30650 of width 13.\n",
      "Using motif -AT1G30650 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.954455\n",
      "#   Estimated pi_0=0.968784\n",
      "Using motif +AT1G31140 of width 15.\n",
      "Using motif -AT1G31140 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7936\n",
      "#   Estimated pi_0=0.801765\n",
      "Using motif +AT1G32150 of width 15.\n",
      "Using motif -AT1G32150 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0007\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G32240 of width 15.\n",
      "Using motif -AT1G32240 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7802\n",
      "#   Estimated pi_0=0.785981\n",
      "Using motif +AT1G32510 of width 21.\n",
      "Using motif -AT1G32510 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944\n",
      "#   Estimated pi_0=0.957258\n",
      "Using motif +AT1G32640 of width 8.\n",
      "Using motif -AT1G32640 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997282\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT1G32770 of width 12.\n",
      "Using motif -AT1G32770 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9374\n",
      "#   Estimated pi_0=0.947358\n",
      "Using motif +AT1G32870 of width 16.\n",
      "Using motif -AT1G32870 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997245\n",
      "#   Estimated pi_0=0.999196\n",
      "Using motif +AT1G33240 of width 10.\n",
      "Using motif -AT1G33240 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.817\n",
      "#   Estimated pi_0=0.823725\n",
      "Using motif +AT1G34180 of width 15.\n",
      "Using motif -AT1G34180 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978627\n",
      "#   Estimated pi_0=0.993375\n",
      "Using motif +AT1G34190 of width 19.\n",
      "Using motif -AT1G34190 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.977\n",
      "#   Estimated pi_0=0.998995\n",
      "Using motif +AT1G34370 of width 15.\n",
      "Using motif -AT1G34370 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993373\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G34670 of width 21.\n",
      "Using motif -AT1G34670 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9234\n",
      "#   Estimated pi_0=0.925631\n",
      "Using motif +AT1G35460 of width 11.\n",
      "Using motif -AT1G35460 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96422\n",
      "#   Estimated pi_0=0.981325\n",
      "Using motif +AT1G35560 of width 10.\n",
      "Using motif -AT1G35560 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G36060 of width 14.\n",
      "Using motif -AT1G36060 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G42990 of width 10.\n",
      "Using motif -AT1G42990 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G43160 of width 15.\n",
      "Using motif -AT1G43160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G43700 of width 12.\n",
      "Using motif -AT1G43700 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.981887\n",
      "#   Estimated pi_0=0.99544\n",
      "Using motif +AT1G44830 of width 15.\n",
      "Using motif -AT1G44830 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G45249 of width 18.\n",
      "Using motif -AT1G45249 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G46264 of width 15.\n",
      "Using motif -AT1G46264 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.983333\n",
      "#   Estimated pi_0=0.988837\n",
      "Using motif +AT1G46768 of width 13.\n",
      "Using motif -AT1G46768 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G47655 of width 11.\n",
      "Using motif -AT1G47655 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937815\n",
      "#   Estimated pi_0=0.942623\n",
      "Using motif +AT1G47870 of width 15.\n",
      "Using motif -AT1G47870 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.97901\n",
      "#   Estimated pi_0=0.991758\n",
      "Using motif +AT1G49010 of width 15.\n",
      "Using motif -AT1G49010 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.852\n",
      "#   Estimated pi_0=0.857476\n",
      "Using motif +AT1G49480 of width 15.\n",
      "Using motif -AT1G49480 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 48 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938759\n",
      "#   Estimated pi_0=0.943934\n",
      "Using motif +AT1G49560 of width 11.\n",
      "Using motif -AT1G49560 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.906535\n",
      "#   Estimated pi_0=0.920336\n",
      "Using motif +AT1G49720 of width 19.\n",
      "Using motif -AT1G49720 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G49950 of width 9.\n",
      "Using motif -AT1G49950 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998783\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G50640 of width 15.\n",
      "Using motif -AT1G50640 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G51140 of width 14.\n",
      "Using motif -AT1G51140 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.97604\n",
      "#   Estimated pi_0=0.983\n",
      "Using motif +AT1G51220 of width 15.\n",
      "Using motif -AT1G51220 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999795\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G51600 of width 19.\n",
      "Using motif -AT1G51600 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.955728\n",
      "#   Estimated pi_0=0.969506\n",
      "Using motif +AT1G51700 of width 21.\n",
      "Using motif -AT1G51700 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 50 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8808\n",
      "#   Estimated pi_0=0.889402\n",
      "Using motif +AT1G52150 of width 15.\n",
      "Using motif -AT1G52150 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8346\n",
      "#   Estimated pi_0=0.8346\n",
      "Using motif +AT1G52880 of width 19.\n",
      "Using motif -AT1G52880 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.932\n",
      "#   Estimated pi_0=0.93549\n",
      "Using motif +AT1G52890 of width 12.\n",
      "Using motif -AT1G52890 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G53160 of width 10.\n",
      "Using motif -AT1G53160 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9738\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G53170 of width 15.\n",
      "Using motif -AT1G53170 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G53230 of width 13.\n",
      "Using motif -AT1G53230 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G53910 of width 21.\n",
      "Using motif -AT1G53910 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G54330 of width 21.\n",
      "Using motif -AT1G54330 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.904356\n",
      "#   Estimated pi_0=0.905234\n",
      "Using motif +AT1G55110 of width 18.\n",
      "Using motif -AT1G55110 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 35 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937087\n",
      "#   Estimated pi_0=0.953704\n",
      "Using motif +AT1G59640 of width 21.\n",
      "Using motif -AT1G59640 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0003\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G59750 of width 8.\n",
      "Using motif -AT1G59750 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G60920 of width 6.\n",
      "Using motif -AT1G60920 of width 6.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT1G61110 of width 8.\n",
      "Using motif -AT1G61110 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00119\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G62300 of width 19.\n",
      "Using motif -AT1G62300 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944423\n",
      "#   Estimated pi_0=0.947455\n",
      "Using motif +AT1G63030 of width 8.\n",
      "Using motif -AT1G63030 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.988947\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G64620 of width 19.\n",
      "Using motif -AT1G64620 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 44 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.860877\n",
      "#   Estimated pi_0=0.866525\n",
      "Using motif +AT1G65620 of width 21.\n",
      "Using motif -AT1G65620 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G65910 of width 16.\n",
      "Using motif -AT1G65910 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.939208\n",
      "#   Estimated pi_0=0.939208\n",
      "Using motif +AT1G66370 of width 11.\n",
      "Using motif -AT1G66370 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9076\n",
      "#   Estimated pi_0=0.916639\n",
      "Using motif +AT1G66420 of width 19.\n",
      "Using motif -AT1G66420 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.872475\n",
      "#   Estimated pi_0=0.875192\n",
      "Using motif +AT1G66600 of width 8.\n",
      "Using motif -AT1G66600 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.980699\n",
      "#   Estimated pi_0=0.994301\n",
      "Using motif +AT1G67260 of width 30.\n",
      "Using motif -AT1G67260 of width 30.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00372\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G67710 of width 10.\n",
      "Using motif -AT1G67710 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.918\n",
      "#   Estimated pi_0=0.926972\n",
      "Using motif +AT1G68200 of width 15.\n",
      "Using motif -AT1G68200 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G68320 of width 15.\n",
      "Using motif -AT1G68320 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9412\n",
      "#   Estimated pi_0=0.946606\n",
      "Using motif +AT1G68550 of width 15.\n",
      "Using motif -AT1G68550 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G68670 of width 13.\n",
      "Using motif -AT1G68670 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9066\n",
      "#   Estimated pi_0=0.921239\n",
      "Using motif +AT1G68840 of width 8.\n",
      "Using motif -AT1G68840 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.989431\n",
      "#   Estimated pi_0=0.996684\n",
      "Using motif +AT1G69010 of width 15.\n",
      "Using motif -AT1G69010 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G69120 of width 12.\n",
      "Using motif -AT1G69120 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.822\n",
      "#   Estimated pi_0=0.836216\n",
      "Using motif +AT1G69180 of width 8.\n",
      "Using motif -AT1G69180 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.887451\n",
      "#   Estimated pi_0=0.887451\n",
      "Using motif +AT1G69310 of width 10.\n",
      "Using motif -AT1G69310 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9012\n",
      "#   Estimated pi_0=0.909143\n",
      "Using motif +AT1G69490 of width 11.\n",
      "Using motif -AT1G69490 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961553\n",
      "#   Estimated pi_0=0.968244\n",
      "Using motif +AT1G69560 of width 13.\n",
      "Using motif -AT1G69560 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8964\n",
      "#   Estimated pi_0=0.905769\n",
      "Using motif +AT1G69570 of width 27.\n",
      "Using motif -AT1G69570 of width 27.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 52 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8722\n",
      "#   Estimated pi_0=0.887581\n",
      "Using motif +AT1G69690 of width 13.\n",
      "Using motif -AT1G69690 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00312\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G69780 of width 11.\n",
      "Using motif -AT1G69780 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8152\n",
      "#   Estimated pi_0=0.818476\n",
      "Using motif +AT1G70000 of width 10.\n",
      "Using motif -AT1G70000 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9364\n",
      "#   Estimated pi_0=0.9364\n",
      "Using motif +AT1G70920 of width 11.\n",
      "Using motif -AT1G70920 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8298\n",
      "#   Estimated pi_0=0.845981\n",
      "Using motif +AT1G71450 of width 20.\n",
      "Using motif -AT1G71450 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997\n",
      "#   Estimated pi_0=0.99809\n",
      "Using motif +AT1G71930 of width 12.\n",
      "Using motif -AT1G71930 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G72010 of width 13.\n",
      "Using motif -AT1G72010 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G72050 of width 19.\n",
      "Using motif -AT1G72050 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94473\n",
      "#   Estimated pi_0=0.950698\n",
      "Using motif +AT1G72360 of width 21.\n",
      "Using motif -AT1G72360 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G72740 of width 15.\n",
      "Using motif -AT1G72740 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96367\n",
      "#   Estimated pi_0=0.981918\n",
      "Using motif +AT1G73360 of width 11.\n",
      "Using motif -AT1G73360 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8324\n",
      "#   Estimated pi_0=0.834902\n",
      "Using motif +AT1G73730 of width 21.\n",
      "Using motif -AT1G73730 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.882\n",
      "#   Estimated pi_0=0.886275\n",
      "Using motif +AT1G74480 of width 19.\n",
      "Using motif -AT1G74480 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.984286\n",
      "#   Estimated pi_0=0.991413\n",
      "Using motif +AT1G74650 of width 15.\n",
      "Using motif -AT1G74650 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.925545\n",
      "#   Estimated pi_0=0.940476\n",
      "Using motif +AT1G74840 of width 11.\n",
      "Using motif -AT1G74840 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8968\n",
      "#   Estimated pi_0=0.90233\n",
      "Using motif +AT1G74930 of width 10.\n",
      "Using motif -AT1G74930 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G75080 of width 20.\n",
      "Using motif -AT1G75080 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G75240 of width 11.\n",
      "Using motif -AT1G75240 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9296\n",
      "#   Estimated pi_0=0.942069\n",
      "Using motif +AT1G75390 of width 15.\n",
      "Using motif -AT1G75390 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G75490 of width 15.\n",
      "Using motif -AT1G75490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G76420 of width 18.\n",
      "Using motif -AT1G76420 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.904272\n",
      "#   Estimated pi_0=0.912321\n",
      "Using motif +AT1G76870 of width 14.\n",
      "Using motif -AT1G76870 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9124\n",
      "#   Estimated pi_0=0.915596\n",
      "Using motif +AT1G76880 of width 16.\n",
      "Using motif -AT1G76880 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8432\n",
      "#   Estimated pi_0=0.8432\n",
      "Using motif +AT1G76890 of width 14.\n",
      "Using motif -AT1G76890 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8098\n",
      "#   Estimated pi_0=0.814175\n",
      "Using motif +AT1G77080 of width 15.\n",
      "Using motif -AT1G77080 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.834\n",
      "#   Estimated pi_0=0.839626\n",
      "Using motif +AT1G77200 of width 14.\n",
      "Using motif -AT1G77200 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G77640 of width 21.\n",
      "Using motif -AT1G77640 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G77920 of width 17.\n",
      "Using motif -AT1G77920 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9868\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G78080 of width 10.\n",
      "Using motif -AT1G78080 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G78700 of width 11.\n",
      "Using motif -AT1G78700 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G79180 of width 21.\n",
      "Using motif -AT1G79180 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.985686\n",
      "#   Estimated pi_0=0.996735\n",
      "Using motif +AT1G79580 of width 21.\n",
      "Using motif -AT1G79580 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9254\n",
      "#   Estimated pi_0=0.94\n",
      "Using motif +AT1G80840 of width 15.\n",
      "Using motif -AT1G80840 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 32 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.955146\n",
      "#   Estimated pi_0=0.977246\n",
      "Using motif +AT2G01060 of width 13.\n",
      "Using motif -AT2G01060 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9216\n",
      "#   Estimated pi_0=0.926346\n",
      "Using motif +AT2G01570 of width 20.\n",
      "Using motif -AT2G01570 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 54 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.958864\n",
      "#   Estimated pi_0=0.962258\n",
      "Using motif +AT2G01760 of width 8.\n",
      "Using motif -AT2G01760 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9708\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G01930 of width 24.\n",
      "Using motif -AT2G01930 of width 24.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 45 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.903307\n",
      "#   Estimated pi_0=0.907105\n",
      "Using motif +AT2G01940 of width 15.\n",
      "Using motif -AT2G01940 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 44 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.914286\n",
      "#   Estimated pi_0=0.927576\n",
      "Using motif +AT2G02070 of width 16.\n",
      "Using motif -AT2G02070 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 34 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.957962\n",
      "#   Estimated pi_0=0.962644\n",
      "Using motif +AT2G02080 of width 17.\n",
      "Using motif -AT2G02080 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 39 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.92434\n",
      "#   Estimated pi_0=0.93\n",
      "Using motif +AT2G02450 of width 19.\n",
      "Using motif -AT2G02450 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G02540 of width 14.\n",
      "Using motif -AT2G02540 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8318\n",
      "#   Estimated pi_0=0.840541\n",
      "Using motif +AT2G02820 of width 15.\n",
      "Using motif -AT2G02820 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G03340 of width 11.\n",
      "Using motif -AT2G03340 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.949043\n",
      "#   Estimated pi_0=0.959538\n",
      "Using motif +AT2G03500 of width 11.\n",
      "Using motif -AT2G03500 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9014\n",
      "#   Estimated pi_0=0.908868\n",
      "Using motif +AT2G03710 of width 18.\n",
      "Using motif -AT2G03710 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8304\n",
      "#   Estimated pi_0=0.832277\n",
      "Using motif +AT2G04038 of width 14.\n",
      "Using motif -AT2G04038 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9974\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G04880 of width 11.\n",
      "Using motif -AT2G04880 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G06200 of width 14.\n",
      "Using motif -AT2G06200 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.952\n",
      "#   Estimated pi_0=0.958\n",
      "Using motif +AT2G15740 of width 11.\n",
      "Using motif -AT2G15740 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9628\n",
      "#   Estimated pi_0=0.974386\n",
      "Using motif +AT2G17180 of width 10.\n",
      "Using motif -AT2G17180 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G17950 of width 11.\n",
      "Using motif -AT2G17950 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8408\n",
      "#   Estimated pi_0=0.849434\n",
      "Using motif +AT2G18060 of width 13.\n",
      "Using motif -AT2G18060 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9424\n",
      "#   Estimated pi_0=0.95301\n",
      "Using motif +AT2G18160 of width 15.\n",
      "Using motif -AT2G18160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G18300 of width 12.\n",
      "Using motif -AT2G18300 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G18350 of width 15.\n",
      "Using motif -AT2G18350 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8426\n",
      "#   Estimated pi_0=0.84819\n",
      "Using motif +AT2G18380 of width 13.\n",
      "Using motif -AT2G18380 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961386\n",
      "#   Estimated pi_0=0.97439\n",
      "Using motif +AT2G20110 of width 15.\n",
      "Using motif -AT2G20110 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.808\n",
      "#   Estimated pi_0=0.812816\n",
      "Using motif +AT2G20180 of width 14.\n",
      "Using motif -AT2G20180 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G20400 of width 15.\n",
      "Using motif -AT2G20400 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9188\n",
      "#   Estimated pi_0=0.931\n",
      "Using motif +AT2G20570 of width 10.\n",
      "Using motif -AT2G20570 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8992\n",
      "#   Estimated pi_0=0.909355\n",
      "Using motif +AT2G21230 of width 10.\n",
      "Using motif -AT2G21230 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G21900 of width 12.\n",
      "Using motif -AT2G21900 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.947778\n",
      "#   Estimated pi_0=0.9525\n",
      "Using motif +AT2G22300 of width 9.\n",
      "Using motif -AT2G22300 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00362\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G22430 of width 11.\n",
      "Using motif -AT2G22430 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.853981\n",
      "#   Estimated pi_0=0.855472\n",
      "Using motif +AT2G22540 of width 15.\n",
      "Using motif -AT2G22540 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.886\n",
      "#   Estimated pi_0=0.886\n",
      "Using motif +AT2G22750 of width 17.\n",
      "Using motif -AT2G22750 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999286\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G23290 of width 14.\n",
      "Using motif -AT2G23290 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.950455\n",
      "#   Estimated pi_0=0.956\n",
      "Using motif +AT2G23320 of width 11.\n",
      "Using motif -AT2G23320 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.952871\n",
      "#   Estimated pi_0=0.961667\n",
      "Using motif +AT2G23340 of width 17.\n",
      "Using motif -AT2G23340 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G24260 of width 10.\n",
      "Using motif -AT2G24260 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9594\n",
      "#   Estimated pi_0=0.974656\n",
      "Using motif +AT2G24430 of width 18.\n",
      "Using motif -AT2G24430 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9312\n",
      "#   Estimated pi_0=0.934643\n",
      "Using motif +AT2G24570 of width 14.\n",
      "Using motif -AT2G24570 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9756\n",
      "#   Estimated pi_0=0.984457\n",
      "Using motif +AT2G25000 of width 11.\n",
      "Using motif -AT2G25000 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.986923\n",
      "#   Estimated pi_0=0.996751\n",
      "Using motif +AT2G26960 of width 11.\n",
      "Using motif -AT2G26960 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8828\n",
      "#   Estimated pi_0=0.887664\n",
      "Using motif +AT2G27300 of width 19.\n",
      "Using motif -AT2G27300 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.980769\n",
      "#   Estimated pi_0=0.996684\n",
      "Using motif +AT2G28550 of width 10.\n",
      "Using motif -AT2G28550 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971\n",
      "#   Estimated pi_0=0.97719\n",
      "Using motif +AT2G28810 of width 21.\n",
      "Using motif -AT2G28810 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 52 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.881333\n",
      "#   Estimated pi_0=0.888613\n",
      "Using motif +AT2G30250 of width 11.\n",
      "Using motif -AT2G30250 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.941165\n",
      "#   Estimated pi_0=0.955\n",
      "Using motif +AT2G30340 of width 17.\n",
      "Using motif -AT2G30340 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G30590 of width 11.\n",
      "Using motif -AT2G30590 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9334\n",
      "#   Estimated pi_0=0.947273\n",
      "Using motif +AT2G31220 of width 14.\n",
      "Using motif -AT2G31220 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G31230 of width 15.\n",
      "Using motif -AT2G31230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0008\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G32460 of width 15.\n",
      "Using motif -AT2G32460 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.965743\n",
      "#   Estimated pi_0=0.967846\n",
      "Using motif +AT2G33550 of width 18.\n",
      "Using motif -AT2G33550 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9054\n",
      "#   Estimated pi_0=0.91\n",
      "Using motif +AT2G33710 of width 15.\n",
      "Using motif -AT2G33710 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G33810 of width 8.\n",
      "Using motif -AT2G33810 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00332\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G33860 of width 10.\n",
      "Using motif -AT2G33860 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G35530 of width 15.\n",
      "Using motif -AT2G35530 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G35700 of width 15.\n",
      "Using motif -AT2G35700 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G36010 of width 14.\n",
      "Using motif -AT2G36010 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G36270 of width 18.\n",
      "Using motif -AT2G36270 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00081\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G36610 of width 13.\n",
      "Using motif -AT2G36610 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8\n",
      "#   Estimated pi_0=0.8\n",
      "Using motif +AT2G37590 of width 15.\n",
      "Using motif -AT2G37590 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 32 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.873725\n",
      "#   Estimated pi_0=0.878\n",
      "Using motif +AT2G38090 of width 21.\n",
      "Using motif -AT2G38090 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8358\n",
      "#   Estimated pi_0=0.848545\n",
      "Using motif +AT2G38300 of width 11.\n",
      "Using motif -AT2G38300 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9138\n",
      "#   Estimated pi_0=0.927377\n",
      "Using motif +AT2G38340 of width 17.\n",
      "Using motif -AT2G38340 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G38470 of width 11.\n",
      "Using motif -AT2G38470 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.947611\n",
      "#   Estimated pi_0=0.951094\n",
      "Using motif +AT2G38880 of width 9.\n",
      "Using motif -AT2G38880 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.980828\n",
      "#   Estimated pi_0=0.984809\n",
      "Using motif +AT2G40220 of width 10.\n",
      "Using motif -AT2G40220 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G40260 of width 15.\n",
      "Using motif -AT2G40260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.849307\n",
      "#   Estimated pi_0=0.849307\n",
      "Using motif +AT2G40340 of width 21.\n",
      "Using motif -AT2G40340 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999698\n",
      "#   Estimated pi_0=0.999698\n",
      "Using motif +AT2G40620 of width 11.\n",
      "Using motif -AT2G40620 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G40740 of width 12.\n",
      "Using motif -AT2G40740 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.968235\n",
      "#   Estimated pi_0=0.983879\n",
      "Using motif +AT2G41690 of width 15.\n",
      "Using motif -AT2G41690 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998985\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G41835 of width 7.\n",
      "Using motif -AT2G41835 of width 7.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921119\n",
      "#   Estimated pi_0=0.921119\n",
      "Using motif +AT2G42200 of width 11.\n",
      "Using motif -AT2G42200 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9822\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G42280 of width 11.\n",
      "Using motif -AT2G42280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.996033\n",
      "#   Estimated pi_0=0.998995\n",
      "Using motif +AT2G42400 of width 12.\n",
      "Using motif -AT2G42400 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G42430 of width 15.\n",
      "Using motif -AT2G42430 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G43000 of width 20.\n",
      "Using motif -AT2G43000 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00302\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G43010 of width 9.\n",
      "Using motif -AT2G43010 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0001\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G44745 of width 10.\n",
      "Using motif -AT2G44745 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0008\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G44840 of width 15.\n",
      "Using motif -AT2G44840 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G44940 of width 15.\n",
      "Using motif -AT2G44940 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G45190 of width 10.\n",
      "Using motif -AT2G45190 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8126\n",
      "#   Estimated pi_0=0.816505\n",
      "Using motif +AT2G45410 of width 15.\n",
      "Using motif -AT2G45410 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G45420 of width 19.\n",
      "Using motif -AT2G45420 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G45650 of width 19.\n",
      "Using motif -AT2G45650 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.768\n",
      "#   Estimated pi_0=0.770891\n",
      "Using motif +AT2G45660 of width 21.\n",
      "Using motif -AT2G45660 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 53 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.960787\n",
      "#   Estimated pi_0=0.962674\n",
      "Using motif +AT2G45680 of width 21.\n",
      "Using motif -AT2G45680 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G46130 of width 14.\n",
      "Using motif -AT2G46130 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.97604\n",
      "#   Estimated pi_0=0.987053\n",
      "Using motif +AT2G46270 of width 15.\n",
      "Using motif -AT2G46270 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G46400 of width 11.\n",
      "Using motif -AT2G46400 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.970693\n",
      "#   Estimated pi_0=0.98\n",
      "Using motif +AT2G46510 of width 8.\n",
      "Using motif -AT2G46510 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G46590 of width 11.\n",
      "Using motif -AT2G46590 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8842\n",
      "#   Estimated pi_0=0.899483\n",
      "Using motif +AT2G46680 of width 13.\n",
      "Using motif -AT2G46680 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.866\n",
      "#   Estimated pi_0=0.872642\n",
      "Using motif +AT2G46770 of width 17.\n",
      "Using motif -AT2G46770 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9168\n",
      "#   Estimated pi_0=0.929649\n",
      "Using motif +AT2G46830 of width 8.\n",
      "Using motif -AT2G46830 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.868317\n",
      "#   Estimated pi_0=0.868627\n",
      "Using motif +AT2G47070 of width 10.\n",
      "Using motif -AT2G47070 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00291\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G47260 of width 8.\n",
      "Using motif -AT2G47260 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.982769\n",
      "#   Estimated pi_0=0.986554\n",
      "Using motif +AT2G47460 of width 10.\n",
      "Using motif -AT2G47460 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9356\n",
      "#   Estimated pi_0=0.936238\n",
      "Using motif +AT3G01220 of width 11.\n",
      "Using motif -AT3G01220 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8132\n",
      "#   Estimated pi_0=0.82283\n",
      "Using motif +AT3G01470 of width 11.\n",
      "Using motif -AT3G01470 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8158\n",
      "#   Estimated pi_0=0.8158\n",
      "Using motif +AT3G01530 of width 14.\n",
      "Using motif -AT3G01530 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9308\n",
      "#   Estimated pi_0=0.932389\n",
      "Using motif +AT3G01970 of width 12.\n",
      "Using motif -AT3G01970 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937426\n",
      "#   Estimated pi_0=0.946333\n",
      "Using motif +AT3G02150 of width 13.\n",
      "Using motif -AT3G02150 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G02940 of width 18.\n",
      "Using motif -AT3G02940 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9314\n",
      "#   Estimated pi_0=0.938966\n",
      "Using motif +AT3G02990 of width 13.\n",
      "Using motif -AT3G02990 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9542\n",
      "#   Estimated pi_0=0.963393\n",
      "Using motif +AT3G03200 of width 16.\n",
      "Using motif -AT3G03200 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.964571\n",
      "#   Estimated pi_0=0.975188\n",
      "Using motif +AT3G03660 of width 11.\n",
      "Using motif -AT3G03660 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.906733\n",
      "#   Estimated pi_0=0.917143\n",
      "Using motif +AT3G04030 of width 14.\n",
      "Using motif -AT3G04030 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9032\n",
      "#   Estimated pi_0=0.909815\n",
      "Using motif +AT3G04060 of width 21.\n",
      "Using motif -AT3G04060 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8994\n",
      "#   Estimated pi_0=0.902476\n",
      "Using motif +AT3G04070 of width 15.\n",
      "Using motif -AT3G04070 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.916275\n",
      "#   Estimated pi_0=0.925\n",
      "Using motif +AT3G06490 of width 15.\n",
      "Using motif -AT3G06490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9462\n",
      "#   Estimated pi_0=0.954454\n",
      "Using motif +AT3G06740 of width 19.\n",
      "Using motif -AT3G06740 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.881765\n",
      "#   Estimated pi_0=0.8925\n",
      "Using motif +AT3G08500 of width 11.\n",
      "Using motif -AT3G08500 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.992935\n",
      "#   Estimated pi_0=0.998593\n",
      "Using motif +AT3G09230 of width 19.\n",
      "Using motif -AT3G09230 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9474\n",
      "#   Estimated pi_0=0.9474\n",
      "Using motif +AT3G09600 of width 12.\n",
      "Using motif -AT3G09600 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.84\n",
      "#   Estimated pi_0=0.843689\n",
      "Using motif +AT3G09735 of width 15.\n",
      "Using motif -AT3G09735 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999394\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT3G10030 of width 14.\n",
      "Using motif -AT3G10030 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.925631\n",
      "#   Estimated pi_0=0.933393\n",
      "Using motif +AT3G10113 of width 15.\n",
      "Using motif -AT3G10113 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.875882\n",
      "#   Estimated pi_0=0.88625\n",
      "Using motif +AT3G10480 of width 16.\n",
      "Using motif -AT3G10480 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.981\n",
      "#   Estimated pi_0=0.995957\n",
      "Using motif +AT3G10490 of width 20.\n",
      "Using motif -AT3G10490 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.963529\n",
      "#   Estimated pi_0=0.980373\n",
      "Using motif +AT3G10500 of width 17.\n",
      "Using motif -AT3G10500 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9116\n",
      "#   Estimated pi_0=0.921835\n",
      "Using motif +AT3G10580 of width 15.\n",
      "Using motif -AT3G10580 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7838\n",
      "#   Estimated pi_0=0.789216\n",
      "Using motif +AT3G10800 of width 15.\n",
      "Using motif -AT3G10800 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G11020 of width 21.\n",
      "Using motif -AT3G11020 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G11280 of width 14.\n",
      "Using motif -AT3G11280 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9224\n",
      "#   Estimated pi_0=0.935806\n",
      "Using motif +AT3G11440 of width 13.\n",
      "Using motif -AT3G11440 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9164\n",
      "#   Estimated pi_0=0.928224\n",
      "Using motif +AT3G12130 of width 14.\n",
      "Using motif -AT3G12130 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913462\n",
      "#   Estimated pi_0=0.926613\n",
      "Using motif +AT3G12250 of width 15.\n",
      "Using motif -AT3G12250 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9518\n",
      "#   Estimated pi_0=0.967179\n",
      "Using motif +AT3G12720 of width 21.\n",
      "Using motif -AT3G12720 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913725\n",
      "#   Estimated pi_0=0.91463\n",
      "Using motif +AT3G12730 of width 14.\n",
      "Using motif -AT3G12730 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917864\n",
      "#   Estimated pi_0=0.929043\n",
      "Using motif +AT3G12820 of width 16.\n",
      "Using motif -AT3G12820 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.936\n",
      "#   Estimated pi_0=0.93802\n",
      "Using motif +AT3G12910 of width 17.\n",
      "Using motif -AT3G12910 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G13040 of width 9.\n",
      "Using motif -AT3G13040 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940345\n",
      "#   Estimated pi_0=0.957093\n",
      "Using motif +AT3G13810 of width 21.\n",
      "Using motif -AT3G13810 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 42 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940927\n",
      "#   Estimated pi_0=0.94481\n",
      "Using motif +AT3G14180 of width 11.\n",
      "Using motif -AT3G14180 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15030 of width 8.\n",
      "Using motif -AT3G15030 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00162\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15170 of width 16.\n",
      "Using motif -AT3G15170 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.963301\n",
      "#   Estimated pi_0=0.981955\n",
      "Using motif +AT3G15210 of width 15.\n",
      "Using motif -AT3G15210 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15270 of width 12.\n",
      "Using motif -AT3G15270 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15500 of width 21.\n",
      "Using motif -AT3G15500 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940792\n",
      "#   Estimated pi_0=0.951402\n",
      "Using motif +AT3G15510 of width 17.\n",
      "Using motif -AT3G15510 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.983\n",
      "#   Estimated pi_0=0.999196\n",
      "Using motif +AT3G16280 of width 14.\n",
      "Using motif -AT3G16280 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G16770 of width 8.\n",
      "Using motif -AT3G16770 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G17730 of width 17.\n",
      "Using motif -AT3G17730 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973861\n",
      "#   Estimated pi_0=0.981955\n",
      "Using motif +AT3G18400 of width 16.\n",
      "Using motif -AT3G18400 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.953271\n",
      "#   Estimated pi_0=0.959153\n",
      "Using motif +AT3G20310 of width 21.\n",
      "Using motif -AT3G20310 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0008\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G20770 of width 15.\n",
      "Using motif -AT3G20770 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917228\n",
      "#   Estimated pi_0=0.9305\n",
      "Using motif +AT3G20840 of width 19.\n",
      "Using motif -AT3G20840 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9594\n",
      "#   Estimated pi_0=0.968037\n",
      "Using motif +AT3G21175 of width 15.\n",
      "Using motif -AT3G21175 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.896275\n",
      "#   Estimated pi_0=0.90672\n",
      "Using motif +AT3G22170 of width 12.\n",
      "Using motif -AT3G22170 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G22760 of width 8.\n",
      "Using motif -AT3G22760 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8044\n",
      "#   Estimated pi_0=0.806337\n",
      "Using motif +AT3G22780 of width 15.\n",
      "Using motif -AT3G22780 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8008\n",
      "#   Estimated pi_0=0.807963\n",
      "Using motif +AT3G22830 of width 15.\n",
      "Using motif -AT3G22830 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.99868\n",
      "#   Estimated pi_0=0.998894\n",
      "Using motif +AT3G23210 of width 23.\n",
      "Using motif -AT3G23210 of width 23.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G23220 of width 15.\n",
      "Using motif -AT3G23220 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G23240 of width 28.\n",
      "Using motif -AT3G23240 of width 28.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G23250 of width 21.\n",
      "Using motif -AT3G23250 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9192\n",
      "#   Estimated pi_0=0.92368\n",
      "Using motif +AT3G23690 of width 15.\n",
      "Using motif -AT3G23690 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G24050 of width 15.\n",
      "Using motif -AT3G24050 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9534\n",
      "#   Estimated pi_0=0.970299\n",
      "Using motif +AT3G24120 of width 8.\n",
      "Using motif -AT3G24120 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.941\n",
      "#   Estimated pi_0=0.944808\n",
      "Using motif +AT3G24520 of width 11.\n",
      "Using motif -AT3G24520 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.988119\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G24650 of width 9.\n",
      "Using motif -AT3G24650 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G25730 of width 8.\n",
      "Using motif -AT3G25730 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948397\n",
      "#   Estimated pi_0=0.950435\n",
      "Using motif +AT3G25990 of width 21.\n",
      "Using motif -AT3G25990 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.960855\n",
      "#   Estimated pi_0=0.970219\n",
      "Using motif +AT3G26620 of width 15.\n",
      "Using motif -AT3G26620 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G26790 of width 15.\n",
      "Using motif -AT3G26790 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.980385\n",
      "#   Estimated pi_0=0.995979\n",
      "Using motif +AT3G27010 of width 21.\n",
      "Using motif -AT3G27010 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G27785 of width 17.\n",
      "Using motif -AT3G27785 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9062\n",
      "#   Estimated pi_0=0.907525\n",
      "Using motif +AT3G28910 of width 15.\n",
      "Using motif -AT3G28910 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9152\n",
      "#   Estimated pi_0=0.925\n",
      "Using motif +AT3G28920 of width 15.\n",
      "Using motif -AT3G28920 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8462\n",
      "#   Estimated pi_0=0.851346\n",
      "Using motif +AT3G30210 of width 14.\n",
      "Using motif -AT3G30210 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9132\n",
      "#   Estimated pi_0=0.924035\n",
      "Using motif +AT3G30530 of width 12.\n",
      "Using motif -AT3G30530 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9802\n",
      "#   Estimated pi_0=0.999698\n",
      "Using motif +AT3G45150 of width 11.\n",
      "Using motif -AT3G45150 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G45170 of width 15.\n",
      "Using motif -AT3G45170 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.972\n",
      "#   Estimated pi_0=0.987674\n",
      "Using motif +AT3G45610 of width 19.\n",
      "Using motif -AT3G45610 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 41 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.870297\n",
      "#   Estimated pi_0=0.877069\n",
      "Using motif +AT3G46070 of width 15.\n",
      "Using motif -AT3G46070 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9186\n",
      "#   Estimated pi_0=0.930534\n",
      "Using motif +AT3G46090 of width 13.\n",
      "Using motif -AT3G46090 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G46640 of width 14.\n",
      "Using motif -AT3G46640 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9096\n",
      "#   Estimated pi_0=0.914019\n",
      "Using motif +AT3G47500 of width 19.\n",
      "Using motif -AT3G47500 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 35 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.881905\n",
      "#   Estimated pi_0=0.888929\n",
      "Using motif +AT3G47600 of width 19.\n",
      "Using motif -AT3G47600 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9302\n",
      "#   Estimated pi_0=0.937913\n",
      "Using motif +AT3G47620 of width 20.\n",
      "Using motif -AT3G47620 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G48160 of width 21.\n",
      "Using motif -AT3G48160 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993069\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G48430 of width 12.\n",
      "Using motif -AT3G48430 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.959322\n",
      "#   Estimated pi_0=0.967059\n",
      "Using motif +AT3G49530 of width 15.\n",
      "Using motif -AT3G49530 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917\n",
      "#   Estimated pi_0=0.921176\n",
      "Using motif +AT3G49690 of width 13.\n",
      "Using motif -AT3G49690 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G49930 of width 7.\n",
      "Using motif -AT3G49930 of width 7.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937097\n",
      "#   Estimated pi_0=0.944531\n",
      "Using motif +AT3G50060 of width 15.\n",
      "Using motif -AT3G50060 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9354\n",
      "#   Estimated pi_0=0.936667\n",
      "Using motif +AT3G50260 of width 15.\n",
      "Using motif -AT3G50260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G50410 of width 21.\n",
      "Using motif -AT3G50410 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 54 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.906863\n",
      "#   Estimated pi_0=0.913394\n",
      "Using motif +AT3G50700 of width 15.\n",
      "Using motif -AT3G50700 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 40 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.912358\n",
      "#   Estimated pi_0=0.917564\n",
      "Using motif +AT3G51060 of width 10.\n",
      "Using motif -AT3G51060 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G51080 of width 13.\n",
      "Using motif -AT3G51080 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.979683\n",
      "#   Estimated pi_0=0.986667\n",
      "Using motif +AT3G52440 of width 11.\n",
      "Using motif -AT3G52440 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9006\n",
      "#   Estimated pi_0=0.903107\n",
      "Using motif +AT3G53200 of width 15.\n",
      "Using motif -AT3G53200 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8694\n",
      "#   Estimated pi_0=0.8694\n",
      "Using motif +AT3G54340 of width 15.\n",
      "Using motif -AT3G54340 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8834\n",
      "#   Estimated pi_0=0.895752\n",
      "Using motif +AT3G54990 of width 8.\n",
      "Using motif -AT3G54990 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G55370 of width 21.\n",
      "Using motif -AT3G55370 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 56 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8486\n",
      "#   Estimated pi_0=0.859853\n",
      "Using motif +AT3G56400 of width 13.\n",
      "Using motif -AT3G56400 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961\n",
      "#   Estimated pi_0=0.970839\n",
      "Using motif +AT3G56850 of width 15.\n",
      "Using motif -AT3G56850 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999697\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G57230 of width 15.\n",
      "Using motif -AT3G57230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.907\n",
      "#   Estimated pi_0=0.914579\n",
      "Using motif +AT3G57600 of width 11.\n",
      "Using motif -AT3G57600 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0003\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G57920 of width 14.\n",
      "Using motif -AT3G57920 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96463\n",
      "#   Estimated pi_0=0.972\n",
      "Using motif +AT3G58630 of width 15.\n",
      "Using motif -AT3G58630 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G58780 of width 18.\n",
      "Using motif -AT3G58780 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9054\n",
      "#   Estimated pi_0=0.912692\n",
      "Using motif +AT3G59060 of width 8.\n",
      "Using motif -AT3G59060 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0003\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G60030 of width 8.\n",
      "Using motif -AT3G60030 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G60490 of width 14.\n",
      "Using motif -AT3G60490 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G60530 of width 15.\n",
      "Using motif -AT3G60530 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.980588\n",
      "#   Estimated pi_0=0.9925\n",
      "Using motif +AT3G60580 of width 8.\n",
      "Using motif -AT3G60580 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.965149\n",
      "#   Estimated pi_0=0.970267\n",
      "Using motif +AT3G61120 of width 18.\n",
      "Using motif -AT3G61120 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8008\n",
      "#   Estimated pi_0=0.807184\n",
      "Using motif +AT3G61150 of width 11.\n",
      "Using motif -AT3G61150 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7816\n",
      "#   Estimated pi_0=0.785941\n",
      "Using motif +AT3G61250 of width 15.\n",
      "Using motif -AT3G61250 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9182\n",
      "#   Estimated pi_0=0.922281\n",
      "Using motif +AT3G61890 of width 10.\n",
      "Using motif -AT3G61890 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.890891\n",
      "#   Estimated pi_0=0.892157\n",
      "Using motif +AT3G61910 of width 13.\n",
      "Using motif -AT3G61910 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9816\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G62420 of width 15.\n",
      "Using motif -AT3G62420 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999394\n",
      "#   Estimated pi_0=0.999497\n",
      "Using motif +AT4G00050 of width 8.\n",
      "Using motif -AT4G00050 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G00250 of width 13.\n",
      "Using motif -AT4G00250 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.907327\n",
      "#   Estimated pi_0=0.912252\n",
      "Using motif +AT4G00270 of width 8.\n",
      "Using motif -AT4G00270 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G00730 of width 11.\n",
      "Using motif -AT4G00730 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8014\n",
      "#   Estimated pi_0=0.806476\n",
      "Using motif +AT4G01250 of width 13.\n",
      "Using motif -AT4G01250 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917426\n",
      "#   Estimated pi_0=0.920374\n",
      "Using motif +AT4G01280 of width 11.\n",
      "Using motif -AT4G01280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.81\n",
      "#   Estimated pi_0=0.81\n",
      "Using motif +AT4G01500 of width 16.\n",
      "Using motif -AT4G01500 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G01540 of width 15.\n",
      "Using motif -AT4G01540 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997045\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT4G01550 of width 15.\n",
      "Using motif -AT4G01550 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997379\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G01680 of width 14.\n",
      "Using motif -AT4G01680 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.995847\n",
      "#   Estimated pi_0=0.999091\n",
      "Using motif +AT4G01720 of width 19.\n",
      "Using motif -AT4G01720 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.931698\n",
      "#   Estimated pi_0=0.947647\n",
      "Using motif +AT4G04450 of width 21.\n",
      "Using motif -AT4G04450 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.958\n",
      "#   Estimated pi_0=0.977037\n",
      "Using motif +AT4G05100 of width 14.\n",
      "Using motif -AT4G05100 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9322\n",
      "#   Estimated pi_0=0.941875\n",
      "Using motif +AT4G06746 of width 19.\n",
      "Using motif -AT4G06746 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G10350 of width 21.\n",
      "Using motif -AT4G10350 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.895644\n",
      "#   Estimated pi_0=0.903585\n",
      "Using motif +AT4G11660 of width 14.\n",
      "Using motif -AT4G11660 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971261\n",
      "#   Estimated pi_0=0.993505\n",
      "Using motif +AT4G14410 of width 10.\n",
      "Using motif -AT4G14410 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G14770 of width 15.\n",
      "Using motif -AT4G14770 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.827885\n",
      "#   Estimated pi_0=0.834414\n",
      "Using motif +AT4G15090 of width 15.\n",
      "Using motif -AT4G15090 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G16110 of width 10.\n",
      "Using motif -AT4G16110 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.929273\n",
      "#   Estimated pi_0=0.93569\n",
      "Using motif +AT4G16150 of width 18.\n",
      "Using motif -AT4G16150 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997524\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G16430 of width 8.\n",
      "Using motif -AT4G16430 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G16750 of width 15.\n",
      "Using motif -AT4G16750 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G17460 of width 10.\n",
      "Using motif -AT4G17460 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.895962\n",
      "#   Estimated pi_0=0.908673\n",
      "Using motif +AT4G17490 of width 21.\n",
      "Using motif -AT4G17490 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G17785 of width 14.\n",
      "Using motif -AT4G17785 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8608\n",
      "#   Estimated pi_0=0.866415\n",
      "Using motif +AT4G17880 of width 8.\n",
      "Using motif -AT4G17880 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993333\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G17980 of width 15.\n",
      "Using motif -AT4G17980 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9492\n",
      "#   Estimated pi_0=0.958525\n",
      "Using motif +AT4G18170 of width 13.\n",
      "Using motif -AT4G18170 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966847\n",
      "#   Estimated pi_0=0.988315\n",
      "Using motif +AT4G18390 of width 10.\n",
      "Using motif -AT4G18390 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G18450 of width 17.\n",
      "Using motif -AT4G18450 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G18770 of width 15.\n",
      "Using motif -AT4G18770 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8994\n",
      "#   Estimated pi_0=0.913982\n",
      "Using motif +AT4G18880 of width 19.\n",
      "Using motif -AT4G18880 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9036\n",
      "#   Estimated pi_0=0.911171\n",
      "Using motif +AT4G18890 of width 11.\n",
      "Using motif -AT4G18890 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G18960 of width 19.\n",
      "Using motif -AT4G18960 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8904\n",
      "#   Estimated pi_0=0.90339\n",
      "Using motif +AT4G21030 of width 19.\n",
      "Using motif -AT4G21030 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 39 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8594\n",
      "#   Estimated pi_0=0.8594\n",
      "Using motif +AT4G21040 of width 14.\n",
      "Using motif -AT4G21040 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8736\n",
      "#   Estimated pi_0=0.885766\n",
      "Using motif +AT4G21080 of width 14.\n",
      "Using motif -AT4G21080 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.864752\n",
      "#   Estimated pi_0=0.868113\n",
      "Using motif +AT4G21330 of width 14.\n",
      "Using motif -AT4G21330 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.963774\n",
      "#   Estimated pi_0=0.981842\n",
      "Using motif +AT4G22070 of width 19.\n",
      "Using motif -AT4G22070 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.960354\n",
      "#   Estimated pi_0=0.979034\n",
      "Using motif +AT4G23550 of width 11.\n",
      "Using motif -AT4G23550 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9594\n",
      "#   Estimated pi_0=0.972\n",
      "Using motif +AT4G23750 of width 8.\n",
      "Using motif -AT4G23750 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00291\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G24020 of width 15.\n",
      "Using motif -AT4G24020 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.992277\n",
      "#   Estimated pi_0=0.999799\n",
      "Using motif +AT4G24240 of width 14.\n",
      "Using motif -AT4G24240 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9016\n",
      "#   Estimated pi_0=0.919174\n",
      "Using motif +AT4G24470 of width 15.\n",
      "Using motif -AT4G24470 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.929703\n",
      "#   Estimated pi_0=0.932727\n",
      "Using motif +AT4G25470 of width 15.\n",
      "Using motif -AT4G25470 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G25480 of width 14.\n",
      "Using motif -AT4G25480 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G25490 of width 12.\n",
      "Using motif -AT4G25490 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G26640 of width 13.\n",
      "Using motif -AT4G26640 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9802\n",
      "#   Estimated pi_0=0.992903\n",
      "Using motif +AT4G27950 of width 21.\n",
      "Using motif -AT4G27950 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0003\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G28110 of width 15.\n",
      "Using motif -AT4G28110 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.914369\n",
      "#   Estimated pi_0=0.926129\n",
      "Using motif +AT4G28140 of width 20.\n",
      "Using motif -AT4G28140 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G28500 of width 15.\n",
      "Using motif -AT4G28500 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933529\n",
      "#   Estimated pi_0=0.943934\n",
      "Using motif +AT4G29000 of width 10.\n",
      "Using motif -AT4G29000 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.857624\n",
      "#   Estimated pi_0=0.870973\n",
      "Using motif +AT4G29230 of width 15.\n",
      "Using motif -AT4G29230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9198\n",
      "#   Estimated pi_0=0.938226\n",
      "Using motif +AT4G30080 of width 21.\n",
      "Using motif -AT4G30080 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.851569\n",
      "#   Estimated pi_0=0.861368\n",
      "Using motif +AT4G30980 of width 15.\n",
      "Using motif -AT4G30980 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G31060 of width 20.\n",
      "Using motif -AT4G31060 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G31550 of width 14.\n",
      "Using motif -AT4G31550 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944902\n",
      "#   Estimated pi_0=0.949091\n",
      "Using motif +AT4G31610 of width 10.\n",
      "Using motif -AT4G31610 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9344\n",
      "#   Estimated pi_0=0.947903\n",
      "Using motif +AT4G31800 of width 11.\n",
      "Using motif -AT4G31800 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938\n",
      "#   Estimated pi_0=0.949313\n",
      "Using motif +AT4G31920 of width 8.\n",
      "Using motif -AT4G31920 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.936311\n",
      "#   Estimated pi_0=0.936311\n",
      "Using motif +AT4G32730 of width 15.\n",
      "Using motif -AT4G32730 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.95\n",
      "#   Estimated pi_0=0.957842\n",
      "Using motif +AT4G32800 of width 16.\n",
      "Using motif -AT4G32800 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G32890 of width 11.\n",
      "Using motif -AT4G32890 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.980946\n",
      "#   Estimated pi_0=0.985432\n",
      "Using motif +AT4G33280 of width 21.\n",
      "Using motif -AT4G33280 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913069\n",
      "#   Estimated pi_0=0.924098\n",
      "Using motif +AT4G34000 of width 8.\n",
      "Using motif -AT4G34000 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT4G34410 of width 15.\n",
      "Using motif -AT4G34410 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G34590 of width 15.\n",
      "Using motif -AT4G34590 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0008\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G35550 of width 10.\n",
      "Using motif -AT4G35550 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.89549\n",
      "#   Estimated pi_0=0.89549\n",
      "Using motif +AT4G35580 of width 9.\n",
      "Using motif -AT4G35580 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.841651\n",
      "#   Estimated pi_0=0.841651\n",
      "Using motif +AT4G36160 of width 15.\n",
      "Using motif -AT4G36160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969333\n",
      "#   Estimated pi_0=0.993176\n",
      "Using motif +AT4G36620 of width 11.\n",
      "Using motif -AT4G36620 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.996639\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G36730 of width 10.\n",
      "Using motif -AT4G36730 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G36740 of width 21.\n",
      "Using motif -AT4G36740 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7612\n",
      "#   Estimated pi_0=0.770291\n",
      "Using motif +AT4G36780 of width 11.\n",
      "Using motif -AT4G36780 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G36900 of width 10.\n",
      "Using motif -AT4G36900 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00352\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G36930 of width 9.\n",
      "Using motif -AT4G36930 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G37180 of width 11.\n",
      "Using motif -AT4G37180 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.889\n",
      "#   Estimated pi_0=0.892593\n",
      "Using motif +AT4G37260 of width 11.\n",
      "Using motif -AT4G37260 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8982\n",
      "#   Estimated pi_0=0.8982\n",
      "Using motif +AT4G37750 of width 21.\n",
      "Using motif -AT4G37750 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.992549\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G37790 of width 10.\n",
      "Using motif -AT4G37790 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9338\n",
      "#   Estimated pi_0=0.957333\n",
      "Using motif +AT4G38000 of width 28.\n",
      "Using motif -AT4G38000 of width 28.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 60 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.828824\n",
      "#   Estimated pi_0=0.839844\n",
      "Using motif +AT4G38620 of width 15.\n",
      "Using motif -AT4G38620 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9122\n",
      "#   Estimated pi_0=0.920917\n",
      "Using motif +AT4G38910 of width 30.\n",
      "Using motif -AT4G38910 of width 30.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 42 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.946282\n",
      "#   Estimated pi_0=0.950857\n",
      "Using motif +AT4G40060 of width 8.\n",
      "Using motif -AT4G40060 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.867525\n",
      "#   Estimated pi_0=0.870811\n",
      "Using motif +AT5G01380 of width 17.\n",
      "Using motif -AT5G01380 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9042\n",
      "#   Estimated pi_0=0.9042\n",
      "Using motif +AT5G01900 of width 8.\n",
      "Using motif -AT5G01900 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961034\n",
      "#   Estimated pi_0=0.961085\n",
      "Using motif +AT5G02320 of width 15.\n",
      "Using motif -AT5G02320 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921\n",
      "#   Estimated pi_0=0.927664\n",
      "Using motif +AT5G02460 of width 21.\n",
      "Using motif -AT5G02460 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 56 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.826733\n",
      "#   Estimated pi_0=0.830857\n",
      "Using motif +AT5G02840 of width 10.\n",
      "Using motif -AT5G02840 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8444\n",
      "#   Estimated pi_0=0.853333\n",
      "Using motif +AT5G03150 of width 20.\n",
      "Using motif -AT5G03150 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 38 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9178\n",
      "#   Estimated pi_0=0.92971\n",
      "Using motif +AT5G03790 of width 13.\n",
      "Using motif -AT5G03790 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7894\n",
      "#   Estimated pi_0=0.793462\n",
      "Using motif +AT5G04340 of width 10.\n",
      "Using motif -AT5G04340 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.905\n",
      "#   Estimated pi_0=0.911111\n",
      "Using motif +AT5G04390 of width 11.\n",
      "Using motif -AT5G04390 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.946078\n",
      "#   Estimated pi_0=0.957414\n",
      "Using motif +AT5G04410 of width 12.\n",
      "Using motif -AT5G04410 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961524\n",
      "#   Estimated pi_0=0.97507\n",
      "Using motif +AT5G04760 of width 10.\n",
      "Using motif -AT5G04760 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.951485\n",
      "#   Estimated pi_0=0.959664\n",
      "Using motif +AT5G05090 of width 11.\n",
      "Using motif -AT5G05090 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.930476\n",
      "#   Estimated pi_0=0.934259\n",
      "Using motif +AT5G05550 of width 15.\n",
      "Using motif -AT5G05550 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G05790 of width 15.\n",
      "Using motif -AT5G05790 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8946\n",
      "#   Estimated pi_0=0.900377\n",
      "Using motif +AT5G06100 of width 11.\n",
      "Using motif -AT5G06100 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8914\n",
      "#   Estimated pi_0=0.900545\n",
      "Using motif +AT5G06839 of width 15.\n",
      "Using motif -AT5G06839 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96466\n",
      "#   Estimated pi_0=0.97009\n",
      "Using motif +AT5G06950 of width 20.\n",
      "Using motif -AT5G06950 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9504\n",
      "#   Estimated pi_0=0.96209\n",
      "Using motif +AT5G06960 of width 14.\n",
      "Using motif -AT5G06960 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9602\n",
      "#   Estimated pi_0=0.98529\n",
      "Using motif +AT5G07100 of width 13.\n",
      "Using motif -AT5G07100 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.977424\n",
      "#   Estimated pi_0=0.98659\n",
      "Using motif +AT5G07310 of width 15.\n",
      "Using motif -AT5G07310 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G07680 of width 21.\n",
      "Using motif -AT5G07680 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.869505\n",
      "#   Estimated pi_0=0.877838\n",
      "Using motif +AT5G08070 of width 11.\n",
      "Using motif -AT5G08070 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G08130 of width 14.\n",
      "Using motif -AT5G08130 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0003\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G08330 of width 14.\n",
      "Using motif -AT5G08330 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0007\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G08520 of width 13.\n",
      "Using motif -AT5G08520 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.875842\n",
      "#   Estimated pi_0=0.882243\n",
      "Using motif +AT5G09410 of width 12.\n",
      "Using motif -AT5G09410 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00312\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G10030 of width 11.\n",
      "Using motif -AT5G10030 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.99068\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G10140 of width 21.\n",
      "Using motif -AT5G10140 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 32 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9148\n",
      "#   Estimated pi_0=0.927969\n",
      "Using motif +AT5G10280 of width 10.\n",
      "Using motif -AT5G10280 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.982419\n",
      "#   Estimated pi_0=0.988619\n",
      "Using motif +AT5G10510 of width 15.\n",
      "Using motif -AT5G10510 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971881\n",
      "#   Estimated pi_0=0.991014\n",
      "Using motif +AT5G11260 of width 15.\n",
      "Using motif -AT5G11260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0007\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G11510 of width 15.\n",
      "Using motif -AT5G11510 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.945243\n",
      "#   Estimated pi_0=0.955536\n",
      "Using motif +AT5G12870 of width 10.\n",
      "Using motif -AT5G12870 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.982453\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G13080 of width 13.\n",
      "Using motif -AT5G13080 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.972115\n",
      "#   Estimated pi_0=0.979444\n",
      "Using motif +AT5G13180 of width 19.\n",
      "Using motif -AT5G13180 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.879604\n",
      "#   Estimated pi_0=0.883883\n",
      "Using motif +AT5G13790 of width 19.\n",
      "Using motif -AT5G13790 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.880198\n",
      "#   Estimated pi_0=0.885385\n",
      "Using motif +AT5G13910 of width 21.\n",
      "Using motif -AT5G13910 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G14340 of width 13.\n",
      "Using motif -AT5G14340 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.941165\n",
      "#   Estimated pi_0=0.957031\n",
      "Using motif +AT5G14960 of width 20.\n",
      "Using motif -AT5G14960 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9202\n",
      "#   Estimated pi_0=0.932069\n",
      "Using motif +AT5G15800 of width 18.\n",
      "Using motif -AT5G15800 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8668\n",
      "#   Estimated pi_0=0.870784\n",
      "Using motif +AT5G15830 of width 15.\n",
      "Using motif -AT5G15830 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G16600 of width 13.\n",
      "Using motif -AT5G16600 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.968679\n",
      "#   Estimated pi_0=0.977551\n",
      "Using motif +AT5G16820 of width 12.\n",
      "Using motif -AT5G16820 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999095\n",
      "#   Estimated pi_0=0.999095\n",
      "Using motif +AT5G17300 of width 14.\n",
      "Using motif -AT5G17300 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8366\n",
      "#   Estimated pi_0=0.846667\n",
      "Using motif +AT5G17430 of width 20.\n",
      "Using motif -AT5G17430 of width 20.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 57 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.901239\n",
      "#   Estimated pi_0=0.90873\n",
      "Using motif +AT5G17800 of width 15.\n",
      "Using motif -AT5G17800 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8874\n",
      "#   Estimated pi_0=0.891373\n",
      "Using motif +AT5G18090 of width 15.\n",
      "Using motif -AT5G18090 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940328\n",
      "#   Estimated pi_0=0.950145\n",
      "Using motif +AT5G18270 of width 19.\n",
      "Using motif -AT5G18270 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9006\n",
      "#   Estimated pi_0=0.905607\n",
      "Using motif +AT5G18450 of width 15.\n",
      "Using motif -AT5G18450 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G18560 of width 15.\n",
      "Using motif -AT5G18560 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G18830 of width 10.\n",
      "Using motif -AT5G18830 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G19790 of width 15.\n",
      "Using motif -AT5G19790 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G20240 of width 14.\n",
      "Using motif -AT5G20240 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 43 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933151\n",
      "#   Estimated pi_0=0.94131\n",
      "Using motif +AT5G22220 of width 10.\n",
      "Using motif -AT5G22220 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G22570 of width 10.\n",
      "Using motif -AT5G22570 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0001\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G22890 of width 18.\n",
      "Using motif -AT5G22890 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.932\n",
      "#   Estimated pi_0=0.938\n",
      "Using motif +AT5G22990 of width 15.\n",
      "Using motif -AT5G22990 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971\n",
      "#   Estimated pi_0=0.996452\n",
      "Using motif +AT5G23280 of width 11.\n",
      "Using motif -AT5G23280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G24110 of width 11.\n",
      "Using motif -AT5G24110 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966139\n",
      "#   Estimated pi_0=0.98773\n",
      "Using motif +AT5G25190 of width 19.\n",
      "Using motif -AT5G25190 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G25390 of width 21.\n",
      "Using motif -AT5G25390 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999192\n",
      "#   Estimated pi_0=0.999196\n",
      "Using motif +AT5G25475 of width 6.\n",
      "Using motif -AT5G25475 of width 6.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G25810 of width 15.\n",
      "Using motif -AT5G25810 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G25830 of width 13.\n",
      "Using motif -AT5G25830 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.959279\n",
      "#   Estimated pi_0=0.967903\n",
      "Using motif +AT5G26170 of width 13.\n",
      "Using motif -AT5G26170 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 33 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94972\n",
      "#   Estimated pi_0=0.955246\n",
      "Using motif +AT5G28300 of width 10.\n",
      "Using motif -AT5G28300 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.949\n",
      "#   Estimated pi_0=0.962569\n",
      "Using motif +AT5G28770 of width 12.\n",
      "Using motif -AT5G28770 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G29000 of width 15.\n",
      "Using motif -AT5G29000 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.974848\n",
      "#   Estimated pi_0=0.995714\n",
      "Using motif +AT5G37020 of width 9.\n",
      "Using motif -AT5G37020 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.963529\n",
      "#   Estimated pi_0=0.97396\n",
      "Using motif +AT5G38800 of width 12.\n",
      "Using motif -AT5G38800 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G38860 of width 18.\n",
      "Using motif -AT5G38860 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G39610 of width 15.\n",
      "Using motif -AT5G39610 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9302\n",
      "#   Estimated pi_0=0.936182\n",
      "Using motif +AT5G39660 of width 10.\n",
      "Using motif -AT5G39660 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.916316\n",
      "#   Estimated pi_0=0.926202\n",
      "Using motif +AT5G39760 of width 22.\n",
      "Using motif -AT5G39760 of width 22.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7774\n",
      "#   Estimated pi_0=0.783077\n",
      "Using motif +AT5G39820 of width 18.\n",
      "Using motif -AT5G39820 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G40330 of width 11.\n",
      "Using motif -AT5G40330 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9216\n",
      "#   Estimated pi_0=0.934915\n",
      "Using motif +AT5G41570 of width 14.\n",
      "Using motif -AT5G41570 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.943107\n",
      "#   Estimated pi_0=0.958649\n",
      "Using motif +AT5G42520 of width 21.\n",
      "Using motif -AT5G42520 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 38 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8812\n",
      "#   Estimated pi_0=0.899433\n",
      "Using motif +AT5G42630 of width 8.\n",
      "Using motif -AT5G42630 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.880185\n",
      "#   Estimated pi_0=0.885766\n",
      "Using motif +AT5G43410 of width 10.\n",
      "Using motif -AT5G43410 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G43840 of width 16.\n",
      "Using motif -AT5G43840 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.979478\n",
      "#   Estimated pi_0=0.995692\n",
      "Using motif +AT5G44210 of width 21.\n",
      "Using motif -AT5G44210 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G45300 of width 15.\n",
      "Using motif -AT5G45300 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G45580 of width 14.\n",
      "Using motif -AT5G45580 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9126\n",
      "#   Estimated pi_0=0.917593\n",
      "Using motif +AT5G46350 of width 15.\n",
      "Using motif -AT5G46350 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.909\n",
      "#   Estimated pi_0=0.911089\n",
      "Using motif +AT5G46590 of width 16.\n",
      "Using motif -AT5G46590 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9106\n",
      "#   Estimated pi_0=0.9106\n",
      "Using motif +AT5G46760 of width 8.\n",
      "Using motif -AT5G46760 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G46830 of width 12.\n",
      "Using motif -AT5G46830 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.98314\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G47220 of width 21.\n",
      "Using motif -AT5G47220 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G47230 of width 15.\n",
      "Using motif -AT5G47230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G47370 of width 11.\n",
      "Using motif -AT5G47370 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.835\n",
      "#   Estimated pi_0=0.840392\n",
      "Using motif +AT5G47390 of width 15.\n",
      "Using motif -AT5G47390 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8568\n",
      "#   Estimated pi_0=0.859038\n",
      "Using motif +AT5G47660 of width 15.\n",
      "Using motif -AT5G47660 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.83604\n",
      "#   Estimated pi_0=0.840583\n",
      "Using motif +AT5G48560 of width 8.\n",
      "Using motif -AT5G48560 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00291\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G48670 of width 6.\n",
      "Using motif -AT5G48670 of width 6.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G49300 of width 27.\n",
      "Using motif -AT5G49300 of width 27.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.914059\n",
      "#   Estimated pi_0=0.922373\n",
      "Using motif +AT5G49330 of width 10.\n",
      "Using motif -AT5G49330 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96902\n",
      "#   Estimated pi_0=0.995337\n",
      "Using motif +AT5G50570 of width 11.\n",
      "Using motif -AT5G50570 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9536\n",
      "#   Estimated pi_0=0.958302\n",
      "Using motif +AT5G51190 of width 15.\n",
      "Using motif -AT5G51190 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G51910 of width 10.\n",
      "Using motif -AT5G51910 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G51990 of width 13.\n",
      "Using motif -AT5G51990 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G52020 of width 15.\n",
      "Using motif -AT5G52020 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G52170 of width 11.\n",
      "Using motif -AT5G52170 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8004\n",
      "#   Estimated pi_0=0.8004\n",
      "Using motif +AT5G52660 of width 11.\n",
      "Using motif -AT5G52660 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.851\n",
      "#   Estimated pi_0=0.852941\n",
      "Using motif +AT5G52830 of width 13.\n",
      "Using motif -AT5G52830 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937822\n",
      "#   Estimated pi_0=0.947681\n",
      "Using motif +AT5G53950 of width 19.\n",
      "Using motif -AT5G53950 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.891089\n",
      "#   Estimated pi_0=0.896038\n",
      "Using motif +AT5G54230 of width 15.\n",
      "Using motif -AT5G54230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8972\n",
      "#   Estimated pi_0=0.907395\n",
      "Using motif +AT5G56110 of width 15.\n",
      "Using motif -AT5G56110 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8962\n",
      "#   Estimated pi_0=0.898584\n",
      "Using motif +AT5G56270 of width 8.\n",
      "Using motif -AT5G56270 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0002\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G56840 of width 15.\n",
      "Using motif -AT5G56840 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8768\n",
      "#   Estimated pi_0=0.890093\n",
      "Using motif +AT5G58080 of width 13.\n",
      "Using motif -AT5G58080 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948269\n",
      "#   Estimated pi_0=0.955372\n",
      "Using motif +AT5G58620 of width 6.\n",
      "Using motif -AT5G58620 of width 6.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G58850 of width 11.\n",
      "Using motif -AT5G58850 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.965905\n",
      "#   Estimated pi_0=0.965905\n",
      "Using motif +AT5G58900 of width 15.\n",
      "Using motif -AT5G58900 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.838\n",
      "#   Estimated pi_0=0.843429\n",
      "Using motif +AT5G59780 of width 10.\n",
      "Using motif -AT5G59780 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8976\n",
      "#   Estimated pi_0=0.903462\n",
      "Using motif +AT5G60120 of width 10.\n",
      "Using motif -AT5G60120 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.903168\n",
      "#   Estimated pi_0=0.903168\n",
      "Using motif +AT5G60130 of width 19.\n",
      "Using motif -AT5G60130 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 39 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7824\n",
      "#   Estimated pi_0=0.7875\n",
      "Using motif +AT5G60850 of width 14.\n",
      "Using motif -AT5G60850 of width 14.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 32 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.852549\n",
      "#   Estimated pi_0=0.852816\n",
      "Using motif +AT5G60970 of width 8.\n",
      "Using motif -AT5G60970 of width 8.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G61270 of width 15.\n",
      "Using motif -AT5G61270 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G61600 of width 19.\n",
      "Using motif -AT5G61600 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G61620 of width 15.\n",
      "Using motif -AT5G61620 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8544\n",
      "#   Estimated pi_0=0.8544\n",
      "Using motif +AT5G61850 of width 19.\n",
      "Using motif -AT5G61850 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9224\n",
      "#   Estimated pi_0=0.924272\n",
      "Using motif +AT5G62000 of width 10.\n",
      "Using motif -AT5G62000 of width 10.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00074\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G62020 of width 12.\n",
      "Using motif -AT5G62020 of width 12.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.871881\n",
      "#   Estimated pi_0=0.871881\n",
      "Using motif +AT5G62165 of width 7.\n",
      "Using motif -AT5G62165 of width 7.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944902\n",
      "#   Estimated pi_0=0.952453\n",
      "Using motif +AT5G62320 of width 13.\n",
      "Using motif -AT5G62320 of width 13.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.914\n",
      "#   Estimated pi_0=0.914038\n",
      "Using motif +AT5G62380 of width 21.\n",
      "Using motif -AT5G62380 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8926\n",
      "#   Estimated pi_0=0.905185\n",
      "Using motif +AT5G62470 of width 18.\n",
      "Using motif -AT5G62470 of width 18.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9192\n",
      "#   Estimated pi_0=0.926111\n",
      "Using motif +AT5G62940 of width 21.\n",
      "Using motif -AT5G62940 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 47 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.892182\n",
      "#   Estimated pi_0=0.899231\n",
      "Using motif +AT5G63090 of width 21.\n",
      "Using motif -AT5G63090 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G63260 of width 11.\n",
      "Using motif -AT5G63260 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.91981\n",
      "#   Estimated pi_0=0.926765\n",
      "Using motif +AT5G64060 of width 21.\n",
      "Using motif -AT5G64060 of width 21.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.93\n",
      "#   Estimated pi_0=0.941207\n",
      "Using motif +AT5G64220 of width 9.\n",
      "Using motif -AT5G64220 of width 9.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00312\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G64750 of width 19.\n",
      "Using motif -AT5G64750 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G65130 of width 15.\n",
      "Using motif -AT5G65130 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G65210 of width 15.\n",
      "Using motif -AT5G65210 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971683\n",
      "#   Estimated pi_0=0.988129\n",
      "Using motif +AT5G65310 of width 11.\n",
      "Using motif -AT5G65310 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.83604\n",
      "#   Estimated pi_0=0.841333\n",
      "Using motif +AT5G65410 of width 15.\n",
      "Using motif -AT5G65410 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.895\n",
      "#   Estimated pi_0=0.895\n",
      "Using motif +AT5G65510 of width 11.\n",
      "Using motif -AT5G65510 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9228\n",
      "#   Estimated pi_0=0.929286\n",
      "Using motif +AT5G66300 of width 16.\n",
      "Using motif -AT5G66300 of width 16.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9298\n",
      "#   Estimated pi_0=0.932593\n",
      "Using motif +AT5G66700 of width 11.\n",
      "Using motif -AT5G66700 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7988\n",
      "#   Estimated pi_0=0.80099\n",
      "Using motif +AT5G66730 of width 17.\n",
      "Using motif -AT5G66730 of width 17.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.905631\n",
      "#   Estimated pi_0=0.907087\n",
      "Using motif +AT5G66940 of width 29.\n",
      "Using motif -AT5G66940 of width 29.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 58 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.860531\n",
      "#   Estimated pi_0=0.864531\n",
      "Using motif +AT5G67000 of width 15.\n",
      "Using motif -AT5G67000 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999598\n",
      "#   Estimated pi_0=0.999598\n",
      "Using motif +AT5G67190 of width 15.\n",
      "Using motif -AT5G67190 of width 15.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G67300 of width 19.\n",
      "Using motif -AT5G67300 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9094\n",
      "#   Estimated pi_0=0.909703\n",
      "Using motif +AT5G67450 of width 11.\n",
      "Using motif -AT5G67450 of width 11.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975207\n",
      "#   Estimated pi_0=0.988987\n",
      "Using motif +AT5G67580 of width 19.\n",
      "Using motif -AT5G67580 of width 19.\n",
      "Finding best site passing the output threshold in each of the 79 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9054\n",
      "#   Estimated pi_0=0.9054\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  fimo --oc \"/data/NLRs_LRRs/fimo_LRRs_up\" \\\n",
    "    \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\" \"/data/NLRs_LRRs/all_LRRs_up_peaks.fasta\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "7ad012cd-a1a3-4ef5-b97d-4117ff7a37db",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using motif +AT1G01060 of width 10.\n",
      "Using motif -AT1G01060 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.900784\n",
      "#   Estimated pi_0=0.904\n",
      "Using motif +AT1G01250 of width 15.\n",
      "Using motif -AT1G01250 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00302\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G01260 of width 8.\n",
      "Using motif -AT1G01260 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0007\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G01720 of width 22.\n",
      "Using motif -AT1G01720 of width 22.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9478\n",
      "#   Estimated pi_0=0.96\n",
      "Using motif +AT1G02065 of width 16.\n",
      "Using motif -AT1G02065 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.833\n",
      "#   Estimated pi_0=0.833\n",
      "Using motif +AT1G02230 of width 14.\n",
      "Using motif -AT1G02230 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.979379\n",
      "#   Estimated pi_0=0.985506\n",
      "Using motif +AT1G02250 of width 19.\n",
      "Using motif -AT1G02250 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.924231\n",
      "#   Estimated pi_0=0.939355\n",
      "Using motif +AT1G03800 of width 21.\n",
      "Using motif -AT1G03800 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G03840 of width 20.\n",
      "Using motif -AT1G03840 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.956667\n",
      "#   Estimated pi_0=0.970292\n",
      "Using motif +AT1G06070 of width 11.\n",
      "Using motif -AT1G06070 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06160 of width 8.\n",
      "Using motif -AT1G06160 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00362\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06180 of width 15.\n",
      "Using motif -AT1G06180 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966731\n",
      "#   Estimated pi_0=0.984048\n",
      "Using motif +AT1G06280 of width 15.\n",
      "Using motif -AT1G06280 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06850 of width 14.\n",
      "Using motif -AT1G06850 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G08000 of width 8.\n",
      "Using motif -AT1G08000 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.994783\n",
      "#   Estimated pi_0=0.998883\n",
      "Using motif +AT1G08010 of width 15.\n",
      "Using motif -AT1G08010 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.989254\n",
      "#   Estimated pi_0=0.998788\n",
      "Using motif +AT1G08320 of width 11.\n",
      "Using motif -AT1G08320 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G08810 of width 19.\n",
      "Using motif -AT1G08810 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9536\n",
      "#   Estimated pi_0=0.961789\n",
      "Using motif +AT1G09030 of width 7.\n",
      "Using motif -AT1G09030 of width 7.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944865\n",
      "#   Estimated pi_0=0.953875\n",
      "Using motif +AT1G09530 of width 10.\n",
      "Using motif -AT1G09530 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G09540 of width 15.\n",
      "Using motif -AT1G09540 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.967895\n",
      "#   Estimated pi_0=0.977333\n",
      "Using motif +AT1G09770 of width 11.\n",
      "Using motif -AT1G09770 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00302\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G10120 of width 21.\n",
      "Using motif -AT1G10120 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G12260 of width 15.\n",
      "Using motif -AT1G12260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9482\n",
      "#   Estimated pi_0=0.962857\n",
      "Using motif +AT1G12610 of width 15.\n",
      "Using motif -AT1G12610 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G12630 of width 21.\n",
      "Using motif -AT1G12630 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G13260 of width 17.\n",
      "Using motif -AT1G13260 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9434\n",
      "#   Estimated pi_0=0.943774\n",
      "Using motif +AT1G13300 of width 15.\n",
      "Using motif -AT1G13300 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9234\n",
      "#   Estimated pi_0=0.931579\n",
      "Using motif +AT1G13450 of width 19.\n",
      "Using motif -AT1G13450 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9142\n",
      "#   Estimated pi_0=0.923925\n",
      "Using motif +AT1G14580 of width 20.\n",
      "Using motif -AT1G14580 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.949618\n",
      "#   Estimated pi_0=0.956163\n",
      "Using motif +AT1G14687 of width 11.\n",
      "Using motif -AT1G14687 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8424\n",
      "#   Estimated pi_0=0.847379\n",
      "Using motif +AT1G15720 of width 8.\n",
      "Using motif -AT1G15720 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT1G16490 of width 15.\n",
      "Using motif -AT1G16490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.947129\n",
      "#   Estimated pi_0=0.952072\n",
      "Using motif +AT1G17950 of width 11.\n",
      "Using motif -AT1G17950 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.939806\n",
      "#   Estimated pi_0=0.956134\n",
      "Using motif +AT1G18330 of width 12.\n",
      "Using motif -AT1G18330 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8602\n",
      "#   Estimated pi_0=0.86396\n",
      "Using motif +AT1G18570 of width 21.\n",
      "Using motif -AT1G18570 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9724\n",
      "#   Estimated pi_0=0.985176\n",
      "Using motif +AT1G18960 of width 22.\n",
      "Using motif -AT1G18960 of width 22.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997857\n",
      "#   Estimated pi_0=0.998995\n",
      "Using motif +AT1G19000 of width 14.\n",
      "Using motif -AT1G19000 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9132\n",
      "#   Estimated pi_0=0.92528\n",
      "Using motif +AT1G19040 of width 16.\n",
      "Using motif -AT1G19040 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993605\n",
      "#   Estimated pi_0=0.998376\n",
      "Using motif +AT1G19210 of width 14.\n",
      "Using motif -AT1G19210 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G19350 of width 11.\n",
      "Using motif -AT1G19350 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G19790 of width 23.\n",
      "Using motif -AT1G19790 of width 23.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8522\n",
      "#   Estimated pi_0=0.8522\n",
      "Using motif +AT1G19850 of width 10.\n",
      "Using motif -AT1G19850 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.990241\n",
      "#   Estimated pi_0=0.995873\n",
      "Using motif +AT1G20640 of width 15.\n",
      "Using motif -AT1G20640 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G20980 of width 11.\n",
      "Using motif -AT1G20980 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9792\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G21910 of width 20.\n",
      "Using motif -AT1G21910 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G22070 of width 14.\n",
      "Using motif -AT1G22070 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.996517\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G22640 of width 11.\n",
      "Using motif -AT1G22640 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G22810 of width 15.\n",
      "Using motif -AT1G22810 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G24260 of width 13.\n",
      "Using motif -AT1G24260 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9518\n",
      "#   Estimated pi_0=0.978767\n",
      "Using motif +AT1G25340 of width 15.\n",
      "Using motif -AT1G25340 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9256\n",
      "#   Estimated pi_0=0.934528\n",
      "Using motif +AT1G25550 of width 11.\n",
      "Using motif -AT1G25550 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.898\n",
      "#   Estimated pi_0=0.90233\n",
      "Using motif +AT1G25560 of width 8.\n",
      "Using motif -AT1G25560 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.985549\n",
      "#   Estimated pi_0=0.992923\n",
      "Using motif +AT1G27360 of width 11.\n",
      "Using motif -AT1G27360 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9825\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G27730 of width 11.\n",
      "Using motif -AT1G27730 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.95\n",
      "#   Estimated pi_0=0.967761\n",
      "Using motif +AT1G28160 of width 15.\n",
      "Using motif -AT1G28160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28300 of width 8.\n",
      "Using motif -AT1G28300 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999899\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT1G28360 of width 8.\n",
      "Using motif -AT1G28360 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00412\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28370 of width 15.\n",
      "Using motif -AT1G28370 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28470 of width 15.\n",
      "Using motif -AT1G28470 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973069\n",
      "#   Estimated pi_0=0.986667\n",
      "Using motif +AT1G29160 of width 21.\n",
      "Using motif -AT1G29160 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.912\n",
      "#   Estimated pi_0=0.918197\n",
      "Using motif +AT1G29280 of width 11.\n",
      "Using motif -AT1G29280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.945631\n",
      "#   Estimated pi_0=0.957031\n",
      "Using motif +AT1G29860 of width 11.\n",
      "Using motif -AT1G29860 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9468\n",
      "#   Estimated pi_0=0.962963\n",
      "Using motif +AT1G30210 of width 12.\n",
      "Using motif -AT1G30210 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G30490 of width 15.\n",
      "Using motif -AT1G30490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.908317\n",
      "#   Estimated pi_0=0.914087\n",
      "Using motif +AT1G30650 of width 13.\n",
      "Using motif -AT1G30650 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.952277\n",
      "#   Estimated pi_0=0.960153\n",
      "Using motif +AT1G31140 of width 15.\n",
      "Using motif -AT1G31140 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8094\n",
      "#   Estimated pi_0=0.824486\n",
      "Using motif +AT1G32150 of width 15.\n",
      "Using motif -AT1G32150 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G32240 of width 15.\n",
      "Using motif -AT1G32240 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8312\n",
      "#   Estimated pi_0=0.840935\n",
      "Using motif +AT1G32510 of width 21.\n",
      "Using motif -AT1G32510 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9486\n",
      "#   Estimated pi_0=0.95875\n",
      "Using motif +AT1G32640 of width 8.\n",
      "Using motif -AT1G32640 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.994706\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G32770 of width 12.\n",
      "Using motif -AT1G32770 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9494\n",
      "#   Estimated pi_0=0.95\n",
      "Using motif +AT1G32870 of width 16.\n",
      "Using motif -AT1G32870 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998889\n",
      "#   Estimated pi_0=0.999598\n",
      "Using motif +AT1G33240 of width 10.\n",
      "Using motif -AT1G33240 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8652\n",
      "#   Estimated pi_0=0.8652\n",
      "Using motif +AT1G34180 of width 15.\n",
      "Using motif -AT1G34180 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993333\n",
      "#   Estimated pi_0=0.999192\n",
      "Using motif +AT1G34190 of width 19.\n",
      "Using motif -AT1G34190 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.99156\n",
      "#   Estimated pi_0=0.996041\n",
      "Using motif +AT1G34370 of width 15.\n",
      "Using motif -AT1G34370 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9854\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G34670 of width 21.\n",
      "Using motif -AT1G34670 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9332\n",
      "#   Estimated pi_0=0.935962\n",
      "Using motif +AT1G35460 of width 11.\n",
      "Using motif -AT1G35460 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975398\n",
      "#   Estimated pi_0=0.979641\n",
      "Using motif +AT1G35560 of width 10.\n",
      "Using motif -AT1G35560 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G36060 of width 14.\n",
      "Using motif -AT1G36060 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G42990 of width 10.\n",
      "Using motif -AT1G42990 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999899\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT1G43160 of width 15.\n",
      "Using motif -AT1G43160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G43700 of width 12.\n",
      "Using motif -AT1G43700 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.974757\n",
      "#   Estimated pi_0=0.997474\n",
      "Using motif +AT1G44830 of width 15.\n",
      "Using motif -AT1G44830 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G45249 of width 18.\n",
      "Using motif -AT1G45249 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999086\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G46264 of width 15.\n",
      "Using motif -AT1G46264 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978485\n",
      "#   Estimated pi_0=0.981882\n",
      "Using motif +AT1G46768 of width 13.\n",
      "Using motif -AT1G46768 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G47655 of width 11.\n",
      "Using motif -AT1G47655 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9306\n",
      "#   Estimated pi_0=0.942581\n",
      "Using motif +AT1G47870 of width 15.\n",
      "Using motif -AT1G47870 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.95514\n",
      "#   Estimated pi_0=0.967939\n",
      "Using motif +AT1G49010 of width 15.\n",
      "Using motif -AT1G49010 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8824\n",
      "#   Estimated pi_0=0.895741\n",
      "Using motif +AT1G49480 of width 15.\n",
      "Using motif -AT1G49480 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 36 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944675\n",
      "#   Estimated pi_0=0.956906\n",
      "Using motif +AT1G49560 of width 11.\n",
      "Using motif -AT1G49560 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937255\n",
      "#   Estimated pi_0=0.950619\n",
      "Using motif +AT1G49720 of width 19.\n",
      "Using motif -AT1G49720 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G49950 of width 9.\n",
      "Using motif -AT1G49950 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.990631\n",
      "#   Estimated pi_0=0.999698\n",
      "Using motif +AT1G50640 of width 15.\n",
      "Using motif -AT1G50640 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G51140 of width 14.\n",
      "Using motif -AT1G51140 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9832\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT1G51220 of width 15.\n",
      "Using motif -AT1G51220 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.99625\n",
      "#   Estimated pi_0=0.999196\n",
      "Using motif +AT1G51600 of width 19.\n",
      "Using motif -AT1G51600 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944874\n",
      "#   Estimated pi_0=0.951449\n",
      "Using motif +AT1G51700 of width 21.\n",
      "Using motif -AT1G51700 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8894\n",
      "#   Estimated pi_0=0.901176\n",
      "Using motif +AT1G52150 of width 15.\n",
      "Using motif -AT1G52150 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8526\n",
      "#   Estimated pi_0=0.858077\n",
      "Using motif +AT1G52880 of width 19.\n",
      "Using motif -AT1G52880 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9376\n",
      "#   Estimated pi_0=0.945505\n",
      "Using motif +AT1G52890 of width 12.\n",
      "Using motif -AT1G52890 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G53160 of width 10.\n",
      "Using motif -AT1G53160 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9692\n",
      "#   Estimated pi_0=0.9692\n",
      "Using motif +AT1G53170 of width 15.\n",
      "Using motif -AT1G53170 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G53230 of width 13.\n",
      "Using motif -AT1G53230 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G53910 of width 21.\n",
      "Using motif -AT1G53910 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G54330 of width 21.\n",
      "Using motif -AT1G54330 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.919811\n",
      "#   Estimated pi_0=0.927027\n",
      "Using motif +AT1G55110 of width 18.\n",
      "Using motif -AT1G55110 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.935882\n",
      "#   Estimated pi_0=0.956711\n",
      "Using motif +AT1G59640 of width 21.\n",
      "Using motif -AT1G59640 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00071\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G59750 of width 8.\n",
      "Using motif -AT1G59750 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G60920 of width 6.\n",
      "Using motif -AT1G60920 of width 6.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT1G61110 of width 8.\n",
      "Using motif -AT1G61110 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00135\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G62300 of width 19.\n",
      "Using motif -AT1G62300 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94729\n",
      "#   Estimated pi_0=0.960169\n",
      "Using motif +AT1G63030 of width 8.\n",
      "Using motif -AT1G63030 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.985505\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G64620 of width 19.\n",
      "Using motif -AT1G64620 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.866931\n",
      "#   Estimated pi_0=0.878939\n",
      "Using motif +AT1G65620 of width 21.\n",
      "Using motif -AT1G65620 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G65910 of width 16.\n",
      "Using motif -AT1G65910 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9596\n",
      "#   Estimated pi_0=0.975385\n",
      "Using motif +AT1G66370 of width 11.\n",
      "Using motif -AT1G66370 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.919\n",
      "#   Estimated pi_0=0.933178\n",
      "Using motif +AT1G66420 of width 19.\n",
      "Using motif -AT1G66420 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8938\n",
      "#   Estimated pi_0=0.912283\n",
      "Using motif +AT1G66600 of width 8.\n",
      "Using motif -AT1G66600 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.987692\n",
      "#   Estimated pi_0=0.992391\n",
      "Using motif +AT1G67260 of width 30.\n",
      "Using motif -AT1G67260 of width 30.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G67710 of width 10.\n",
      "Using motif -AT1G67710 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933905\n",
      "#   Estimated pi_0=0.948174\n",
      "Using motif +AT1G68200 of width 15.\n",
      "Using motif -AT1G68200 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G68320 of width 15.\n",
      "Using motif -AT1G68320 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.934902\n",
      "#   Estimated pi_0=0.946207\n",
      "Using motif +AT1G68550 of width 15.\n",
      "Using motif -AT1G68550 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G68670 of width 13.\n",
      "Using motif -AT1G68670 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.923107\n",
      "#   Estimated pi_0=0.92434\n",
      "Using motif +AT1G68840 of width 8.\n",
      "Using motif -AT1G68840 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993628\n",
      "#   Estimated pi_0=0.998794\n",
      "Using motif +AT1G69010 of width 15.\n",
      "Using motif -AT1G69010 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G69120 of width 12.\n",
      "Using motif -AT1G69120 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8538\n",
      "#   Estimated pi_0=0.861481\n",
      "Using motif +AT1G69180 of width 8.\n",
      "Using motif -AT1G69180 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.895534\n",
      "#   Estimated pi_0=0.90193\n",
      "Using motif +AT1G69310 of width 10.\n",
      "Using motif -AT1G69310 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9204\n",
      "#   Estimated pi_0=0.930667\n",
      "Using motif +AT1G69490 of width 11.\n",
      "Using motif -AT1G69490 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.964314\n",
      "#   Estimated pi_0=0.977778\n",
      "Using motif +AT1G69560 of width 13.\n",
      "Using motif -AT1G69560 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9064\n",
      "#   Estimated pi_0=0.91\n",
      "Using motif +AT1G69570 of width 27.\n",
      "Using motif -AT1G69570 of width 27.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 32 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.882353\n",
      "#   Estimated pi_0=0.892903\n",
      "Using motif +AT1G69690 of width 13.\n",
      "Using motif -AT1G69690 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G69780 of width 11.\n",
      "Using motif -AT1G69780 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8238\n",
      "#   Estimated pi_0=0.826471\n",
      "Using motif +AT1G70000 of width 10.\n",
      "Using motif -AT1G70000 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.958\n",
      "#   Estimated pi_0=0.977763\n",
      "Using motif +AT1G70920 of width 11.\n",
      "Using motif -AT1G70920 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8948\n",
      "#   Estimated pi_0=0.8948\n",
      "Using motif +AT1G71450 of width 20.\n",
      "Using motif -AT1G71450 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999797\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT1G71930 of width 12.\n",
      "Using motif -AT1G71930 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G72010 of width 13.\n",
      "Using motif -AT1G72010 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G72050 of width 19.\n",
      "Using motif -AT1G72050 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 35 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.957687\n",
      "#   Estimated pi_0=0.966738\n",
      "Using motif +AT1G72360 of width 21.\n",
      "Using motif -AT1G72360 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G72740 of width 15.\n",
      "Using motif -AT1G72740 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969608\n",
      "#   Estimated pi_0=0.986857\n",
      "Using motif +AT1G73360 of width 11.\n",
      "Using motif -AT1G73360 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8776\n",
      "#   Estimated pi_0=0.879612\n",
      "Using motif +AT1G73730 of width 21.\n",
      "Using motif -AT1G73730 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.906863\n",
      "#   Estimated pi_0=0.906863\n",
      "Using motif +AT1G74480 of width 19.\n",
      "Using motif -AT1G74480 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.977321\n",
      "#   Estimated pi_0=0.989735\n",
      "Using motif +AT1G74650 of width 15.\n",
      "Using motif -AT1G74650 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9336\n",
      "#   Estimated pi_0=0.946341\n",
      "Using motif +AT1G74840 of width 11.\n",
      "Using motif -AT1G74840 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.931089\n",
      "#   Estimated pi_0=0.944098\n",
      "Using motif +AT1G74930 of width 10.\n",
      "Using motif -AT1G74930 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G75080 of width 20.\n",
      "Using motif -AT1G75080 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G75240 of width 11.\n",
      "Using motif -AT1G75240 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9304\n",
      "#   Estimated pi_0=0.932308\n",
      "Using motif +AT1G75390 of width 15.\n",
      "Using motif -AT1G75390 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G75490 of width 15.\n",
      "Using motif -AT1G75490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G76420 of width 18.\n",
      "Using motif -AT1G76420 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.928\n",
      "#   Estimated pi_0=0.935192\n",
      "Using motif +AT1G76870 of width 14.\n",
      "Using motif -AT1G76870 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9388\n",
      "#   Estimated pi_0=0.951552\n",
      "Using motif +AT1G76880 of width 16.\n",
      "Using motif -AT1G76880 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8586\n",
      "#   Estimated pi_0=0.865825\n",
      "Using motif +AT1G76890 of width 14.\n",
      "Using motif -AT1G76890 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.83901\n",
      "#   Estimated pi_0=0.845905\n",
      "Using motif +AT1G77080 of width 15.\n",
      "Using motif -AT1G77080 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8674\n",
      "#   Estimated pi_0=0.878302\n",
      "Using motif +AT1G77200 of width 14.\n",
      "Using motif -AT1G77200 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G77640 of width 21.\n",
      "Using motif -AT1G77640 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G77920 of width 17.\n",
      "Using motif -AT1G77920 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975514\n",
      "#   Estimated pi_0=0.991351\n",
      "Using motif +AT1G78080 of width 10.\n",
      "Using motif -AT1G78080 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G78700 of width 11.\n",
      "Using motif -AT1G78700 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G79180 of width 21.\n",
      "Using motif -AT1G79180 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.986179\n",
      "#   Estimated pi_0=0.998071\n",
      "Using motif +AT1G79580 of width 21.\n",
      "Using motif -AT1G79580 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.950495\n",
      "#   Estimated pi_0=0.962899\n",
      "Using motif +AT1G80840 of width 15.\n",
      "Using motif -AT1G80840 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.955\n",
      "#   Estimated pi_0=0.967402\n",
      "Using motif +AT2G01060 of width 13.\n",
      "Using motif -AT2G01060 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.936\n",
      "#   Estimated pi_0=0.936\n",
      "Using motif +AT2G01570 of width 20.\n",
      "Using motif -AT2G01570 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 43 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.974012\n",
      "#   Estimated pi_0=0.978191\n",
      "Using motif +AT2G01760 of width 8.\n",
      "Using motif -AT2G01760 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9856\n",
      "#   Estimated pi_0=0.997306\n",
      "Using motif +AT2G01930 of width 24.\n",
      "Using motif -AT2G01930 of width 24.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921565\n",
      "#   Estimated pi_0=0.926946\n",
      "Using motif +AT2G01940 of width 15.\n",
      "Using motif -AT2G01940 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94\n",
      "#   Estimated pi_0=0.948987\n",
      "Using motif +AT2G02070 of width 16.\n",
      "Using motif -AT2G02070 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.954737\n",
      "#   Estimated pi_0=0.968824\n",
      "Using motif +AT2G02080 of width 17.\n",
      "Using motif -AT2G02080 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.946055\n",
      "#   Estimated pi_0=0.964286\n",
      "Using motif +AT2G02450 of width 19.\n",
      "Using motif -AT2G02450 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G02540 of width 14.\n",
      "Using motif -AT2G02540 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8414\n",
      "#   Estimated pi_0=0.8414\n",
      "Using motif +AT2G02820 of width 15.\n",
      "Using motif -AT2G02820 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G03340 of width 11.\n",
      "Using motif -AT2G03340 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.970275\n",
      "#   Estimated pi_0=0.978769\n",
      "Using motif +AT2G03500 of width 11.\n",
      "Using motif -AT2G03500 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.92\n",
      "#   Estimated pi_0=0.928288\n",
      "Using motif +AT2G03710 of width 18.\n",
      "Using motif -AT2G03710 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8606\n",
      "#   Estimated pi_0=0.863725\n",
      "Using motif +AT2G04038 of width 14.\n",
      "Using motif -AT2G04038 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.991509\n",
      "#   Estimated pi_0=0.998462\n",
      "Using motif +AT2G04880 of width 11.\n",
      "Using motif -AT2G04880 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G06200 of width 14.\n",
      "Using motif -AT2G06200 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978491\n",
      "#   Estimated pi_0=0.994194\n",
      "Using motif +AT2G15740 of width 11.\n",
      "Using motif -AT2G15740 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971683\n",
      "#   Estimated pi_0=0.992262\n",
      "Using motif +AT2G17180 of width 10.\n",
      "Using motif -AT2G17180 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G17950 of width 11.\n",
      "Using motif -AT2G17950 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8875\n",
      "#   Estimated pi_0=0.89322\n",
      "Using motif +AT2G18060 of width 13.\n",
      "Using motif -AT2G18060 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9756\n",
      "#   Estimated pi_0=0.999497\n",
      "Using motif +AT2G18160 of width 15.\n",
      "Using motif -AT2G18160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.995326\n",
      "#   Estimated pi_0=0.999598\n",
      "Using motif +AT2G18300 of width 12.\n",
      "Using motif -AT2G18300 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G18350 of width 15.\n",
      "Using motif -AT2G18350 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.895294\n",
      "#   Estimated pi_0=0.90487\n",
      "Using motif +AT2G18380 of width 13.\n",
      "Using motif -AT2G18380 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.98\n",
      "#   Estimated pi_0=0.997113\n",
      "Using motif +AT2G20110 of width 15.\n",
      "Using motif -AT2G20110 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.842353\n",
      "#   Estimated pi_0=0.84973\n",
      "Using motif +AT2G20180 of width 14.\n",
      "Using motif -AT2G20180 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G20400 of width 15.\n",
      "Using motif -AT2G20400 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.915\n",
      "#   Estimated pi_0=0.929302\n",
      "Using motif +AT2G20570 of width 10.\n",
      "Using motif -AT2G20570 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.952642\n",
      "#   Estimated pi_0=0.96359\n",
      "Using motif +AT2G21230 of width 10.\n",
      "Using motif -AT2G21230 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.990495\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G21900 of width 12.\n",
      "Using motif -AT2G21900 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9544\n",
      "#   Estimated pi_0=0.97295\n",
      "Using motif +AT2G22300 of width 9.\n",
      "Using motif -AT2G22300 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G22430 of width 11.\n",
      "Using motif -AT2G22430 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8568\n",
      "#   Estimated pi_0=0.860577\n",
      "Using motif +AT2G22540 of width 15.\n",
      "Using motif -AT2G22540 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.904118\n",
      "#   Estimated pi_0=0.904158\n",
      "Using motif +AT2G22750 of width 17.\n",
      "Using motif -AT2G22750 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G23290 of width 14.\n",
      "Using motif -AT2G23290 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9546\n",
      "#   Estimated pi_0=0.973649\n",
      "Using motif +AT2G23320 of width 11.\n",
      "Using motif -AT2G23320 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.955294\n",
      "#   Estimated pi_0=0.965442\n",
      "Using motif +AT2G23340 of width 17.\n",
      "Using motif -AT2G23340 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G24260 of width 10.\n",
      "Using motif -AT2G24260 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96233\n",
      "#   Estimated pi_0=0.974351\n",
      "Using motif +AT2G24430 of width 18.\n",
      "Using motif -AT2G24430 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9302\n",
      "#   Estimated pi_0=0.934364\n",
      "Using motif +AT2G24570 of width 14.\n",
      "Using motif -AT2G24570 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969256\n",
      "#   Estimated pi_0=0.983396\n",
      "Using motif +AT2G25000 of width 11.\n",
      "Using motif -AT2G25000 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.983651\n",
      "#   Estimated pi_0=0.999391\n",
      "Using motif +AT2G26960 of width 11.\n",
      "Using motif -AT2G26960 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9118\n",
      "#   Estimated pi_0=0.922143\n",
      "Using motif +AT2G27300 of width 19.\n",
      "Using motif -AT2G27300 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.991591\n",
      "#   Estimated pi_0=0.996869\n",
      "Using motif +AT2G28550 of width 10.\n",
      "Using motif -AT2G28550 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.988317\n",
      "#   Estimated pi_0=0.99959\n",
      "Using motif +AT2G28810 of width 21.\n",
      "Using motif -AT2G28810 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 33 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.876885\n",
      "#   Estimated pi_0=0.879062\n",
      "Using motif +AT2G30250 of width 11.\n",
      "Using motif -AT2G30250 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.951845\n",
      "#   Estimated pi_0=0.959646\n",
      "Using motif +AT2G30340 of width 17.\n",
      "Using motif -AT2G30340 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G30590 of width 11.\n",
      "Using motif -AT2G30590 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9456\n",
      "#   Estimated pi_0=0.963357\n",
      "Using motif +AT2G31220 of width 14.\n",
      "Using motif -AT2G31220 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G31230 of width 15.\n",
      "Using motif -AT2G31230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G32460 of width 15.\n",
      "Using motif -AT2G32460 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9604\n",
      "#   Estimated pi_0=0.965902\n",
      "Using motif +AT2G33550 of width 18.\n",
      "Using motif -AT2G33550 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9248\n",
      "#   Estimated pi_0=0.926796\n",
      "Using motif +AT2G33710 of width 15.\n",
      "Using motif -AT2G33710 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G33810 of width 8.\n",
      "Using motif -AT2G33810 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00312\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G33860 of width 10.\n",
      "Using motif -AT2G33860 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G35530 of width 15.\n",
      "Using motif -AT2G35530 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G35700 of width 15.\n",
      "Using motif -AT2G35700 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G36010 of width 14.\n",
      "Using motif -AT2G36010 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00362\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G36270 of width 18.\n",
      "Using motif -AT2G36270 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G36610 of width 13.\n",
      "Using motif -AT2G36610 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8304\n",
      "#   Estimated pi_0=0.833663\n",
      "Using motif +AT2G37590 of width 15.\n",
      "Using motif -AT2G37590 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.872952\n",
      "#   Estimated pi_0=0.873585\n",
      "Using motif +AT2G38090 of width 21.\n",
      "Using motif -AT2G38090 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8676\n",
      "#   Estimated pi_0=0.884746\n",
      "Using motif +AT2G38300 of width 11.\n",
      "Using motif -AT2G38300 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.942281\n",
      "#   Estimated pi_0=0.951034\n",
      "Using motif +AT2G38340 of width 17.\n",
      "Using motif -AT2G38340 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G38470 of width 11.\n",
      "Using motif -AT2G38470 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.943585\n",
      "#   Estimated pi_0=0.949825\n",
      "Using motif +AT2G38880 of width 9.\n",
      "Using motif -AT2G38880 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.99082\n",
      "#   Estimated pi_0=0.994082\n",
      "Using motif +AT2G40220 of width 10.\n",
      "Using motif -AT2G40220 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G40260 of width 15.\n",
      "Using motif -AT2G40260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8768\n",
      "#   Estimated pi_0=0.880588\n",
      "Using motif +AT2G40340 of width 21.\n",
      "Using motif -AT2G40340 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0003\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G40620 of width 11.\n",
      "Using motif -AT2G40620 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G40740 of width 12.\n",
      "Using motif -AT2G40740 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961923\n",
      "#   Estimated pi_0=0.985663\n",
      "Using motif +AT2G41690 of width 15.\n",
      "Using motif -AT2G41690 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.996614\n",
      "#   Estimated pi_0=0.999397\n",
      "Using motif +AT2G41835 of width 7.\n",
      "Using motif -AT2G41835 of width 7.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937483\n",
      "#   Estimated pi_0=0.937483\n",
      "Using motif +AT2G42200 of width 11.\n",
      "Using motif -AT2G42200 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9822\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G42280 of width 11.\n",
      "Using motif -AT2G42280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.968393\n",
      "#   Estimated pi_0=0.976159\n",
      "Using motif +AT2G42400 of width 12.\n",
      "Using motif -AT2G42400 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G42430 of width 15.\n",
      "Using motif -AT2G42430 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G43000 of width 20.\n",
      "Using motif -AT2G43000 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G43010 of width 9.\n",
      "Using motif -AT2G43010 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G44745 of width 10.\n",
      "Using motif -AT2G44745 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0007\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G44840 of width 15.\n",
      "Using motif -AT2G44840 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G44940 of width 15.\n",
      "Using motif -AT2G44940 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G45190 of width 10.\n",
      "Using motif -AT2G45190 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8134\n",
      "#   Estimated pi_0=0.81505\n",
      "Using motif +AT2G45410 of width 15.\n",
      "Using motif -AT2G45410 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G45420 of width 19.\n",
      "Using motif -AT2G45420 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G45650 of width 19.\n",
      "Using motif -AT2G45650 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8124\n",
      "#   Estimated pi_0=0.8124\n",
      "Using motif +AT2G45660 of width 21.\n",
      "Using motif -AT2G45660 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 34 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96578\n",
      "#   Estimated pi_0=0.967955\n",
      "Using motif +AT2G45680 of width 21.\n",
      "Using motif -AT2G45680 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G46130 of width 14.\n",
      "Using motif -AT2G46130 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966535\n",
      "#   Estimated pi_0=0.981333\n",
      "Using motif +AT2G46270 of width 15.\n",
      "Using motif -AT2G46270 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G46400 of width 11.\n",
      "Using motif -AT2G46400 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978218\n",
      "#   Estimated pi_0=0.997263\n",
      "Using motif +AT2G46510 of width 8.\n",
      "Using motif -AT2G46510 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G46590 of width 11.\n",
      "Using motif -AT2G46590 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9106\n",
      "#   Estimated pi_0=0.921913\n",
      "Using motif +AT2G46680 of width 13.\n",
      "Using motif -AT2G46680 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.890495\n",
      "#   Estimated pi_0=0.901455\n",
      "Using motif +AT2G46770 of width 17.\n",
      "Using motif -AT2G46770 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.932816\n",
      "#   Estimated pi_0=0.932816\n",
      "Using motif +AT2G46830 of width 8.\n",
      "Using motif -AT2G46830 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.891\n",
      "#   Estimated pi_0=0.895455\n",
      "Using motif +AT2G47070 of width 10.\n",
      "Using motif -AT2G47070 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT2G47260 of width 8.\n",
      "Using motif -AT2G47260 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.985385\n",
      "#   Estimated pi_0=0.996923\n",
      "Using motif +AT2G47460 of width 10.\n",
      "Using motif -AT2G47460 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948515\n",
      "#   Estimated pi_0=0.965528\n",
      "Using motif +AT3G01220 of width 11.\n",
      "Using motif -AT3G01220 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.847\n",
      "#   Estimated pi_0=0.855091\n",
      "Using motif +AT3G01470 of width 11.\n",
      "Using motif -AT3G01470 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8358\n",
      "#   Estimated pi_0=0.845333\n",
      "Using motif +AT3G01530 of width 14.\n",
      "Using motif -AT3G01530 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9196\n",
      "#   Estimated pi_0=0.935\n",
      "Using motif +AT3G01970 of width 12.\n",
      "Using motif -AT3G01970 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.934312\n",
      "#   Estimated pi_0=0.944806\n",
      "Using motif +AT3G02150 of width 13.\n",
      "Using motif -AT3G02150 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G02940 of width 18.\n",
      "Using motif -AT3G02940 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94367\n",
      "#   Estimated pi_0=0.956429\n",
      "Using motif +AT3G02990 of width 13.\n",
      "Using motif -AT3G02990 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.964952\n",
      "#   Estimated pi_0=0.981829\n",
      "Using motif +AT3G03200 of width 16.\n",
      "Using motif -AT3G03200 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973274\n",
      "#   Estimated pi_0=0.986364\n",
      "Using motif +AT3G03660 of width 11.\n",
      "Using motif -AT3G03660 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.923636\n",
      "#   Estimated pi_0=0.923636\n",
      "Using motif +AT3G04030 of width 14.\n",
      "Using motif -AT3G04030 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9212\n",
      "#   Estimated pi_0=0.937638\n",
      "Using motif +AT3G04060 of width 21.\n",
      "Using motif -AT3G04060 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9052\n",
      "#   Estimated pi_0=0.913458\n",
      "Using motif +AT3G04070 of width 15.\n",
      "Using motif -AT3G04070 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.93902\n",
      "#   Estimated pi_0=0.947769\n",
      "Using motif +AT3G06490 of width 15.\n",
      "Using motif -AT3G06490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.964752\n",
      "#   Estimated pi_0=0.979296\n",
      "Using motif +AT3G06740 of width 19.\n",
      "Using motif -AT3G06740 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.904314\n",
      "#   Estimated pi_0=0.913645\n",
      "Using motif +AT3G08500 of width 11.\n",
      "Using motif -AT3G08500 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.995319\n",
      "#   Estimated pi_0=0.997186\n",
      "Using motif +AT3G09230 of width 19.\n",
      "Using motif -AT3G09230 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9436\n",
      "#   Estimated pi_0=0.951456\n",
      "Using motif +AT3G09600 of width 12.\n",
      "Using motif -AT3G09600 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.873333\n",
      "#   Estimated pi_0=0.87581\n",
      "Using motif +AT3G09735 of width 15.\n",
      "Using motif -AT3G09735 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999289\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G10030 of width 14.\n",
      "Using motif -AT3G10030 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.936275\n",
      "#   Estimated pi_0=0.938269\n",
      "Using motif +AT3G10113 of width 15.\n",
      "Using motif -AT3G10113 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.900364\n",
      "#   Estimated pi_0=0.907333\n",
      "Using motif +AT3G10480 of width 16.\n",
      "Using motif -AT3G10480 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.987593\n",
      "#   Estimated pi_0=0.999598\n",
      "Using motif +AT3G10490 of width 20.\n",
      "Using motif -AT3G10490 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94729\n",
      "#   Estimated pi_0=0.957727\n",
      "Using motif +AT3G10500 of width 17.\n",
      "Using motif -AT3G10500 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9376\n",
      "#   Estimated pi_0=0.94541\n",
      "Using motif +AT3G10580 of width 15.\n",
      "Using motif -AT3G10580 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8038\n",
      "#   Estimated pi_0=0.809709\n",
      "Using motif +AT3G10800 of width 15.\n",
      "Using motif -AT3G10800 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997876\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G11020 of width 21.\n",
      "Using motif -AT3G11020 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G11280 of width 14.\n",
      "Using motif -AT3G11280 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9372\n",
      "#   Estimated pi_0=0.948806\n",
      "Using motif +AT3G11440 of width 13.\n",
      "Using motif -AT3G11440 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921188\n",
      "#   Estimated pi_0=0.929455\n",
      "Using motif +AT3G12130 of width 14.\n",
      "Using motif -AT3G12130 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.90781\n",
      "#   Estimated pi_0=0.937452\n",
      "Using motif +AT3G12250 of width 15.\n",
      "Using motif -AT3G12250 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9806\n",
      "#   Estimated pi_0=0.994556\n",
      "Using motif +AT3G12720 of width 21.\n",
      "Using motif -AT3G12720 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9018\n",
      "#   Estimated pi_0=0.907619\n",
      "Using motif +AT3G12730 of width 14.\n",
      "Using motif -AT3G12730 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9134\n",
      "#   Estimated pi_0=0.915769\n",
      "Using motif +AT3G12820 of width 16.\n",
      "Using motif -AT3G12820 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948302\n",
      "#   Estimated pi_0=0.954775\n",
      "Using motif +AT3G12910 of width 17.\n",
      "Using motif -AT3G12910 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G13040 of width 9.\n",
      "Using motif -AT3G13040 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.956104\n",
      "#   Estimated pi_0=0.961272\n",
      "Using motif +AT3G13810 of width 21.\n",
      "Using motif -AT3G13810 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9598\n",
      "#   Estimated pi_0=0.966199\n",
      "Using motif +AT3G14180 of width 11.\n",
      "Using motif -AT3G14180 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15030 of width 8.\n",
      "Using motif -AT3G15030 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15170 of width 16.\n",
      "Using motif -AT3G15170 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.972075\n",
      "#   Estimated pi_0=0.977311\n",
      "Using motif +AT3G15210 of width 15.\n",
      "Using motif -AT3G15210 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15270 of width 12.\n",
      "Using motif -AT3G15270 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9924\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G15500 of width 21.\n",
      "Using motif -AT3G15500 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9402\n",
      "#   Estimated pi_0=0.95453\n",
      "Using motif +AT3G15510 of width 17.\n",
      "Using motif -AT3G15510 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.977945\n",
      "#   Estimated pi_0=0.991458\n",
      "Using motif +AT3G16280 of width 14.\n",
      "Using motif -AT3G16280 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0008\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G16770 of width 8.\n",
      "Using motif -AT3G16770 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00372\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G17730 of width 17.\n",
      "Using motif -AT3G17730 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997143\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G18400 of width 16.\n",
      "Using motif -AT3G18400 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.958077\n",
      "#   Estimated pi_0=0.962569\n",
      "Using motif +AT3G20310 of width 21.\n",
      "Using motif -AT3G20310 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999482\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G20770 of width 15.\n",
      "Using motif -AT3G20770 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9494\n",
      "#   Estimated pi_0=0.958291\n",
      "Using motif +AT3G20840 of width 19.\n",
      "Using motif -AT3G20840 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969167\n",
      "#   Estimated pi_0=0.975659\n",
      "Using motif +AT3G21175 of width 15.\n",
      "Using motif -AT3G21175 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.893267\n",
      "#   Estimated pi_0=0.900177\n",
      "Using motif +AT3G22170 of width 12.\n",
      "Using motif -AT3G22170 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G22760 of width 8.\n",
      "Using motif -AT3G22760 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8554\n",
      "#   Estimated pi_0=0.861538\n",
      "Using motif +AT3G22780 of width 15.\n",
      "Using motif -AT3G22780 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8368\n",
      "#   Estimated pi_0=0.841154\n",
      "Using motif +AT3G22830 of width 15.\n",
      "Using motif -AT3G22830 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999793\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G23210 of width 23.\n",
      "Using motif -AT3G23210 of width 23.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT3G23220 of width 15.\n",
      "Using motif -AT3G23220 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G23240 of width 28.\n",
      "Using motif -AT3G23240 of width 28.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G23250 of width 21.\n",
      "Using motif -AT3G23250 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.916275\n",
      "#   Estimated pi_0=0.92\n",
      "Using motif +AT3G23690 of width 15.\n",
      "Using motif -AT3G23690 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00031\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G24050 of width 15.\n",
      "Using motif -AT3G24050 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.958018\n",
      "#   Estimated pi_0=0.969863\n",
      "Using motif +AT3G24120 of width 8.\n",
      "Using motif -AT3G24120 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961538\n",
      "#   Estimated pi_0=0.982712\n",
      "Using motif +AT3G24520 of width 11.\n",
      "Using motif -AT3G24520 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998307\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G24650 of width 9.\n",
      "Using motif -AT3G24650 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G25730 of width 8.\n",
      "Using motif -AT3G25730 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948952\n",
      "#   Estimated pi_0=0.960458\n",
      "Using motif +AT3G25990 of width 21.\n",
      "Using motif -AT3G25990 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.957308\n",
      "#   Estimated pi_0=0.969919\n",
      "Using motif +AT3G26620 of width 15.\n",
      "Using motif -AT3G26620 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G26790 of width 15.\n",
      "Using motif -AT3G26790 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978992\n",
      "#   Estimated pi_0=0.988276\n",
      "Using motif +AT3G27010 of width 21.\n",
      "Using motif -AT3G27010 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G27785 of width 17.\n",
      "Using motif -AT3G27785 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913663\n",
      "#   Estimated pi_0=0.925586\n",
      "Using motif +AT3G28910 of width 15.\n",
      "Using motif -AT3G28910 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.937925\n",
      "#   Estimated pi_0=0.94431\n",
      "Using motif +AT3G28920 of width 15.\n",
      "Using motif -AT3G28920 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8542\n",
      "#   Estimated pi_0=0.866355\n",
      "Using motif +AT3G30210 of width 14.\n",
      "Using motif -AT3G30210 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9266\n",
      "#   Estimated pi_0=0.932885\n",
      "Using motif +AT3G30530 of width 12.\n",
      "Using motif -AT3G30530 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998692\n",
      "#   Estimated pi_0=0.999095\n",
      "Using motif +AT3G45150 of width 11.\n",
      "Using motif -AT3G45150 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G45170 of width 15.\n",
      "Using motif -AT3G45170 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.983827\n",
      "#   Estimated pi_0=0.991217\n",
      "Using motif +AT3G45610 of width 19.\n",
      "Using motif -AT3G45610 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.896757\n",
      "#   Estimated pi_0=0.906446\n",
      "Using motif +AT3G46070 of width 15.\n",
      "Using motif -AT3G46070 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.939223\n",
      "#   Estimated pi_0=0.954702\n",
      "Using motif +AT3G46090 of width 13.\n",
      "Using motif -AT3G46090 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G46640 of width 14.\n",
      "Using motif -AT3G46640 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.914231\n",
      "#   Estimated pi_0=0.921982\n",
      "Using motif +AT3G47500 of width 19.\n",
      "Using motif -AT3G47500 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8952\n",
      "#   Estimated pi_0=0.898276\n",
      "Using motif +AT3G47600 of width 19.\n",
      "Using motif -AT3G47600 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.959\n",
      "#   Estimated pi_0=0.975405\n",
      "Using motif +AT3G47620 of width 20.\n",
      "Using motif -AT3G47620 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G48160 of width 21.\n",
      "Using motif -AT3G48160 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969703\n",
      "#   Estimated pi_0=0.983556\n",
      "Using motif +AT3G48430 of width 12.\n",
      "Using motif -AT3G48430 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 30 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969524\n",
      "#   Estimated pi_0=0.977049\n",
      "Using motif +AT3G49530 of width 15.\n",
      "Using motif -AT3G49530 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.928\n",
      "#   Estimated pi_0=0.935556\n",
      "Using motif +AT3G49690 of width 13.\n",
      "Using motif -AT3G49690 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00191\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G49930 of width 7.\n",
      "Using motif -AT3G49930 of width 7.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.942656\n",
      "#   Estimated pi_0=0.951128\n",
      "Using motif +AT3G50060 of width 15.\n",
      "Using motif -AT3G50060 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9194\n",
      "#   Estimated pi_0=0.921961\n",
      "Using motif +AT3G50260 of width 15.\n",
      "Using motif -AT3G50260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G50410 of width 21.\n",
      "Using motif -AT3G50410 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 34 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.902574\n",
      "#   Estimated pi_0=0.908654\n",
      "Using motif +AT3G50700 of width 15.\n",
      "Using motif -AT3G50700 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.945124\n",
      "#   Estimated pi_0=0.952327\n",
      "Using motif +AT3G51060 of width 10.\n",
      "Using motif -AT3G51060 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00241\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G51080 of width 13.\n",
      "Using motif -AT3G51080 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.977282\n",
      "#   Estimated pi_0=0.987037\n",
      "Using motif +AT3G52440 of width 11.\n",
      "Using motif -AT3G52440 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.898039\n",
      "#   Estimated pi_0=0.902617\n",
      "Using motif +AT3G53200 of width 15.\n",
      "Using motif -AT3G53200 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8906\n",
      "#   Estimated pi_0=0.904545\n",
      "Using motif +AT3G54340 of width 15.\n",
      "Using motif -AT3G54340 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8956\n",
      "#   Estimated pi_0=0.901132\n",
      "Using motif +AT3G54990 of width 8.\n",
      "Using motif -AT3G54990 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G55370 of width 21.\n",
      "Using motif -AT3G55370 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 33 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.868571\n",
      "#   Estimated pi_0=0.875217\n",
      "Using motif +AT3G56400 of width 13.\n",
      "Using motif -AT3G56400 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.964348\n",
      "#   Estimated pi_0=0.971026\n",
      "Using motif +AT3G56850 of width 15.\n",
      "Using motif -AT3G56850 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00051\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G57230 of width 15.\n",
      "Using motif -AT3G57230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921386\n",
      "#   Estimated pi_0=0.933178\n",
      "Using motif +AT3G57600 of width 11.\n",
      "Using motif -AT3G57600 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G57920 of width 14.\n",
      "Using motif -AT3G57920 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.98451\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G58630 of width 15.\n",
      "Using motif -AT3G58630 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00291\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G58780 of width 18.\n",
      "Using motif -AT3G58780 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9378\n",
      "#   Estimated pi_0=0.948155\n",
      "Using motif +AT3G59060 of width 8.\n",
      "Using motif -AT3G59060 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999179\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G60030 of width 8.\n",
      "Using motif -AT3G60030 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00079\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G60490 of width 14.\n",
      "Using motif -AT3G60490 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G60530 of width 15.\n",
      "Using motif -AT3G60530 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9686\n",
      "#   Estimated pi_0=0.984586\n",
      "Using motif +AT3G60580 of width 8.\n",
      "Using motif -AT3G60580 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9556\n",
      "#   Estimated pi_0=0.9556\n",
      "Using motif +AT3G61120 of width 18.\n",
      "Using motif -AT3G61120 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8272\n",
      "#   Estimated pi_0=0.830095\n",
      "Using motif +AT3G61150 of width 11.\n",
      "Using motif -AT3G61150 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7956\n",
      "#   Estimated pi_0=0.7956\n",
      "Using motif +AT3G61250 of width 15.\n",
      "Using motif -AT3G61250 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9252\n",
      "#   Estimated pi_0=0.938291\n",
      "Using motif +AT3G61890 of width 10.\n",
      "Using motif -AT3G61890 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913905\n",
      "#   Estimated pi_0=0.921085\n",
      "Using motif +AT3G61910 of width 13.\n",
      "Using motif -AT3G61910 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00162\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT3G62420 of width 15.\n",
      "Using motif -AT3G62420 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0008\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G00050 of width 8.\n",
      "Using motif -AT4G00050 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G00250 of width 13.\n",
      "Using motif -AT4G00250 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9204\n",
      "#   Estimated pi_0=0.929672\n",
      "Using motif +AT4G00270 of width 8.\n",
      "Using motif -AT4G00270 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G00730 of width 11.\n",
      "Using motif -AT4G00730 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8218\n",
      "#   Estimated pi_0=0.8218\n",
      "Using motif +AT4G01250 of width 13.\n",
      "Using motif -AT4G01250 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940561\n",
      "#   Estimated pi_0=0.952698\n",
      "Using motif +AT4G01280 of width 11.\n",
      "Using motif -AT4G01280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8406\n",
      "#   Estimated pi_0=0.845049\n",
      "Using motif +AT4G01500 of width 16.\n",
      "Using motif -AT4G01500 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999898\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G01540 of width 15.\n",
      "Using motif -AT4G01540 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997409\n",
      "#   Estimated pi_0=0.998894\n",
      "Using motif +AT4G01550 of width 15.\n",
      "Using motif -AT4G01550 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999095\n",
      "#   Estimated pi_0=0.999095\n",
      "Using motif +AT4G01680 of width 14.\n",
      "Using motif -AT4G01680 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.996598\n",
      "#   Estimated pi_0=0.998794\n",
      "Using motif +AT4G01720 of width 19.\n",
      "Using motif -AT4G01720 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9268\n",
      "#   Estimated pi_0=0.930526\n",
      "Using motif +AT4G04450 of width 21.\n",
      "Using motif -AT4G04450 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944078\n",
      "#   Estimated pi_0=0.956825\n",
      "Using motif +AT4G05100 of width 14.\n",
      "Using motif -AT4G05100 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9496\n",
      "#   Estimated pi_0=0.957876\n",
      "Using motif +AT4G06746 of width 19.\n",
      "Using motif -AT4G06746 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G10350 of width 21.\n",
      "Using motif -AT4G10350 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9024\n",
      "#   Estimated pi_0=0.906275\n",
      "Using motif +AT4G11660 of width 14.\n",
      "Using motif -AT4G11660 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.968976\n",
      "#   Estimated pi_0=0.980914\n",
      "Using motif +AT4G14410 of width 10.\n",
      "Using motif -AT4G14410 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G14770 of width 15.\n",
      "Using motif -AT4G14770 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.858627\n",
      "#   Estimated pi_0=0.860194\n",
      "Using motif +AT4G15090 of width 15.\n",
      "Using motif -AT4G15090 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G16110 of width 10.\n",
      "Using motif -AT4G16110 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94354\n",
      "#   Estimated pi_0=0.950709\n",
      "Using motif +AT4G16150 of width 18.\n",
      "Using motif -AT4G16150 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.988113\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G16430 of width 8.\n",
      "Using motif -AT4G16430 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G16750 of width 15.\n",
      "Using motif -AT4G16750 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G17460 of width 10.\n",
      "Using motif -AT4G17460 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9156\n",
      "#   Estimated pi_0=0.929559\n",
      "Using motif +AT4G17490 of width 21.\n",
      "Using motif -AT4G17490 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G17785 of width 14.\n",
      "Using motif -AT4G17785 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8932\n",
      "#   Estimated pi_0=0.907627\n",
      "Using motif +AT4G17880 of width 8.\n",
      "Using motif -AT4G17880 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.989808\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G17980 of width 15.\n",
      "Using motif -AT4G17980 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.965\n",
      "#   Estimated pi_0=0.972083\n",
      "Using motif +AT4G18170 of width 13.\n",
      "Using motif -AT4G18170 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9758\n",
      "#   Estimated pi_0=0.986625\n",
      "Using motif +AT4G18390 of width 10.\n",
      "Using motif -AT4G18390 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G18450 of width 17.\n",
      "Using motif -AT4G18450 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G18770 of width 15.\n",
      "Using motif -AT4G18770 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.918824\n",
      "#   Estimated pi_0=0.928036\n",
      "Using motif +AT4G18880 of width 19.\n",
      "Using motif -AT4G18880 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.920392\n",
      "#   Estimated pi_0=0.930476\n",
      "Using motif +AT4G18890 of width 11.\n",
      "Using motif -AT4G18890 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G18960 of width 19.\n",
      "Using motif -AT4G18960 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8876\n",
      "#   Estimated pi_0=0.8876\n",
      "Using motif +AT4G21030 of width 19.\n",
      "Using motif -AT4G21030 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8608\n",
      "#   Estimated pi_0=0.861569\n",
      "Using motif +AT4G21040 of width 14.\n",
      "Using motif -AT4G21040 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.905\n",
      "#   Estimated pi_0=0.915798\n",
      "Using motif +AT4G21080 of width 14.\n",
      "Using motif -AT4G21080 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8876\n",
      "#   Estimated pi_0=0.902353\n",
      "Using motif +AT4G21330 of width 14.\n",
      "Using motif -AT4G21330 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.967129\n",
      "#   Estimated pi_0=0.975188\n",
      "Using motif +AT4G22070 of width 19.\n",
      "Using motif -AT4G22070 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9754\n",
      "#   Estimated pi_0=0.990601\n",
      "Using motif +AT4G23550 of width 11.\n",
      "Using motif -AT4G23550 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9592\n",
      "#   Estimated pi_0=0.971639\n",
      "Using motif +AT4G23750 of width 8.\n",
      "Using motif -AT4G23750 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00372\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G24020 of width 15.\n",
      "Using motif -AT4G24020 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9762\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G24240 of width 14.\n",
      "Using motif -AT4G24240 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.927048\n",
      "#   Estimated pi_0=0.941583\n",
      "Using motif +AT4G24470 of width 15.\n",
      "Using motif -AT4G24470 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9334\n",
      "#   Estimated pi_0=0.942957\n",
      "Using motif +AT4G25470 of width 15.\n",
      "Using motif -AT4G25470 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G25480 of width 14.\n",
      "Using motif -AT4G25480 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00182\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G25490 of width 12.\n",
      "Using motif -AT4G25490 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G26640 of width 13.\n",
      "Using motif -AT4G26640 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.957843\n",
      "#   Estimated pi_0=0.957843\n",
      "Using motif +AT4G27950 of width 21.\n",
      "Using motif -AT4G27950 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00121\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G28110 of width 15.\n",
      "Using motif -AT4G28110 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.93098\n",
      "#   Estimated pi_0=0.940185\n",
      "Using motif +AT4G28140 of width 20.\n",
      "Using motif -AT4G28140 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00291\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G28500 of width 15.\n",
      "Using motif -AT4G28500 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9248\n",
      "#   Estimated pi_0=0.938992\n",
      "Using motif +AT4G29000 of width 10.\n",
      "Using motif -AT4G29000 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8876\n",
      "#   Estimated pi_0=0.900893\n",
      "Using motif +AT4G29230 of width 15.\n",
      "Using motif -AT4G29230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94819\n",
      "#   Estimated pi_0=0.95328\n",
      "Using motif +AT4G30080 of width 21.\n",
      "Using motif -AT4G30080 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.884854\n",
      "#   Estimated pi_0=0.888037\n",
      "Using motif +AT4G30980 of width 15.\n",
      "Using motif -AT4G30980 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G31060 of width 20.\n",
      "Using motif -AT4G31060 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G31550 of width 14.\n",
      "Using motif -AT4G31550 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9344\n",
      "#   Estimated pi_0=0.950333\n",
      "Using motif +AT4G31610 of width 10.\n",
      "Using motif -AT4G31610 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.95\n",
      "#   Estimated pi_0=0.95339\n",
      "Using motif +AT4G31800 of width 11.\n",
      "Using motif -AT4G31800 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9392\n",
      "#   Estimated pi_0=0.944696\n",
      "Using motif +AT4G31920 of width 8.\n",
      "Using motif -AT4G31920 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.963019\n",
      "#   Estimated pi_0=0.982317\n",
      "Using motif +AT4G32730 of width 15.\n",
      "Using motif -AT4G32730 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9468\n",
      "#   Estimated pi_0=0.96386\n",
      "Using motif +AT4G32800 of width 16.\n",
      "Using motif -AT4G32800 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G32890 of width 11.\n",
      "Using motif -AT4G32890 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.976543\n",
      "#   Estimated pi_0=0.98236\n",
      "Using motif +AT4G33280 of width 21.\n",
      "Using motif -AT4G33280 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9072\n",
      "#   Estimated pi_0=0.92\n",
      "Using motif +AT4G34000 of width 8.\n",
      "Using motif -AT4G34000 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G34410 of width 15.\n",
      "Using motif -AT4G34410 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00251\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G34590 of width 15.\n",
      "Using motif -AT4G34590 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.996739\n",
      "#   Estimated pi_0=0.999598\n",
      "Using motif +AT4G35550 of width 10.\n",
      "Using motif -AT4G35550 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9\n",
      "#   Estimated pi_0=0.901765\n",
      "Using motif +AT4G35580 of width 9.\n",
      "Using motif -AT4G35580 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.873578\n",
      "#   Estimated pi_0=0.879273\n",
      "Using motif +AT4G36160 of width 15.\n",
      "Using motif -AT4G36160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.977009\n",
      "#   Estimated pi_0=0.989102\n",
      "Using motif +AT4G36620 of width 11.\n",
      "Using motif -AT4G36620 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.985487\n",
      "#   Estimated pi_0=0.999167\n",
      "Using motif +AT4G36730 of width 10.\n",
      "Using motif -AT4G36730 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G36740 of width 21.\n",
      "Using motif -AT4G36740 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8104\n",
      "#   Estimated pi_0=0.8104\n",
      "Using motif +AT4G36780 of width 11.\n",
      "Using motif -AT4G36780 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G36900 of width 10.\n",
      "Using motif -AT4G36900 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00342\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G36930 of width 9.\n",
      "Using motif -AT4G36930 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G37180 of width 11.\n",
      "Using motif -AT4G37180 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9138\n",
      "#   Estimated pi_0=0.920561\n",
      "Using motif +AT4G37260 of width 11.\n",
      "Using motif -AT4G37260 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9034\n",
      "#   Estimated pi_0=0.907037\n",
      "Using motif +AT4G37750 of width 21.\n",
      "Using motif -AT4G37750 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT4G37790 of width 10.\n",
      "Using motif -AT4G37790 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948235\n",
      "#   Estimated pi_0=0.964628\n",
      "Using motif +AT4G38000 of width 28.\n",
      "Using motif -AT4G38000 of width 28.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 42 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.851346\n",
      "#   Estimated pi_0=0.862903\n",
      "Using motif +AT4G38620 of width 15.\n",
      "Using motif -AT4G38620 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.942376\n",
      "#   Estimated pi_0=0.942376\n",
      "Using motif +AT4G38910 of width 30.\n",
      "Using motif -AT4G38910 of width 30.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.950548\n",
      "#   Estimated pi_0=0.958444\n",
      "Using motif +AT4G40060 of width 8.\n",
      "Using motif -AT4G40060 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.887129\n",
      "#   Estimated pi_0=0.898151\n",
      "Using motif +AT5G01380 of width 17.\n",
      "Using motif -AT5G01380 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933654\n",
      "#   Estimated pi_0=0.949437\n",
      "Using motif +AT5G01900 of width 8.\n",
      "Using motif -AT5G01900 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966607\n",
      "#   Estimated pi_0=0.976589\n",
      "Using motif +AT5G02320 of width 15.\n",
      "Using motif -AT5G02320 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9312\n",
      "#   Estimated pi_0=0.93283\n",
      "Using motif +AT5G02460 of width 21.\n",
      "Using motif -AT5G02460 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 33 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.846789\n",
      "#   Estimated pi_0=0.849189\n",
      "Using motif +AT5G02840 of width 10.\n",
      "Using motif -AT5G02840 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.884466\n",
      "#   Estimated pi_0=0.894261\n",
      "Using motif +AT5G03150 of width 20.\n",
      "Using motif -AT5G03150 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.906731\n",
      "#   Estimated pi_0=0.90955\n",
      "Using motif +AT5G03790 of width 13.\n",
      "Using motif -AT5G03790 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.819604\n",
      "#   Estimated pi_0=0.823529\n",
      "Using motif +AT5G04340 of width 10.\n",
      "Using motif -AT5G04340 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9328\n",
      "#   Estimated pi_0=0.944259\n",
      "Using motif +AT5G04390 of width 11.\n",
      "Using motif -AT5G04390 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.954653\n",
      "#   Estimated pi_0=0.966441\n",
      "Using motif +AT5G04410 of width 12.\n",
      "Using motif -AT5G04410 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.977282\n",
      "#   Estimated pi_0=0.997172\n",
      "Using motif +AT5G04760 of width 10.\n",
      "Using motif -AT5G04760 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.949661\n",
      "#   Estimated pi_0=0.950085\n",
      "Using motif +AT5G05090 of width 11.\n",
      "Using motif -AT5G05090 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.934038\n",
      "#   Estimated pi_0=0.955152\n",
      "Using motif +AT5G05550 of width 15.\n",
      "Using motif -AT5G05550 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G05790 of width 15.\n",
      "Using motif -AT5G05790 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.916634\n",
      "#   Estimated pi_0=0.931774\n",
      "Using motif +AT5G06100 of width 11.\n",
      "Using motif -AT5G06100 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.894455\n",
      "#   Estimated pi_0=0.898846\n",
      "Using motif +AT5G06839 of width 15.\n",
      "Using motif -AT5G06839 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997761\n",
      "#   Estimated pi_0=0.999799\n",
      "Using motif +AT5G06950 of width 20.\n",
      "Using motif -AT5G06950 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.956111\n",
      "#   Estimated pi_0=0.965161\n",
      "Using motif +AT5G06960 of width 14.\n",
      "Using motif -AT5G06960 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969615\n",
      "#   Estimated pi_0=0.979559\n",
      "Using motif +AT5G07100 of width 13.\n",
      "Using motif -AT5G07100 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.976508\n",
      "#   Estimated pi_0=0.98593\n",
      "Using motif +AT5G07310 of width 15.\n",
      "Using motif -AT5G07310 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0009\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G07680 of width 21.\n",
      "Using motif -AT5G07680 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9046\n",
      "#   Estimated pi_0=0.914906\n",
      "Using motif +AT5G08070 of width 11.\n",
      "Using motif -AT5G08070 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G08130 of width 14.\n",
      "Using motif -AT5G08130 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999899\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G08330 of width 14.\n",
      "Using motif -AT5G08330 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G08520 of width 13.\n",
      "Using motif -AT5G08520 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 23 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.920377\n",
      "#   Estimated pi_0=0.927143\n",
      "Using motif +AT5G09410 of width 12.\n",
      "Using motif -AT5G09410 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00322\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G10030 of width 11.\n",
      "Using motif -AT5G10030 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.992308\n",
      "#   Estimated pi_0=0.997423\n",
      "Using motif +AT5G10140 of width 21.\n",
      "Using motif -AT5G10140 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 28 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9186\n",
      "#   Estimated pi_0=0.922991\n",
      "Using motif +AT5G10280 of width 10.\n",
      "Using motif -AT5G10280 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.974\n",
      "#   Estimated pi_0=0.990939\n",
      "Using motif +AT5G10510 of width 15.\n",
      "Using motif -AT5G10510 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.981273\n",
      "#   Estimated pi_0=0.993896\n",
      "Using motif +AT5G11260 of width 15.\n",
      "Using motif -AT5G11260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G11510 of width 15.\n",
      "Using motif -AT5G11510 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9448\n",
      "#   Estimated pi_0=0.96\n",
      "Using motif +AT5G12870 of width 10.\n",
      "Using motif -AT5G12870 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G13080 of width 13.\n",
      "Using motif -AT5G13080 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975\n",
      "#   Estimated pi_0=0.988902\n",
      "Using motif +AT5G13180 of width 19.\n",
      "Using motif -AT5G13180 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9046\n",
      "#   Estimated pi_0=0.914766\n",
      "Using motif +AT5G13790 of width 19.\n",
      "Using motif -AT5G13790 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 29 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.887327\n",
      "#   Estimated pi_0=0.890385\n",
      "Using motif +AT5G13910 of width 21.\n",
      "Using motif -AT5G13910 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G14340 of width 13.\n",
      "Using motif -AT5G14340 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.939802\n",
      "#   Estimated pi_0=0.952063\n",
      "Using motif +AT5G14960 of width 20.\n",
      "Using motif -AT5G14960 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94\n",
      "#   Estimated pi_0=0.95\n",
      "Using motif +AT5G15800 of width 18.\n",
      "Using motif -AT5G15800 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.907\n",
      "#   Estimated pi_0=0.910196\n",
      "Using motif +AT5G15830 of width 15.\n",
      "Using motif -AT5G15830 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999898\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G16600 of width 13.\n",
      "Using motif -AT5G16600 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973036\n",
      "#   Estimated pi_0=0.992203\n",
      "Using motif +AT5G16820 of width 12.\n",
      "Using motif -AT5G16820 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998\n",
      "#   Estimated pi_0=0.999196\n",
      "Using motif +AT5G17300 of width 14.\n",
      "Using motif -AT5G17300 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.85505\n",
      "#   Estimated pi_0=0.855686\n",
      "Using motif +AT5G17430 of width 20.\n",
      "Using motif -AT5G17430 of width 20.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 36 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.923016\n",
      "#   Estimated pi_0=0.932651\n",
      "Using motif +AT5G17800 of width 15.\n",
      "Using motif -AT5G17800 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9042\n",
      "#   Estimated pi_0=0.9042\n",
      "Using motif +AT5G18090 of width 15.\n",
      "Using motif -AT5G18090 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.955968\n",
      "#   Estimated pi_0=0.968588\n",
      "Using motif +AT5G18270 of width 19.\n",
      "Using motif -AT5G18270 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9354\n",
      "#   Estimated pi_0=0.944211\n",
      "Using motif +AT5G18450 of width 15.\n",
      "Using motif -AT5G18450 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0002\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G18560 of width 15.\n",
      "Using motif -AT5G18560 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G18830 of width 10.\n",
      "Using motif -AT5G18830 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G19790 of width 15.\n",
      "Using motif -AT5G19790 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00271\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G20240 of width 14.\n",
      "Using motif -AT5G20240 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 31 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.952231\n",
      "#   Estimated pi_0=0.959497\n",
      "Using motif +AT5G22220 of width 10.\n",
      "Using motif -AT5G22220 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00362\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G22570 of width 10.\n",
      "Using motif -AT5G22570 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999698\n",
      "#   Estimated pi_0=0.999698\n",
      "Using motif +AT5G22890 of width 18.\n",
      "Using motif -AT5G22890 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9402\n",
      "#   Estimated pi_0=0.952061\n",
      "Using motif +AT5G22990 of width 15.\n",
      "Using motif -AT5G22990 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973267\n",
      "#   Estimated pi_0=0.999192\n",
      "Using motif +AT5G23280 of width 11.\n",
      "Using motif -AT5G23280 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G24110 of width 11.\n",
      "Using motif -AT5G24110 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.967723\n",
      "#   Estimated pi_0=0.996632\n",
      "Using motif +AT5G25190 of width 19.\n",
      "Using motif -AT5G25190 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G25390 of width 21.\n",
      "Using motif -AT5G25390 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999898\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G25475 of width 6.\n",
      "Using motif -AT5G25475 of width 6.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G25810 of width 15.\n",
      "Using motif -AT5G25810 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G25830 of width 13.\n",
      "Using motif -AT5G25830 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966667\n",
      "#   Estimated pi_0=0.9875\n",
      "Using motif +AT5G26170 of width 13.\n",
      "Using motif -AT5G26170 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9368\n",
      "#   Estimated pi_0=0.952031\n",
      "Using motif +AT5G28300 of width 10.\n",
      "Using motif -AT5G28300 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.960792\n",
      "#   Estimated pi_0=0.97937\n",
      "Using motif +AT5G28770 of width 12.\n",
      "Using motif -AT5G28770 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00181\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G29000 of width 15.\n",
      "Using motif -AT5G29000 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978993\n",
      "#   Estimated pi_0=0.991383\n",
      "Using motif +AT5G37020 of width 9.\n",
      "Using motif -AT5G37020 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.980588\n",
      "#   Estimated pi_0=0.986857\n",
      "Using motif +AT5G38800 of width 12.\n",
      "Using motif -AT5G38800 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00112\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G38860 of width 18.\n",
      "Using motif -AT5G38860 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00211\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G39610 of width 15.\n",
      "Using motif -AT5G39610 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9444\n",
      "#   Estimated pi_0=0.959016\n",
      "Using motif +AT5G39660 of width 10.\n",
      "Using motif -AT5G39660 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.94283\n",
      "#   Estimated pi_0=0.947612\n",
      "Using motif +AT5G39760 of width 22.\n",
      "Using motif -AT5G39760 of width 22.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 22 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8152\n",
      "#   Estimated pi_0=0.816471\n",
      "Using motif +AT5G39820 of width 18.\n",
      "Using motif -AT5G39820 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00302\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G40330 of width 11.\n",
      "Using motif -AT5G40330 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933861\n",
      "#   Estimated pi_0=0.942143\n",
      "Using motif +AT5G41570 of width 14.\n",
      "Using motif -AT5G41570 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9578\n",
      "#   Estimated pi_0=0.962407\n",
      "Using motif +AT5G42520 of width 21.\n",
      "Using motif -AT5G42520 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 33 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.920333\n",
      "#   Estimated pi_0=0.928194\n",
      "Using motif +AT5G42630 of width 8.\n",
      "Using motif -AT5G42630 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921524\n",
      "#   Estimated pi_0=0.939528\n",
      "Using motif +AT5G43410 of width 10.\n",
      "Using motif -AT5G43410 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00342\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G43840 of width 16.\n",
      "Using motif -AT5G43840 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.986471\n",
      "#   Estimated pi_0=0.994011\n",
      "Using motif +AT5G44210 of width 21.\n",
      "Using motif -AT5G44210 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0005\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G45300 of width 15.\n",
      "Using motif -AT5G45300 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G45580 of width 14.\n",
      "Using motif -AT5G45580 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.924615\n",
      "#   Estimated pi_0=0.933333\n",
      "Using motif +AT5G46350 of width 15.\n",
      "Using motif -AT5G46350 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 25 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9144\n",
      "#   Estimated pi_0=0.920566\n",
      "Using motif +AT5G46590 of width 16.\n",
      "Using motif -AT5G46590 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 20 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9178\n",
      "#   Estimated pi_0=0.93\n",
      "Using motif +AT5G46760 of width 8.\n",
      "Using motif -AT5G46760 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 1 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0003\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G46830 of width 12.\n",
      "Using motif -AT5G46830 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9798\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G47220 of width 21.\n",
      "Using motif -AT5G47220 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999899\n",
      "#   Estimated pi_0=0.999899\n",
      "Using motif +AT5G47230 of width 15.\n",
      "Using motif -AT5G47230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0007\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G47370 of width 11.\n",
      "Using motif -AT5G47370 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8536\n",
      "#   Estimated pi_0=0.867238\n",
      "Using motif +AT5G47390 of width 15.\n",
      "Using motif -AT5G47390 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 17 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913091\n",
      "#   Estimated pi_0=0.921774\n",
      "Using motif +AT5G47660 of width 15.\n",
      "Using motif -AT5G47660 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8794\n",
      "#   Estimated pi_0=0.888468\n",
      "Using motif +AT5G48560 of width 8.\n",
      "Using motif -AT5G48560 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G48670 of width 6.\n",
      "Using motif -AT5G48670 of width 6.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G49300 of width 27.\n",
      "Using motif -AT5G49300 of width 27.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 26 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938696\n",
      "#   Estimated pi_0=0.944487\n",
      "Using motif +AT5G49330 of width 10.\n",
      "Using motif -AT5G49330 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9678\n",
      "#   Estimated pi_0=0.978832\n",
      "Using motif +AT5G50570 of width 11.\n",
      "Using motif -AT5G50570 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9542\n",
      "#   Estimated pi_0=0.955842\n",
      "Using motif +AT5G51190 of width 15.\n",
      "Using motif -AT5G51190 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00141\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G51910 of width 10.\n",
      "Using motif -AT5G51910 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G51990 of width 13.\n",
      "Using motif -AT5G51990 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G52020 of width 15.\n",
      "Using motif -AT5G52020 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00231\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G52170 of width 11.\n",
      "Using motif -AT5G52170 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.7956\n",
      "#   Estimated pi_0=0.802745\n",
      "Using motif +AT5G52660 of width 11.\n",
      "Using motif -AT5G52660 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8558\n",
      "#   Estimated pi_0=0.8558\n",
      "Using motif +AT5G52830 of width 13.\n",
      "Using motif -AT5G52830 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.930891\n",
      "#   Estimated pi_0=0.935\n",
      "Using motif +AT5G53950 of width 19.\n",
      "Using motif -AT5G53950 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.931\n",
      "#   Estimated pi_0=0.936545\n",
      "Using motif +AT5G54230 of width 15.\n",
      "Using motif -AT5G54230 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.907525\n",
      "#   Estimated pi_0=0.910857\n",
      "Using motif +AT5G56110 of width 15.\n",
      "Using motif -AT5G56110 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 13 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.926337\n",
      "#   Estimated pi_0=0.935172\n",
      "Using motif +AT5G56270 of width 8.\n",
      "Using motif -AT5G56270 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999495\n",
      "#   Estimated pi_0=0.999799\n",
      "Using motif +AT5G56840 of width 15.\n",
      "Using motif -AT5G56840 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.908571\n",
      "#   Estimated pi_0=0.918195\n",
      "Using motif +AT5G58080 of width 13.\n",
      "Using motif -AT5G58080 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.954118\n",
      "#   Estimated pi_0=0.966711\n",
      "Using motif +AT5G58620 of width 6.\n",
      "Using motif -AT5G58620 of width 6.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 0 sequences.\n",
      "Computing q-values.\n",
      "Using motif +AT5G58850 of width 11.\n",
      "Using motif -AT5G58850 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9744\n",
      "#   Estimated pi_0=0.996168\n",
      "Using motif +AT5G58900 of width 15.\n",
      "Using motif -AT5G58900 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 21 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8556\n",
      "#   Estimated pi_0=0.860194\n",
      "Using motif +AT5G59780 of width 10.\n",
      "Using motif -AT5G59780 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9062\n",
      "#   Estimated pi_0=0.914528\n",
      "Using motif +AT5G60120 of width 10.\n",
      "Using motif -AT5G60120 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9288\n",
      "#   Estimated pi_0=0.932233\n",
      "Using motif +AT5G60130 of width 19.\n",
      "Using motif -AT5G60130 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 32 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.823762\n",
      "#   Estimated pi_0=0.837699\n",
      "Using motif +AT5G60850 of width 14.\n",
      "Using motif -AT5G60850 of width 14.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.890693\n",
      "#   Estimated pi_0=0.903566\n",
      "Using motif +AT5G60970 of width 8.\n",
      "Using motif -AT5G60970 of width 8.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 7 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00302\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G61270 of width 15.\n",
      "Using motif -AT5G61270 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 4 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00111\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G61600 of width 19.\n",
      "Using motif -AT5G61600 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G61620 of width 15.\n",
      "Using motif -AT5G61620 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.905577\n",
      "#   Estimated pi_0=0.911379\n",
      "Using motif +AT5G61850 of width 19.\n",
      "Using motif -AT5G61850 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 2 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9136\n",
      "#   Estimated pi_0=0.925\n",
      "Using motif +AT5G62000 of width 10.\n",
      "Using motif -AT5G62000 of width 10.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998519\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G62020 of width 12.\n",
      "Using motif -AT5G62020 of width 12.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 11 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9124\n",
      "#   Estimated pi_0=0.922453\n",
      "Using motif +AT5G62165 of width 7.\n",
      "Using motif -AT5G62165 of width 7.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 19 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.963137\n",
      "#   Estimated pi_0=0.986258\n",
      "Using motif +AT5G62320 of width 13.\n",
      "Using motif -AT5G62320 of width 13.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.912157\n",
      "#   Estimated pi_0=0.918727\n",
      "Using motif +AT5G62380 of width 21.\n",
      "Using motif -AT5G62380 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9062\n",
      "#   Estimated pi_0=0.92736\n",
      "Using motif +AT5G62470 of width 18.\n",
      "Using motif -AT5G62470 of width 18.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.931215\n",
      "#   Estimated pi_0=0.931927\n",
      "Using motif +AT5G62940 of width 21.\n",
      "Using motif -AT5G62940 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 24 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.872632\n",
      "#   Estimated pi_0=0.873333\n",
      "Using motif +AT5G63090 of width 21.\n",
      "Using motif -AT5G63090 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 5 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999391\n",
      "#   Estimated pi_0=0.999799\n",
      "Using motif +AT5G63260 of width 11.\n",
      "Using motif -AT5G63260 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.925283\n",
      "#   Estimated pi_0=0.932212\n",
      "Using motif +AT5G64060 of width 21.\n",
      "Using motif -AT5G64060 of width 21.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.953097\n",
      "#   Estimated pi_0=0.961138\n",
      "Using motif +AT5G64220 of width 9.\n",
      "Using motif -AT5G64220 of width 9.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00261\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G64750 of width 19.\n",
      "Using motif -AT5G64750 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00312\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G65130 of width 15.\n",
      "Using motif -AT5G65130 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 8 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G65210 of width 15.\n",
      "Using motif -AT5G65210 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.978716\n",
      "#   Estimated pi_0=0.988024\n",
      "Using motif +AT5G65310 of width 11.\n",
      "Using motif -AT5G65310 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 12 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8596\n",
      "#   Estimated pi_0=0.859604\n",
      "Using motif +AT5G65410 of width 15.\n",
      "Using motif -AT5G65410 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 10 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9032\n",
      "#   Estimated pi_0=0.918862\n",
      "Using motif +AT5G65510 of width 11.\n",
      "Using motif -AT5G65510 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 3 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.941\n",
      "#   Estimated pi_0=0.957398\n",
      "Using motif +AT5G66300 of width 16.\n",
      "Using motif -AT5G66300 of width 16.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 16 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9362\n",
      "#   Estimated pi_0=0.945565\n",
      "Using motif +AT5G66700 of width 11.\n",
      "Using motif -AT5G66700 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 18 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8344\n",
      "#   Estimated pi_0=0.8344\n",
      "Using motif +AT5G66730 of width 17.\n",
      "Using motif -AT5G66730 of width 17.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 27 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.911068\n",
      "#   Estimated pi_0=0.927778\n",
      "Using motif +AT5G66940 of width 29.\n",
      "Using motif -AT5G66940 of width 29.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 38 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8714\n",
      "#   Estimated pi_0=0.882362\n",
      "Using motif +AT5G67000 of width 15.\n",
      "Using motif -AT5G67000 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 6 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G67190 of width 15.\n",
      "Using motif -AT5G67190 of width 15.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT5G67300 of width 19.\n",
      "Using motif -AT5G67300 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 14 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9122\n",
      "#   Estimated pi_0=0.921321\n",
      "Using motif +AT5G67450 of width 11.\n",
      "Using motif -AT5G67450 of width 11.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 15 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975221\n",
      "#   Estimated pi_0=0.999485\n",
      "Using motif +AT5G67580 of width 19.\n",
      "Using motif -AT5G67580 of width 19.\n",
      "Finding best site passing the output threshold in each of the 84 sequences.\n",
      "Found a best site in 9 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.931\n",
      "#   Estimated pi_0=0.946016\n"
     ]
    }
   ],
   "source": [
    "! sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d:/data memesuite/memesuite \\\n",
    "  fimo --oc \"/data/NLRs_LRRs/fimo_NLRs_up\" \\\n",
    "    \"/data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\" \"/data/NLRs_LRRs/all_NLRs_up_peaks.fasta\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a67ca9f7-3b08-47e0-aac9-739ef6a50667",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:work2] *",
   "language": "python",
   "name": "conda-env-work2-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
