{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "387496df-c5fd-48b6-a68e-67a4ed6152dd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import glob\n",
    "import os\n",
    "from os.path import exists\n",
    "import re\n",
    "import pandas as pd\n",
    "from multiprocessing import Pool\n",
    "import pyreadr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "92ecc9f4-6933-4005-af23-eef97df9b9a4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "WGCNA_path = './WGCNA_ATAC_integration'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "7abef394-e052-4ed7-ae7a-0e7e2a46d721",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24266"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_variable_peaks_in_RNA_seq = pyreadr.read_r(os.path.join(WGCNA_path, \"non_variable_peaks_in_RNA_seq.rds\"))\n",
    "non_variable_peaks_in_RNA_seq = non_variable_peaks_in_RNA_seq[None]\n",
    "len(non_variable_peaks_in_RNA_seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "e6e16b74-88b9-4623-93d8-da141bdce8d5",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Load the RDS file into Python\n",
    "result = pyreadr.read_r(os.path.join(WGCNA_path, \"connectivity_and_peak_info_DM1_DM2d.rds\"))\n",
    "df = result[None]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e3b3238-3d27-444e-aa3c-8dc83db1f8d2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "c049af4b-f4dc-436a-b43a-0cc9ce039120",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Create an empty dictionary to store the subsets\n",
    "module_dict = {}\n",
    "\n",
    "# Loop through the unique values in the 'module' column\n",
    "for module in df['module'].unique():\n",
    "    # Subset the dataframe where 'module' equals the current unique value\n",
    "    module_dict[module] = df[df['module'] == module]\n",
    "\n",
    "# Now 'module_dict' contains each module as a key with its corresponding subset dataframe as its value.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "4e42d30f-0490-4713-9882-c31ef45dcecb",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>connectivity_rowSums_adj</th>\n",
       "      <th>module</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>DM6/DM7</th>\n",
       "      <th>DM1/DM2d</th>\n",
       "      <th>DM10/DM11</th>\n",
       "      <th>AvrRpt2-6h</th>\n",
       "      <th>AvrRpt2-12h</th>\n",
       "      <th>AvrRpt2-20h</th>\n",
       "      <th>...</th>\n",
       "      <th>Fold</th>\n",
       "      <th>p.value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneChr</th>\n",
       "      <th>geneStart</th>\n",
       "      <th>geneEnd</th>\n",
       "      <th>geneLength</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>distanceToTSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G63720</td>\n",
       "      <td>1954.528305</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>AT1G63720</td>\n",
       "      <td>1.935779</td>\n",
       "      <td>3.970436</td>\n",
       "      <td>6.033027</td>\n",
       "      <td>0.887911</td>\n",
       "      <td>1.510569</td>\n",
       "      <td>3.775730</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.406409</td>\n",
       "      <td>2.593965e-07</td>\n",
       "      <td>1.476102e-06</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>23635745</td>\n",
       "      <td>23637577</td>\n",
       "      <td>1833</td>\n",
       "      <td>2</td>\n",
       "      <td>699.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT3G54150</td>\n",
       "      <td>1952.887628</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>EFD</td>\n",
       "      <td>3.815566</td>\n",
       "      <td>5.122284</td>\n",
       "      <td>8.439073</td>\n",
       "      <td>1.157111</td>\n",
       "      <td>3.410757</td>\n",
       "      <td>5.334764</td>\n",
       "      <td>...</td>\n",
       "      <td>0.520665</td>\n",
       "      <td>5.630124e-08</td>\n",
       "      <td>3.492766e-07</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>3</td>\n",
       "      <td>20050564</td>\n",
       "      <td>20052931</td>\n",
       "      <td>2368</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AT5G48657</td>\n",
       "      <td>1947.970337</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>NOI10</td>\n",
       "      <td>3.501474</td>\n",
       "      <td>4.454391</td>\n",
       "      <td>8.090660</td>\n",
       "      <td>1.011472</td>\n",
       "      <td>2.203670</td>\n",
       "      <td>3.545495</td>\n",
       "      <td>...</td>\n",
       "      <td>0.270921</td>\n",
       "      <td>1.657224e-04</td>\n",
       "      <td>6.039474e-04</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>19733882</td>\n",
       "      <td>19735726</td>\n",
       "      <td>1845</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AT5G48657</td>\n",
       "      <td>1947.970337</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>NOI10</td>\n",
       "      <td>3.501474</td>\n",
       "      <td>4.454391</td>\n",
       "      <td>8.090660</td>\n",
       "      <td>1.011472</td>\n",
       "      <td>2.203670</td>\n",
       "      <td>3.545495</td>\n",
       "      <td>...</td>\n",
       "      <td>0.222130</td>\n",
       "      <td>2.069287e-03</td>\n",
       "      <td>5.910763e-03</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>19733882</td>\n",
       "      <td>19735726</td>\n",
       "      <td>1845</td>\n",
       "      <td>2</td>\n",
       "      <td>838.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AT4G23700</td>\n",
       "      <td>1947.947682</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>CHX17</td>\n",
       "      <td>6.202520</td>\n",
       "      <td>6.294546</td>\n",
       "      <td>9.241291</td>\n",
       "      <td>0.900020</td>\n",
       "      <td>3.321431</td>\n",
       "      <td>6.899931</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000460</td>\n",
       "      <td>2.808698e-32</td>\n",
       "      <td>8.609768e-31</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>4</td>\n",
       "      <td>12342376</td>\n",
       "      <td>12346051</td>\n",
       "      <td>3676</td>\n",
       "      <td>2</td>\n",
       "      <td>-708.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6067</th>\n",
       "      <td>AT5G61790</td>\n",
       "      <td>664.442523</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>CNX1</td>\n",
       "      <td>1.128610</td>\n",
       "      <td>1.005367</td>\n",
       "      <td>1.013328</td>\n",
       "      <td>0.093627</td>\n",
       "      <td>1.096341</td>\n",
       "      <td>0.998485</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.510068</td>\n",
       "      <td>3.789737e-12</td>\n",
       "      <td>3.583663e-11</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>24827063</td>\n",
       "      <td>24829900</td>\n",
       "      <td>2838</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6073</th>\n",
       "      <td>AT4G24190</td>\n",
       "      <td>603.862556</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>SHD</td>\n",
       "      <td>1.396126</td>\n",
       "      <td>0.295383</td>\n",
       "      <td>0.784250</td>\n",
       "      <td>-0.194620</td>\n",
       "      <td>0.850157</td>\n",
       "      <td>0.795426</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.625392</td>\n",
       "      <td>3.304481e-30</td>\n",
       "      <td>9.103409e-29</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>4</td>\n",
       "      <td>12551552</td>\n",
       "      <td>12556058</td>\n",
       "      <td>4507</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6074</th>\n",
       "      <td>AT3G41768</td>\n",
       "      <td>571.912538</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>AT3G41768</td>\n",
       "      <td>-0.812099</td>\n",
       "      <td>1.071715</td>\n",
       "      <td>-0.107859</td>\n",
       "      <td>1.196581</td>\n",
       "      <td>1.036284</td>\n",
       "      <td>-0.368824</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.598999</td>\n",
       "      <td>1.260891e-06</td>\n",
       "      <td>6.518600e-06</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>3</td>\n",
       "      <td>14197677</td>\n",
       "      <td>14199484</td>\n",
       "      <td>1808</td>\n",
       "      <td>1</td>\n",
       "      <td>443.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6075</th>\n",
       "      <td>AT2G16660</td>\n",
       "      <td>513.021525</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>AT2G16660</td>\n",
       "      <td>2.005898</td>\n",
       "      <td>-0.166410</td>\n",
       "      <td>-0.635497</td>\n",
       "      <td>-0.099147</td>\n",
       "      <td>0.005229</td>\n",
       "      <td>0.512974</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.474237</td>\n",
       "      <td>4.214350e-10</td>\n",
       "      <td>3.318071e-09</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>7218654</td>\n",
       "      <td>7221890</td>\n",
       "      <td>3237</td>\n",
       "      <td>2</td>\n",
       "      <td>-1727.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6076</th>\n",
       "      <td>AT2G16660</td>\n",
       "      <td>513.021525</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>AT2G16660</td>\n",
       "      <td>2.005898</td>\n",
       "      <td>-0.166410</td>\n",
       "      <td>-0.635497</td>\n",
       "      <td>-0.099147</td>\n",
       "      <td>0.005229</td>\n",
       "      <td>0.512974</td>\n",
       "      <td>...</td>\n",
       "      <td>0.359387</td>\n",
       "      <td>2.768488e-03</td>\n",
       "      <td>7.660940e-03</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>2</td>\n",
       "      <td>7218654</td>\n",
       "      <td>7221890</td>\n",
       "      <td>3237</td>\n",
       "      <td>2</td>\n",
       "      <td>-3609.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2859 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene  connectivity_rowSums_adj  module  gene_name   DM6/DM7  \\\n",
       "0     AT1G63720               1954.528305  MEcyan  AT1G63720  1.935779   \n",
       "1     AT3G54150               1952.887628  MEcyan        EFD  3.815566   \n",
       "2     AT5G48657               1947.970337  MEcyan      NOI10  3.501474   \n",
       "3     AT5G48657               1947.970337  MEcyan      NOI10  3.501474   \n",
       "4     AT4G23700               1947.947682  MEcyan      CHX17  6.202520   \n",
       "...         ...                       ...     ...        ...       ...   \n",
       "6067  AT5G61790                664.442523  MEcyan       CNX1  1.128610   \n",
       "6073  AT4G24190                603.862556  MEcyan        SHD  1.396126   \n",
       "6074  AT3G41768                571.912538  MEcyan  AT3G41768 -0.812099   \n",
       "6075  AT2G16660                513.021525  MEcyan  AT2G16660  2.005898   \n",
       "6076  AT2G16660                513.021525  MEcyan  AT2G16660  2.005898   \n",
       "\n",
       "      DM1/DM2d  DM10/DM11  AvrRpt2-6h  AvrRpt2-12h  AvrRpt2-20h  ...  \\\n",
       "0     3.970436   6.033027    0.887911     1.510569     3.775730  ...   \n",
       "1     5.122284   8.439073    1.157111     3.410757     5.334764  ...   \n",
       "2     4.454391   8.090660    1.011472     2.203670     3.545495  ...   \n",
       "3     4.454391   8.090660    1.011472     2.203670     3.545495  ...   \n",
       "4     6.294546   9.241291    0.900020     3.321431     6.899931  ...   \n",
       "...        ...        ...         ...          ...          ...  ...   \n",
       "6067  1.005367   1.013328    0.093627     1.096341     0.998485  ...   \n",
       "6073  0.295383   0.784250   -0.194620     0.850157     0.795426  ...   \n",
       "6074  1.071715  -0.107859    1.196581     1.036284    -0.368824  ...   \n",
       "6075 -0.166410  -0.635497   -0.099147     0.005229     0.512974  ...   \n",
       "6076 -0.166410  -0.635497   -0.099147     0.005229     0.512974  ...   \n",
       "\n",
       "          Fold       p.value           FDR         annotation  geneChr  \\\n",
       "0    -0.406409  2.593965e-07  1.476102e-06   Promoter (<=1kb)        1   \n",
       "1     0.520665  5.630124e-08  3.492766e-07   Promoter (<=1kb)        3   \n",
       "2     0.270921  1.657224e-04  6.039474e-04   Promoter (<=1kb)        5   \n",
       "3     0.222130  2.069287e-03  5.910763e-03   Promoter (<=1kb)        5   \n",
       "4     1.000460  2.808698e-32  8.609768e-31   Promoter (<=1kb)        4   \n",
       "...        ...           ...           ...                ...      ...   \n",
       "6067 -0.510068  3.789737e-12  3.583663e-11   Promoter (<=1kb)        5   \n",
       "6073 -0.625392  3.304481e-30  9.103409e-29   Promoter (<=1kb)        4   \n",
       "6074 -0.598999  1.260891e-06  6.518600e-06   Promoter (<=1kb)        3   \n",
       "6075 -0.474237  4.214350e-10  3.318071e-09   Promoter (1-2kb)        2   \n",
       "6076  0.359387  2.768488e-03  7.660940e-03  Distal Intergenic        2   \n",
       "\n",
       "      geneStart   geneEnd  geneLength  geneStrand  distanceToTSS  \n",
       "0      23635745  23637577        1833           2          699.0  \n",
       "1      20050564  20052931        2368           2            0.0  \n",
       "2      19733882  19735726        1845           2            0.0  \n",
       "3      19733882  19735726        1845           2          838.0  \n",
       "4      12342376  12346051        3676           2         -708.0  \n",
       "...         ...       ...         ...         ...            ...  \n",
       "6067   24827063  24829900        2838           2            0.0  \n",
       "6073   12551552  12556058        4507           2            0.0  \n",
       "6074   14197677  14199484        1808           1          443.0  \n",
       "6075    7218654   7221890        3237           2        -1727.0  \n",
       "6076    7218654   7221890        3237           2        -3609.0  \n",
       "\n",
       "[2859 rows x 34 columns]"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "module_dict['MEcyan']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "395bb399-abbe-4671-ac3f-faec9d410ccc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "71c59fa1-29f7-47e7-be47-c3fae159cd94",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MEcyan'"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "module_dict['MEcyan']['module'].unique().tolist()[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "ceb832ad-e117-43d5-981e-02e0bb8e5e6c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>connectivity_rowSums_adj</th>\n",
       "      <th>module</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>DM6/DM7</th>\n",
       "      <th>DM1/DM2d</th>\n",
       "      <th>DM10/DM11</th>\n",
       "      <th>AvrRpt2-6h</th>\n",
       "      <th>AvrRpt2-12h</th>\n",
       "      <th>AvrRpt2-20h</th>\n",
       "      <th>...</th>\n",
       "      <th>Fold</th>\n",
       "      <th>p.value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneChr</th>\n",
       "      <th>geneStart</th>\n",
       "      <th>geneEnd</th>\n",
       "      <th>geneLength</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>distanceToTSS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AT1G63720</td>\n",
       "      <td>1954.528305</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>AT1G63720</td>\n",
       "      <td>1.935779</td>\n",
       "      <td>3.970436</td>\n",
       "      <td>6.033027</td>\n",
       "      <td>0.887911</td>\n",
       "      <td>1.510569</td>\n",
       "      <td>3.775730</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.406409</td>\n",
       "      <td>2.593965e-07</td>\n",
       "      <td>1.476102e-06</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>23635745</td>\n",
       "      <td>23637577</td>\n",
       "      <td>1833</td>\n",
       "      <td>2</td>\n",
       "      <td>699.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AT3G54150</td>\n",
       "      <td>1952.887628</td>\n",
       "      <td>MEcyan</td>\n",
       "      <td>EFD</td>\n",
       "      <td>3.815566</td>\n",
       "      <td>5.122284</td>\n",
       "      <td>8.439073</td>\n",
       "      <td>1.157111</td>\n",
       "      <td>3.410757</td>\n",
       "      <td>5.334764</td>\n",
       "      <td>...</td>\n",
       "      <td>0.520665</td>\n",
       "      <td>5.630124e-08</td>\n",
       "      <td>3.492766e-07</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>3</td>\n",
       "      <td>20050564</td>\n",
       "      <td>20052931</td>\n",
       "      <td>2368</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene  connectivity_rowSums_adj  module  gene_name   DM6/DM7  DM1/DM2d  \\\n",
       "0  AT1G63720               1954.528305  MEcyan  AT1G63720  1.935779  3.970436   \n",
       "1  AT3G54150               1952.887628  MEcyan        EFD  3.815566  5.122284   \n",
       "\n",
       "   DM10/DM11  AvrRpt2-6h  AvrRpt2-12h  AvrRpt2-20h  ...      Fold  \\\n",
       "0   6.033027    0.887911     1.510569     3.775730  ... -0.406409   \n",
       "1   8.439073    1.157111     3.410757     5.334764  ...  0.520665   \n",
       "\n",
       "        p.value           FDR        annotation  geneChr  geneStart   geneEnd  \\\n",
       "0  2.593965e-07  1.476102e-06  Promoter (<=1kb)        1   23635745  23637577   \n",
       "1  5.630124e-08  3.492766e-07  Promoter (<=1kb)        3   20050564  20052931   \n",
       "\n",
       "   geneLength  geneStrand  distanceToTSS  \n",
       "0        1833           2          699.0  \n",
       "1        2368           2            0.0  \n",
       "\n",
       "[2 rows x 34 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#first 2 rows\n",
    "module_dict['MEcyan'].iloc[0:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "65b80278-e947-4887-af4f-79b45469c173",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       AT1G63720\n",
       "1       AT3G54150\n",
       "2       AT5G48657\n",
       "3       AT5G48657\n",
       "4       AT4G23700\n",
       "          ...    \n",
       "6067    AT5G61790\n",
       "6073    AT4G24190\n",
       "6074    AT3G41768\n",
       "6075    AT2G16660\n",
       "6076    AT2G16660\n",
       "Name: gene, Length: 2859, dtype: object"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get \n",
    "module_dict['MEcyan']['gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "4a43903b-b31f-4ca8-af10-68e4a13f3540",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'./WGCNA_ATAC_integration'"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "WGCNA_path"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d7633181-e538-49ab-a85c-b70589632fc7",
   "metadata": {
    "tags": []
   },
   "source": [
    "def process_csv_to_bed(df, output_directory = output_directory):\n",
    "\n",
    "\n",
    "    base_filename = df['module'].unique().tolist()[0]\n",
    "    # Filter rows based on the conditions\n",
    "    df = df[ (df['FDR'] < 0.05)]\n",
    " \n",
    "    # Select the required columns and save to BED format\n",
    "    bed_file = os.path.join(output_directory, f\"{base_filename}.bed\")\n",
    "\n",
    "    df[['seqnames', 'start', 'end', 'width', 'strand']].to_csv(\n",
    "        bed_file, sep='\\t', header=False, index=False)\n",
    "\n",
    "    print(f\"Generated {bed_file} \\n\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "1102f021-e3a2-4142-afd6-a12ca389d491",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEcyan.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEblue.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEpurple.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEdarkred.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEgreenyellow.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEtan.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEdarkgreen.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEsalmon.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEmidnightblue.bed\n",
      "Generated BED file at: ./WGCNA_ATAC_integration/bed_files/MEdarkturquoise.bed\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "def process_all_csv_to_single_bed_dir(module_dict, df, WGCNA_path):\n",
    "    # Define the bed files directory\n",
    "    bed_files_dir = os.path.join(WGCNA_path, 'bed_files')\n",
    "    \n",
    "    # Ensure the bed files directory exists\n",
    "    os.makedirs(bed_files_dir, exist_ok=True)\n",
    "\n",
    "    # Iterate over all keys in the module_dict\n",
    "    for module_name in module_dict.keys():\n",
    "        # Filter the DataFrame for the current module and FDR < 0.05\n",
    "        module_df = df[(df['module'] == module_name) & (df['FDR'] < 0.05)]\n",
    "\n",
    "        # Proceed only if there are entries to write\n",
    "        if not module_df.empty:\n",
    "            # Construct the path for the BED file within the 'bed_files' directory\n",
    "            bed_file = os.path.join(bed_files_dir, f\"{module_name}.bed\")\n",
    "\n",
    "            # Select the required columns and save to BED format\n",
    "            module_df[['seqnames', 'start', 'end', 'width', 'strand']].to_csv(\n",
    "                bed_file, sep='\\t', header=False, index=False)\n",
    "\n",
    "            print(f\"Generated BED file at: {bed_file}\")\n",
    "\n",
    "\n",
    "process_all_csv_to_single_bed_dir(module_dict, df, WGCNA_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "b8f26434-2887-449d-ad92-8646c83403c8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEcyan.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEblue.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEpurple.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEdarkred.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEgreenyellow.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEtan.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEdarkgreen.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEsalmon.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEmidnightblue.fasta\n",
      "Generated fasta file at: ./WGCNA_ATAC_integration/fasta_files/MEdarkturquoise.fasta\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "def process_all_csv_to_fasta_dir(module_dict, df, WGCNA_path, genome_path):\n",
    "    # Define the fasta files directory\n",
    "    fasta_files_dir = os.path.join(WGCNA_path, 'fasta_files')\n",
    "    \n",
    "    # Ensure the fasta files directory exists\n",
    "    os.makedirs(fasta_files_dir, exist_ok=True)\n",
    "    \n",
    "    # Load the genome fasta file\n",
    "    genome_sequences = SeqIO.to_dict(SeqIO.parse(genome_path, \"fasta\"))\n",
    "\n",
    "    # Iterate over all keys in the module_dict\n",
    "    for module_name, genes_df in module_dict.items():\n",
    "        # Filter the DataFrame for the current module and FDR < 0.05\n",
    "        module_df = genes_df[(genes_df['module'] == module_name) & (genes_df['FDR'] < 0.05)]\n",
    "\n",
    "        # Proceed only if there are entries to write\n",
    "        if not module_df.empty:\n",
    "            # Construct the path for the fasta file within the 'fasta_files' directory\n",
    "            fasta_file = os.path.join(fasta_files_dir, f\"{module_name}.fasta\")\n",
    "\n",
    "            # Open the file in write mode\n",
    "            with open(fasta_file, 'w') as f:\n",
    "                # Iterate over rows in the module dataframe\n",
    "                for index, row in module_df.iterrows():\n",
    "                    # Extract sequence details\n",
    "                    seq_name = row['seqnames']\n",
    "                    start = int(row['start'])\n",
    "                    end = int(row['end'])\n",
    "                    gene_name = row['gene_name']\n",
    "                    # strand = row['strand']\n",
    "                    \n",
    "                    # Get the corresponding sequence from the genome\n",
    "                    gene_seq = genome_sequences[seq_name].seq[start-1:end]\n",
    "                    \n",
    "                    # # Reverse complement sequence if on negative strand\n",
    "                    # if strand == '-':\n",
    "                    #     gene_seq = gene_seq.reverse_complement()\n",
    "                    \n",
    "                    # Write to the fasta file\n",
    "                    f.write(f\">{gene_name}_{seq_name}:{start}-{end}\\n{gene_seq}\\n\")\n",
    "\n",
    "            print(f\"Generated fasta file at: {fasta_file}\")\n",
    "\n",
    "# Define paths and DataFrame here\n",
    "genome_path = os.path.join('/media/HDD2/Genomes/Ath_Ensembl56', 'Arabidopsis_thaliana.TAIR10.dna.toplevel.fa')\n",
    "\n",
    "# Call the function with the necessary arguments\n",
    "process_all_csv_to_fasta_dir(module_dict, df, WGCNA_path, genome_path)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4be790b5-955d-435c-bd02-d17c927862ff",
   "metadata": {
    "tags": []
   },
   "source": [
    "### for AME"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "b803bb63-168a-4b87-8ee4-02c3f96d7530",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>module</th>\n",
       "      <th>seq_name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>peak_sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MEcyan</td>\n",
       "      <td>1</td>\n",
       "      <td>23636274</td>\n",
       "      <td>23636878</td>\n",
       "      <td>AT1G63720</td>\n",
       "      <td>AT1G63720</td>\n",
       "      <td>ATTCTTGATCTGATGATGCTTCAGGGAACGTTGTTCTTAGCTTCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MEcyan</td>\n",
       "      <td>3</td>\n",
       "      <td>20052576</td>\n",
       "      <td>20053358</td>\n",
       "      <td>AT3G54150</td>\n",
       "      <td>EFD</td>\n",
       "      <td>AGTATTATAAAGCTTACCCCAATAGCAGCCTGACCATTACCAGTTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MEcyan</td>\n",
       "      <td>5</td>\n",
       "      <td>19735475</td>\n",
       "      <td>19736700</td>\n",
       "      <td>AT5G48657</td>\n",
       "      <td>NOI10</td>\n",
       "      <td>ACTAAAACCACACCCAAAAGAAAAAAAAAAACTTACTGCCATATTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEcyan</td>\n",
       "      <td>5</td>\n",
       "      <td>19734229</td>\n",
       "      <td>19734888</td>\n",
       "      <td>AT5G48657</td>\n",
       "      <td>NOI10</td>\n",
       "      <td>TACCTTGGGACTAGAAGGTAATGGTTGATGATGTTGGTCGATGATG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MEcyan</td>\n",
       "      <td>4</td>\n",
       "      <td>12346759</td>\n",
       "      <td>12347372</td>\n",
       "      <td>AT4G23700</td>\n",
       "      <td>CHX17</td>\n",
       "      <td>AACATATAATTATATAACATTAGTATATGAGATGACATAGAAATTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7085</th>\n",
       "      <td>MEdarkturquoise</td>\n",
       "      <td>2</td>\n",
       "      <td>527915</td>\n",
       "      <td>529292</td>\n",
       "      <td>AT2G02100</td>\n",
       "      <td>LCR69</td>\n",
       "      <td>GATAAAAATACTCTATCCCGTGACTATTCCACTTGTTGTTACTTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7086</th>\n",
       "      <td>MEdarkturquoise</td>\n",
       "      <td>2</td>\n",
       "      <td>529597</td>\n",
       "      <td>529856</td>\n",
       "      <td>AT2G02100</td>\n",
       "      <td>LCR69</td>\n",
       "      <td>AGGATTAGATTAGGATTACAACATGACTTGAGTGAAGCCACCAAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7087</th>\n",
       "      <td>MEdarkturquoise</td>\n",
       "      <td>4</td>\n",
       "      <td>8020565</td>\n",
       "      <td>8021053</td>\n",
       "      <td>AT4G13850</td>\n",
       "      <td>GR-RBP2</td>\n",
       "      <td>CATCAACGACAATTATCCGTTTAACATGGATTTTGAAGCCCAGCAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7088</th>\n",
       "      <td>MEdarkturquoise</td>\n",
       "      <td>4</td>\n",
       "      <td>13106367</td>\n",
       "      <td>13107055</td>\n",
       "      <td>AT4G25730</td>\n",
       "      <td>TRM7B</td>\n",
       "      <td>CGTGCAGAACCAAATTAAAAGCACTAACACCATGCTGCTCCATCAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7089</th>\n",
       "      <td>MEdarkturquoise</td>\n",
       "      <td>4</td>\n",
       "      <td>7502772</td>\n",
       "      <td>7503086</td>\n",
       "      <td>AT4G12750</td>\n",
       "      <td>AT4G12750</td>\n",
       "      <td>GGAGATACATCCCTCCTGTTTGGATTCATTACACGCCATACAGTCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7090 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               module seq_name     start       end       gene  gene_name  \\\n",
       "0              MEcyan        1  23636274  23636878  AT1G63720  AT1G63720   \n",
       "1              MEcyan        3  20052576  20053358  AT3G54150        EFD   \n",
       "2              MEcyan        5  19735475  19736700  AT5G48657      NOI10   \n",
       "3              MEcyan        5  19734229  19734888  AT5G48657      NOI10   \n",
       "4              MEcyan        4  12346759  12347372  AT4G23700      CHX17   \n",
       "...               ...      ...       ...       ...        ...        ...   \n",
       "7085  MEdarkturquoise        2    527915    529292  AT2G02100      LCR69   \n",
       "7086  MEdarkturquoise        2    529597    529856  AT2G02100      LCR69   \n",
       "7087  MEdarkturquoise        4   8020565   8021053  AT4G13850    GR-RBP2   \n",
       "7088  MEdarkturquoise        4  13106367  13107055  AT4G25730      TRM7B   \n",
       "7089  MEdarkturquoise        4   7502772   7503086  AT4G12750  AT4G12750   \n",
       "\n",
       "                                          peak_sequence  \n",
       "0     ATTCTTGATCTGATGATGCTTCAGGGAACGTTGTTCTTAGCTTCTG...  \n",
       "1     AGTATTATAAAGCTTACCCCAATAGCAGCCTGACCATTACCAGTTC...  \n",
       "2     ACTAAAACCACACCCAAAAGAAAAAAAAAAACTTACTGCCATATTT...  \n",
       "3     TACCTTGGGACTAGAAGGTAATGGTTGATGATGTTGGTCGATGATG...  \n",
       "4     AACATATAATTATATAACATTAGTATATGAGATGACATAGAAATTT...  \n",
       "...                                                 ...  \n",
       "7085  GATAAAAATACTCTATCCCGTGACTATTCCACTTGTTGTTACTTTT...  \n",
       "7086  AGGATTAGATTAGGATTACAACATGACTTGAGTGAAGCCACCAAAA...  \n",
       "7087  CATCAACGACAATTATCCGTTTAACATGGATTTTGAAGCCCAGCAA...  \n",
       "7088  CGTGCAGAACCAAATTAAAAGCACTAACACCATGCTGCTCCATCAC...  \n",
       "7089  GGAGATACATCCCTCCTGTTTGGATTCATTACACGCCATACAGTCA...  \n",
       "\n",
       "[7090 rows x 7 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "def extract_sequences_to_df(module_dict, genome_path):\n",
    "    # Load the genome fasta file\n",
    "    genome_sequences = SeqIO.to_dict(SeqIO.parse(genome_path, \"fasta\"))\n",
    "\n",
    "    # Create a list to store sequence data\n",
    "    sequences_data = []\n",
    "\n",
    "    # Iterate over all keys in the module_dict\n",
    "    for module_name, genes_df in module_dict.items():\n",
    "        # Filter the DataFrame for the current module and FDR < 0.05\n",
    "        module_df = genes_df[(genes_df['module'] == module_name) & (genes_df['FDR'] < 0.05)]\n",
    "\n",
    "        # Proceed only if there are entries to add\n",
    "        if not module_df.empty:\n",
    "            # Iterate over rows in the module dataframe\n",
    "            for index, row in module_df.iterrows():\n",
    "                # Extract sequence details\n",
    "                seq_name = row['seqnames']\n",
    "                start = int(row['start'])\n",
    "                end = int(row['end'])\n",
    "                gene_name = row['gene_name']\n",
    "                gene = row['gene']\n",
    "                \n",
    "                # Get the corresponding sequence from the genome\n",
    "                gene_seq = str(genome_sequences[seq_name].seq[start-1:end])\n",
    "                \n",
    "                # Store the sequence data in the list\n",
    "                sequences_data.append({\n",
    "                    \"module\": module_name,\n",
    "                    \"seq_name\": seq_name,\n",
    "                    \"start\": start,\n",
    "                    \"end\": end,\n",
    "                    \"gene\" :gene,\n",
    "                    \"gene_name\": gene_name,\n",
    "                    \"peak_sequence\": gene_seq\n",
    "                })\n",
    "\n",
    "    # Create a DataFrame from the list of sequence data\n",
    "    sequences_df = pd.DataFrame(sequences_data)\n",
    "\n",
    "    return sequences_df\n",
    "\n",
    "# Define paths here\n",
    "genome_path = os.path.join('/media/HDD2/Genomes/Ath_Ensembl56', 'Arabidopsis_thaliana.TAIR10.dna.toplevel.fa')\n",
    "\n",
    "# Call the function with the necessary arguments and print the resulting DataFrame\n",
    "sequences_df = extract_sequences_to_df(module_dict, genome_path)\n",
    "sequences_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "04d617b5-614e-4d3a-b606-ebcaa719ab0c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "def extract_non_variable_sequences_to_df(df, genome_path):\n",
    "    # Load the genome fasta file\n",
    "    genome_sequences = SeqIO.to_dict(SeqIO.parse(genome_path, \"fasta\"))\n",
    "    \n",
    "    # Initialize a list to store sequences\n",
    "    sequences = []\n",
    "    \n",
    "    # Iterate through each row of the DataFrame\n",
    "    for idx, row in df.iterrows():\n",
    "        # Extract sequence details\n",
    "        seq_name = row['seqnames']\n",
    "        start = int(row['start'])\n",
    "        end = int(row['end'])\n",
    "        # gene_name = row['geneId']\n",
    "        # gene = row['gene']\n",
    "        \n",
    "        # Extract the sequence from the genome fasta\n",
    "        sequence = str(genome_sequences[seq_name].seq[start-1:end])  # -1 because Python is 0-based\n",
    "        \n",
    "        # Append the sequence to the list\n",
    "        sequences.append(sequence)\n",
    "    \n",
    "    # Add the sequences as a new column to the DataFrame\n",
    "    df['sequence'] = sequences\n",
    "    \n",
    "    return df\n",
    "\n",
    "# Usage example:\n",
    "# Assuming 'dataframe' is your existing DataFrame and 'path_to_genome.fasta' is the path to your fasta file\n",
    "non_variable_peaks_in_RNA_seq2 = extract_non_variable_sequences_to_df(non_variable_peaks_in_RNA_seq, genome_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "d1245887-909a-4022-8962-a11b41b252f1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>strand</th>\n",
       "      <th>Conc</th>\n",
       "      <th>Conc_DM1.DM2d</th>\n",
       "      <th>Conc_Col.0_12DAS</th>\n",
       "      <th>Fold</th>\n",
       "      <th>p.value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>annotation</th>\n",
       "      <th>geneChr</th>\n",
       "      <th>geneStart</th>\n",
       "      <th>geneEnd</th>\n",
       "      <th>geneLength</th>\n",
       "      <th>geneStrand</th>\n",
       "      <th>geneId</th>\n",
       "      <th>distanceToTSS</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>5236541</td>\n",
       "      <td>5237157</td>\n",
       "      <td>617</td>\n",
       "      <td>*</td>\n",
       "      <td>10.907429</td>\n",
       "      <td>9.407313</td>\n",
       "      <td>11.626810</td>\n",
       "      <td>-2.200833</td>\n",
       "      <td>6.460956e-278</td>\n",
       "      <td>1.128907e-273</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>5236634</td>\n",
       "      <td>5236879</td>\n",
       "      <td>246</td>\n",
       "      <td>1</td>\n",
       "      <td>AT5G02495</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AATTTCATAGTTAGTCATCTTCACCACATGTACAACAAAGATGAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>7104355</td>\n",
       "      <td>7104488</td>\n",
       "      <td>134</td>\n",
       "      <td>*</td>\n",
       "      <td>10.252167</td>\n",
       "      <td>8.418776</td>\n",
       "      <td>11.034068</td>\n",
       "      <td>-2.588420</td>\n",
       "      <td>1.021448e-256</td>\n",
       "      <td>8.923750e-253</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>3</td>\n",
       "      <td>7103817</td>\n",
       "      <td>7104250</td>\n",
       "      <td>434</td>\n",
       "      <td>1</td>\n",
       "      <td>AT3G20362</td>\n",
       "      <td>538.0</td>\n",
       "      <td>GCAATATGTCTACGCTGGTTCAAATCCAGCTCGGCCCAATAATTAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>17828736</td>\n",
       "      <td>17829549</td>\n",
       "      <td>814</td>\n",
       "      <td>*</td>\n",
       "      <td>11.143519</td>\n",
       "      <td>9.809886</td>\n",
       "      <td>11.824502</td>\n",
       "      <td>-1.997740</td>\n",
       "      <td>3.602299e-250</td>\n",
       "      <td>2.517683e-246</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>17829618</td>\n",
       "      <td>17831336</td>\n",
       "      <td>1719</td>\n",
       "      <td>2</td>\n",
       "      <td>AT5G44260</td>\n",
       "      <td>1787.0</td>\n",
       "      <td>ACCGTAGTACCGTACGTACGTGGTCCTCACTCTCACATCTCAAAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>17504350</td>\n",
       "      <td>17505043</td>\n",
       "      <td>694</td>\n",
       "      <td>*</td>\n",
       "      <td>10.551486</td>\n",
       "      <td>8.694952</td>\n",
       "      <td>11.337129</td>\n",
       "      <td>-2.614031</td>\n",
       "      <td>5.171035e-250</td>\n",
       "      <td>3.285534e-246</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>2</td>\n",
       "      <td>17504551</td>\n",
       "      <td>17504932</td>\n",
       "      <td>382</td>\n",
       "      <td>2</td>\n",
       "      <td>AT2G09320</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AGTTTGGTCATGAGTTCGATGAAGTGATGGAGAGTTAAGAGTAGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>8598368</td>\n",
       "      <td>8599538</td>\n",
       "      <td>1171</td>\n",
       "      <td>*</td>\n",
       "      <td>11.514378</td>\n",
       "      <td>10.492546</td>\n",
       "      <td>12.106545</td>\n",
       "      <td>-1.601076</td>\n",
       "      <td>2.816934e-228</td>\n",
       "      <td>1.514449e-224</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>8598624</td>\n",
       "      <td>8598990</td>\n",
       "      <td>367</td>\n",
       "      <td>1</td>\n",
       "      <td>AT1G05863</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AGTCTAAAGCATAAGAAAAAAAAAGGTTAGTAAGAAAGAAAGAAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24261</th>\n",
       "      <td>1</td>\n",
       "      <td>4036334</td>\n",
       "      <td>4037134</td>\n",
       "      <td>801</td>\n",
       "      <td>*</td>\n",
       "      <td>9.993201</td>\n",
       "      <td>10.086007</td>\n",
       "      <td>9.894011</td>\n",
       "      <td>0.164856</td>\n",
       "      <td>2.362045e-02</td>\n",
       "      <td>4.995784e-02</td>\n",
       "      <td>Promoter (1-2kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>4034479</td>\n",
       "      <td>4039065</td>\n",
       "      <td>4587</td>\n",
       "      <td>2</td>\n",
       "      <td>AT1G11950</td>\n",
       "      <td>1931.0</td>\n",
       "      <td>TGTTAAGAGCATTTCGAACAATCACTGGTTCACCTTTGGACCAATG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24262</th>\n",
       "      <td>5</td>\n",
       "      <td>3658888</td>\n",
       "      <td>3658978</td>\n",
       "      <td>91</td>\n",
       "      <td>*</td>\n",
       "      <td>7.613293</td>\n",
       "      <td>7.426264</td>\n",
       "      <td>7.778834</td>\n",
       "      <td>-0.264245</td>\n",
       "      <td>2.362478e-02</td>\n",
       "      <td>4.996483e-02</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>5</td>\n",
       "      <td>3656582</td>\n",
       "      <td>3659025</td>\n",
       "      <td>2444</td>\n",
       "      <td>2</td>\n",
       "      <td>AT5G11460</td>\n",
       "      <td>47.0</td>\n",
       "      <td>CGTATATATCTGTGCAAGTCTTGTGAGGTACTCTTCTCTCTCTCTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24263</th>\n",
       "      <td>5</td>\n",
       "      <td>13056298</td>\n",
       "      <td>13056771</td>\n",
       "      <td>474</td>\n",
       "      <td>*</td>\n",
       "      <td>9.006034</td>\n",
       "      <td>9.140994</td>\n",
       "      <td>8.857133</td>\n",
       "      <td>0.222461</td>\n",
       "      <td>2.362877e-02</td>\n",
       "      <td>4.996939e-02</td>\n",
       "      <td>Distal Intergenic</td>\n",
       "      <td>5</td>\n",
       "      <td>13061035</td>\n",
       "      <td>13062230</td>\n",
       "      <td>1196</td>\n",
       "      <td>1</td>\n",
       "      <td>AT5G34869</td>\n",
       "      <td>-4264.0</td>\n",
       "      <td>ATCTTAGGTTCATCTAAACCACGAAGACCAAACTTCTCCTCCACAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24264</th>\n",
       "      <td>1</td>\n",
       "      <td>29390905</td>\n",
       "      <td>29391031</td>\n",
       "      <td>127</td>\n",
       "      <td>*</td>\n",
       "      <td>7.891752</td>\n",
       "      <td>7.722579</td>\n",
       "      <td>8.043153</td>\n",
       "      <td>-0.246531</td>\n",
       "      <td>2.363869e-02</td>\n",
       "      <td>4.998885e-02</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>1</td>\n",
       "      <td>29391630</td>\n",
       "      <td>29393334</td>\n",
       "      <td>1705</td>\n",
       "      <td>1</td>\n",
       "      <td>AT1G78110</td>\n",
       "      <td>-599.0</td>\n",
       "      <td>TATATAAGTATATAGTTTGGTGATATAGTTAGAAGGTAGTTAAATC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24265</th>\n",
       "      <td>3</td>\n",
       "      <td>6428336</td>\n",
       "      <td>6429592</td>\n",
       "      <td>1257</td>\n",
       "      <td>*</td>\n",
       "      <td>10.305931</td>\n",
       "      <td>10.382822</td>\n",
       "      <td>10.224711</td>\n",
       "      <td>0.139538</td>\n",
       "      <td>2.364269e-02</td>\n",
       "      <td>4.999580e-02</td>\n",
       "      <td>Promoter (&lt;=1kb)</td>\n",
       "      <td>3</td>\n",
       "      <td>6430084</td>\n",
       "      <td>6430211</td>\n",
       "      <td>128</td>\n",
       "      <td>1</td>\n",
       "      <td>AT3G03745</td>\n",
       "      <td>-492.0</td>\n",
       "      <td>TGTATTCAGCTTTGAGTTCTTCCTATACATAAGTATAATACCCGAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24266 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      seqnames     start       end  width strand       Conc  Conc_DM1.DM2d  \\\n",
       "0            5   5236541   5237157    617      *  10.907429       9.407313   \n",
       "1            3   7104355   7104488    134      *  10.252167       8.418776   \n",
       "2            5  17828736  17829549    814      *  11.143519       9.809886   \n",
       "3            2  17504350  17505043    694      *  10.551486       8.694952   \n",
       "4            1   8598368   8599538   1171      *  11.514378      10.492546   \n",
       "...        ...       ...       ...    ...    ...        ...            ...   \n",
       "24261        1   4036334   4037134    801      *   9.993201      10.086007   \n",
       "24262        5   3658888   3658978     91      *   7.613293       7.426264   \n",
       "24263        5  13056298  13056771    474      *   9.006034       9.140994   \n",
       "24264        1  29390905  29391031    127      *   7.891752       7.722579   \n",
       "24265        3   6428336   6429592   1257      *  10.305931      10.382822   \n",
       "\n",
       "       Conc_Col.0_12DAS      Fold        p.value            FDR  \\\n",
       "0             11.626810 -2.200833  6.460956e-278  1.128907e-273   \n",
       "1             11.034068 -2.588420  1.021448e-256  8.923750e-253   \n",
       "2             11.824502 -1.997740  3.602299e-250  2.517683e-246   \n",
       "3             11.337129 -2.614031  5.171035e-250  3.285534e-246   \n",
       "4             12.106545 -1.601076  2.816934e-228  1.514449e-224   \n",
       "...                 ...       ...            ...            ...   \n",
       "24261          9.894011  0.164856   2.362045e-02   4.995784e-02   \n",
       "24262          7.778834 -0.264245   2.362478e-02   4.996483e-02   \n",
       "24263          8.857133  0.222461   2.362877e-02   4.996939e-02   \n",
       "24264          8.043153 -0.246531   2.363869e-02   4.998885e-02   \n",
       "24265         10.224711  0.139538   2.364269e-02   4.999580e-02   \n",
       "\n",
       "              annotation  geneChr  geneStart   geneEnd  geneLength  \\\n",
       "0       Promoter (<=1kb)        5    5236634   5236879         246   \n",
       "1       Promoter (<=1kb)        3    7103817   7104250         434   \n",
       "2       Promoter (1-2kb)        5   17829618  17831336        1719   \n",
       "3       Promoter (<=1kb)        2   17504551  17504932         382   \n",
       "4       Promoter (<=1kb)        1    8598624   8598990         367   \n",
       "...                  ...      ...        ...       ...         ...   \n",
       "24261   Promoter (1-2kb)        1    4034479   4039065        4587   \n",
       "24262   Promoter (<=1kb)        5    3656582   3659025        2444   \n",
       "24263  Distal Intergenic        5   13061035  13062230        1196   \n",
       "24264   Promoter (<=1kb)        1   29391630  29393334        1705   \n",
       "24265   Promoter (<=1kb)        3    6430084   6430211         128   \n",
       "\n",
       "       geneStrand     geneId  distanceToTSS  \\\n",
       "0               1  AT5G02495            0.0   \n",
       "1               1  AT3G20362          538.0   \n",
       "2               2  AT5G44260         1787.0   \n",
       "3               2  AT2G09320            0.0   \n",
       "4               1  AT1G05863            0.0   \n",
       "...           ...        ...            ...   \n",
       "24261           2  AT1G11950         1931.0   \n",
       "24262           2  AT5G11460           47.0   \n",
       "24263           1  AT5G34869        -4264.0   \n",
       "24264           1  AT1G78110         -599.0   \n",
       "24265           1  AT3G03745         -492.0   \n",
       "\n",
       "                                                sequence  \n",
       "0      AATTTCATAGTTAGTCATCTTCACCACATGTACAACAAAGATGAAT...  \n",
       "1      GCAATATGTCTACGCTGGTTCAAATCCAGCTCGGCCCAATAATTAG...  \n",
       "2      ACCGTAGTACCGTACGTACGTGGTCCTCACTCTCACATCTCAAAGA...  \n",
       "3      AGTTTGGTCATGAGTTCGATGAAGTGATGGAGAGTTAAGAGTAGAT...  \n",
       "4      AGTCTAAAGCATAAGAAAAAAAAAGGTTAGTAAGAAAGAAAGAAAG...  \n",
       "...                                                  ...  \n",
       "24261  TGTTAAGAGCATTTCGAACAATCACTGGTTCACCTTTGGACCAATG...  \n",
       "24262  CGTATATATCTGTGCAAGTCTTGTGAGGTACTCTTCTCTCTCTCTC...  \n",
       "24263  ATCTTAGGTTCATCTAAACCACGAAGACCAAACTTCTCCTCCACAT...  \n",
       "24264  TATATAAGTATATAGTTTGGTGATATAGTTAGAAGGTAGTTAAATC...  \n",
       "24265  TGTATTCAGCTTTGAGTTCTTCCTATACATAAGTATAATACCCGAA...  \n",
       "\n",
       "[24266 rows x 20 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "non_variable_peaks_in_RNA_seq2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "fb98c25e-9261-4499-a89e-b682428c9eb6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def write_sequences_to_fasta(df, output_path):\n",
    "    # Open the file in write mode\n",
    "    with open(output_path, 'w') as fasta_file:\n",
    "        # Iterate through the DataFrame\n",
    "        for index, seq in df.iterrows():\n",
    "            # Create the header\n",
    "            header = f\">{seq['geneId']}_{seq['seqnames']}_{seq['start']}_{seq['end']}\"\n",
    "            # Retrieve the sequence\n",
    "            sequence = seq['sequence']  # Make sure 'sequence' is the correct column name where sequences are stored\n",
    "            \n",
    "            # Write the header and sequence to the file\n",
    "            fasta_file.write(f\"{header}\\n{sequence}\\n\")\n",
    "\n",
    "# Usage:\n",
    "output_fasta_path = os.path.join(fasta_files_dir, f\"non_variable_peaks_in_RNA_seq.fasta\")\n",
    "write_sequences_to_fasta(non_variable_peaks_in_RNA_seq2, output_fasta_path)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "2502db6a-3a36-4137-8002-5ff49ca064c0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7090  sequences written to file\n"
     ]
    }
   ],
   "source": [
    "sequences_df\n",
    "\n",
    "with open(os.path.join(fasta_files_dir, f\"all_WGCNA_seq.fasta\"), 'w') as fasta_file:\n",
    "    # Iterate through the DataFrame\n",
    "    n = 0\n",
    "    for index, seq in sequences_df.iterrows():\n",
    "        # Create the header\n",
    "        header = f\">{seq['gene']}_{seq['gene_name']}_{seq['seq_name']}_{seq['start']}_{seq['end']}\"\n",
    "        # Retrieve the sequence\n",
    "        sequence = seq['peak_sequence']  # Make sure 'sequence' is the correct column name where sequences are stored\n",
    "        # Write the header and sequence to the file\n",
    "        fasta_file.write(f\"{header}\\n{sequence}\\n\")\n",
    "        n += 1\n",
    "    print(n, \" sequences written to file\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "8b096db6-ce2a-4f07-8a21-1e4da960a969",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Module MEcyan has 2859 rows\n",
      "Module MEblue has 3219 rows\n",
      "Module MEpurple has 332 rows\n",
      "Module MEdarkred has 132 rows\n",
      "Module MEgreenyellow has 117 rows\n",
      "Module MEtan has 92 rows\n",
      "Module MEdarkgreen has 129 rows\n",
      "Module MEsalmon has 103 rows\n",
      "Module MEmidnightblue has 74 rows\n",
      "Module MEdarkturquoise has 33 rows\n"
     ]
    }
   ],
   "source": [
    "# Assuming 'sequences_df' is your DataFrame\n",
    "\n",
    "# Get the unique modules\n",
    "unique_modules = sequences_df['module'].unique()\n",
    "\n",
    "# Print the number of rows for each module\n",
    "for module in unique_modules:\n",
    "    n_rows = sequences_df[sequences_df['module'] == module].shape[0]\n",
    "    print(f\"Module {module} has {n_rows} rows\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "bd717649-1e74-41af-abcf-62eaf83ab856",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Module MEcyan: 2859 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEcyan.fasta\n",
      "Module MEblue: 3219 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEblue.fasta\n",
      "Module MEpurple: 332 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEpurple.fasta\n",
      "Module MEdarkred: 132 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEdarkred.fasta\n",
      "Module MEgreenyellow: 117 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEgreenyellow.fasta\n",
      "Module MEtan: 92 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEtan.fasta\n",
      "Module MEdarkgreen: 129 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEdarkgreen.fasta\n",
      "Module MEsalmon: 103 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEsalmon.fasta\n",
      "Module MEmidnightblue: 74 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEmidnightblue.fasta\n",
      "Module MEdarkturquoise: 33 sequences\n",
      "Control sequences written to ./WGCNA_ATAC_integration/fasta_files_and_control/control_for_MEdarkturquoise.fasta\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import random\n",
    "\n",
    "# Function to write sequences to a fasta file\n",
    "def write_to_fasta(filename, sequences):\n",
    "    with open(filename, 'w') as f:\n",
    "        for seq in sequences:\n",
    "            header = f\">{seq['gene']}{seq['gene_name']}_{seq['seq_name']}_{seq['start']}_{seq['end']}\"\n",
    "            sequence = seq['peak_sequence']\n",
    "            f.write(f\"{header}\\n{sequence}\\n\")\n",
    "\n",
    "# Base directory for the fasta files\n",
    "fasta_files_dir = './WGCNA_ATAC_integration/fasta_files_and_control/'\n",
    "os.makedirs(fasta_files_dir, exist_ok=True)\n",
    "\n",
    "# Get the unique modules\n",
    "unique_modules = sequences_df['module'].unique()\n",
    "\n",
    "# Dictionary to keep track of control sequences for each module\n",
    "control_sequences = {module: [] for module in unique_modules}\n",
    "\n",
    "# Iterate over each module and create fasta files\n",
    "for module in unique_modules:\n",
    "    module_df = sequences_df[sequences_df['module'] == module]\n",
    "    n_sequences = module_df.shape[0]\n",
    "\n",
    "    # Write the current module sequences to a fasta file\n",
    "    fasta_filename = os.path.join(fasta_files_dir, f\"{module}.fasta\")\n",
    "    write_to_fasta(fasta_filename, module_df.to_dict('records'))\n",
    "\n",
    "    # Select random sequences from other modules\n",
    "    other_modules_df = sequences_df[sequences_df['module'] != module]\n",
    "    control_seqs = other_modules_df.sample(n_sequences).to_dict('records')\n",
    "    control_sequences[module].extend(control_seqs)\n",
    "\n",
    "    # Write the control sequences to a fasta file\n",
    "    control_fasta_filename = os.path.join(fasta_files_dir, f\"control_for_{module}.fasta\")\n",
    "    write_to_fasta(control_fasta_filename, control_seqs)\n",
    "\n",
    "    print(f\"Module {module}: {n_sequences} sequences\")\n",
    "    print(f\"Control sequences written to {control_fasta_filename}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f21dfef3-eb27-4ab4-8791-f6edd3e21198",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "22b2c625-d2a8-47f2-b8b4-910fec9148b4",
   "metadata": {},
   "source": [
    "### the bed file of peaks from Chipseeker"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "984f22cf-32d4-4c0a-aacc-35b357804d24",
   "metadata": {},
   "source": [
    "## homer2 compare enriched genes vs random background"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "26ccc277-12c5-41fc-8a81-687329b0b7dd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#using homer provided motifs\n",
    "\n",
    "def find_motifs_task(bed_file):\n",
    "    # Assume bed_file path is like '.../WGCNA_integrated/bed_files/MEcyan.bed'\n",
    "    module_name = os.path.splitext(os.path.basename(bed_file))[0]\n",
    "    WGCNA_path = os.path.dirname(os.path.dirname(bed_file)) # Go up two levels to the WGCNA_integrated directory\n",
    "\n",
    "    # Define unique output path for the module\n",
    "    output_path = os.path.join(WGCNA_path, 'homer_using_homer_database_plants' ,module_name + '_motifs')\n",
    "    \n",
    "    # Create the output directory if it doesn't exist\n",
    "    os.makedirs(output_path, exist_ok=True)\n",
    "    \n",
    "    #Program will find de novo and known motifs in regions in the genome\n",
    "    cmd = [\n",
    "        \"findMotifsGenome.pl\",\n",
    "        bed_file,\n",
    "         os.path.join('/media/HDD2/Genomes/Ath_Ensembl56', 'Arabidopsis_thaliana.TAIR10.dna.toplevel.fa'),\n",
    "        output_path,\n",
    "        \"-p\", \"20\",\n",
    "        \"-mset\", \"plants\"\n",
    "    ]\n",
    "    result = subprocess.run(cmd, capture_output=True, text=True)\n",
    "    \n",
    "    # Log file named after the module\n",
    "    log_file_path = os.path.join(output_path, f\"{module_name}_HOMER.log\")\n",
    "    \n",
    "    # Append both stderr and stdout to the log file\n",
    "    with open(log_file_path, \"a\") as log_file:\n",
    "        log_file.write(result.stdout)\n",
    "        log_file.write(result.stderr)\n",
    "\n",
    "def run_find_motifs(WGCNA_path):\n",
    "    # Define the directory where all bed files are located\n",
    "    bed_files_dir = os.path.join(WGCNA_path, 'bed_files')\n",
    "    # Gather all bed files\n",
    "    bed_files = [os.path.join(bed_files_dir, f) for f in os.listdir(bed_files_dir) if f.endswith('.bed')]\n",
    "\n",
    "    # Use Pool to run the tasks in parallel\n",
    "    with Pool(6) as p:  # Adjust number of processes if necessary\n",
    "        p.map(find_motifs_task, bed_files)\n",
    "\n",
    "# Usage:\n",
    "run_find_motifs(WGCNA_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "e6b59edd-5e94-4990-b570-a3087e9fb8e9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "##using PlantTFDB motifs\n",
    "\n",
    "def find_motifs_task(bed_file):\n",
    "    # Assume bed_file path is like '.../WGCNA_integrated/bed_files/MEcyan.bed'\n",
    "    module_name = os.path.splitext(os.path.basename(bed_file))[0]\n",
    "    WGCNA_path = os.path.dirname(os.path.dirname(bed_file)) # Go up two levels to the WGCNA_integrated directory\n",
    "\n",
    "    # Define unique output path for the module\n",
    "    output_path = os.path.join(WGCNA_path, 'homer_plantTFDB', module_name + '_motifs')\n",
    "    \n",
    "    # Create the output directory if it doesn't exist\n",
    "    os.makedirs(output_path, exist_ok=True)\n",
    "    \n",
    "    #Program will find de novo and known motifs in regions in the genome\n",
    "    cmd = [\n",
    "        \"findMotifsGenome.pl\",\n",
    "        bed_file,\n",
    "         os.path.join('/media/HDD2/Genomes/Ath_Ensembl56', 'Arabidopsis_thaliana.TAIR10.dna.toplevel.fa'),\n",
    "        output_path,\n",
    "        \"-p\", \"20\",\n",
    "        # \"-mset\", \"plants\"\n",
    "        '-mknown', '/media/HDD2/donghui/bulk_ATAC_DM1_DM2d/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
    "    ]\n",
    "    result = subprocess.run(cmd, capture_output=True, text=True)\n",
    "    \n",
    "    # Log file named after the module\n",
    "    log_file_path = os.path.join(output_path, f\"{module_name}_HOMER.log\")\n",
    "    \n",
    "    # Append both stderr and stdout to the log file\n",
    "    with open(log_file_path, \"a\") as log_file:\n",
    "        log_file.write(result.stdout)\n",
    "        log_file.write(result.stderr)\n",
    "\n",
    "def run_find_motifs(WGCNA_path):\n",
    "    # Define the directory where all bed files are located\n",
    "    bed_files_dir = os.path.join(WGCNA_path, 'bed_files')\n",
    "    # Gather all bed files\n",
    "    bed_files = [os.path.join(bed_files_dir, f) for f in os.listdir(bed_files_dir) if f.endswith('.bed')]\n",
    "\n",
    "    # Use Pool to run the tasks in parallel\n",
    "    with Pool(6) as p:  # Adjust number of processes if necessary\n",
    "        p.map(find_motifs_task, bed_files)\n",
    "\n",
    "# Usage:\n",
    "run_find_motifs(WGCNA_path)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8e0f8dc-bab2-40f8-9053-d3c55ef94998",
   "metadata": {
    "tags": []
   },
   "source": [
    "## MEME AME enrichment of motifs from peaks2"
   ]
  },
  {
   "cell_type": "raw",
   "id": "cbededc6-b795-4ece-8aea-e591c5d7e2e0",
   "metadata": {
    "tags": []
   },
   "source": [
    "import os\n",
    "import subprocess\n",
    "from multiprocessing import Pool\n",
    "import glob\n",
    "\n",
    "def run_ame(fasta_file, motif_file_path, ame_results_dir):\n",
    "    # Get the module name from the fasta file name\n",
    "    module_name = os.path.basename(fasta_file).replace('.fasta', '')\n",
    "    \n",
    "    # Define the output directory for this module's AME results\n",
    "    output_dir = os.path.join(ame_results_dir, module_name)\n",
    "    \n",
    "    # Ensure the output directory exists\n",
    "    # os.makedirs(output_dir, exist_ok=True)\n",
    "    \n",
    "    # Construct the AME command\n",
    "    ame_command = [\n",
    "        \"ame\",\n",
    "        fasta_file,\n",
    "        motif_file_path,\n",
    "        \"--oc\", output_dir,\n",
    "         \"--control --shuffle--\",\n",
    "        \"--kmer\", \"2\",\n",
    "        \"--method\", \"fisher\",\n",
    "        \"--scoring\", \"avg\",\n",
    "    ]\n",
    "    \n",
    "    # Run the AME command\n",
    "    subprocess.run(ame_command, check=True)\n",
    "    \n",
    "    print(f\"AME analysis completed for module: {module_name}\")\n",
    "\n",
    "# Define paths\n",
    "\n",
    "motif_file_path = './scan_motif/Ath_TF_binding_motifs_plantTFDB.meme'\n",
    "fasta_files_dir = os.path.join(WGCNA_path, 'fasta_files')\n",
    "ame_results_dir = os.path.join(WGCNA_path, 'ame_results')\n",
    "\n",
    "# Ensure the AME results directory exists\n",
    "os.makedirs(ame_results_dir, exist_ok=True)\n",
    "\n",
    "# Use glob to find all fasta files\n",
    "fasta_files = glob.glob(os.path.join(fasta_files_dir, '*.fasta'))\n",
    "\n",
    "# Create a list of tuples for the arguments to pass to run_ame\n",
    "ame_args = [(fasta_file, motif_file_path, ame_results_dir) for fasta_file in fasta_files]\n",
    "print(ame_args)\n",
    "# Number of processes\n",
    "n_processes = 4\n",
    "\n",
    "# Create a pool of workers and map the run_ame function to fasta files\n",
    "if __name__ == '__main__':  # Safe guard for multiprocessing\n",
    "    with Pool(n_processes) as pool:\n",
    "        pool.starmap(run_ame, ame_args)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "de1c2670-03c9-474d-875d-2b671b73c13c",
   "metadata": {},
   "source": [
    "### MEME installed from conda seems to be broken, use docker version instead"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "7fb39120-83d0-488a-8ae5-503b958e14db",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[sudo] password for hu: \n"
     ]
    }
   ],
   "source": [
    "! sudo chmod -R 777 /media/HDD2/donghui/bulk_ATAC_DM1_DM2d/WGCNA_ATAC_integration/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b4a63a7b-dd7e-470f-bae3-36fa1909bc4a",
   "metadata": {
    "tags": []
   },
   "source": [
    "docker takes sudo, can run visudo and add \"yourusername ALL=(ALL) NOPASSWD: /usr/bin/docker\"\n",
    "then run the script docker_mem_ame.sh, which loops over all fasta files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28220abf-9f99-4cea-a7f1-c364a6783ddd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "4487bab7-e876-4c18-b46c-8bd0ce357ae8",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (1206664557.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Cell \u001b[0;32mIn[146], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d/:/data memesuite/memesuite ame --oc /data/WGCNA_ATAC_integration/ame_WGCNA_vs_non_variable_peaks/ --control /data/WGCNA_ATAC_integration/fasta_files_and_control/non_variable_peaks_in_RNA_seq.fasta /data/WGCNA_ATAC_integration/fasta_files_and_control/all_WGCNA_seq.fasta /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\u001b[0m\n\u001b[0m         ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "sudo docker run -v /media/HDD2/donghui/bulk_ATAC_DM1_DM2d/:/data memesuite/memesuite ame --oc /data/WGCNA_ATAC_integration/ame_WGCNA_vs_non_variable_peaks/ --control /data/WGCNA_ATAC_integration/fasta_files_and_control/non_variable_peaks_in_RNA_seq.fasta /data/WGCNA_ATAC_integration/fasta_files_and_control/all_WGCNA_seq.fasta /data/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f253142-6a6a-4279-b931-f0c4d5e9f1a0",
   "metadata": {},
   "source": [
    "/media/HDD2/donghui/bulk_ATAC_DM1_DM2d » mkdir WGCNA_ATAC_integration/ame_results_with_shuffle/                      hu@chaelab2\n",
    "(work2) -------------------------------------------------------------------------------------------------------------------------\n",
    "/media/HDD2/donghui/bulk_ATAC_DM1_DM2d » chmod -R 777 WGCNA_ATAC_integration/ame_results_with_shuffle/   \n",
    "/media/HDD2/donghui/bulk_ATAC_DM1_DM2d » bash ./docker_mem_ame_with_shuffle.sh    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39621c33-94fa-4ed1-a0b3-5107f8d4c08d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "b1b8c996-8131-4024-8a9e-53850f7fbc88",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using motif +AT1G01060 of width 10.\n",
      "Using motif -AT1G01060 of width 10.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 714 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.886139\n",
      "#   Estimated pi_0=0.900893\n",
      "Using motif +AT1G01250 of width 15.\n",
      "Using motif -AT1G01250 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 510 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00161\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G01260 of width 8.\n",
      "Using motif -AT1G01260 of width 8.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 360 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G01720 of width 22.\n",
      "Using motif -AT1G01720 of width 22.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 300 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9606\n",
      "#   Estimated pi_0=0.968305\n",
      "Using motif +AT1G02065 of width 16.\n",
      "Using motif -AT1G02065 of width 16.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 461 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.839\n",
      "#   Estimated pi_0=0.84283\n",
      "Using motif +AT1G02230 of width 14.\n",
      "Using motif -AT1G02230 of width 14.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 441 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.962157\n",
      "#   Estimated pi_0=0.966935\n",
      "Using motif +AT1G02250 of width 19.\n",
      "Using motif -AT1G02250 of width 19.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 540 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.926602\n",
      "#   Estimated pi_0=0.938268\n",
      "Using motif +AT1G03800 of width 21.\n",
      "Using motif -AT1G03800 of width 21.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 655 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999798\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G03840 of width 20.\n",
      "Using motif -AT1G03840 of width 20.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 1089 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944298\n",
      "#   Estimated pi_0=0.957515\n",
      "Using motif +AT1G06070 of width 11.\n",
      "Using motif -AT1G06070 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 258 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06160 of width 8.\n",
      "Using motif -AT1G06160 of width 8.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 338 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06180 of width 15.\n",
      "Using motif -AT1G06180 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 631 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.954118\n",
      "#   Estimated pi_0=0.962203\n",
      "Using motif +AT1G06280 of width 15.\n",
      "Using motif -AT1G06280 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 326 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00151\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G06850 of width 14.\n",
      "Using motif -AT1G06850 of width 14.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 288 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00201\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G08000 of width 8.\n",
      "Using motif -AT1G08000 of width 8.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 400 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00101\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G08010 of width 15.\n",
      "Using motif -AT1G08010 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 398 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9826\n",
      "#   Estimated pi_0=0.992258\n",
      "Using motif +AT1G08320 of width 11.\n",
      "Using motif -AT1G08320 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 292 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.994706\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G08810 of width 19.\n",
      "Using motif -AT1G08810 of width 19.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 829 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938\n",
      "#   Estimated pi_0=0.948033\n",
      "Using motif +AT1G09030 of width 7.\n",
      "Using motif -AT1G09030 of width 7.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 561 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.936757\n",
      "#   Estimated pi_0=0.945625\n",
      "Using motif +AT1G09530 of width 10.\n",
      "Using motif -AT1G09530 of width 10.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 602 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G09540 of width 15.\n",
      "Using motif -AT1G09540 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 596 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.970896\n",
      "#   Estimated pi_0=0.98117\n",
      "Using motif +AT1G09770 of width 11.\n",
      "Using motif -AT1G09770 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 122 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00221\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G10120 of width 21.\n",
      "Using motif -AT1G10120 of width 21.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 478 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00131\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G12260 of width 15.\n",
      "Using motif -AT1G12260 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 399 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9446\n",
      "#   Estimated pi_0=0.951171\n",
      "Using motif +AT1G12610 of width 15.\n",
      "Using motif -AT1G12610 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 342 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0004\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G12630 of width 21.\n",
      "Using motif -AT1G12630 of width 21.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 517 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G13260 of width 17.\n",
      "Using motif -AT1G13260 of width 17.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 979 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940189\n",
      "#   Estimated pi_0=0.950141\n",
      "Using motif +AT1G13300 of width 15.\n",
      "Using motif -AT1G13300 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 587 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.910784\n",
      "#   Estimated pi_0=0.913962\n",
      "Using motif +AT1G13450 of width 19.\n",
      "Using motif -AT1G13450 of width 19.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 417 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8996\n",
      "#   Estimated pi_0=0.908704\n",
      "Using motif +AT1G14580 of width 20.\n",
      "Using motif -AT1G14580 of width 20.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 1470 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.942628\n",
      "#   Estimated pi_0=0.950184\n",
      "Using motif +AT1G14687 of width 11.\n",
      "Using motif -AT1G14687 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 427 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8692\n",
      "#   Estimated pi_0=0.873786\n",
      "Using motif +AT1G15720 of width 8.\n",
      "Using motif -AT1G15720 of width 8.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 178 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00281\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G16490 of width 15.\n",
      "Using motif -AT1G16490 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 665 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966019\n",
      "#   Estimated pi_0=0.980423\n",
      "Using motif +AT1G17950 of width 11.\n",
      "Using motif -AT1G17950 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 354 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9464\n",
      "#   Estimated pi_0=0.950841\n",
      "Using motif +AT1G18330 of width 12.\n",
      "Using motif -AT1G18330 of width 12.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 676 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8574\n",
      "#   Estimated pi_0=0.868319\n",
      "Using motif +AT1G18570 of width 21.\n",
      "Using motif -AT1G18570 of width 21.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 646 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9646\n",
      "#   Estimated pi_0=0.974331\n",
      "Using motif +AT1G18960 of width 22.\n",
      "Using motif -AT1G18960 of width 22.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 960 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.991786\n",
      "#   Estimated pi_0=0.997387\n",
      "Using motif +AT1G19000 of width 14.\n",
      "Using motif -AT1G19000 of width 14.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 668 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9152\n",
      "#   Estimated pi_0=0.924237\n",
      "Using motif +AT1G19040 of width 16.\n",
      "Using motif -AT1G19040 of width 16.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 447 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.993765\n",
      "#   Estimated pi_0=0.997653\n",
      "Using motif +AT1G19210 of width 14.\n",
      "Using motif -AT1G19210 of width 14.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 492 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00171\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G19350 of width 11.\n",
      "Using motif -AT1G19350 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 477 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G19790 of width 23.\n",
      "Using motif -AT1G19790 of width 23.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 678 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.8316\n",
      "#   Estimated pi_0=0.838252\n",
      "Using motif +AT1G19850 of width 10.\n",
      "Using motif -AT1G19850 of width 10.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 441 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.986441\n",
      "#   Estimated pi_0=0.990378\n",
      "Using motif +AT1G20640 of width 15.\n",
      "Using motif -AT1G20640 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 468 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0006\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G20980 of width 11.\n",
      "Using motif -AT1G20980 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 239 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.983103\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G21910 of width 20.\n",
      "Using motif -AT1G21910 of width 20.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 612 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999698\n",
      "#   Estimated pi_0=0.999698\n",
      "Using motif +AT1G22070 of width 14.\n",
      "Using motif -AT1G22070 of width 14.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 274 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.992549\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G22640 of width 11.\n",
      "Using motif -AT1G22640 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 245 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.959\n",
      "#   Estimated pi_0=0.959\n",
      "Using motif +AT1G22810 of width 15.\n",
      "Using motif -AT1G22810 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 715 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998081\n",
      "#   Estimated pi_0=0.999196\n",
      "Using motif +AT1G24260 of width 13.\n",
      "Using motif -AT1G24260 of width 13.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 868 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.942\n",
      "#   Estimated pi_0=0.95215\n",
      "Using motif +AT1G25340 of width 15.\n",
      "Using motif -AT1G25340 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 318 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9124\n",
      "#   Estimated pi_0=0.922752\n",
      "Using motif +AT1G25550 of width 11.\n",
      "Using motif -AT1G25550 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 587 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.9082\n",
      "#   Estimated pi_0=0.918584\n",
      "Using motif +AT1G25560 of width 8.\n",
      "Using motif -AT1G25560 of width 8.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 427 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.965225\n",
      "#   Estimated pi_0=0.97681\n",
      "Using motif +AT1G27360 of width 11.\n",
      "Using motif -AT1G27360 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 457 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.962569\n",
      "#   Estimated pi_0=0.964202\n",
      "Using motif +AT1G27730 of width 11.\n",
      "Using motif -AT1G27730 of width 11.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 710 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.935714\n",
      "#   Estimated pi_0=0.946129\n",
      "Using motif +AT1G28160 of width 15.\n",
      "Using motif -AT1G28160 of width 15.\n",
      "Finding best site passing the output threshold in each of the 3219 sequences.\n",
      "Found a best site in 512 sequences.\n",
      "Computing q-values.\n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from a uniformly sampled set of 10000 p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.0001\n",
      "#   Estimated pi_0=1\n",
      "Using motif +AT1G28300 of width 8.\n",
      "Using motif -AT1G28300 of width 8.\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "! fimo --oc ./WGCNA_ATAC_integration/fimo_results/MEblue\\\n",
    "/media/HDD2/donghui/bulk_ATAC_DM1_DM2d/scan_motif/Ath_TF_binding_motifs_plantTFDB.meme\\\n",
    "./WGCNA_ATAC_integration/fasta_files_and_control/MEblue.fasta "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "594e71df-b509-4af8-b20f-4e40dfb33ffc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97a07aff-64fb-47f4-b53d-7aac02e4f478",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a437c07b-66e6-47d2-b169-a23e85df274c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "b1b0d54f-2f5e-490f-a5a7-0cb4b8ab929f",
   "metadata": {},
   "source": [
    "## extract peak fastas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "db188476-0651-4f35-b744-81ec5ab28500",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Load the RDS file into Python\n",
    "peakAnno_tb_DM1_DM2d = pyreadr.read_r(os.path.join('./fimo/peakAnno_tb_DM1_DM2d.rds'))\n",
    "peakAnno_tb_DM1_DM2d = peakAnno_tb_DM1_DM2d[None]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b6ff5dd-943d-4324-aeb6-25a9382fabe3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "77cce321-3d52-4985-882a-8e58e1423869",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      seqnames     start       end  width  \\\n",
      "0            4   1499634   1501162   1529   \n",
      "1            5  17549546  17550412    867   \n",
      "2            1  28051844  28052977   1134   \n",
      "3            5   5236541   5237157    617   \n",
      "4            4  18125528  18126153    626   \n",
      "...        ...       ...       ...    ...   \n",
      "31351        1  28403669  28404892   1224   \n",
      "31352        5  21383686  21384233    548   \n",
      "31353        5  13056298  13056771    474   \n",
      "31354        1  29390905  29391031    127   \n",
      "31355        3   6428336   6429592   1257   \n",
      "\n",
      "                                                sequence  \n",
      "0      CCTTCGTAGCGTCTCGCTTTGGACCTTGCAGGCATTTTCCGACACA...  \n",
      "1      ACGATAAGACCCCATAGATAATGAACTGTCTAAAGACAAAACCTTT...  \n",
      "2      TCTCTTGAGCGGTGTAGGAAGGTGAAGTAATAAATGAAGAGGATAC...  \n",
      "3      AATTTCATAGTTAGTCATCTTCACCACATGTACAACAAAGATGAAT...  \n",
      "4      AGAGCAGAAAGGAGTCAGTGATGAGTATATATAGATAAGAACATTA...  \n",
      "...                                                  ...  \n",
      "31351  GACTCTACACCCTTCGAGCATGAAAACAACATCAGCCATAGACGGA...  \n",
      "31352  AACTTTTACCTAGAGAACTTGCTTGAGAAGTTCTTATGACCGAAGA...  \n",
      "31353  ATCTTAGGTTCATCTAAACCACGAAGACCAAACTTCTCCTCCACAT...  \n",
      "31354  TATATAAGTATATAGTTTGGTGATATAGTTAGAAGGTAGTTAAATC...  \n",
      "31355  TGTATTCAGCTTTGAGTTCTTCCTATACATAAGTATAATACCCGAA...  \n",
      "\n",
      "[31356 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "def extract_sequences(df, genome_path):\n",
    "    # Load the genome fasta file as a dictionary\n",
    "    genome_sequences = SeqIO.to_dict(SeqIO.parse(genome_path, \"fasta\"))\n",
    "\n",
    "    # Add a new column to the DataFrame for storing sequences\n",
    "    df['sequence'] = ''\n",
    "\n",
    "    # Iterate over rows in the DataFrame\n",
    "    for index, row in df.iterrows():\n",
    "        seq_name = row['seqnames']\n",
    "        start = row['start']\n",
    "        end = row['end']\n",
    "\n",
    "        # Extract the sequence and store it in the DataFrame\n",
    "        gene_seq = str(genome_sequences[str(seq_name)].seq[start-1:end])\n",
    "        df.at[index, 'sequence'] = gene_seq\n",
    "\n",
    "    return df\n",
    "\n",
    "# Define the path to the FASTA file\n",
    "genome_path = '/media/HDD2/Genomes/Ath_Ensembl56/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa'\n",
    "\n",
    "# Assuming 'peakAnno_tb_DM1_DM2d' is your DataFrame, call the function\n",
    "# Update 'peakAnno_tb_DM1_DM2d' with your DataFrame variable name\n",
    "sequences_df = extract_sequences(peakAnno_tb_DM1_DM2d, genome_path)\n",
    "print(sequences_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "8fb7b053-dc19-4a0a-9416-fa0fc940b575",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqnames</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>width</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4</td>\n",
       "      <td>1499634</td>\n",
       "      <td>1501162</td>\n",
       "      <td>1529</td>\n",
       "      <td>CCTTCGTAGCGTCTCGCTTTGGACCTTGCAGGCATTTTCCGACACA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>17549546</td>\n",
       "      <td>17550412</td>\n",
       "      <td>867</td>\n",
       "      <td>ACGATAAGACCCCATAGATAATGAACTGTCTAAAGACAAAACCTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>28051844</td>\n",
       "      <td>28052977</td>\n",
       "      <td>1134</td>\n",
       "      <td>TCTCTTGAGCGGTGTAGGAAGGTGAAGTAATAAATGAAGAGGATAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>5236541</td>\n",
       "      <td>5237157</td>\n",
       "      <td>617</td>\n",
       "      <td>AATTTCATAGTTAGTCATCTTCACCACATGTACAACAAAGATGAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>18125528</td>\n",
       "      <td>18126153</td>\n",
       "      <td>626</td>\n",
       "      <td>AGAGCAGAAAGGAGTCAGTGATGAGTATATATAGATAAGAACATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31351</th>\n",
       "      <td>1</td>\n",
       "      <td>28403669</td>\n",
       "      <td>28404892</td>\n",
       "      <td>1224</td>\n",
       "      <td>GACTCTACACCCTTCGAGCATGAAAACAACATCAGCCATAGACGGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31352</th>\n",
       "      <td>5</td>\n",
       "      <td>21383686</td>\n",
       "      <td>21384233</td>\n",
       "      <td>548</td>\n",
       "      <td>AACTTTTACCTAGAGAACTTGCTTGAGAAGTTCTTATGACCGAAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31353</th>\n",
       "      <td>5</td>\n",
       "      <td>13056298</td>\n",
       "      <td>13056771</td>\n",
       "      <td>474</td>\n",
       "      <td>ATCTTAGGTTCATCTAAACCACGAAGACCAAACTTCTCCTCCACAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31354</th>\n",
       "      <td>1</td>\n",
       "      <td>29390905</td>\n",
       "      <td>29391031</td>\n",
       "      <td>127</td>\n",
       "      <td>TATATAAGTATATAGTTTGGTGATATAGTTAGAAGGTAGTTAAATC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31355</th>\n",
       "      <td>3</td>\n",
       "      <td>6428336</td>\n",
       "      <td>6429592</td>\n",
       "      <td>1257</td>\n",
       "      <td>TGTATTCAGCTTTGAGTTCTTCCTATACATAAGTATAATACCCGAA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31356 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      seqnames     start       end  width  \\\n",
       "0            4   1499634   1501162   1529   \n",
       "1            5  17549546  17550412    867   \n",
       "2            1  28051844  28052977   1134   \n",
       "3            5   5236541   5237157    617   \n",
       "4            4  18125528  18126153    626   \n",
       "...        ...       ...       ...    ...   \n",
       "31351        1  28403669  28404892   1224   \n",
       "31352        5  21383686  21384233    548   \n",
       "31353        5  13056298  13056771    474   \n",
       "31354        1  29390905  29391031    127   \n",
       "31355        3   6428336   6429592   1257   \n",
       "\n",
       "                                                sequence  \n",
       "0      CCTTCGTAGCGTCTCGCTTTGGACCTTGCAGGCATTTTCCGACACA...  \n",
       "1      ACGATAAGACCCCATAGATAATGAACTGTCTAAAGACAAAACCTTT...  \n",
       "2      TCTCTTGAGCGGTGTAGGAAGGTGAAGTAATAAATGAAGAGGATAC...  \n",
       "3      AATTTCATAGTTAGTCATCTTCACCACATGTACAACAAAGATGAAT...  \n",
       "4      AGAGCAGAAAGGAGTCAGTGATGAGTATATATAGATAAGAACATTA...  \n",
       "...                                                  ...  \n",
       "31351  GACTCTACACCCTTCGAGCATGAAAACAACATCAGCCATAGACGGA...  \n",
       "31352  AACTTTTACCTAGAGAACTTGCTTGAGAAGTTCTTATGACCGAAGA...  \n",
       "31353  ATCTTAGGTTCATCTAAACCACGAAGACCAAACTTCTCCTCCACAT...  \n",
       "31354  TATATAAGTATATAGTTTGGTGATATAGTTAGAAGGTAGTTAAATC...  \n",
       "31355  TGTATTCAGCTTTGAGTTCTTCCTATACATAAGTATAATACCCGAA...  \n",
       "\n",
       "[31356 rows x 5 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "8ad04b19-8b04-4043-bafd-2a27954ff8dc",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def write_sequences_to_fasta(df, output_path):\n",
    "    with open(output_path, 'w') as fasta_file:\n",
    "        for index, row in df.iterrows():\n",
    "            # Create a header using seqnames, start, and end\n",
    "            header = f\">{row['seqnames']}_{row['start']}_{row['end']}\"\n",
    "            sequence = row['sequence']\n",
    "\n",
    "            # Write the header and sequence to the FASTA file\n",
    "            fasta_file.write(f\"{header}\\n{sequence}\\n\")\n",
    "\n",
    "# Usage example\n",
    "output_fasta_path = \"./fimo/sequences_for_all_peaks_DM1_DM2d.fasta\"\n",
    "write_sequences_to_fasta(sequences_df, output_fasta_path)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93921728-1678-448d-8c89-cffae1eab053",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:work2] *",
   "language": "python",
   "name": "conda-env-work2-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
